{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "training set ~ 3k, ~300 unique genes\n",
    "classes, genes highly uneven\n",
    "* Simple baseline model assigning classes proportional to training set : 1.8\n",
    "* Bag of words + Naive Bayes: ~12\n",
    "* Bag of genes models with logistic regression: 1.1\n",
    "* Bag of genes models with random forest: 1.3\n",
    "* Bag of words for filtered words (between 10-90%ile) and logistic regression: 1.0\n",
    "* Bag of words for filtered words (between 10-90%ile) and random forest: 1.17\n",
    "* Bag of words for filtered words in abstract and logistic regression: 1.2\n",
    "* Bag of words for filtered words, gene info, and type of mutation: Accuracy on test set: 0.99\n",
    "* Default XGBoost: .99 vs .58 train/test\n",
    "\n",
    "* lowering sentences seems to be helpful\n",
    "* averaging XGBoost and LR seems to help slightly\n",
    "* finding sentences for g + v separately is good, then use both\n",
    "* for cutoffs, use nEmpty instead of len(merged).\n",
    "* will use bigrams based on g (lower gene!)\n",
    "\n",
    "\n",
    "* Using stems of words is not helpful (slightly worse performance)\n",
    "* filtering bigrams for stopwords is worse.\n",
    "\n",
    "\n",
    "\n",
    "Abstract: ~250 words, maybe up to a weird junction (period no space, next section, etc.), any sentences that include the allele, maybe discussion?\n",
    "\n",
    "Thoughts:\n",
    "* Class 1: loss of function causes increased resistance to tamoxifen-- deletion/truncation?\n",
    "* Class 2: loh (recessive, lf?) causes increased motility, viability.\n",
    "* Class 3: non-causal, non-transforming\n",
    "* Class 4: benign\n",
    "* Class 5: destabilizing? Potentially higher function?\n",
    "* Class 6: resistant to chemo, dominant?\n",
    "* Class 7: \n",
    "* Class 8: \n",
    "* Class 9: \n",
    "\n",
    "\n",
    "1. Likely Loss-of-function  568\n",
    "2. Likely Gain-of-function  452\n",
    "3. Neutral 89\n",
    "4. Loss-of-function 686\n",
    "5. Likely Neutral  242\n",
    "6. Inconclusive  275\n",
    "7. Gain-of-function 953\n",
    "8. Likely Switch-of-function  19\n",
    "9. Switch-of-function  37\n",
    "\n",
    "1, 4 lf combo: 1254\n",
    "2, 7 gf combo: 1405\n",
    "3, 5 neutral combo: 331\n",
    "8, 9 switch combo: 56\n",
    "6, inconclusive: 275\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import nltk\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import _pickle as pickle\n",
    "import sklearn\n",
    "import string\n",
    "import os\n",
    "from nltk.classify import NaiveBayesClassifier\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "import xgboost as xgb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/hjohnsen/Desktop\n"
     ]
    }
   ],
   "source": [
    "%cd \"/Users/hjohnsen/Desktop\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>sentenceWordBag</th>\n",
       "      <th>bigrams</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>wordfeatures</th>\n",
       "      <th>mutType</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>bigramfeatures</th>\n",
       "      <th>all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...</td>\n",
       "      <td>[(Cyclin, M,), (M,, an), (an, orphan), (orphan...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>15 Apart from N454D, all residues affected by ...</td>\n",
       "      <td>[15, Apart, from, N454D, ,, all, residues, aff...</td>\n",
       "      <td>{'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...</td>\n",
       "      <td>[(15, Apart), (Apart, from), (from, N454D,), (...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>Finally, the third group constituted mutatio...</td>\n",
       "      <td>[Finally, ,, the, third, group, constituted, m...</td>\n",
       "      <td>{'Finally': 1, 'third': 1, 'group': 1, 'consti...</td>\n",
       "      <td>[(Finally,, the), (the, third), (third, group)...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene                                               Text  \\\n",
       "0   0  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   1     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "2   2     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "3   3     CBL  Recent evidence has demonstrated that acquired...   \n",
       "4   4     CBL  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "              Variation  Class  \\\n",
       "0  Truncating Mutations      1   \n",
       "1                 W802*      2   \n",
       "2                 Q249E      2   \n",
       "3                 N454D      3   \n",
       "4                 L399V      4   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "1   Using select c-CBL somatic mutations such as ...   \n",
       "2   Using select c-CBL somatic mutations such as ...   \n",
       "3  15 Apart from N454D, all residues affected by ...   \n",
       "4    Finally, the third group constituted mutatio...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "2  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "3  [15, Apart, from, N454D, ,, all, residues, aff...   \n",
       "4  [Finally, ,, the, third, group, constituted, m...   \n",
       "\n",
       "                                     sentenceWordBag  \\\n",
       "0  {'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...   \n",
       "1  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "2  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "3  {'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...   \n",
       "4  {'Finally': 1, 'third': 1, 'group': 1, 'consti...   \n",
       "\n",
       "                                             bigrams  \\\n",
       "0  [(Cyclin, M,), (M,, an), (an, orphan), (orphan...   \n",
       "1  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "2  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "3  [(15, Apart), (Apart, from), (from, N454D,), (...   \n",
       "4  [(Finally,, the), (the, third), (third, group)...   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        wordfeatures                mutType  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 1, 0, 0]   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 1]   \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "4  [1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      bigramfeatures  \\\n",
       "0  [1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...   \n",
       "1  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                                 all  \n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...  \n",
       "4  [1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if not os.path.exists('processedmerged.pkl'):\n",
    "    print(\"File does not exist; building new tokenized file\")\n",
    "    merged = pd.read_csv('combotraining.csv', sep=\",\")\n",
    "    merged.rename(columns={'Unnamed: 0' : 'ID'},inplace=True)\n",
    "    merged[\"split\"]=np.random.random([len(merged)])<0.8\n",
    "    merged[\"tokenized\"]=merged[\"Text\"].map(nltk.word_tokenize)\n",
    "    with open(\"tokenized.pkl\",'wb') as f:\n",
    "        pickle.dump(merged, f)\n",
    "else:\n",
    "    merged=pickle.load(open('processedmerged.pkl','rb'))\n",
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"split\"]=np.random.random([len(merged)])<0.8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "if not os.path.exists('tokenized.pkl'):\n",
    "    print(\"File does not exist; building new tokenized file\")\n",
    "    tvarsdf = pd.read_csv('training_variants', sep=\",\")\n",
    "    tvarsdf[\"split\"]=np.random.random([len(tvarsdf)])<0.8\n",
    "    ttextdf = pd.read_csv('training_text', sep=\"\\|\\|\")\n",
    "    ttextdf.rename(columns={'ID,Text': \"Text\"}, inplace = True)\n",
    "    ttextdf[\"ID\"]=ttextdf.index\n",
    "    merged = pd.merge(tvarsdf,ttextdf, how=\"inner\")\n",
    "    merged[\"tokenized\"]=merged[\"Text\"].map(nltk.word_tokenize)\n",
    "    with open(\"tokenized.pkl\",'wb') as f:\n",
    "        pickle.dump(merged, f)\n",
    "else:\n",
    "    merged=pickle.load(open('tokenized.pkl','rb'))\n",
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def logistictester(df, X, Y, C=1e-2):\n",
    "    training = df[df[\"split\"]]\n",
    "    test = df[df[\"split\"]==False]\n",
    "    X_train = training[X].tolist()\n",
    "    y_train = training[Y].tolist()\n",
    "    X_test=test[X].tolist()\n",
    "\n",
    "    lr = LogisticRegression( solver = \"lbfgs\",\n",
    "                                multi_class=\"multinomial\",\n",
    "                                penalty='l2',\n",
    "                                C=C,\n",
    "                                fit_intercept=True\n",
    "                                )\n",
    "    lr.fit(X_train, y_train)\n",
    "    y_pred = lr.predict_proba(X_test) \n",
    "    \n",
    "    ls = []\n",
    "    for i in range(len(y_pred)):\n",
    "        c = test.iloc[i][Y]\n",
    "        prob = y_pred[i][c-1]\n",
    "        ls.append(-np.log(prob+1e-6))\n",
    "\n",
    "    print(\"Accuracy on test set:\")\n",
    "    print(np.mean(ls))\n",
    "\n",
    "    ytrain_pred = lr.predict_proba(X_train)\n",
    "    ls = []\n",
    "    for i in range(len(ytrain_pred)):\n",
    "        c = training.iloc[i][\"Class\"]\n",
    "        prob = ytrain_pred[i][c-1]\n",
    "        ls.append(-np.log(prob+1e-6))\n",
    "    print(\"Accuracy on training set:\")\n",
    "    print(np.mean(ls))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "269"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniqueGenes = merged[\"Gene\"].unique()\n",
    "len(uniqueGenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BRCA1       293\n",
       "TP53        182\n",
       "EGFR        157\n",
       "PTEN        140\n",
       "BRCA2       139\n",
       "KIT         110\n",
       "BRAF        104\n",
       "ALK          77\n",
       "ERBB2        76\n",
       "PDGFRA       67\n",
       "PIK3CA       62\n",
       "CDKN2A       58\n",
       "FGFR2        55\n",
       "FLT3         54\n",
       "TSC2         52\n",
       "MTOR         50\n",
       "KRAS         49\n",
       "MAP2K1       47\n",
       "RET          45\n",
       "VHL          45\n",
       "FGFR3        43\n",
       "MLH1         39\n",
       "JAK2         37\n",
       "SMAD4        37\n",
       "MET          36\n",
       "NOTCH1       35\n",
       "AKT1         32\n",
       "ABL1         29\n",
       "PTPN11       29\n",
       "CTNNB1       28\n",
       "           ... \n",
       "ERRFI1        1\n",
       "PIK3R3        1\n",
       "MEN1          1\n",
       "NCOR1         1\n",
       "LATS2         1\n",
       "AXIN1         1\n",
       "STAG2         1\n",
       "PAX8          1\n",
       "RNF43         1\n",
       "PPM1D         1\n",
       "FGF3          1\n",
       "RARA          1\n",
       "EPCAM         1\n",
       "SDHB          1\n",
       "IL7R          1\n",
       "RAD54L        1\n",
       "AXL           1\n",
       "HIST1H1C      1\n",
       "DUSP4         1\n",
       "CEBPA         1\n",
       "DIS3          1\n",
       "PMS1          1\n",
       "FAM58A        1\n",
       "IKZF1         1\n",
       "ATR           1\n",
       "HLA-B         1\n",
       "VEGFA         1\n",
       "RAD51D        1\n",
       "CDKN2C        1\n",
       "KLF4          1\n",
       "Name: Gene, Length: 269, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[\"Gene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Class\n",
       "1     662\n",
       "2     498\n",
       "3      96\n",
       "4     751\n",
       "5     267\n",
       "6     297\n",
       "7    1054\n",
       "8      21\n",
       "9      43\n",
       "Name: ID, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.groupby(\"Class\").count()[\"ID\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Simple baseline model assigning classes proportional to training set : 1.8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ttr=merged[merged[\"split\"]]\n",
    "tt=merged[merged[\"split\"]==False]\n",
    "counts=ttr.groupby(\"Class\").count()\n",
    "probs = {}\n",
    "for i in range(len(counts)):\n",
    "    probs[i+1]=counts.iloc[i][\"ID\"]/len(ttr)\n",
    "print(probs)\n",
    "tt[\"Class\"].map(lambda c: -np.log(probs[c])).mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# bag of words: terrible."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def build_bag_of_words_features_filtered(words):\n",
    "    bag = {}\n",
    "    useless_words = nltk.corpus.stopwords.words(\"english\") + list(string.punctuation)\n",
    "    for word in words:\n",
    "        if not word in useless_words:\n",
    "            bag[word]=1\n",
    "    return bag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# PROB ALREADY IN THE PICKLE\n",
    "#merged[\"wordbag\"]=merged[\"tokenized\"].map(build_bag_of_words_features_filtered)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# bag of genes, gene identity models with logistic regression: 1.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def build_bag_of_genes(words):\n",
    "    bag={}\n",
    "    for word in words:\n",
    "        if word.upper() in uniqueGenes:\n",
    "            bag[word.upper()]=1\n",
    "    return bag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# PROB ALREADY IN THE PICKLE\n",
    "#merged[\"genebag\"]=merged[\"tokenized\"].map(build_bag_of_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Most Informative Features\n",
      "                   SRSF2 = 1                   9 : 4      =    107.1 : 1.0\n",
      "                    RARA = 1                   8 : 4      =     87.3 : 1.0\n",
      "                  PDGFRB = 1                   3 : 4      =     76.4 : 1.0\n",
      "                   U2AF1 = 1                   9 : 6      =     71.4 : 1.0\n",
      "                    BCOR = 1                   8 : 2      =     69.9 : 1.0\n",
      "                    IDH1 = 1                   8 : 6      =     58.2 : 1.0\n",
      "                    TET1 = 1                   9 : 7      =     54.9 : 1.0\n",
      "                   AURKB = 1                   9 : 7      =     42.4 : 1.0\n",
      "                   SF3B1 = 1                   9 : 4      =     41.2 : 1.0\n",
      "                    ATRX = 1                   8 : 4      =     40.7 : 1.0\n"
     ]
    }
   ],
   "source": [
    "#training = merged[merged[\"split\"]]\n",
    "#test = merged[merged[\"split\"]==False]\n",
    "#trainbags = [(training[\"bag\"].iloc[i], training[\"Class\"].iloc[i]) for i in range(len(training))]\n",
    "#sentiment_classifier = NaiveBayesClassifier.train(trainbags)\n",
    "#sentiment_classifier.show_most_informative_features()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#del merged[\"genefeature\"]\n",
    "#del merged[\"features\"]\n",
    "#del merged[\"combofeatures\"]\n",
    "#del merged[\"combowords\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>split</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>genefeatures</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>True</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>False</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>True</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>N454D</td>\n",
       "      <td>False</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>L399V</td>\n",
       "      <td>True</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID  Class    Gene                                               Text  \\\n",
       "0   0      1  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   1      2     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "2   2      2     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "3   3      3     CBL  Recent evidence has demonstrated that acquired...   \n",
       "4   4      4     CBL  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "              Variation  split  \\\n",
       "0  Truncating Mutations   True   \n",
       "1                 W802*  False   \n",
       "2                 Q249E   True   \n",
       "3                 N454D  False   \n",
       "4                 L399V   True   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        genefeatures  \n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[\"genesmentioned\"]=merged[\"genebag\"].map(lambda b: [int(gene in b) for gene in uniqueGenes])\n",
    "merged[\"geneofinterest\"]=merged[\"Gene\"].map(lambda b: [int(gene == b) for gene in uniqueGenes])\n",
    "merged[\"genefeatures\"]=merged[\"genesmentioned\"]+merged[\"geneofinterest\"]\n",
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(\"tokenized.pkl\",'wb') as f:\n",
    "    pickle.dump(merged, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.58832785636\n",
      "Accuracy on training set:\n",
      "1.45186516533\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"genefeatures\", \"Class\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# gene features plus random forest: 1.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "lr = RandomForestClassifier(n_estimators=1000)\n",
    "lr.fit(X_train, y_train)\n",
    "y_pred = lr.predict_proba(X_test)\n",
    " \n",
    "ls = []\n",
    "for i in range(len(y_pred)):\n",
    "    c = test.iloc[i][\"Class\"]\n",
    "    prob = y_pred[i][c-1]\n",
    "    ls.append(-np.log(prob+1e-6))\n",
    "np.mean(ls)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bag of words for filtered words (between 10-90%ile) and logistic regression: 1.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "wordcounts = {}\n",
    "for i in merged[\"tokenized\"]:\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.9 and v>len(merged)*.1:\n",
    "        abridgedsorted[k]=v\n",
    "merged[\"wordfeatures\"]=merged[\"wordbag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "323789"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(wordcounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('in', 3683),\n",
       " (',', 3683),\n",
       " ('.', 3683),\n",
       " ('and', 3683),\n",
       " ('of', 3683),\n",
       " ('the', 3683),\n",
       " ('(', 3682),\n",
       " (')', 3682),\n",
       " ('a', 3682),\n",
       " ('to', 3680),\n",
       " ('that', 3679),\n",
       " ('for', 3678),\n",
       " ('with', 3677),\n",
       " ('is', 3677),\n",
       " ('The', 3674),\n",
       " ('as', 3670),\n",
       " ('an', 3669),\n",
       " ('by', 3669),\n",
       " ('at', 3667),\n",
       " ('or', 3666),\n",
       " ('were', 3665),\n",
       " ('are', 3663),\n",
       " ('on', 3662),\n",
       " ('be', 3661),\n",
       " ('was', 3661),\n",
       " ('not', 3661),\n",
       " ('this', 3658),\n",
       " ('have', 3655),\n",
       " ('from', 3652),\n",
       " ('1', 3651),\n",
       " ('we', 3649),\n",
       " ('also', 3646),\n",
       " ('%', 3646),\n",
       " ('In', 3645),\n",
       " ('3', 3645),\n",
       " ('which', 3645),\n",
       " ('these', 3644),\n",
       " ('2', 3642),\n",
       " (';', 3639),\n",
       " ('been', 3629),\n",
       " ('other', 3623),\n",
       " (':', 3610),\n",
       " ('both', 3609),\n",
       " ('all', 3602),\n",
       " ('4', 3601),\n",
       " ('5', 3598),\n",
       " ('has', 3596),\n",
       " ('cells', 3592),\n",
       " ('A', 3588),\n",
       " ('We', 3585),\n",
       " ('mutations', 3578),\n",
       " ('but', 3576),\n",
       " ('analysis', 3575),\n",
       " ('protein', 3574),\n",
       " ('may', 3573),\n",
       " ('These', 3562),\n",
       " ('mutation', 3561),\n",
       " ('data', 3560),\n",
       " ('using', 3560),\n",
       " ('shown', 3550),\n",
       " ('This', 3529),\n",
       " ('than', 3527),\n",
       " ('To', 3521),\n",
       " ('only', 3519),\n",
       " ('10', 3515),\n",
       " ('used', 3512),\n",
       " ('results', 3509),\n",
       " ('found', 3505),\n",
       " ('one', 3504),\n",
       " ('our', 3498),\n",
       " ('6', 3496),\n",
       " ('it', 3495),\n",
       " ('no', 3490),\n",
       " ('between', 3487),\n",
       " ('expression', 3484),\n",
       " ('two', 3481),\n",
       " ('more', 3477),\n",
       " ('most', 3477),\n",
       " ('gene', 3473),\n",
       " ('can', 3463),\n",
       " ('into', 3462),\n",
       " ('cell', 3460),\n",
       " ('previously', 3453),\n",
       " ('well', 3445),\n",
       " ('observed', 3444),\n",
       " ('However', 3443),\n",
       " ('study', 3442),\n",
       " ('described', 3431),\n",
       " ('identified', 3430),\n",
       " ('their', 3429),\n",
       " ('activity', 3425),\n",
       " ('reported', 3420),\n",
       " ('including', 3416),\n",
       " ('each', 3404),\n",
       " ('domain', 3403),\n",
       " ('addition', 3396),\n",
       " ('different', 3390),\n",
       " ('such', 3377),\n",
       " ('three', 3372),\n",
       " ('similar', 3370),\n",
       " ('studies', 3370),\n",
       " ('its', 3368),\n",
       " ('compared', 3359),\n",
       " ('had', 3333),\n",
       " ('20', 3318),\n",
       " ('either', 3316),\n",
       " ('same', 3291),\n",
       " ('DNA', 3288),\n",
       " ('mutant', 3287),\n",
       " ('As', 3279),\n",
       " ('could', 3279),\n",
       " ('number', 3274),\n",
       " ('human', 3263),\n",
       " ('known', 3260),\n",
       " ('proteins', 3257),\n",
       " ('Table', 3255),\n",
       " ('7', 3243),\n",
       " ('associated', 3242),\n",
       " ('For', 3238),\n",
       " ('presence', 3237),\n",
       " ('within', 3225),\n",
       " ('further', 3225),\n",
       " ('Although', 3214),\n",
       " ('cancer', 3211),\n",
       " ('some', 3208),\n",
       " ('after', 3200),\n",
       " ('12', 3199),\n",
       " ('those', 3197),\n",
       " ('performed', 3196),\n",
       " ('It', 3195),\n",
       " ('effect', 3192),\n",
       " ('function', 3188),\n",
       " ('All', 3181),\n",
       " ('15', 3177),\n",
       " ('control', 3176),\n",
       " ('wild-type', 3175),\n",
       " ('additional', 3172),\n",
       " ('likely', 3166),\n",
       " ('important', 3165),\n",
       " ('present', 3157),\n",
       " ('respectively', 3155),\n",
       " ('Figure', 3150),\n",
       " ('sequence', 3141),\n",
       " ('did', 3141),\n",
       " ('expressed', 3139),\n",
       " ('8', 3138),\n",
       " ('several', 3133),\n",
       " ('showed', 3131),\n",
       " ('levels', 3122),\n",
       " ('functional', 3118),\n",
       " ('amino', 3118),\n",
       " ('when', 3112),\n",
       " ('9', 3109),\n",
       " ('binding', 3096),\n",
       " ('role', 3093),\n",
       " ('activation', 3092),\n",
       " ('result', 3089),\n",
       " ('high', 3089),\n",
       " ('whether', 3085),\n",
       " ('increased', 3083),\n",
       " ('50', 3077),\n",
       " ('indicated', 3075),\n",
       " ('suggest', 3072),\n",
       " ('significant', 3070),\n",
       " ('patients', 3068),\n",
       " ('tumor', 3068),\n",
       " ('Our', 3064),\n",
       " ('specific', 3063),\n",
       " ('analyzed', 3059),\n",
       " ('through', 3057),\n",
       " ('contrast', 3056),\n",
       " ('genes', 3052),\n",
       " ('they', 3051),\n",
       " ('assay', 3041),\n",
       " ('acid', 3037),\n",
       " ('because', 3034),\n",
       " ('containing', 3034),\n",
       " ('region', 3032),\n",
       " ('first', 3031),\n",
       " ('detected', 3030),\n",
       " ('14', 3028),\n",
       " ('effects', 3027),\n",
       " ('determined', 3025),\n",
       " ('whereas', 3022),\n",
       " ('obtained', 3019),\n",
       " ('then', 3005),\n",
       " ('higher', 3005),\n",
       " ('sequencing', 3004),\n",
       " ('11', 3001),\n",
       " ('30', 2997),\n",
       " ('16', 2990),\n",
       " ('site', 2988),\n",
       " ('13', 2977),\n",
       " ('show', 2975),\n",
       " ('vitro', 2972),\n",
       " ('any', 2971),\n",
       " ('highly', 2966),\n",
       " ('suggesting', 2953),\n",
       " ('total', 2953),\n",
       " ('multiple', 2947),\n",
       " ('clinical', 2945),\n",
       " ('would', 2940),\n",
       " ('growth', 2934),\n",
       " ('resulting', 2933),\n",
       " ('lines', 2933),\n",
       " ('small', 2930),\n",
       " ('cases', 2912),\n",
       " ('independent', 2906),\n",
       " ('loss', 2906),\n",
       " ('due', 2905),\n",
       " ('18', 2904),\n",
       " ('type', 2900),\n",
       " ('assays', 2896),\n",
       " ('development', 2892),\n",
       " ('common', 2888),\n",
       " ('24', 2886),\n",
       " ('19', 2885),\n",
       " ('least', 2880),\n",
       " ('single', 2880),\n",
       " ('potential', 2877),\n",
       " ('previous', 2871),\n",
       " ('there', 2870),\n",
       " ('possible', 2870),\n",
       " ('tumors', 2860),\n",
       " ('significantly', 2857),\n",
       " ('although', 2851),\n",
       " ('provide', 2847),\n",
       " ('tested', 2845),\n",
       " ('less', 2845),\n",
       " ('Thus', 2843),\n",
       " ('mutants', 2841),\n",
       " ('involved', 2841),\n",
       " ('will', 2830),\n",
       " ('line', 2829),\n",
       " ('25', 2826),\n",
       " ('Results', 2824),\n",
       " ('Fig', 2822),\n",
       " ('17', 2822),\n",
       " ('samples', 2818),\n",
       " ('low', 2817),\n",
       " ('genetic', 2813),\n",
       " ('experiments', 2810),\n",
       " ('required', 2807),\n",
       " ('B', 2804),\n",
       " ('indicate', 2801),\n",
       " ('might', 2799),\n",
       " ('based', 2792),\n",
       " ('C', 2781),\n",
       " ('21', 2779),\n",
       " ('do', 2777),\n",
       " ('evidence', 2770),\n",
       " ('disease', 2763),\n",
       " ('revealed', 2762),\n",
       " ('model', 2760),\n",
       " ('suggests', 2758),\n",
       " ('many', 2749),\n",
       " ('100', 2748),\n",
       " ('large', 2746),\n",
       " ('confirmed', 2744),\n",
       " ('level', 2743),\n",
       " ('four', 2740),\n",
       " ('against', 2734),\n",
       " ('domains', 2732),\n",
       " ('under', 2729),\n",
       " ('molecular', 2727),\n",
       " ('changes', 2723),\n",
       " ('kinase', 2723),\n",
       " ('affect', 2722),\n",
       " ('determine', 2720),\n",
       " ('according', 2718),\n",
       " ('absence', 2715),\n",
       " ('available', 2712),\n",
       " ('reduced', 2710),\n",
       " ('without', 2710),\n",
       " ('consistent', 2710),\n",
       " ('lower', 2709),\n",
       " ('generated', 2709),\n",
       " ('22', 2706),\n",
       " ('following', 2701),\n",
       " ('normal', 2696),\n",
       " ('target', 2696),\n",
       " ('mutated', 2695),\n",
       " ('residues', 2687),\n",
       " ('cancers', 2681),\n",
       " ('ability', 2674),\n",
       " ('Mutations', 2663),\n",
       " ('[', 2652),\n",
       " ('sites', 2651),\n",
       " (']', 2651),\n",
       " ('directly', 2648),\n",
       " ('family', 2645),\n",
       " ('therefore', 2635),\n",
       " ('response', 2634),\n",
       " ('et', 2627),\n",
       " ('identify', 2625),\n",
       " ('structure', 2620),\n",
       " ('examined', 2617),\n",
       " ('thus', 2616),\n",
       " ('located', 2612),\n",
       " ('primary', 2612),\n",
       " ('case', 2609),\n",
       " ('vector', 2609),\n",
       " ('above', 2608),\n",
       " ('recently', 2607),\n",
       " ('40', 2605),\n",
       " ('test', 2602),\n",
       " ('complex', 2600),\n",
       " ('>', 2597),\n",
       " ('factor', 2595),\n",
       " ('analyses', 2593),\n",
       " ('expressing', 2591),\n",
       " ('One', 2590),\n",
       " ('out', 2589),\n",
       " ('26', 2585),\n",
       " ('findings', 2583),\n",
       " ('Furthermore', 2581),\n",
       " ('negative', 2564),\n",
       " ('point', 2561),\n",
       " ('predicted', 2560),\n",
       " ('group', 2559),\n",
       " ('PCR', 2555),\n",
       " ('distinct', 2553),\n",
       " ('where', 2553),\n",
       " ('indicating', 2547),\n",
       " ('expected', 2544),\n",
       " ('patient', 2538),\n",
       " ('<', 2538),\n",
       " ('27', 2528),\n",
       " ('new', 2525),\n",
       " ('mechanisms', 2523),\n",
       " ('included', 2522),\n",
       " ('phosphorylation', 2520),\n",
       " ('majority', 2518),\n",
       " ('23', 2512),\n",
       " ('form', 2509),\n",
       " ('Interestingly', 2509),\n",
       " ('interaction', 2502),\n",
       " ('change', 2497),\n",
       " ('if', 2494),\n",
       " ('very', 2494),\n",
       " ('increase', 2491),\n",
       " ('regions', 2490),\n",
       " ('Discussion', 2490),\n",
       " ('missense', 2489),\n",
       " ('use', 2486),\n",
       " ('mechanism', 2481),\n",
       " ('critical', 2475),\n",
       " ('cellular', 2473),\n",
       " ('relative', 2470),\n",
       " ('29', 2470),\n",
       " ('28', 2470),\n",
       " ('corresponding', 2468),\n",
       " ('signaling', 2460),\n",
       " ('transfected', 2459),\n",
       " ('and/or', 2457),\n",
       " ('occur', 2457),\n",
       " ('suggested', 2456),\n",
       " ('differences', 2455),\n",
       " ('Cell', 2444),\n",
       " ('system', 2440),\n",
       " ('positive', 2439),\n",
       " ('however', 2439),\n",
       " ('pathway', 2437),\n",
       " ('part', 2436),\n",
       " ('sequences', 2436),\n",
       " ('Cancer', 2436),\n",
       " ('report', 2431),\n",
       " ('treatment', 2431),\n",
       " ('five', 2428),\n",
       " ('downstream', 2428),\n",
       " ('second', 2427),\n",
       " ('assessed', 2424),\n",
       " ('time', 2423),\n",
       " ('represent', 2423),\n",
       " ('Methods', 2421),\n",
       " ('35', 2421),\n",
       " ('selected', 2418),\n",
       " ('even', 2410),\n",
       " ('antibody', 2409),\n",
       " ('does', 2408),\n",
       " ('conserved', 2407),\n",
       " ('types', 2405),\n",
       " ('followed', 2403),\n",
       " ('together', 2403),\n",
       " ('should', 2402),\n",
       " ('individual', 2398),\n",
       " ('measured', 2398),\n",
       " ('inhibitor', 2393),\n",
       " ('residue', 2392),\n",
       " ('position', 2389),\n",
       " ('acids', 2387),\n",
       " ('frequency', 2387),\n",
       " ('characterized', 2386),\n",
       " ('inhibition', 2382),\n",
       " ('affected', 2381),\n",
       " ('remains', 2379),\n",
       " ('proliferation', 2376),\n",
       " ('various', 2372),\n",
       " ('s', 2372),\n",
       " ('contribute', 2369),\n",
       " ('conditions', 2365),\n",
       " ('active', 2362),\n",
       " ('example', 2362),\n",
       " ('another', 2361),\n",
       " ('over', 2359),\n",
       " ('fact', 2358),\n",
       " ('being', 2353),\n",
       " ('among', 2352),\n",
       " ('see', 2351),\n",
       " ('per', 2351),\n",
       " ('set', 2350),\n",
       " ('demonstrated', 2348),\n",
       " ('showing', 2348),\n",
       " ('sample', 2344),\n",
       " ('32', 2343),\n",
       " ('alterations', 2342),\n",
       " ('derived', 2338),\n",
       " ('direct', 2335),\n",
       " ('novel', 2334),\n",
       " ('include', 2333),\n",
       " ('recent', 2329),\n",
       " ('resulted', 2327),\n",
       " ('31', 2327),\n",
       " ('factors', 2322),\n",
       " ('After', 2313),\n",
       " ('complete', 2312),\n",
       " ('range', 2309),\n",
       " ('vivo', 2307),\n",
       " ('Introduction', 2304),\n",
       " ('formation', 2302),\n",
       " ('genomic', 2298),\n",
       " ('', 2294),\n",
       " ('before', 2293),\n",
       " ('considered', 2292),\n",
       " ('while', 2291),\n",
       " ('Cells', 2289),\n",
       " ('frequently', 2288),\n",
       " ('during', 2278),\n",
       " ('pathways', 2277),\n",
       " ('difference', 2274),\n",
       " ('major', 2272),\n",
       " ('Therefore', 2265),\n",
       " ('strong', 2262),\n",
       " ('signal', 2258),\n",
       " ('lead', 2255),\n",
       " ('support', 2255),\n",
       " ('overall', 2252),\n",
       " ('antibodies', 2250),\n",
       " ('cause', 2249),\n",
       " ('functions', 2244),\n",
       " ('lack', 2241),\n",
       " ('structural', 2240),\n",
       " ('33', 2239),\n",
       " ('encoding', 2234),\n",
       " ('approximately', 2234),\n",
       " ('shows', 2229),\n",
       " ('inhibitors', 2228),\n",
       " ('induced', 2226),\n",
       " ('Analysis', 2224),\n",
       " ('them', 2223),\n",
       " ('possibility', 2223),\n",
       " ('Because', 2221),\n",
       " ('Moreover', 2219),\n",
       " ('observation', 2216),\n",
       " ('substitution', 2215),\n",
       " ('particular', 2213),\n",
       " ('provided', 2211),\n",
       " ('here', 2210),\n",
       " ('37', 2208),\n",
       " ('developed', 2207),\n",
       " ('34', 2202),\n",
       " ('Two', 2202),\n",
       " ('significance', 2199),\n",
       " ('treated', 2198),\n",
       " ('controls', 2185),\n",
       " ('48', 2183),\n",
       " ('Of', 2178),\n",
       " ('unknown', 2177),\n",
       " ('deletion', 2176),\n",
       " ('able', 2174),\n",
       " ('strongly', 2169),\n",
       " ('substitutions', 2169),\n",
       " ('somatic', 2159),\n",
       " ('status', 2158),\n",
       " ('cDNA', 2158),\n",
       " ('variants', 2158),\n",
       " ('sensitivity', 2155),\n",
       " ('al', 2155),\n",
       " ('transcription', 2152),\n",
       " ('standard', 2149),\n",
       " ('key', 2144),\n",
       " ('tissue', 2144),\n",
       " ('phenotype', 2143),\n",
       " ('D', 2142),\n",
       " ('groups', 2141),\n",
       " ('=', 2140),\n",
       " ('indicates', 2139),\n",
       " ('basis', 2139),\n",
       " ('', 2138),\n",
       " ('receptor', 2137),\n",
       " ('An', 2135),\n",
       " ('assess', 2130),\n",
       " ('Both', 2129),\n",
       " ('', 2129),\n",
       " ('established', 2128),\n",
       " ('subset', 2125),\n",
       " ('clear', 2123),\n",
       " ('Expression', 2122),\n",
       " ('mouse', 2120),\n",
       " ('Here', 2119),\n",
       " ('forms', 2117),\n",
       " ('activated', 2114),\n",
       " ('activating', 2112),\n",
       " ('targets', 2110),\n",
       " ('relatively', 2104),\n",
       " ('leading', 2103),\n",
       " ('limited', 2102),\n",
       " ('P', 2100),\n",
       " ('transfection', 2096),\n",
       " ('interactions', 2095),\n",
       " ('sensitive', 2094),\n",
       " ('observations', 2092),\n",
       " ('given', 2092),\n",
       " ('Finally', 2086),\n",
       " ('caused', 2086),\n",
       " ('yet', 2085),\n",
       " ('mutagenesis', 2085),\n",
       " ('comparison', 2083),\n",
       " ('oncogenic', 2082),\n",
       " ('values', 2080),\n",
       " ('36', 2078),\n",
       " ('No', 2072),\n",
       " ('1A', 2071),\n",
       " ('decreased', 2070),\n",
       " ('seen', 2067),\n",
       " ('models', 2063),\n",
       " ('information', 2063),\n",
       " ('association', 2061),\n",
       " ('evaluated', 2059),\n",
       " ('method', 2058),\n",
       " ('finding', 2055),\n",
       " ('still', 2054),\n",
       " ('related', 2053),\n",
       " ('60', 2050),\n",
       " ('alone', 2049),\n",
       " ('about', 2044),\n",
       " ('investigated', 2042),\n",
       " ('six', 2039),\n",
       " ('importance', 2038),\n",
       " ('exon', 2035),\n",
       " ('constructs', 2033),\n",
       " ('allele', 2027),\n",
       " ('Data', 2025),\n",
       " ('demonstrate', 2023),\n",
       " ('defined', 2023),\n",
       " ('targeted', 2021),\n",
       " (\"'s\", 2019),\n",
       " ('On', 2018),\n",
       " ('panel', 2017),\n",
       " ('rare', 2016),\n",
       " ('Western', 2013),\n",
       " ('survival', 2011),\n",
       " ('altered', 2009),\n",
       " ('few', 2007),\n",
       " ('I', 2006),\n",
       " ('confirm', 2002),\n",
       " ('1B', 2001),\n",
       " ('incubated', 1999),\n",
       " ('essential', 1995),\n",
       " ('via', 1994),\n",
       " ('calculated', 1991),\n",
       " ('2A', 1986),\n",
       " ('Using', 1984),\n",
       " ('isolated', 1982),\n",
       " ('identification', 1981),\n",
       " ('approach', 1979),\n",
       " ('Several', 1977),\n",
       " ('rate', 1975),\n",
       " ('studied', 1971),\n",
       " ('methods', 1969),\n",
       " ('published', 1968),\n",
       " ('stable', 1966),\n",
       " ('variant', 1966),\n",
       " ('sequenced', 1965),\n",
       " ('exons', 1964),\n",
       " ('medium', 1961),\n",
       " ('tyrosine', 1960),\n",
       " ('carried', 1958),\n",
       " ('culture', 1958),\n",
       " ('contains', 1957),\n",
       " ('sufficient', 1957),\n",
       " ('RNA', 1954),\n",
       " ('breast', 1952),\n",
       " ('times', 1952),\n",
       " ('series', 1951),\n",
       " ('often', 1951),\n",
       " ('relevant', 1949),\n",
       " ('regulatory', 1947),\n",
       " ('coding', 1946),\n",
       " ('regulation', 1941),\n",
       " ('remaining', 1939),\n",
       " ('There', 1937),\n",
       " ('therapeutic', 1937),\n",
       " ('At', 1935),\n",
       " ('mean', 1931),\n",
       " ('Materials', 1929),\n",
       " ('secondary', 1928),\n",
       " ('members', 1927),\n",
       " ('population', 1919),\n",
       " ('unique', 1918),\n",
       " ('nucleotide', 1917),\n",
       " ('reaction', 1915),\n",
       " ('features', 1913),\n",
       " ('led', 1913),\n",
       " ('phosphorylated', 1912),\n",
       " ('When', 1912),\n",
       " ('3A', 1910),\n",
       " ('Protein', 1909),\n",
       " ('amplification', 1908),\n",
       " ('size', 1907),\n",
       " ('C-terminal', 1907),\n",
       " ('mice', 1904),\n",
       " ('2B', 1900),\n",
       " ('rather', 1898),\n",
       " ('progression', 1894),\n",
       " ('concentrations', 1894),\n",
       " ('serum', 1892),\n",
       " ('buffer', 1891),\n",
       " ('S1', 1890),\n",
       " ('hypothesis', 1884),\n",
       " ('upon', 1883),\n",
       " ('38', 1881),\n",
       " ('activate', 1880),\n",
       " ('manufacturer', 1879),\n",
       " ('bind', 1878),\n",
       " ('end', 1877),\n",
       " ('induce', 1874),\n",
       " ('impact', 1871),\n",
       " ('enhanced', 1870),\n",
       " ('leads', 1869),\n",
       " ('added', 1868),\n",
       " ('II', 1866),\n",
       " ('Mutation', 1865),\n",
       " ('reports', 1864),\n",
       " ('reduction', 1863),\n",
       " ('days', 1860),\n",
       " ('prepared', 1859),\n",
       " ('larger', 1857),\n",
       " ('value', 1857),\n",
       " ('so', 1854),\n",
       " ('39', 1853),\n",
       " ('who', 1850),\n",
       " ('Each', 1849),\n",
       " ('red', 1848),\n",
       " ('occurs', 1848),\n",
       " ('general', 1847),\n",
       " ('ratio', 1847),\n",
       " ('Go', 1846),\n",
       " ('made', 1844),\n",
       " ('investigate', 1844),\n",
       " ('others', 1842),\n",
       " ('primers', 1842),\n",
       " ('endogenous', 1841),\n",
       " ('suppressor', 1840),\n",
       " ('involving', 1840),\n",
       " ('displayed', 1839),\n",
       " ('next', 1837),\n",
       " ('greater', 1835),\n",
       " ('90', 1832),\n",
       " ('42', 1831),\n",
       " ('45', 1828),\n",
       " ('purified', 1825),\n",
       " ('failed', 1823),\n",
       " ('context', 1822),\n",
       " ('combination', 1822),\n",
       " ('provides', 1820),\n",
       " ('0', 1817),\n",
       " ('80', 1817),\n",
       " ('Among', 1815),\n",
       " ('work', 1814),\n",
       " ('responsible', 1813),\n",
       " ('products', 1812),\n",
       " ('chain', 1812),\n",
       " ('WT', 1811),\n",
       " ('pattern', 1810),\n",
       " ('Supplementary', 1804),\n",
       " ('selection', 1804),\n",
       " ('amplified', 1798),\n",
       " ('G', 1798),\n",
       " ('comparable', 1796),\n",
       " ('41', 1796),\n",
       " ('43', 1795),\n",
       " ('mutational', 1794),\n",
       " ('combined', 1792),\n",
       " ('activities', 1792),\n",
       " ('detection', 1791),\n",
       " ('dependent', 1790),\n",
       " ('specifically', 1790),\n",
       " ('potentially', 1790),\n",
       " ('kinases', 1789),\n",
       " ('correlation', 1787),\n",
       " ('Indeed', 1786),\n",
       " ('presented', 1778),\n",
       " ('up', 1777),\n",
       " ('cultured', 1776),\n",
       " ('below', 1775),\n",
       " ('manner', 1774),\n",
       " ('0.5', 1773),\n",
       " ('kit', 1773),\n",
       " ('confer', 1773),\n",
       " ('events', 1772),\n",
       " ('inhibit', 1771),\n",
       " ('necessary', 1769),\n",
       " ('except', 1763),\n",
       " ('particularly', 1761),\n",
       " ('Similarly', 1761),\n",
       " ('hours', 1760),\n",
       " ('properties', 1759),\n",
       " ('average', 1759),\n",
       " ('contained', 1758),\n",
       " ('initial', 1758),\n",
       " ('seven', 1757),\n",
       " ('concentration', 1756),\n",
       " ('develop', 1754),\n",
       " ('detect', 1753),\n",
       " ('generate', 1752),\n",
       " ('Most', 1752),\n",
       " ('positions', 1750),\n",
       " ('alternative', 1750),\n",
       " ('Further', 1750),\n",
       " ('biological', 1749),\n",
       " ('carcinoma', 1749),\n",
       " ('b', 1748),\n",
       " ('years', 1748),\n",
       " ('noted', 1747),\n",
       " ('70', 1746),\n",
       " ('subjected', 1745),\n",
       " ('currently', 1745),\n",
       " ('T', 1744),\n",
       " ('play', 1743),\n",
       " ('proposed', 1742),\n",
       " ('far', 1742),\n",
       " ('3B', 1742),\n",
       " ('frequent', 1742),\n",
       " ('therapy', 1740),\n",
       " ('blue', 1739),\n",
       " ('prior', 1739),\n",
       " ('collected', 1737),\n",
       " ('lysates', 1731),\n",
       " ('staining', 1730),\n",
       " ('plasmid', 1730),\n",
       " ('chromosome', 1729),\n",
       " ('inhibited', 1729),\n",
       " ('green', 1726),\n",
       " ('occurred', 1726),\n",
       " ('Since', 1725),\n",
       " ('targeting', 1725),\n",
       " ('express', 1723),\n",
       " ('lung', 1721),\n",
       " ('mRNA', 1719),\n",
       " ('appears', 1715),\n",
       " ('require', 1714),\n",
       " ('Consistent', 1713),\n",
       " ('produced', 1712),\n",
       " ('constitutive', 1712),\n",
       " ('Three', 1710),\n",
       " ('blot', 1709),\n",
       " ('much', 1704),\n",
       " ('eight', 1703),\n",
       " ('plates', 1703),\n",
       " ('mediated', 1702),\n",
       " ('how', 1698),\n",
       " ('carrying', 1697),\n",
       " ('contain', 1696),\n",
       " ('decrease', 1695),\n",
       " ('deletions', 1695),\n",
       " ('h', 1695),\n",
       " ('early', 1694),\n",
       " ('', 1693),\n",
       " ('S2', 1691),\n",
       " ('http', 1691),\n",
       " ('subsequent', 1690),\n",
       " ('University', 1689),\n",
       " ('variety', 1688),\n",
       " ('resistance', 1687),\n",
       " ('increasing', 1687),\n",
       " ('stability', 1685),\n",
       " ('note', 1683),\n",
       " ('cycle', 1680),\n",
       " ('database', 1679),\n",
       " ('current', 1677),\n",
       " ('understanding', 1676),\n",
       " ('certain', 1673),\n",
       " ('alleles', 1673),\n",
       " ('E', 1669),\n",
       " ('extracted', 1667),\n",
       " ('conformation', 1667),\n",
       " ('codon', 1664),\n",
       " ('0.1', 1661),\n",
       " ('harboring', 1655),\n",
       " ('little', 1655),\n",
       " ('wild', 1654),\n",
       " ('binds', 1653),\n",
       " ('marked', 1650),\n",
       " ('construct', 1650),\n",
       " ('N-terminal', 1648),\n",
       " ('grown', 1648),\n",
       " ('commonly', 1648),\n",
       " ('software', 1647),\n",
       " ('affecting', 1645),\n",
       " ('need', 1640),\n",
       " ('now', 1639),\n",
       " ('effective', 1638),\n",
       " ('Overall', 1637),\n",
       " ('represents', 1636),\n",
       " ('tissues', 1636),\n",
       " ('distribution', 1636),\n",
       " ('experimental', 1636),\n",
       " ('CA', 1634),\n",
       " ('consequences', 1632),\n",
       " ('4A', 1631),\n",
       " ('full-length', 1631),\n",
       " ('44', 1631),\n",
       " ('despite', 1631),\n",
       " ('exhibited', 1630),\n",
       " ('gel', 1630),\n",
       " ('crystal', 1629),\n",
       " ('left', 1627),\n",
       " ('mammalian', 1627),\n",
       " ('overexpression', 1626),\n",
       " ('right', 1624),\n",
       " ('alter', 1623),\n",
       " ('substrate', 1622),\n",
       " ('diagnosis', 1622),\n",
       " ('close', 1621),\n",
       " ('RESULTS', 1621),\n",
       " ('versus', 1621),\n",
       " ('product', 1616),\n",
       " ('central', 1616),\n",
       " ('almost', 1615),\n",
       " ('specificity', 1614),\n",
       " ('numbers', 1612),\n",
       " ('4B', 1612),\n",
       " ('individuals', 1611),\n",
       " ('1C', 1611),\n",
       " ('implicated', 1611),\n",
       " ('min', 1610),\n",
       " ('By', 1608),\n",
       " ('DISCUSSION', 1607),\n",
       " ('Molecular', 1607),\n",
       " ('correlated', 1606),\n",
       " ('introduced', 1602),\n",
       " ('...', 1601),\n",
       " ('screening', 1600),\n",
       " ('across', 1599),\n",
       " ('Invitrogen', 1595),\n",
       " ('supported', 1592),\n",
       " ('us', 1591),\n",
       " ('none', 1590),\n",
       " ('nuclear', 1586),\n",
       " ('experiment', 1583),\n",
       " ('clearly', 1581),\n",
       " ('date', 1581),\n",
       " ('drug', 1580),\n",
       " ('phase', 1580),\n",
       " ('amount', 1579),\n",
       " ('transcriptional', 1577),\n",
       " ('unclear', 1576),\n",
       " ('process', 1575),\n",
       " ('Other', 1574),\n",
       " ('help', 1574),\n",
       " ('classified', 1573),\n",
       " ('predict', 1572),\n",
       " ('reveal', 1572),\n",
       " ('molecules', 1571),\n",
       " ('motif', 1569),\n",
       " ('fully', 1567),\n",
       " ('final', 1566),\n",
       " ('Recently', 1563),\n",
       " ('membrane', 1563),\n",
       " ('plays', 1562),\n",
       " ('signals', 1562),\n",
       " ('apoptosis', 1562),\n",
       " ('examine', 1562),\n",
       " ('fraction', 1560),\n",
       " ('biochemical', 1556),\n",
       " ('entire', 1555),\n",
       " ('bound', 1554),\n",
       " ('state', 1553),\n",
       " ('interact', 1550),\n",
       " ('identical', 1549),\n",
       " ('If', 1548),\n",
       " ('probably', 1548),\n",
       " ('completely', 1547),\n",
       " ('formed', 1546),\n",
       " ('testing', 1545),\n",
       " ('percentage', 1545),\n",
       " ('whose', 1544),\n",
       " ('appropriate', 1543),\n",
       " ('supplemented', 1541),\n",
       " ('age', 1541),\n",
       " ('germline', 1541),\n",
       " ('partial', 1541),\n",
       " ('functionally', 1540),\n",
       " ('clinically', 1539),\n",
       " ('clones', 1539),\n",
       " ('transformation', 1538),\n",
       " ('order', 1538),\n",
       " ('structures', 1537),\n",
       " ('version', 1536),\n",
       " ('disrupt', 1536),\n",
       " ('promoter', 1536),\n",
       " ('full', 1535),\n",
       " ('evaluate', 1534),\n",
       " ('protocol', 1532),\n",
       " ('tumorigenesis', 1531),\n",
       " ('loop', 1531),\n",
       " ('allow', 1529),\n",
       " ('characterization', 1529),\n",
       " ('like', 1528),\n",
       " ('especially', 1528),\n",
       " ('occurring', 1528),\n",
       " ('half', 1527),\n",
       " ('surface', 1527),\n",
       " ('normalized', 1525),\n",
       " ('nature', 1524),\n",
       " ('proportion', 1524),\n",
       " ('2000', 1523),\n",
       " ('encodes', 1523),\n",
       " ('inactive', 1523),\n",
       " ('inactivation', 1522),\n",
       " ('interest', 1519),\n",
       " ('blood', 1518),\n",
       " ('approved', 1517),\n",
       " ('0.05', 1517),\n",
       " ('appear', 1515),\n",
       " ('side', 1515),\n",
       " ('rates', 1514),\n",
       " ('includes', 1513),\n",
       " ('catalytic', 1513),\n",
       " ('spectrum', 1512),\n",
       " ('While', 1511),\n",
       " ('thought', 1511),\n",
       " ('risk', 1508),\n",
       " ('having', 1506),\n",
       " ('Similar', 1506),\n",
       " ('lysed', 1501),\n",
       " ('explain', 1498),\n",
       " ('resistant', 1497),\n",
       " ('degree', 1496),\n",
       " ('along', 1495),\n",
       " ('Another', 1494),\n",
       " ('components', 1494),\n",
       " ('representative', 1490),\n",
       " ('thereby', 1488),\n",
       " ('affinity', 1488),\n",
       " ('account', 1488),\n",
       " ('c', 1482),\n",
       " ('promote', 1480),\n",
       " ('2C', 1480),\n",
       " ('later', 1479),\n",
       " ('Research', 1479),\n",
       " ('reverse', 1478),\n",
       " ('top', 1478),\n",
       " ('fusion', 1476),\n",
       " ('location', 1475),\n",
       " ('linked', 1474),\n",
       " ('underlying', 1472),\n",
       " ('detectable', 1468),\n",
       " ('class', 1465),\n",
       " ('200', 1463),\n",
       " ('highest', 1463),\n",
       " ('exhibit', 1461),\n",
       " ('increases', 1458),\n",
       " ('Importantly', 1458),\n",
       " ('taken', 1454),\n",
       " ('unable', 1453),\n",
       " ('extent', 1453),\n",
       " ('Signaling', 1453),\n",
       " ('Given', 1451),\n",
       " ('elevated', 1451),\n",
       " ('alteration', 1451),\n",
       " ('already', 1449),\n",
       " ('modified', 1449),\n",
       " ('Based', 1449),\n",
       " ('hand', 1447),\n",
       " ('characteristics', 1447),\n",
       " ('roles', 1445),\n",
       " ('whole', 1444),\n",
       " ('stably', 1444),\n",
       " ('third', 1442),\n",
       " ('cloned', 1441),\n",
       " ('First', 1439),\n",
       " ('conclusion', 1438),\n",
       " ('applied', 1438),\n",
       " ('mM', 1437),\n",
       " ('generally', 1437),\n",
       " ('Human', 1437),\n",
       " ('washed', 1435),\n",
       " ('polymerase', 1435),\n",
       " ...]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sortedwords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3862"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(abridgedsorted)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "collapsed": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#abridgedsorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>split</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>genefeatures</th>\n",
       "      <th>wordfeatures</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>True</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>False</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>True</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>N454D</td>\n",
       "      <td>False</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>L399V</td>\n",
       "      <td>True</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID  Class    Gene                                               Text  \\\n",
       "0   0      1  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   1      2     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "2   2      2     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "3   3      3     CBL  Recent evidence has demonstrated that acquired...   \n",
       "4   4      4     CBL  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "              Variation  split  \\\n",
       "0  Truncating Mutations   True   \n",
       "1                 W802*  False   \n",
       "2                 Q249E   True   \n",
       "3                 N454D  False   \n",
       "4                 L399V   True   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        genefeatures  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        wordfeatures  \n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...  \n",
       "4  [1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.994950052555\n",
      "Accuracy on training set:\n",
      "0.468286259757\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"wordfeatures\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"wordsgenes\"]=merged[\"wordfeatures\"]+merged[\"genefeatures\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.990991804236\n",
      "Accuracy on training set:\n",
      "0.464804794285\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"wordsgenes\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0598844931121063"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lr = RandomForestClassifier(n_estimators=10000, max_depth=15)\n",
    "lr.fit(X_train, y_train)\n",
    "y_pred = lr.predict_proba(X_test)\n",
    " \n",
    "ls = []\n",
    "for i in range(len(y_pred)):\n",
    "    c = test.iloc[i][\"Class\"]\n",
    "    prob = y_pred[i][c-1]\n",
    "    ls.append(-np.log(prob+1e-6))\n",
    "np.mean(ls)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.38207888497823161"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ytrain_pred = lr.predict_proba(X_train)\n",
    "ls = []\n",
    "for i in range(len(ytrain_pred)):\n",
    "    c = training.iloc[i][\"Class\"]\n",
    "    prob = ytrain_pred[i][c-1]\n",
    "    ls.append(-np.log(prob+1e-6))\n",
    "np.mean(ls)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using just abstracts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "abstracts = merged[[\"ID\", \"Gene\", \"Variation\", \"Class\", \"split\", \"tokenized\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "abstracts[\"tokenizedabst\"] = abstracts[\"tokenized\"].map(lambda x: x[:250])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "abstracts[\"wordbag\"]=abstracts[\"tokenizedabst\"].map(build_bag_of_words_features_filtered)\n",
    "abstracts[\"wordfeatures\"]=abstracts[\"wordbag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.21293397387\n",
      "Accuracy on training set:\n",
      "1.02414901996\n"
     ]
    }
   ],
   "source": [
    "logistictester(abstracts, \"wordfeatures\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "abstracts[\"geneofinterest\"]=abstracts[\"Gene\"].map(lambda b: [int(gene == b) for gene in uniqueGenes])\n",
    "abstracts[\"wordsgeneOI\"]=abstracts[\"wordfeatures\"]+merged[\"geneofinterest\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "logistictester(abstracts, \"wordsgeneOI\", \"Class\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Using allele type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>split</th>\n",
       "      <th>Text</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>bag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>wordfeatures</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene             Variation  Class  split  \\\n",
       "0   0  FAM58A  Truncating Mutations      1   True   \n",
       "1   1     CBL                 W802*      2  False   \n",
       "2   2     CBL                 Q249E      2   True   \n",
       "3   3     CBL                 N454D      3   True   \n",
       "4   4     CBL                 L399V      4   True   \n",
       "\n",
       "                                                Text  \\\n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   Abstract Background  Non-small cell lung canc...   \n",
       "2   Abstract Background  Non-small cell lung canc...   \n",
       "3  Recent evidence has demonstrated that acquired...   \n",
       "4  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                                 bag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                                        wordfeatures  \n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...  \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...  \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "4  [1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3096</th>\n",
       "      <td>1_2009trunc</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3097</th>\n",
       "      <td>2010_2471trunc</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2123</th>\n",
       "      <td>256_286trunc</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351</th>\n",
       "      <td>3' Deletion</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>385_418del</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1707</th>\n",
       "      <td>422_605trunc</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>533_534del</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>534_536del</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3035</th>\n",
       "      <td>550_592del</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>926</th>\n",
       "      <td>560_561insER</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>596_619splice</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1148</th>\n",
       "      <td>963_D1010splice</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>981_1028splice</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>A113_splice</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3119</th>\n",
       "      <td>A11_G12insGA</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3022</th>\n",
       "      <td>A502_Y503dup</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3002</th>\n",
       "      <td>A504_Y505ins</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>A750_E758del</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>A750_E758delinsP</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>A763_Y764insFQEA</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>A767_V769del</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>A767_V769dup</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>A859_L883delinsV</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1222</th>\n",
       "      <td>ACPP-PIK3CB Fusion</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2720</th>\n",
       "      <td>AGK-BRAF Fusion</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2752</th>\n",
       "      <td>AKAP9-BRAF Fusion</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1786</th>\n",
       "      <td>ARv567es</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2335</th>\n",
       "      <td>ATF7IP-JAK2 Fusion</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>941</th>\n",
       "      <td>ATF7IP-PDGFRB Fusion</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3166</th>\n",
       "      <td>ATG7-RAF1 Fusion</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2934</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2362</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2904</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1911</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1826</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2403</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2414</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1840</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1851</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1908</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2142</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2421</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1505</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1910</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1595</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2146</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2285</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2281</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2111</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2114</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1094</th>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>600 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Variation  Class\n",
       "3096           1_2009trunc      1\n",
       "3097        2010_2471trunc      2\n",
       "2123          256_286trunc      7\n",
       "351            3' Deletion      4\n",
       "986             385_418del      4\n",
       "1707          422_605trunc      7\n",
       "368             533_534del      2\n",
       "361             534_536del      2\n",
       "3035            550_592del      7\n",
       "926           560_561insER      7\n",
       "868          596_619splice      1\n",
       "1148       963_D1010splice      7\n",
       "1138        981_1028splice      7\n",
       "285            A113_splice      7\n",
       "3119          A11_G12insGA      7\n",
       "3022          A502_Y503dup      7\n",
       "3002          A504_Y505ins      2\n",
       "174           A750_E758del      7\n",
       "184       A750_E758delinsP      7\n",
       "226       A763_Y764insFQEA      7\n",
       "248           A767_V769del      2\n",
       "236           A767_V769dup      7\n",
       "171       A859_L883delinsV      2\n",
       "1222    ACPP-PIK3CB Fusion      2\n",
       "2720       AGK-BRAF Fusion      7\n",
       "2752     AKAP9-BRAF Fusion      7\n",
       "1786              ARv567es      7\n",
       "2335    ATF7IP-JAK2 Fusion      7\n",
       "941   ATF7IP-PDGFRB Fusion      2\n",
       "3166      ATG7-RAF1 Fusion      2\n",
       "...                    ...    ...\n",
       "641   Truncating Mutations      1\n",
       "2934  Truncating Mutations      1\n",
       "2362  Truncating Mutations      1\n",
       "2904  Truncating Mutations      1\n",
       "650   Truncating Mutations      1\n",
       "694   Truncating Mutations      1\n",
       "1911  Truncating Mutations      1\n",
       "1826  Truncating Mutations      1\n",
       "835   Truncating Mutations      1\n",
       "830   Truncating Mutations      1\n",
       "1015  Truncating Mutations      1\n",
       "2403  Truncating Mutations      1\n",
       "826   Truncating Mutations      1\n",
       "2414  Truncating Mutations      1\n",
       "1840  Truncating Mutations      1\n",
       "1851  Truncating Mutations      1\n",
       "1080  Truncating Mutations      1\n",
       "1908  Truncating Mutations      1\n",
       "2142  Truncating Mutations      1\n",
       "2421  Truncating Mutations      1\n",
       "1505  Truncating Mutations      1\n",
       "1910  Truncating Mutations      1\n",
       "1595  Truncating Mutations      1\n",
       "2146  Truncating Mutations      1\n",
       "1902  Truncating Mutations      1\n",
       "2285  Truncating Mutations      1\n",
       "2281  Truncating Mutations      1\n",
       "2111  Truncating Mutations      1\n",
       "2114  Truncating Mutations      1\n",
       "1094  Truncating Mutations      6\n",
       "\n",
       "[600 rows x 2 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].map(len)>6][[\"Variation\", \"Class\"]].sort_values(\"Variation\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Class</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variation\n",
       "Class           \n",
       "1              2\n",
       "2             78\n",
       "4              3\n",
       "7             93\n",
       "8              5\n",
       "9              1"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].str.contains(\"Fusion\")][[\"Variation\", \"Class\"]].groupby(\"Class\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Class</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variation\n",
       "Class           \n",
       "1             90\n",
       "2              2\n",
       "4              1\n",
       "6              2"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].str.contains(\"Truncating\")][[\"Variation\", \"Class\"]].groupby(\"Class\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows', 1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variation</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Truncating Mutations</th>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Deletion</th>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amplification</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fusions</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Overexpression</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Promoter Hypermethylation</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMPRSS2-ETV1 Fusion</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ETV6-NTRK3 Fusion</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-ETV1 Fusion</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P573_D579del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAX8-PPAR? Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAX5-JAK2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAPSS1-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P577_W582delinsPYD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P577_D579del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P551_V555del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P551_W557delinsL</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDE4DIP-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P551_E554del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P490_Q494del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P449_L455del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P291Qfs*51</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P2415del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NSD1-NUP98 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPM-ALK Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NIN-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCM1-JAK2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTPRZ1-MET Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRKG2-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N771_H773dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S1297del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RUNX1-RUNX1T1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RUNX1-EVI1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ROS1-CD74 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RET-CCDC6 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RANBP2-ALK Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RANBP1-ALK Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R506_K507insVLR</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R177Pfs*126</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q58_Q59insL</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q58_E62del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q579_L581del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q56_V60del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q556_K558del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q347_A348del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q2405Rfs*17</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Q1756fs</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Promoter Mutations</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PVT1-MYC Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NFASC-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_2009trunc</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S453fs*</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L485_P490delinsF</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L747_P753del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L747_E749del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L747_A750delinsP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L747_A750del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L611_E612insCSSDNEYFYVDFREYEYDLKWEFPRENL</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L601_K602insREYEYDL</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L576del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L485_Q494del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L485_P490delinsY</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L485_P490del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L747_T751del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L283_D294del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIF5B-RET Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIF5B-PDGFRA Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIF5B-ALK Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIAA1549-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KIAA1509-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDR-PDGFRA Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDELR2-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KANK1-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L747_P753delinsS</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>L747_T751delinsP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N564_Y578del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL-TET1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N542_E543del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N486_P490del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N480del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N198_F199delinsI</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N1333Gfs*</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N1068fs*4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYC-nick</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSN-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MPRIP-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MKRN1-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LIMA1-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR143-NOTCH1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAGI3-AKT3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M774_A775insAYVM</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M552_W557del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M552_K558del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M1_E165DEL</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M199del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LRIG3-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LMNA-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S267_D273dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S566_E571delinsR</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S459del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V559_V560del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W557_V559delinsC</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W557_K558del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W290_I291delinsC</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W237_Y242del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V769_D770insGVV</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V600D_K601insFGLAT</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V569_L576del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V561_I562insER</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V559del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V555_V559del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W603_E604insDREYEYDLKW</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V555_L576del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V544_L545insAVLVLLVIVIISLI</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V487_P492delinsA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V422del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V1688del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V1605del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V1578del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V128del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>V1075Yfs*2</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W557_V560del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WDR48-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S501_A502dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y553_Q556del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF198-FGFR1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZC3H7B-BCOR Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YAP1-MAMLD1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YAP1-FAM118B Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y772_A775dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y599_D600insSTDNEYFYVDFREYEY</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y599_D600insPAPQIMSTSTLISENMNIA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y599_D600insGLYVDFREYEY</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y599_D600insEYEYEYEY</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y553_K558del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Wildtype</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y418_D419del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y375_K455del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X963_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X582_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X475_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X434_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X1009_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X1008_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X1007_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncating Mutations in the PEST Domain</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Truncating Mutations Upstream of Transactivation Domain</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRKAIII Splice Variant</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAG9-JAK2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T319del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T244_I245insCPT</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1481fs</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T1151dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STRN-PDGFRA Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STRN-ALK Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SSBP2-JAK2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRGAP3-RAF1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPTBN1-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SND1-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRIM24-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC34A2-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC16A1-NOTCH1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SDC4-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S840_N841insGS</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S768_V769insVAS</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S768_D770dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S752_I759del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S70fsX93</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K558delinsNP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T34_A289del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T417_D419delinsI</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T417_D419delinsRG</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T488_P492del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRB-NKX2-1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRA-NKX2-1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TPR-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TPM3-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53BP1-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMPRSS2-ETV5 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMPRSS2-ETV4 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMPRSS2-ERG Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TGFBR1*6A</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFG-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFG-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TEL-RUNX1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TEL-JAK2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T599dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T599_V600insV</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T599_V600insEAT</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T576del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T574insTQLPYD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>T574_R588delinsL</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K745_A750del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K550_W557del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K558_V559del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D419del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D842_M844del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D842_H845del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D835del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D770_P772dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D770_N771insVDSVDNP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D770_N771insNPG</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D770_N771insD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D600_L601insFREYEYD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D579del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D289del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD74-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D289_D292del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Copy Number Loss</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUX1-FGFR1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUL1-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CPEB1-NTRK3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIC-DUX4 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHTOP-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CEP85L-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CEP85L-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNA binding domain deletions</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNA binding domain insertions</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNA binding domain missense mutations</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNMT3B7</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E746_T751delinsA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E746_S752delinsI</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E746_S752delinsA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E746_A750delinsQ</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E746_A750del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E709_T710delinsD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E632_L633del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E612_F613insGYVDFREYEYDLKWEFRPRENLEF</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E598_Y599insGLVQVTGSSDNEYFYVDFREYE</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E598_Y599insDVDFREYE</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E554_V559del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E554_K558del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E554_I571del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E439del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E311_K312del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E161del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E1552del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E102_I103del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Delta-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CEP110-FGFR1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD74-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E746_T751insIP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981_1028splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A767_V769dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A767_V769del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A763_Y764insFQEA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A750_E758delinsP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A750_E758del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A504_Y505ins</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A502_Y503dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A11_G12insGA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A113_splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>963_D1010splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCND1-IGH Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596_619splice</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560_561insER</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>550_592del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534_536del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>533_534del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422_605trunc</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385_418del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3' Deletion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256_286trunc</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A859_L883delinsV</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACPP-PIK3CB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AGK-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKAP9-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC6-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CAD-ALK Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C456_R481del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C456_N468del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C450_K451insMIEWMI</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BTBD1-NTRK3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRD4-NUT Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BIN2-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCR-PDGFRA Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCR-JAK2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCR-FGFR1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCR-ABL1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCOR-RARA Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCOR-CCNB3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BCAN-NTRK1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATG7-RAF1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF7IP-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF7IP-JAK2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARv567es</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E746_T751delinsVA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EBF1-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K558_E562del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G106_R108del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H773_V774insH</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GPIAP1-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GOLGA4-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GIT2-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G778_P780dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G776delinsVC</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G776delinsLC</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G776_V777insYVMA</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G469del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FUS-ERG Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2-AHCYL1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLT3 internal tandem duplications</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIG-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR3 - BAIAP2L1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2?PPHLN1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2-TACC3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2-MGEA5 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2-KIAA1967 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2-FAM76A Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2-CCDC6 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H773dup</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H773insLGNP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H845_N848delinsP</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIP1-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2010_2471trunc</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K550_V555delinsI</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K550_K558del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K459_S460delinsN</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K442Nfs*</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGL-MYC Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGK-MYC Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGH-NKX2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGH-MYC Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGH-FGFR3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGH-BCL2 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I843del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I843_D846del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I836del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I744_K745delinsKIPVAI</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I563_L576del</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I559_D560insDKRMNS</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hypermethylation</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HMGA2-RAD51B Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR2-BICC1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR1OP1-FGFR1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR-KDD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ESRP1-RAF1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-FEV Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-ETV4 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-ERG Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-DDIT3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-CREB1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-ATF1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ETV6-PDGFRB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ETV6-PDGFRA Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ETV6-FLT3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ESR1-YAP1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGFR1-TACC1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ESR1-CCDC170 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERLIN2?FGFR1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EP300-MOZ Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EP300-MLL Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFRvIV</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFRvIII</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFRvII</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR-RAD51 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR-PURB Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-FLI1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-NR4A3 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-WT1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EWSR1-YY1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAM131B-BRAF Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F594_R595insSDNEYFYVDF</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F537_K539delinsL</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F1088Sfs*2</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F1088Lfs*5</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 9 mutations</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 20 insertions/deletions</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 20 insertions</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 20 insertion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 2 mutations</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 19 insertion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 19 deletion/insertion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 19 deletion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 13 deletion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 11 mutations</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Exon 1 mutations</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Epigenetic Silencing</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EZR-ROS1 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EZR-ERBB4 Fusion</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>p61BRAF</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    Class\n",
       "Variation                                                \n",
       "Truncating Mutations                                   93\n",
       "Deletion                                               74\n",
       "Amplification                                          71\n",
       "Fusions                                                34\n",
       "Overexpression                                          6\n",
       "Promoter Hypermethylation                               2\n",
       "TMPRSS2-ETV1 Fusion                                     2\n",
       "ETV6-NTRK3 Fusion                                       2\n",
       "EWSR1-ETV1 Fusion                                       2\n",
       "P573_D579del                                            1\n",
       "PAX8-PPAR? Fusion                                       1\n",
       "PAX5-JAK2 Fusion                                        1\n",
       "PAPSS1-BRAF Fusion                                      1\n",
       "P577_W582delinsPYD                                      1\n",
       "P577_D579del                                            1\n",
       "P551_V555del                                            1\n",
       "P551_W557delinsL                                        1\n",
       "PDE4DIP-PDGFRB Fusion                                   1\n",
       "P551_E554del                                            1\n",
       "P490_Q494del                                            1\n",
       "P449_L455del                                            1\n",
       "P291Qfs*51                                              1\n",
       "P2415del                                                1\n",
       "NSD1-NUP98 Fusion                                       1\n",
       "NPM-ALK Fusion                                          1\n",
       "NIN-PDGFRB Fusion                                       1\n",
       "PCM1-JAK2 Fusion                                        1\n",
       "PTPRZ1-MET Fusion                                       1\n",
       "PRKG2-PDGFRB Fusion                                     1\n",
       "N771_H773dup                                            1\n",
       "S1297del                                                1\n",
       "RUNX1-RUNX1T1 Fusion                                    1\n",
       "RUNX1-EVI1 Fusion                                       1\n",
       "ROS1-CD74 Fusion                                        1\n",
       "RET-CCDC6 Fusion                                        1\n",
       "RANBP2-ALK Fusion                                       1\n",
       "RANBP1-ALK Fusion                                       1\n",
       "R506_K507insVLR                                         1\n",
       "R177Pfs*126                                             1\n",
       "Q58_Q59insL                                             1\n",
       "Q58_E62del                                              1\n",
       "Q579_L581del                                            1\n",
       "Q56_V60del                                              1\n",
       "Q556_K558del                                            1\n",
       "Q347_A348del                                            1\n",
       "Q2405Rfs*17                                             1\n",
       "Q1756fs                                                 1\n",
       "Promoter Mutations                                      1\n",
       "PVT1-MYC Fusion                                         1\n",
       "NFASC-NTRK1 Fusion                                      1\n",
       "1_2009trunc                                             1\n",
       "S453fs*                                                 1\n",
       "L485_P490delinsF                                        1\n",
       "L747_P753del                                            1\n",
       "L747_E749del                                            1\n",
       "L747_A750delinsP                                        1\n",
       "L747_A750del                                            1\n",
       "L611_E612insCSSDNEYFYVDFREYEYDLKWEFPRENL                1\n",
       "L601_K602insREYEYDL                                     1\n",
       "L576del                                                 1\n",
       "L485_Q494del                                            1\n",
       "L485_P490delinsY                                        1\n",
       "L485_P490del                                            1\n",
       "L747_T751del                                            1\n",
       "L283_D294del                                            1\n",
       "KIF5B-RET Fusion                                        1\n",
       "KIF5B-PDGFRA Fusion                                     1\n",
       "KIF5B-ALK Fusion                                        1\n",
       "KIAA1549-BRAF Fusion                                    1\n",
       "KIAA1509-PDGFRB Fusion                                  1\n",
       "KDR-PDGFRA Fusion                                       1\n",
       "KDELR2-ROS1 Fusion                                      1\n",
       "KANK1-PDGFRB Fusion                                     1\n",
       "L747_P753delinsS                                        1\n",
       "L747_T751delinsP                                        1\n",
       "N564_Y578del                                            1\n",
       "MLL-TET1 Fusion                                         1\n",
       "N542_E543del                                            1\n",
       "N486_P490del                                            1\n",
       "N480del                                                 1\n",
       "N198_F199delinsI                                        1\n",
       "N1333Gfs*                                               1\n",
       "N1068fs*4                                               1\n",
       "MYC-nick                                                1\n",
       "MSN-ROS1 Fusion                                         1\n",
       "MPRIP-NTRK1 Fusion                                      1\n",
       "MKRN1-BRAF Fusion                                       1\n",
       "LIMA1-ROS1 Fusion                                       1\n",
       "MIR143-NOTCH1 Fusion                                    1\n",
       "MAGI3-AKT3 Fusion                                       1\n",
       "M774_A775insAYVM                                        1\n",
       "M552_W557del                                            1\n",
       "M552_K558del                                            1\n",
       "M1_E165DEL                                              1\n",
       "M199del                                                 1\n",
       "LRIG3-ROS1 Fusion                                       1\n",
       "LMNA-NTRK1 Fusion                                       1\n",
       "S267_D273dup                                            1\n",
       "S566_E571delinsR                                        1\n",
       "S459del                                                 1\n",
       "V559_V560del                                            1\n",
       "W557_V559delinsC                                        1\n",
       "W557_K558del                                            1\n",
       "W290_I291delinsC                                        1\n",
       "W237_Y242del                                            1\n",
       "V769_D770insGVV                                         1\n",
       "V600D_K601insFGLAT                                      1\n",
       "V569_L576del                                            1\n",
       "V561_I562insER                                          1\n",
       "V559del                                                 1\n",
       "V555_V559del                                            1\n",
       "W603_E604insDREYEYDLKW                                  1\n",
       "V555_L576del                                            1\n",
       "V544_L545insAVLVLLVIVIISLI                              1\n",
       "V487_P492delinsA                                        1\n",
       "V422del                                                 1\n",
       "V1688del                                                1\n",
       "V1605del                                                1\n",
       "V1578del                                                1\n",
       "V128del                                                 1\n",
       "V1075Yfs*2                                              1\n",
       "W557_V560del                                            1\n",
       "WDR48-PDGFRB Fusion                                     1\n",
       "S501_A502dup                                            1\n",
       "Y553_Q556del                                            1\n",
       "ZNF198-FGFR1 Fusion                                     1\n",
       "ZC3H7B-BCOR Fusion                                      1\n",
       "YAP1-MAMLD1 Fusion                                      1\n",
       "YAP1-FAM118B Fusion                                     1\n",
       "Y772_A775dup                                            1\n",
       "Y599_D600insSTDNEYFYVDFREYEY                            1\n",
       "Y599_D600insPAPQIMSTSTLISENMNIA                         1\n",
       "Y599_D600insGLYVDFREYEY                                 1\n",
       "Y599_D600insEYEYEYEY                                    1\n",
       "Y553_K558del                                            1\n",
       "Wildtype                                                1\n",
       "Y418_D419del                                            1\n",
       "Y375_K455del                                            1\n",
       "X963_splice                                             1\n",
       "X582_splice                                             1\n",
       "X475_splice                                             1\n",
       "X434_splice                                             1\n",
       "X1009_splice                                            1\n",
       "X1008_splice                                            1\n",
       "X1007_splice                                            1\n",
       "Truncating Mutations in the PEST Domain                 1\n",
       "Truncating Mutations Upstream of Transactivatio...      1\n",
       "TRKAIII Splice Variant                                  1\n",
       "SPAG9-JAK2 Fusion                                       1\n",
       "T319del                                                 1\n",
       "T244_I245insCPT                                         1\n",
       "T1481fs                                                 1\n",
       "T1151dup                                                1\n",
       "STRN-PDGFRA Fusion                                      1\n",
       "STRN-ALK Fusion                                         1\n",
       "SSBP2-JAK2 Fusion                                       1\n",
       "SRGAP3-RAF1 Fusion                                      1\n",
       "SPTBN1-PDGFRB Fusion                                    1\n",
       "SND1-BRAF Fusion                                        1\n",
       "TRIM24-BRAF Fusion                                      1\n",
       "SLC34A2-ROS1 Fusion                                     1\n",
       "SEC16A1-NOTCH1 Fusion                                   1\n",
       "SDC4-ROS1 Fusion                                        1\n",
       "S840_N841insGS                                          1\n",
       "S768_V769insVAS                                         1\n",
       "S768_D770dup                                            1\n",
       "S752_I759del                                            1\n",
       "S70fsX93                                                1\n",
       "K558delinsNP                                            1\n",
       "T34_A289del                                             1\n",
       "T417_D419delinsI                                        1\n",
       "T417_D419delinsRG                                       1\n",
       "T488_P492del                                            1\n",
       "TRB-NKX2-1 Fusion                                       1\n",
       "TRA-NKX2-1 Fusion                                       1\n",
       "TPR-NTRK1 Fusion                                        1\n",
       "TPM3-NTRK1 Fusion                                       1\n",
       "TP53BP1-PDGFRB Fusion                                   1\n",
       "TMPRSS2-ETV5 Fusion                                     1\n",
       "TMPRSS2-ETV4 Fusion                                     1\n",
       "TMPRSS2-ERG Fusion                                      1\n",
       "TGFBR1*6A                                               1\n",
       "TFG-ROS1 Fusion                                         1\n",
       "TFG-NTRK1 Fusion                                        1\n",
       "TEL-RUNX1 Fusion                                        1\n",
       "TEL-JAK2 Fusion                                         1\n",
       "T599dup                                                 1\n",
       "T599_V600insV                                           1\n",
       "T599_V600insEAT                                         1\n",
       "T576del                                                 1\n",
       "T574insTQLPYD                                           1\n",
       "T574_R588delinsL                                        1\n",
       "K745_A750del                                            1\n",
       "K550_W557del                                            1\n",
       "K558_V559del                                            1\n",
       "D419del                                                 1\n",
       "D842_M844del                                            1\n",
       "D842_H845del                                            1\n",
       "D835del                                                 1\n",
       "D770_P772dup                                            1\n",
       "D770_N771insVDSVDNP                                     1\n",
       "D770_N771insNPG                                         1\n",
       "D770_N771insD                                           1\n",
       "D600_L601insFREYEYD                                     1\n",
       "D579del                                                 1\n",
       "D289del                                                 1\n",
       "CD74-ROS1 Fusion                                        1\n",
       "D289_D292del                                            1\n",
       "Copy Number Loss                                        1\n",
       "CUX1-FGFR1 Fusion                                       1\n",
       "CUL1-BRAF Fusion                                        1\n",
       "CPEB1-NTRK3 Fusion                                      1\n",
       "CIC-DUX4 Fusion                                         1\n",
       "CHTOP-NTRK1 Fusion                                      1\n",
       "CEP85L-ROS1 Fusion                                      1\n",
       "CEP85L-PDGFRB Fusion                                    1\n",
       "DNA binding domain deletions                            1\n",
       "DNA binding domain insertions                           1\n",
       "DNA binding domain missense mutations                   1\n",
       "DNMT3B7                                                 1\n",
       "E746_T751delinsA                                        1\n",
       "E746_S752delinsI                                        1\n",
       "E746_S752delinsA                                        1\n",
       "E746_A750delinsQ                                        1\n",
       "E746_A750del                                            1\n",
       "E709_T710delinsD                                        1\n",
       "E632_L633del                                            1\n",
       "E612_F613insGYVDFREYEYDLKWEFRPRENLEF                    1\n",
       "E598_Y599insGLVQVTGSSDNEYFYVDFREYE                      1\n",
       "E598_Y599insDVDFREYE                                    1\n",
       "E554_V559del                                            1\n",
       "E554_K558del                                            1\n",
       "E554_I571del                                            1\n",
       "E439del                                                 1\n",
       "E311_K312del                                            1\n",
       "E161del                                                 1\n",
       "E1552del                                                1\n",
       "E102_I103del                                            1\n",
       "Delta-NTRK1 Fusion                                      1\n",
       "CEP110-FGFR1 Fusion                                     1\n",
       "CD74-NTRK1 Fusion                                       1\n",
       "E746_T751insIP                                          1\n",
       "981_1028splice                                          1\n",
       "A767_V769dup                                            1\n",
       "A767_V769del                                            1\n",
       "A763_Y764insFQEA                                        1\n",
       "A750_E758delinsP                                        1\n",
       "A750_E758del                                            1\n",
       "A504_Y505ins                                            1\n",
       "A502_Y503dup                                            1\n",
       "A11_G12insGA                                            1\n",
       "A113_splice                                             1\n",
       "963_D1010splice                                         1\n",
       "CCND1-IGH Fusion                                        1\n",
       "596_619splice                                           1\n",
       "560_561insER                                            1\n",
       "550_592del                                              1\n",
       "534_536del                                              1\n",
       "533_534del                                              1\n",
       "422_605trunc                                            1\n",
       "385_418del                                              1\n",
       "3' Deletion                                             1\n",
       "256_286trunc                                            1\n",
       "A859_L883delinsV                                        1\n",
       "ACPP-PIK3CB Fusion                                      1\n",
       "AGK-BRAF Fusion                                         1\n",
       "AKAP9-BRAF Fusion                                       1\n",
       "CCDC6-ROS1 Fusion                                       1\n",
       "CAD-ALK Fusion                                          1\n",
       "C456_R481del                                            1\n",
       "C456_N468del                                            1\n",
       "C450_K451insMIEWMI                                      1\n",
       "BTBD1-NTRK3 Fusion                                      1\n",
       "BRD4-NUT Fusion                                         1\n",
       "BIN2-PDGFRB Fusion                                      1\n",
       "BCR-PDGFRA Fusion                                       1\n",
       "BCR-JAK2 Fusion                                         1\n",
       "BCR-FGFR1 Fusion                                        1\n",
       "BCR-ABL1 Fusion                                         1\n",
       "BCOR-RARA Fusion                                        1\n",
       "BCOR-CCNB3 Fusion                                       1\n",
       "BCAN-NTRK1 Fusion                                       1\n",
       "ATG7-RAF1 Fusion                                        1\n",
       "ATF7IP-PDGFRB Fusion                                    1\n",
       "ATF7IP-JAK2 Fusion                                      1\n",
       "ARv567es                                                1\n",
       "E746_T751delinsVA                                       1\n",
       "EBF1-PDGFRB Fusion                                      1\n",
       "K558_E562del                                            1\n",
       "G106_R108del                                            1\n",
       "H773_V774insH                                           1\n",
       "GPIAP1-PDGFRB Fusion                                    1\n",
       "GOLGA4-PDGFRB Fusion                                    1\n",
       "GIT2-PDGFRB Fusion                                      1\n",
       "G778_P780dup                                            1\n",
       "G776delinsVC                                            1\n",
       "G776delinsLC                                            1\n",
       "G776_V777insYVMA                                        1\n",
       "G469del                                                 1\n",
       "FUS-ERG Fusion                                          1\n",
       "FGFR2-AHCYL1 Fusion                                     1\n",
       "FLT3 internal tandem duplications                       1\n",
       "FIG-ROS1 Fusion                                         1\n",
       "FGFR3 - BAIAP2L1 Fusion                                 1\n",
       "FGFR2?PPHLN1 Fusion                                     1\n",
       "FGFR2-TACC3 Fusion                                      1\n",
       "FGFR2-MGEA5 Fusion                                      1\n",
       "FGFR2-KIAA1967 Fusion                                   1\n",
       "FGFR2-FAM76A Fusion                                     1\n",
       "FGFR2-CCDC6 Fusion                                      1\n",
       "H773dup                                                 1\n",
       "H773insLGNP                                             1\n",
       "H845_N848delinsP                                        1\n",
       "HIP1-PDGFRB Fusion                                      1\n",
       "2010_2471trunc                                          1\n",
       "K550_V555delinsI                                        1\n",
       "K550_K558del                                            1\n",
       "K459_S460delinsN                                        1\n",
       "K442Nfs*                                                1\n",
       "IGL-MYC Fusion                                          1\n",
       "IGK-MYC Fusion                                          1\n",
       "IGH-NKX2 Fusion                                         1\n",
       "IGH-MYC Fusion                                          1\n",
       "IGH-FGFR3 Fusion                                        1\n",
       "IGH-BCL2 Fusion                                         1\n",
       "I843del                                                 1\n",
       "I843_D846del                                            1\n",
       "I836del                                                 1\n",
       "I744_K745delinsKIPVAI                                   1\n",
       "I563_L576del                                            1\n",
       "I559_D560insDKRMNS                                      1\n",
       "Hypermethylation                                        1\n",
       "HMGA2-RAD51B Fusion                                     1\n",
       "FGFR2-BICC1 Fusion                                      1\n",
       "FGFR1OP1-FGFR1 Fusion                                   1\n",
       "EGFR-KDD                                                1\n",
       "ESRP1-RAF1 Fusion                                       1\n",
       "EWSR1-FEV Fusion                                        1\n",
       "EWSR1-ETV4 Fusion                                       1\n",
       "EWSR1-ERG Fusion                                        1\n",
       "EWSR1-DDIT3 Fusion                                      1\n",
       "EWSR1-CREB1 Fusion                                      1\n",
       "EWSR1-ATF1 Fusion                                       1\n",
       "ETV6-PDGFRB Fusion                                      1\n",
       "ETV6-PDGFRA Fusion                                      1\n",
       "ETV6-FLT3 Fusion                                        1\n",
       "ESR1-YAP1 Fusion                                        1\n",
       "FGFR1-TACC1 Fusion                                      1\n",
       "ESR1-CCDC170 Fusion                                     1\n",
       "ERLIN2?FGFR1 Fusion                                     1\n",
       "EP300-MOZ Fusion                                        1\n",
       "EP300-MLL Fusion                                        1\n",
       "EGFRvIV                                                 1\n",
       "EGFRvIII                                                1\n",
       "EGFRvII                                                 1\n",
       "EGFR-RAD51 Fusion                                       1\n",
       "EGFR-PURB Fusion                                        1\n",
       "EWSR1-FLI1 Fusion                                       1\n",
       "EWSR1-NR4A3 Fusion                                      1\n",
       "EWSR1-WT1 Fusion                                        1\n",
       "EWSR1-YY1 Fusion                                        1\n",
       "FAM131B-BRAF Fusion                                     1\n",
       "F594_R595insSDNEYFYVDF                                  1\n",
       "F537_K539delinsL                                        1\n",
       "F1088Sfs*2                                              1\n",
       "F1088Lfs*5                                              1\n",
       "Exon 9 mutations                                        1\n",
       "Exon 20 insertions/deletions                            1\n",
       "Exon 20 insertions                                      1\n",
       "Exon 20 insertion                                       1\n",
       "Exon 2 mutations                                        1\n",
       "Exon 19 insertion                                       1\n",
       "Exon 19 deletion/insertion                              1\n",
       "Exon 19 deletion                                        1\n",
       "Exon 13 deletion                                        1\n",
       "Exon 11 mutations                                       1\n",
       "Exon 1 mutations                                        1\n",
       "Epigenetic Silencing                                    1\n",
       "EZR-ROS1 Fusion                                         1\n",
       "EZR-ERBB4 Fusion                                        1\n",
       "p61BRAF                                                 1"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].map(len)>6][[\"Variation\", \"Class\"]].groupby(\"Variation\").count().sort_values(\"Class\", ascending = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Class</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variation\n",
       "Class           \n",
       "1              4\n",
       "2             38\n",
       "4             14\n",
       "5              3\n",
       "7             69"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].str.contains(\"del\")][[\"Variation\", \"Class\"]].groupby(\"Class\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Class</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variation\n",
       "Class           \n",
       "7             13"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].str.contains(\"dup\")][[\"Variation\", \"Class\"]].groupby(\"Class\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Class</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variation\n",
       "Class           \n",
       "1              1\n",
       "2             11\n",
       "4              4\n",
       "7             59"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].str.contains(\"ins\")][[\"Variation\", \"Class\"]].groupby(\"Class\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Class</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variation\n",
       "Class           \n",
       "1             90\n",
       "2              2\n",
       "4              1\n",
       "6              2"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].str.contains(\"Trunc\")][[\"Variation\", \"Class\"]].groupby(\"Class\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Class</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variation\n",
       "Class           \n",
       "1             12\n",
       "2              5\n",
       "3              2\n",
       "4             39\n",
       "6              1\n",
       "7              5"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged[\"Variation\"].str.contains(\"\\*\")][[\"Variation\", \"Class\"]].groupby(\"Class\").count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "<input>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "<ipython-input-37-904a186f9aa6>:1: DeprecationWarning: invalid escape sequence \\*\n",
      "  types = [\"fusion\", \"del\", \"dup\", \"ins\", \"trunc\", \"splice\", \"\\*\"]\n",
      "<ipython-input-37-904a186f9aa6>:11: DeprecationWarning: invalid escape sequence \\*\n",
      "  merged[\"is\\*\"]))\n"
     ]
    }
   ],
   "source": [
    "types = [\"fusion\", \"del\", \"dup\", \"ins\", \"trunc\", \"splice\", \"\\*\"]\n",
    "for i in types:\n",
    "    merged[\"is\"+i] = merged[\"Variation\"].str.lower().str.contains(i).astype(int)\n",
    "\n",
    "merged[\"mutType\"]=list(zip(merged[\"isfusion\"], \n",
    "                           merged[\"isdel\"], \n",
    "                           merged[\"isdup\"], \n",
    "                           merged[\"isins\"], \n",
    "                           merged[\"istrunc\"], \n",
    "                           merged[\"issplice\"], \n",
    "                           merged[\"is\\*\"]))\n",
    "merged[\"mutType\"]=merged[\"mutType\"].map(list)\n",
    "\n",
    "for i in types:\n",
    "    del merged[\"is\"+i]\n",
    "\n",
    "merged[\"mutwordsgenes\"]=merged[\"mutType\"]+merged[\"wordsgenes\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.76793602509\n",
      "Accuracy on training set:\n",
      "1.80661590455\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutType\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wordfeatures\n",
      "3851\n",
      "mutType\n",
      "7\n",
      "genesmentioned\n",
      "264\n",
      "geneofinterest\n",
      "264\n",
      "genefeatures\n",
      "528\n",
      "wordsgenes\n",
      "4379\n",
      "mutwordsgenes\n",
      "4386\n"
     ]
    }
   ],
   "source": [
    "for i in [\"wordfeatures\", \"mutType\", \"genesmentioned\", \"geneofinterest\", \"genefeatures\", \"wordsgenes\", \"mutwordsgenes\"]:\n",
    "    print(i)\n",
    "    print(len(merged[i][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"mutwordsgenes\"]=merged[\"mutType\"]+merged[\"wordsgenes\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.984817905389\n",
      "Accuracy on training set:\n",
      "0.462702302959\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutwordsgenes\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.00455416402\n",
      "Accuracy on training set:\n",
      "0.452566742175\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"wordsgenes\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>split</th>\n",
       "      <th>Text</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>bag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>wordfeatures</th>\n",
       "      <th>mutType</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>genefeatures</th>\n",
       "      <th>wordsgenes</th>\n",
       "      <th>mutwordsgenes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0]</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1]</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene             Variation  Class  split  \\\n",
       "0   0  FAM58A  Truncating Mutations      1   True   \n",
       "1   1     CBL                 W802*      2  False   \n",
       "2   2     CBL                 Q249E      2   True   \n",
       "3   3     CBL                 N454D      3   True   \n",
       "4   4     CBL                 L399V      4   True   \n",
       "\n",
       "                                                Text  \\\n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   Abstract Background  Non-small cell lung canc...   \n",
       "2   Abstract Background  Non-small cell lung canc...   \n",
       "3  Recent evidence has demonstrated that acquired...   \n",
       "4  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                                 bag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                                        wordfeatures                mutType  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 1, 0, 0]   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...  [0, 0, 0, 0, 0, 0, 1]   \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "4  [1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        genefeatures  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                          wordsgenes  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...   \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...   \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "4  [1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "\n",
       "                                       mutwordsgenes  \n",
       "0  [0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "1  [0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, ...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using relevant sentences: 0.95"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "def getSentences(gene, variation, text):\n",
    "    if variation.lower() not in text.lower():\n",
    "        variation = gene\n",
    "    sentences = \"\"\n",
    "    splitText = text.split(\".\")\n",
    "    for sentence in splitText:\n",
    "        if variation.lower() in sentence.lower():\n",
    "            sentences += sentence\n",
    "    return sentences\n",
    "    \n",
    "def applyGetSentences(row):\n",
    "    gene = row[\"Gene\"]\n",
    "    variation = row[\"Variation\"]\n",
    "    text = row[\"Text\"]\n",
    "    return getSentences(gene, variation, text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getvSentences(gene, variation, text):\n",
    "    sentences = \"\"\n",
    "    splitText = text.split(\".\")\n",
    "    for sentence in splitText:\n",
    "        if variation.lower() in sentence.lower():\n",
    "            sentences += sentence\n",
    "    return sentences.lower()\n",
    "    \n",
    "def applyvGetSentences(row):\n",
    "    gene = row[\"Gene\"]\n",
    "    variation = row[\"Variation\"]\n",
    "    text = row[\"Text\"]\n",
    "    return getvSentences(gene, variation, text)\n",
    "\n",
    "def getgSentences(gene, variation, text):\n",
    "    variation = gene\n",
    "    sentences = \"\"\n",
    "    splitText = text.split(\".\")\n",
    "    for sentence in splitText:\n",
    "        if variation.lower() in sentence.lower():\n",
    "            sentences += sentence\n",
    "    return sentences.lower()\n",
    "    \n",
    "def applygGetSentences(row):\n",
    "    gene = row[\"Gene\"]\n",
    "    variation = row[\"Variation\"]\n",
    "    text = row[\"Text\"]\n",
    "    return getgSentences(gene, variation, text)\n",
    "\n",
    "merged[\"gSentences\"] = merged.apply(applygGetSentences, axis = 1)\n",
    "merged[\"vSentences\"] = merged.apply(applyvGetSentences, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"Sentences\"] = merged.apply(applyGetSentences, axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "count = 0\n",
    "for i in range(1000):\n",
    "    if getSentences(merged[\"Gene\"][i], merged[\"Variation\"][i], merged[\"Text\"][i]) == \"\":\n",
    "        print(i)\n",
    "        print(merged.iloc[i])\n",
    "        count +=1;\n",
    "print(count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"tokenizedSentences\"]=merged[\"Sentences\"].map(nltk.word_tokenize)\n",
    "merged[\"sentenceWordBag\"]=merged[\"tokenizedSentences\"].map(build_bag_of_words_features_filtered)\n",
    "\n",
    "wordcounts = {}\n",
    "for i in merged[\"tokenizedSentences\"]:\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.9 and v>len(merged)*.1:\n",
    "        abridgedsorted[k]=v\n",
    "merged[\"sentenceWordFeatures\"]=merged[\"sentenceWordBag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])\n",
    "merged[\"mutwordgenesSentences\"]= merged[\"mutwordsgenes\"]+merged[\"sentenceWordFeatures\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merged[\"tokenizedvSentences\"]=merged[\"vSentences\"].map(nltk.word_tokenize)\n",
    "#merged[\"sentencevWordBag\"]=merged[\"tokenizedvSentences\"].map(build_bag_of_words_features_filtered)\n",
    "\n",
    "nEmpty = 0\n",
    "wordcounts = {}\n",
    "for i in merged[\"tokenizedvSentences\"]:\n",
    "    if len(i)>0:\n",
    "        nEmpty+=1\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< nEmpty*.9 and v>nEmpty*.1:\n",
    "        abridgedsorted[k]=v\n",
    "merged[\"sentencevWordFeatures\"]=merged[\"sentencevWordBag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])\n",
    "\n",
    "#merged[\"tokenizedgSentences\"]=merged[\"gSentences\"].map(nltk.word_tokenize)\n",
    "#merged[\"sentencegWordBag\"]=merged[\"tokenizedgSentences\"].map(build_bag_of_words_features_filtered)\n",
    "\n",
    "nEmpty = 0\n",
    "wordcounts = {}\n",
    "for i in merged[\"tokenizedgSentences\"]:\n",
    "    if len(i)>0:\n",
    "        nEmpty+=1\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< nEmpty*.9 and v>nEmpty*.1:\n",
    "        abridgedsorted[k]=v\n",
    "merged[\"sentencegWordFeatures\"]=merged[\"sentencegWordBag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0                                                   []\n",
       "1    [using, select, c-cbl, somatic, mutations, suc...\n",
       "2    [using, select, c-cbl, somatic, mutations, suc...\n",
       "3    [15, apart, from, n454d, ,, all, residues, aff...\n",
       "4    [finally, ,, the, third, group, constituted, m...\n",
       "5    [finally, ,, the, third, group, constituted, m...\n",
       "6    [the, second, group, of, mutants, (, m374v, ,,...\n",
       "7    [results, :, we, identified, 3, of, 279, aml/m...\n",
       "8    [we, investigated, the, mechanism, by, which, ...\n",
       "9    [we, investigated, the, mechanism, by, which, ...\n",
       "Name: tokenizedvSentences, dtype: object"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[\"tokenizedvSentences\"].head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.951012080346\n",
      "Accuracy on training set:\n",
      "0.412082914614\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutwordgenesSentences\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.999496304599\n",
      "Accuracy on training set:\n",
      "0.453242040004\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutwordsgenes\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>split</th>\n",
       "      <th>Text</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>bag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>...</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>genefeatures</th>\n",
       "      <th>wordsgenes</th>\n",
       "      <th>mutwordsgenes</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>sentenceWordBag</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>mutwordgenesSentences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene             Variation  Class  split  \\\n",
       "0   0  FAM58A  Truncating Mutations      1   True   \n",
       "\n",
       "                                                Text  \\\n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "\n",
       "                                               bag  \\\n",
       "0  {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "\n",
       "                                           genebag  \\\n",
       "0  {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        genefeatures  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                          wordsgenes  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "\n",
       "                                       mutwordsgenes  \\\n",
       "0  [0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "\n",
       "                                     sentenceWordBag  \\\n",
       "0  {'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                               mutwordgenesSentences  \n",
       "0  [0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "\n",
       "[1 rows x 22 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.23525815466\n",
      "Accuracy on training set:\n",
      "1.09183879716\n"
     ]
    }
   ],
   "source": [
    "merged[\"mutgenesSentences\"]=merged[\"mutType\"]+merged[\"genefeatures\"]+merged[\"sentenceWordFeatures\"]\n",
    "logistictester(merged, \"mutgenesSentences\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.951012080346\n",
      "Accuracy on training set:\n",
      "0.412082914614\n",
      "Accuracy on test set:\n",
      "1.06434176886\n",
      "Accuracy on training set:\n",
      "0.176049128449\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutwordgenesSentences\", \"Class\")\n",
    "logistictester(merged, \"mutwordgenesSentences\", \"Class\", C=0.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.951012080346\n",
      "Accuracy on training set:\n",
      "0.412082914614\n",
      "Accuracy on test set:\n",
      "0.982925552612\n",
      "Accuracy on training set:\n",
      "0.542167925135\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutwordgenesSentences\", \"Class\")\n",
    "logistictester(merged, \"mutwordgenesSentences\", \"Class\", C=5e-3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>split</th>\n",
       "      <th>Text</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>bag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>...</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>genefeatures</th>\n",
       "      <th>wordsgenes</th>\n",
       "      <th>mutwordsgenes</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>sentenceWordBag</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>mutwordgenesSentences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>15 Apart from N454D, all residues affected by ...</td>\n",
       "      <td>[15, Apart, from, N454D, ,, all, residues, aff...</td>\n",
       "      <td>{'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, ...</td>\n",
       "      <td>Finally, the third group constituted mutatio...</td>\n",
       "      <td>[Finally, ,, the, third, group, constituted, m...</td>\n",
       "      <td>{'Finally': 1, 'third': 1, 'group': 1, 'consti...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene             Variation  Class  split  \\\n",
       "0   0  FAM58A  Truncating Mutations      1   True   \n",
       "1   1     CBL                 W802*      2  False   \n",
       "2   2     CBL                 Q249E      2   True   \n",
       "3   3     CBL                 N454D      3   True   \n",
       "4   4     CBL                 L399V      4   True   \n",
       "\n",
       "                                                Text  \\\n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   Abstract Background  Non-small cell lung canc...   \n",
       "2   Abstract Background  Non-small cell lung canc...   \n",
       "3  Recent evidence has demonstrated that acquired...   \n",
       "4  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                                 bag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "1                        ...                           \n",
       "2                        ...                           \n",
       "3                        ...                           \n",
       "4                        ...                           \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        genefeatures  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                          wordsgenes  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...   \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, ...   \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "4  [1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "\n",
       "                                       mutwordsgenes  \\\n",
       "0  [0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, ...   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "1   Using select c-CBL somatic mutations such as ...   \n",
       "2   Using select c-CBL somatic mutations such as ...   \n",
       "3  15 Apart from N454D, all residues affected by ...   \n",
       "4    Finally, the third group constituted mutatio...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "2  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "3  [15, Apart, from, N454D, ,, all, residues, aff...   \n",
       "4  [Finally, ,, the, third, group, constituted, m...   \n",
       "\n",
       "                                     sentenceWordBag  \\\n",
       "0  {'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...   \n",
       "1  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "2  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "3  {'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...   \n",
       "4  {'Finally': 1, 'third': 1, 'group': 1, 'consti...   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                               mutwordgenesSentences  \n",
       "0  [0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "1  [0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...  \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, ...  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n",
       "       colsample_bytree=1, eval_metric='mlogloss', gamma=0,\n",
       "       learning_rate=0.1, max_delta_step=0, max_depth=6,\n",
       "       min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,\n",
       "       nthread=None, objective='multi:softprob', random_state=0,\n",
       "       reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,\n",
       "       silent=True, subsample=1)"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model = xgb.XGBClassifier(eval_metric=\"mlogloss\", max_depth=6)\n",
    "\n",
    "training = merged[merged[\"split\"]]\n",
    "test = merged[merged[\"split\"]==False]\n",
    "X_train = np.array(training[\"mutwordgenesSentencesbgs\"].tolist())\n",
    "y_train = np.array(training[\"Class\"].tolist())\n",
    "\n",
    "model.fit(X_train, y_train)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy: 70.20%\n"
     ]
    }
   ],
   "source": [
    "from sklearn.metrics import accuracy_score\n",
    "X_test=np.array(test[\"mutwordgenesSentencesbgs\"].tolist())\n",
    "y_test=np.array(test[\"Class\"].tolist())\n",
    "y_pred = model.predict(X_test)\n",
    "accuracy = accuracy_score(y_test, y_pred)\n",
    "print(\"Accuracy: %.2f%%\" % (accuracy * 100.0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.888651341029\n"
     ]
    }
   ],
   "source": [
    "y_pred = model.predict_proba(X_test) \n",
    "    \n",
    "ls = []\n",
    "for i in range(len(y_pred)):\n",
    "    c = y_test[i]\n",
    "    prob = y_pred[i][c-1]\n",
    "    ls.append(-np.log(prob+1e-6))\n",
    "\n",
    "print(\"Accuracy on test set:\")\n",
    "print(np.mean(ls))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on training set:\n",
      "0.192435000389\n"
     ]
    }
   ],
   "source": [
    "y_pred = model.predict_proba(X_train) \n",
    "    \n",
    "ls = []\n",
    "for i in range(len(y_pred)):\n",
    "    c = y_train[i]\n",
    "    prob = y_pred[i][c-1]\n",
    "    ls.append(-np.log(prob+1e-6))\n",
    "\n",
    "print(\"Accuracy on training set:\")\n",
    "print(np.mean(ls))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def logistictester2(df, X, Y, C=1e-2):\n",
    "    training = df[df[\"split\"]]\n",
    "    test = df[df[\"split\"]==False]\n",
    "    X_train = training[X].tolist()\n",
    "    y_train = training[Y].tolist()\n",
    "    X_test=test[X].tolist()\n",
    "\n",
    "    lr = LogisticRegression( solver = \"lbfgs\",\n",
    "                                multi_class=\"multinomial\",\n",
    "                                penalty='l2',\n",
    "                                C=C,\n",
    "                                fit_intercept=True\n",
    "                                )\n",
    "    lr.fit(X_train, y_train)\n",
    "    y_pred = lr.predict(X_test)\n",
    "    from sklearn.metrics import accuracy_score\n",
    "    accuracy = accuracy_score(test[Y].tolist(), y_pred)\n",
    "    print(\"Accuracy: %.2f%%\" % (accuracy * 100.0))\n",
    " \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy: 67.19%\n",
      "Accuracy on test set:\n",
      "0.951012080346\n",
      "Accuracy on training set:\n",
      "0.412082914614\n"
     ]
    }
   ],
   "source": [
    "logistictester2(merged, \"mutwordgenesSentences\", \"Class\", C=1e-2)\n",
    "logistictester(merged, \"mutwordgenesSentences\", \"Class\", C=1e-2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(\"tokenized.pkl\",'wb') as f:\n",
    "    pickle.dump(merged, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1976\n"
     ]
    }
   ],
   "source": [
    "wordcounts = {}\n",
    "for i in merged[\"tokenized\"]:\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.8 and v>len(merged)*.2:\n",
    "        abridgedsorted[k]=v\n",
    "print(len(abridgedsorted))\n",
    "merged[\"wordfeatures2\"]=merged[\"wordbag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1860\n"
     ]
    }
   ],
   "source": [
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.7 and v>len(merged)*.2:\n",
    "        abridgedsorted[k]=v\n",
    "print(len(abridgedsorted))\n",
    "merged[\"wordfeatures3\"]=merged[\"wordbag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.994950052555\n",
      "Accuracy on training set:\n",
      "0.468286259757\n",
      "Accuracy on test set:\n",
      "1.0229567602\n",
      "Accuracy on training set:\n",
      "0.556894315989\n",
      "Accuracy on test set:\n",
      "1.0293525844\n",
      "Accuracy on training set:\n",
      "0.578159807015\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"wordfeatures\", \"Class\")\n",
    "logistictester(merged, \"wordfeatures2\", \"Class\")\n",
    "logistictester(merged, \"wordfeatures3\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6378\n",
      "Accuracy on test set:\n",
      "0.987867217001\n",
      "Accuracy on training set:\n",
      "0.429010908948\n"
     ]
    }
   ],
   "source": [
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.95 and v>len(merged)*.05:\n",
    "        abridgedsorted[k]=v\n",
    "print(len(abridgedsorted))\n",
    "merged[\"wordfeatures.5\"]=merged[\"wordbag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])\n",
    "logistictester(merged, \"wordfeatures.5\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20710\n",
      "Accuracy on test set:\n",
      "0.984324915752\n",
      "Accuracy on training set:\n",
      "0.394470366961\n"
     ]
    }
   ],
   "source": [
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.99 and v>len(merged)*.01:\n",
    "        abridgedsorted[k]=v\n",
    "print(len(abridgedsorted))\n",
    "merged[\"wordfeatures.01\"]=merged[\"wordbag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])\n",
    "logistictester(merged, \"wordfeatures.01\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6342\n",
      "Accuracy on test set:\n",
      "0.989575009243\n",
      "Accuracy on training set:\n",
      "0.429792229857\n"
     ]
    }
   ],
   "source": [
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.9 and v>len(merged)*.05:\n",
    "        abridgedsorted[k]=v\n",
    "print(len(abridgedsorted))\n",
    "merged[\"wordfeatures9001\"]=merged[\"wordbag\"].map(lambda b: [int(word in b.keys()) for word in abridgedsorted.keys()])\n",
    "logistictester(merged, \"wordfeatures9001\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "del merged[\"wordfeatures9001\"]\n",
    "del merged[\"wordfeatures.01\"]\n",
    "del merged[\"wordfeatures3\"]\n",
    "del merged[\"wordfeatures2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "useless_words = nltk.corpus.stopwords.words(\"english\") + list(string.punctuation)\n",
    "def find_bigrams(input_sentence):\n",
    "    input_list = input_sentence.split()\n",
    "    input_list = [i for i in input_list if i not in useless_words]\n",
    "    return list(zip(input_list, input_list[1:]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' two brca1 mutations (p1812a, p25t), and a brca2 mutation (5164del4) have been detected in naj high-risk families we conclude that the p1812a and p25t brca1 and 5164del4 brca2 mutations are not likely to be founder mutations in naj high-risk families we also assessed the pathogenicity of the brca1 p1812a mutation in vitro using reporter gene assays in yeast and mammalian cells we found that the brca1 p1812a variant activity assays yielded a slightly reduced reporter gene activity thus, there is some uncertainty as to the pathogenicity of brca1 p1812a'"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[\"Sentences\"].iloc[3680]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('two', 'brca1'),\n",
       " ('brca1', 'mutations'),\n",
       " ('mutations', '(p1812a,'),\n",
       " ('(p1812a,', 'p25t),'),\n",
       " ('p25t),', 'brca2'),\n",
       " ('brca2', 'mutation'),\n",
       " ('mutation', '(5164del4)'),\n",
       " ('(5164del4)', 'detected'),\n",
       " ('detected', 'naj'),\n",
       " ('naj', 'high-risk'),\n",
       " ('high-risk', 'families'),\n",
       " ('families', 'conclude'),\n",
       " ('conclude', 'p1812a'),\n",
       " ('p1812a', 'p25t'),\n",
       " ('p25t', 'brca1'),\n",
       " ('brca1', '5164del4'),\n",
       " ('5164del4', 'brca2'),\n",
       " ('brca2', 'mutations'),\n",
       " ('mutations', 'likely'),\n",
       " ('likely', 'founder'),\n",
       " ('founder', 'mutations'),\n",
       " ('mutations', 'naj'),\n",
       " ('naj', 'high-risk'),\n",
       " ('high-risk', 'families'),\n",
       " ('families', 'also'),\n",
       " ('also', 'assessed'),\n",
       " ('assessed', 'pathogenicity'),\n",
       " ('pathogenicity', 'brca1'),\n",
       " ('brca1', 'p1812a'),\n",
       " ('p1812a', 'mutation'),\n",
       " ('mutation', 'vitro'),\n",
       " ('vitro', 'using'),\n",
       " ('using', 'reporter'),\n",
       " ('reporter', 'gene'),\n",
       " ('gene', 'assays'),\n",
       " ('assays', 'yeast'),\n",
       " ('yeast', 'mammalian'),\n",
       " ('mammalian', 'cells'),\n",
       " ('cells', 'found'),\n",
       " ('found', 'brca1'),\n",
       " ('brca1', 'p1812a'),\n",
       " ('p1812a', 'variant'),\n",
       " ('variant', 'activity'),\n",
       " ('activity', 'assays'),\n",
       " ('assays', 'yielded'),\n",
       " ('yielded', 'slightly'),\n",
       " ('slightly', 'reduced'),\n",
       " ('reduced', 'reporter'),\n",
       " ('reporter', 'gene'),\n",
       " ('gene', 'activity'),\n",
       " ('activity', 'thus,'),\n",
       " ('thus,', 'uncertainty'),\n",
       " ('uncertainty', 'pathogenicity'),\n",
       " ('pathogenicity', 'brca1'),\n",
       " ('brca1', 'p1812a')]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "find_bigrams(merged[\"Sentences\"].iloc[3680])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"bigrams\"]=merged[\"Sentences\"].map(find_bigrams)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"bigramsfiltered\"]=merged[\"Sentences\"].map(find_bigrams)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('cell', 'lines'), 823)\n",
      "(('et', 'al'), 780)\n",
      "(('amino', 'acid'), 673)\n",
      "(('cells', 'expressing'), 655)\n",
      "(('cell', 'line'), 619)\n",
      "(('kinase', 'domain'), 541)\n",
      "(('go', 'to:'), 513)\n",
      "(('missense', 'mutations'), 510)\n",
      "(('site-directed', 'mutagenesis'), 480)\n",
      "(('table', '1'), 469)\n",
      "(('wild', 'type'), 455)\n",
      "(('(data', 'shown)'), 443)\n",
      "(('kinase', 'activity'), 435)\n",
      "(('cells', 'transfected'), 430)\n",
      "(('figure', '1'), 427)\n",
      "(('amino', 'acids'), 422)\n",
      "(('figure', '2'), 422)\n",
      "(('previously', 'reported'), 411)\n",
      "(('mutations', 'found'), 410)\n",
      "(('point', 'mutations'), 409)\n",
      "(('tyrosine', 'kinase'), 382)\n",
      "(('determine', 'whether'), 380)\n",
      "(('figure', '3'), 379)\n",
      "(('mutations', 'identified'), 366)\n",
      "(('figure', '4'), 360)\n",
      "(('breast', 'cancer'), 357)\n",
      "(('previously', 'described'), 350)\n",
      "(('growth', 'factor'), 347)\n",
      "(('somatic', 'mutations'), 344)\n",
      "(('(supplementary', 'fig'), 334)\n",
      "(('cell', 'lines,'), 327)\n",
      "(('1', 'figure'), 319)\n",
      "(('to:', 'results'), 318)\n",
      "(('activating', 'mutations'), 315)\n",
      "(('data', 'suggest'), 313)\n",
      "(('cancer', 'cell'), 310)\n",
      "(('2', 'figure'), 286)\n",
      "(('compared', 'wild-type'), 285)\n",
      "(('results', 'suggest'), 284)\n",
      "(('(table', '1)'), 281)\n",
      "(('western', 'blot'), 281)\n",
      "(('cell', 'growth'), 280)\n",
      "(('lung', 'cancer'), 279)\n",
      "(('cells', '(fig'), 274)\n",
      "(('3', 'figure'), 273)\n",
      "(('protein', 'expression'), 265)\n",
      "(('mutations', 'may'), 263)\n",
      "(('mutations', 'associated'), 257)\n",
      "(('4', 'figure'), 256)\n",
      "(('taken', 'together,'), 254)\n",
      "(('copy', 'number'), 253)\n",
      "(('cell', 'proliferation'), 253)\n",
      "(('wild-type', '(wt)'), 253)\n",
      "(('cell', 'cycle'), 242)\n",
      "(('expression', 'levels'), 242)\n",
      "(('cancer', 'cells'), 239)\n",
      "(('tumor', 'suppressor'), 236)\n",
      "(('highly', 'conserved'), 234)\n",
      "(('to:', 'discussion'), 234)\n",
      "(('materials', 'methods'), 232)\n",
      "(('(supplementary', 'table'), 231)\n",
      "(('1', 'table'), 231)\n",
      "(('(p', '0'), 230)\n",
      "(('tumor', 'cells'), 227)\n",
      "(('transiently', 'transfected'), 224)\n",
      "(('wild-type', 'mutant'), 224)\n",
      "(('acid', 'substitutions'), 224)\n",
      "(('constitutive', 'activation'), 223)\n",
      "(('also', 'observed'), 221)\n",
      "(('table', '2'), 220)\n",
      "(('missense', 'mutation'), 218)\n",
      "(('cell', 'line,'), 217)\n",
      "(('blot', 'analysis'), 217)\n",
      "(('genomic', 'dna'), 216)\n",
      "(('results', 'indicate'), 215)\n",
      "(('gene', 'expression'), 215)\n",
      "(('dna', 'binding'), 214)\n",
      "(('ba/f3', 'cells'), 213)\n",
      "(('factor', 'receptor'), 212)\n",
      "(('mutation', 'status'), 212)\n",
      "(('coding', 'sequence'), 209)\n",
      "(('kinase', 'inhibitor'), 208)\n",
      "(('et', 'al,'), 208)\n",
      "(('colony', 'formation'), 208)\n",
      "(('data', 'indicate'), 208)\n",
      "(('previous', 'studies'), 206)\n",
      "(('domain', 'mutations'), 206)\n",
      "(('crystal', 'structure'), 205)\n",
      "(('functional', 'consequences'), 203)\n",
      "(('expression', 'vector'), 203)\n",
      "(('ovarian', 'cancer'), 201)\n",
      "(('cells', 'stably'), 200)\n",
      "(('also', 'found'), 198)\n",
      "(('signaling', 'pathway'), 198)\n",
      "(('activation', 'loop'), 198)\n",
      "(('constitutively', 'active'), 197)\n",
      "(('supplementary', 'table'), 196)\n",
      "(('downstream', 'signaling'), 195)\n",
      "(('shown', 'figure'), 195)\n",
      "(('kinase', 'inhibitors'), 194)\n",
      "(('significantly', 'higher'), 193)\n",
      "(('protein', 'kinase'), 193)\n",
      "(('receptor', 'tyrosine'), 192)\n",
      "(('statistically', 'significant'), 189)\n",
      "(('analysis', 'revealed'), 189)\n",
      "(('empty', 'vector'), 189)\n",
      "(('mutations', 'also'), 188)\n",
      "(('mutations,', 'including'), 187)\n",
      "(('described', 'previously'), 187)\n",
      "(('stably', 'expressing'), 187)\n",
      "(('point', 'mutation'), 186)\n",
      "(('mutagenesis', 'kit'), 185)\n",
      "(('cells', '(figure'), 184)\n",
      "(('human', 'cancers'), 182)\n",
      "(('western', 'blotting'), 182)\n",
      "(('missense', 'variants'), 181)\n",
      "(('mutations', 'reported'), 180)\n",
      "(('binding', 'domain'), 180)\n",
      "(('mutations', 'detected'), 179)\n",
      "(('figure', '5'), 179)\n",
      "(('(table', '2)'), 177)\n",
      "(('binding', 'site'), 176)\n",
      "(('cells', 'transiently'), 175)\n",
      "(('signaling', 'pathways'), 174)\n",
      "(('vitro', 'vivo'), 174)\n",
      "(('important', 'role'), 174)\n",
      "(('may', 'also'), 173)\n",
      "(('least', 'one'), 173)\n",
      "(('together,', 'data'), 170)\n",
      "(('expressing', 'wild-type'), 169)\n",
      "(('increased', 'risk'), 169)\n",
      "(('tumor', 'growth'), 169)\n",
      "(('high', 'levels'), 168)\n",
      "(('present', 'study,'), 168)\n",
      "(('coding', 'region'), 167)\n",
      "(('extracellular', 'domain'), 167)\n",
      "(('mutations', 'occur'), 166)\n",
      "(('mutant', 'proteins'), 165)\n",
      "(('critical', 'role'), 165)\n",
      "(('tumor', 'samples'), 164)\n",
      "(('results', 'obtained'), 164)\n",
      "(('overall', 'survival'), 164)\n",
      "(('clinical', 'trials'), 164)\n",
      "(('breast', 'ovarian'), 163)\n",
      "(('table', '1)'), 161)\n",
      "(('transcription', 'factors'), 161)\n",
      "(('transcriptional', 'activity'), 161)\n",
      "(('mutations', 'within'), 161)\n",
      "(('cells', 'treated'), 160)\n",
      "(('293t', 'cells'), 158)\n",
      "(('also', 'showed'), 157)\n",
      "(('mutually', 'exclusive'), 156)\n",
      "(('cell', 'lysates'), 155)\n",
      "(('(fig', 'results'), 154)\n",
      "(('using', 'quikchange'), 154)\n",
      "(('previously', 'identified'), 154)\n",
      "(('gene', 'amplification'), 154)\n",
      "(('polymerase', 'chain'), 154)\n",
      "(('mutational', 'analysis'), 153)\n",
      "(('transcriptional', 'activation'), 153)\n",
      "(('present', 'study'), 152)\n",
      "(('chain', 'reaction'), 152)\n",
      "(('previously', 'shown'), 152)\n",
      "(('mutations', 'exon'), 151)\n",
      "(('exon', '11'), 151)\n",
      "(('small', 'molecule'), 151)\n",
      "(('tyrosine', 'phosphorylation'), 150)\n",
      "(('to:', 'introduction'), 150)\n",
      "(('mutations', 'patients'), 150)\n",
      "(('also', 'identified'), 148)\n",
      "(('kinase', 'domain,'), 148)\n",
      "(('reporter', 'gene'), 147)\n",
      "(('cells', 'showed'), 146)\n",
      "(('mutations', 'observed'), 145)\n",
      "(('de', 'novo'), 145)\n",
      "(('expression', 'vectors'), 145)\n",
      "(('dna', 'repair'), 145)\n",
      "(('mutation', 'exon'), 144)\n",
      "(('mutations', 'located'), 143)\n",
      "(('cancer', 'patients'), 143)\n",
      "(('soft', 'agar'), 142)\n",
      "(('presence', 'absence'), 142)\n",
      "(('sanger', 'sequencing'), 142)\n",
      "(('cells', 'harboring'), 142)\n",
      "(('mutation', 'found'), 141)\n",
      "(('(fig', 'however,'), 141)\n",
      "(('positive', 'control'), 141)\n",
      "(('(fig', 'contrast,'), 140)\n",
      "(('signal', 'transduction'), 140)\n",
      "(('tumor', 'cell'), 140)\n",
      "(('performed', 'using'), 140)\n",
      "(('may', 'contribute'), 139)\n",
      "(('bone', 'marrow'), 138)\n",
      "(('mutations', 'could'), 138)\n",
      "(('mutation', 'analysis'), 138)\n",
      "(('human', 'cancer'), 138)\n",
      "(('mutations', 'affecting'), 138)\n",
      "(('situ', 'hybridization'), 138)\n",
      "(('significant', 'difference'), 138)\n",
      "(('transient', 'transfection'), 138)\n",
      "(('similar', 'results'), 136)\n",
      "(('gene', 'mutations'), 136)\n",
      "(('5', 'figure'), 135)\n",
      "(('two', 'patients'), 135)\n",
      "(('fusion', 'protein'), 135)\n",
      "(('mutations', 'might'), 134)\n",
      "(('acid', 'substitution'), 134)\n",
      "(('two', 'mutations'), 134)\n",
      "(('protein', 'levels'), 134)\n",
      "(('squamous', 'cell'), 133)\n",
      "(('fusion', 'proteins'), 133)\n",
      "(('lines', 'expressing'), 133)\n",
      "(('mouse', 'model'), 133)\n",
      "(('cell', 'signaling'), 132)\n",
      "(('(fig', 'figure'), 132)\n",
      "(('mutations', 'significantly'), 132)\n",
      "(('binding', 'activity'), 132)\n",
      "(('plays', 'important'), 131)\n",
      "(('sequence', 'alignment'), 131)\n",
      "(('mutations', 'confer'), 131)\n",
      "(('recently', 'reported'), 130)\n",
      "(('(supplementary', 'figure'), 130)\n",
      "(('target', 'genes'), 130)\n",
      "(('compared', 'wt'), 129)\n",
      "(('data', 'show'), 129)\n",
      "(('investigate', 'whether'), 129)\n",
      "(('patients', 'whose'), 129)\n",
      "(('mutation', 'identified'), 128)\n",
      "(('expressing', 'mutant'), 128)\n",
      "(('epidermal', 'growth'), 128)\n",
      "(('also', 'reported'), 128)\n",
      "(('high', 'frequency'), 128)\n",
      "(('family', 'history'), 128)\n",
      "(('together,', 'results'), 127)\n",
      "(('(figure', '3a)'), 127)\n",
      "(('acquired', 'resistance'), 127)\n",
      "(('cell', 'lung'), 126)\n",
      "(('sequence', 'analysis'), 126)\n",
      "(('mutant', 'forms'), 125)\n",
      "(('containing', 'wild-type'), 125)\n",
      "(('(figure', '2b)'), 125)\n",
      "(('clinical', 'significance'), 125)\n",
      "(('functional', 'analysis'), 124)\n",
      "(('results', 'show'), 124)\n",
      "(('fluorescence', 'situ'), 124)\n",
      "(('mutation', 'may'), 124)\n",
      "(('significant', 'increase'), 124)\n",
      "(('constitutively', 'activated'), 124)\n",
      "(('genetic', 'alterations'), 124)\n",
      "(('1', '2'), 123)\n",
      "(('germline', 'mutations'), 123)\n",
      "(('stably', 'transfected'), 122)\n",
      "(('higher', 'levels'), 122)\n",
      "(('ectopic', 'expression'), 122)\n",
      "(('recent', 'study'), 122)\n",
      "(('cells', 'using'), 122)\n",
      "(('several', 'studies'), 121)\n",
      "(('consistent', 'previous'), 121)\n",
      "(('recent', 'studies'), 121)\n",
      "(('schematic', 'representation'), 121)\n",
      "(('two', 'different'), 120)\n",
      "(('vast', 'majority'), 120)\n",
      "(('tumor', 'types'), 120)\n",
      "(('figure', '6'), 120)\n",
      "(('low', 'levels'), 120)\n",
      "(('mutant', 'protein'), 120)\n",
      "(('tumors', 'harboring'), 120)\n",
      "(('significantly', 'increased'), 119)\n",
      "(('located', 'within'), 119)\n",
      "(('analysis', 'performed'), 118)\n",
      "(('gastrointestinal', 'stromal'), 118)\n",
      "(('significantly', 'reduced'), 118)\n",
      "(('one', 'two'), 118)\n",
      "(('findings', 'suggest'), 117)\n",
      "(('one', 'patient'), 117)\n",
      "(('single', 'amino'), 117)\n",
      "(('table', '3'), 115)\n",
      "(('cell', 'surface'), 115)\n",
      "(('pathway', 'activation'), 115)\n",
      "(('highly', 'sensitive'), 115)\n",
      "(('transfected', 'cells'), 114)\n",
      "(('la', 'jolla,'), 114)\n",
      "(('ovarian', 'cancers'), 114)\n",
      "(('activation', 'downstream'), 114)\n",
      "(('lung', 'adenocarcinoma'), 114)\n",
      "(('mutations', 'affect'), 114)\n",
      "(('family', 'members'), 113)\n",
      "(('mutational', 'status'), 113)\n",
      "(('gain-of-function', 'mutations'), 113)\n",
      "(('downstream', 'targets'), 113)\n",
      "(('exon', '9'), 112)\n",
      "(('mutations', 'human'), 112)\n",
      "(('tumor', 'development'), 112)\n",
      "(('in-frame', 'deletion'), 112)\n",
      "(('kinase', 'domains'), 112)\n",
      "(('expression', 'genes'), 112)\n",
      "(('three', 'different'), 111)\n",
      "(('compared', 'control'), 111)\n",
      "(('supplementary', 'figure'), 111)\n",
      "(('peripheral', 'blood'), 111)\n",
      "(('clinical', 'response'), 111)\n",
      "(('cancer', 'susceptibility'), 110)\n",
      "(('(fig', 'thus,'), 110)\n",
      "(('direct', 'sequencing'), 110)\n",
      "(('p', '0'), 110)\n",
      "(('significantly', 'lower'), 110)\n",
      "(('(figure', '1a)'), 109)\n",
      "(('tyrosine', 'kinases'), 109)\n",
      "(('mutations', 'previously'), 109)\n",
      "(('cell', 'carcinoma'), 109)\n",
      "(('transcription', 'factor'), 109)\n",
      "(('epithelial', 'cells'), 109)\n",
      "(('primers', 'used'), 109)\n",
      "(('cells', 'compared'), 109)\n",
      "(('egfr', 'mutations'), 109)\n",
      "(('mutations', 'cause'), 108)\n",
      "(('according', \"manufacturer's\"), 108)\n",
      "(('mutations', 'result'), 108)\n",
      "(('dna', 'sequencing'), 108)\n",
      "(('growth', 'inhibition'), 108)\n",
      "(('structural', 'analysis'), 108)\n",
      "(('results', 'discussion'), 108)\n",
      "(('test', 'whether'), 108)\n",
      "(('dna-binding', 'domain'), 108)\n",
      "(('protein', 'structure'), 108)\n",
      "(('activity', 'brca1'), 108)\n",
      "(('transforming', 'activity'), 107)\n",
      "(('even', 'though'), 107)\n",
      "(('using', 'primers'), 107)\n",
      "(('primary', 'tumors'), 107)\n",
      "(('domain', 'mutants'), 107)\n",
      "(('mutants', 'tested'), 107)\n",
      "(('whose', 'tumors'), 107)\n",
      "(('anchorage-independent', 'growth'), 107)\n",
      "(('lung', 'cancers'), 106)\n",
      "(('large', 'number'), 106)\n",
      "(('colorectal', 'cancer'), 106)\n",
      "(('effects', 'mutations'), 106)\n",
      "(('ic50', 'values'), 106)\n",
      "(('confer', 'resistance'), 106)\n",
      "(('wild-type', 'brca1'), 106)\n",
      "(('amino', 'acids,'), 105)\n",
      "(('(figure', '1b)'), 105)\n",
      "(('mutations', 'present'), 105)\n",
      "(('analysis', 'using'), 105)\n",
      "(('expressing', 'wt'), 105)\n",
      "(('stop', 'codon'), 105)\n",
      "(('mouse', 'models'), 105)\n",
      "(('(data', 'shown),'), 104)\n",
      "(('table', '2)'), 104)\n",
      "(('expression', 'level'), 104)\n",
      "(('disease', 'progression'), 104)\n",
      "(('discussion', 'study,'), 104)\n",
      "(('(amino', 'acids'), 104)\n",
      "(('control', 'cells'), 103)\n",
      "(('may', 'provide'), 103)\n",
      "(('identified', 'two'), 103)\n",
      "(('2', 'mutations'), 103)\n",
      "(('cells', 'transduced'), 102)\n",
      "(('compared', 'cells'), 102)\n",
      "(('mutations', 'likely'), 102)\n",
      "(('mitogen-activated', 'protein'), 102)\n",
      "(('activity', '(fig'), 102)\n",
      "(('either', 'wild-type'), 102)\n",
      "(('crystal', 'structures'), 102)\n",
      "(('using', 'vitro'), 102)\n",
      "(('patients', 'treated'), 102)\n",
      "(('patients', 'without'), 101)\n",
      "(('dna', 'damage'), 101)\n",
      "(('nih3t3', 'cells'), 100)\n",
      "(('4,', '5,'), 100)\n",
      "(('cell', 'death'), 100)\n",
      "(('cancer-associated', 'mutations'), 100)\n",
      "(('kinase', 'activity,'), 100)\n",
      "(('tumor', 'progression'), 100)\n",
      "(('stromal', 'tumors'), 100)\n",
      "(('data', 'available'), 100)\n",
      "(('oncogenic', 'mutations'), 100)\n",
      "(('expressing', 'either'), 100)\n",
      "(('variants', 'brca1'), 100)\n",
      "(('cell', 'viability'), 99)\n",
      "(('showed', 'increased'), 99)\n",
      "(('mutation', 'also'), 99)\n",
      "(('(figure', '4a)'), 99)\n",
      "(('ligand', 'binding'), 99)\n",
      "(('protein', 'level'), 99)\n",
      "(('activating', 'mutation'), 99)\n",
      "(('cell', 'proliferation,'), 98)\n",
      "(('cancer', 'genome'), 98)\n",
      "(('play', 'role'), 98)\n",
      "(('frequently', 'mutated'), 98)\n",
      "(('wt', 'mutant'), 98)\n",
      "(('(figure', '1)'), 98)\n",
      "(('acid', 'residues'), 98)\n",
      "(('(figure', '2)'), 98)\n",
      "(('somatic', 'mutation'), 97)\n",
      "(('cells', 'express'), 97)\n",
      "(('studies', 'indicate'), 97)\n",
      "(('mutant', 'allele'), 97)\n",
      "(('patients', 'harboring'), 97)\n",
      "(('data', 'support'), 97)\n",
      "(('platelet-derived', 'growth'), 97)\n",
      "(('cells,', 'whereas'), 97)\n",
      "(('myeloid', 'leukemia'), 96)\n",
      "(('pcr', 'products'), 96)\n",
      "(('showed', 'significant'), 96)\n",
      "(('figure', 'legend'), 96)\n",
      "(('least', 'two'), 96)\n",
      "(('mutations', 'increase'), 96)\n",
      "(('previous', 'reports'), 96)\n",
      "(('reported', 'previously'), 96)\n",
      "(('data', 'bank'), 96)\n",
      "(('history', 'breast'), 96)\n",
      "(('brca1', 'mutations'), 96)\n",
      "(('hek293', 'cells'), 95)\n",
      "(('(stratagene,', 'la'), 95)\n",
      "(('loss', 'function'), 95)\n",
      "(('missense', 'mutations,'), 95)\n",
      "(('may', 'represent'), 95)\n",
      "(('less', 'sensitive'), 95)\n",
      "(('different', 'mutations'), 95)\n",
      "(('positive', 'control,'), 95)\n",
      "(('may', 'result'), 94)\n",
      "(('tested', 'whether'), 94)\n",
      "(('growth', 'advantage'), 94)\n",
      "(('also', 'shown'), 94)\n",
      "(('phase', 'ii'), 94)\n",
      "(('exon', '19'), 94)\n",
      "(('mutations', 'introduced'), 94)\n",
      "(('molecular', 'weight'), 94)\n",
      "(('brca1', 'variants'), 94)\n",
      "(('single', 'nucleotide'), 93)\n",
      "(('clinical', 'outcome'), 93)\n",
      "(('investigated', 'whether'), 93)\n",
      "(('wild-type', 'allele'), 93)\n",
      "(('therapeutic', 'target'), 93)\n",
      "(('one', 'additional'), 93)\n",
      "(('mutation', 'human'), 92)\n",
      "(('cell', 'carcinomas'), 92)\n",
      "(('2', 'table'), 92)\n",
      "(('cells', 'also'), 92)\n",
      "(('exon', '20'), 92)\n",
      "(('cells', 'overexpressing'), 91)\n",
      "(('dna', 'extracted'), 91)\n",
      "(('(figure', '4b)'), 91)\n",
      "(('mutations', 'one'), 91)\n",
      "(('lung', 'adenocarcinomas'), 91)\n",
      "(('tumor', 'suppression'), 91)\n",
      "(('transforming', 'potential'), 91)\n",
      "(('fusion', 'gene'), 91)\n",
      "(('vector', 'control'), 91)\n",
      "(('three', 'patients'), 90)\n",
      "(('(figure', '1c)'), 90)\n",
      "(('next', 'examined'), 90)\n",
      "(('figure', 'options'), 90)\n",
      "(('inhibition', 'cell'), 90)\n",
      "(('tumor', 'tissue'), 90)\n",
      "(('secondary', 'mutation'), 90)\n",
      "(('results', 'also'), 90)\n",
      "(('nih', '3t3'), 90)\n",
      "(('protein', 'expression,'), 90)\n",
      "(('quikchange', 'site-directed'), 90)\n",
      "(('exon', '18'), 90)\n",
      "(('dominant', 'negative'), 89)\n",
      "(('cancer', 'types'), 89)\n",
      "(('number', 'patients'), 89)\n",
      "(('truncating', 'mutations'), 89)\n",
      "(('breast', 'cancers'), 89)\n",
      "(('catalytic', 'activity'), 89)\n",
      "(('(figure', '2a)'), 89)\n",
      "(('cancer', 'risk'), 89)\n",
      "(('nuclear', 'localization'), 89)\n",
      "(('response', 'rate'), 89)\n",
      "(('egfr', 'mutation'), 89)\n",
      "(('santa', 'cruz'), 89)\n",
      "(('used', 'previously'), 89)\n",
      "(('samples', 'obtained'), 88)\n",
      "(('deletion', 'exon'), 88)\n",
      "(('increased', 'expression'), 88)\n",
      "(('open', 'figure'), 88)\n",
      "(('first', 'time'), 88)\n",
      "(('(fig', 'consistent'), 88)\n",
      "(('cell', 'culture'), 88)\n",
      "(('shown', 'fig'), 88)\n",
      "(('negative', 'control'), 88)\n",
      "(('mutations', 'shown'), 88)\n",
      "(('generated', 'using'), 88)\n",
      "(('class', '3'), 88)\n",
      "(('mutants', 'showed'), 87)\n",
      "(('transfected', 'wild-type'), 87)\n",
      "(('significant', 'differences'), 87)\n",
      "(('functional', 'effect'), 87)\n",
      "(('cell', 'survival'), 87)\n",
      "(('observations', 'suggest'), 87)\n",
      "(('complete', 'loss'), 87)\n",
      "(('mammalian', 'cells'), 87)\n",
      "(('mapk', 'pathway'), 87)\n",
      "(('two', 'distinct'), 87)\n",
      "(('risk', 'breast'), 87)\n",
      "(('significantly', 'different'), 87)\n",
      "(('mutant', 'receptors'), 87)\n",
      "(('much', 'higher'), 87)\n",
      "(('cruz', 'biotechnology,'), 87)\n",
      "(('exon', '14'), 87)\n",
      "(('activity', 'may'), 87)\n",
      "(('loss-of-function', 'mutations'), 86)\n",
      "(('mrna', 'expression'), 86)\n",
      "(('double', 'mutant'), 86)\n",
      "(('patient', 'samples'), 86)\n",
      "(('clinical', 'features'), 86)\n",
      "(('levels', 'phosphorylated'), 86)\n",
      "(('recently', 'identified'), 86)\n",
      "(('mutations', '(fig'), 86)\n",
      "(('mutations', 'appear'), 86)\n",
      "(('activate', 'transcription'), 86)\n",
      "(('high', 'level'), 86)\n",
      "(('mutant', 'egfr'), 86)\n",
      "(('leading', 'increased'), 86)\n",
      "(('variants', 'identified'), 86)\n",
      "(('peptide', 'binding'), 86)\n",
      "(('repeat', 'region'), 86)\n",
      "(('expression', 'mutant'), 85)\n",
      "(('mutation', '(fig'), 85)\n",
      "(('small', 'number'), 85)\n",
      "(('data', 'demonstrate'), 85)\n",
      "(('may', 'explain'), 85)\n",
      "(('prostate', 'cancer'), 85)\n",
      "(('compared', 'patients'), 85)\n",
      "(('current', 'study'), 85)\n",
      "(('xenograft', 'models'), 85)\n",
      "(('identified', 'novel'), 85)\n",
      "(('nuclear', 'protein'), 85)\n",
      "(('brca1', 'missense'), 85)\n",
      "(('one', 'case'), 84)\n",
      "(('mutations', 'described'), 84)\n",
      "(('mutations', '(table'), 84)\n",
      "(('mutation', 'frequency'), 84)\n",
      "(('6', 'figure'), 84)\n",
      "(('phosphatase', 'activity'), 84)\n",
      "(('conformational', 'change'), 84)\n",
      "(('full', 'figure'), 84)\n",
      "(('gene', 'mutation'), 84)\n",
      "(('variants', 'detected'), 84)\n",
      "(('known', 'functional'), 84)\n",
      "(('mutations', 'brca1'), 84)\n",
      "(('level', 'expression'), 83)\n",
      "(('data', 'provide'), 83)\n",
      "(('stem', 'cell'), 83)\n",
      "(('results', 'demonstrate'), 83)\n",
      "(('side', 'chain'), 83)\n",
      "(('solid', 'tumors'), 83)\n",
      "(('significantly', 'affect'), 83)\n",
      "(('3t3', 'cells'), 83)\n",
      "(('assay', 'assess'), 83)\n",
      "(('first', 'identified'), 83)\n",
      "(('within', 'exon'), 82)\n",
      "(('expression', 'constructs'), 82)\n",
      "(('studies', 'shown'), 82)\n",
      "(('mutations', 'known'), 82)\n",
      "(('analysis', 'showed'), 82)\n",
      "(('(figure', '4)'), 82)\n",
      "(('exon', '21'), 82)\n",
      "(('molecular', 'mechanisms'), 82)\n",
      "(('immunoblot', 'analysis'), 82)\n",
      "(('cancer', 'samples'), 82)\n",
      "(('mutation', 'one'), 82)\n",
      "(('mutant', 'wild-type'), 82)\n",
      "(('mutation', 'detected'), 82)\n",
      "(('mismatch', 'repair'), 82)\n",
      "(('drug', 'resistance'), 82)\n",
      "(('lines', 'harboring'), 82)\n",
      "(('clinical', 'trial'), 82)\n",
      "(('due', 'lack'), 82)\n",
      "(('jolla,', 'ca)'), 81)\n",
      "(('data', 'shown)'), 81)\n",
      "(('assays', 'performed'), 81)\n",
      "(('mutations', 'frequently'), 81)\n",
      "(('three-dimensional', 'structure'), 81)\n",
      "(('enzymatic', 'activity'), 81)\n",
      "(('strongly', 'suggest'), 81)\n",
      "(('breast', 'cancer,'), 81)\n",
      "(('luciferase', 'reporter'), 81)\n",
      "(('tyrosine', 'residues'), 81)\n",
      "(('three', 'mutations'), 81)\n",
      "(('two', 'additional'), 81)\n",
      "(('cooh', 'terminus'), 81)\n",
      "(('function', 'brca1'), 81)\n",
      "(('acute', 'myeloid'), 80)\n",
      "(('(fig', 'furthermore,'), 80)\n",
      "(('activation', 'cell'), 80)\n",
      "(('mutations', 'cancer'), 80)\n",
      "(('absence', 'presence'), 80)\n",
      "(('head', 'neck'), 80)\n",
      "(('here,', 'report'), 80)\n",
      "(('mutation', 'reported'), 80)\n",
      "(('tumors', '(fig'), 80)\n",
      "(('2', '3'), 80)\n",
      "(('stem', 'cells'), 80)\n",
      "(('showed', 'mutations'), 80)\n",
      "(('may', 'play'), 80)\n",
      "(('harboring', 'mutations'), 80)\n",
      "(('kras', 'mutations'), 80)\n",
      "(('active', 'conformation'), 80)\n",
      "(('clinical', 'data'), 80)\n",
      "(('discussion', 'here,'), 80)\n",
      "(('structural', 'functional'), 80)\n",
      "(('activation', 'domain'), 80)\n",
      "(('nsclc', 'cell'), 79)\n",
      "(('cells', 'infected'), 79)\n",
      "(('figure', '1)'), 79)\n",
      "(('catalytic', 'domain'), 79)\n",
      "(('variants', 'associated'), 79)\n",
      "(('respectively', '(fig'), 79)\n",
      "(('presence', 'mutations'), 79)\n",
      "(('patients', 'mutations'), 79)\n",
      "(('mutations', 'exons'), 79)\n",
      "(('resistance', 'mutations'), 79)\n",
      "(('listed', 'table'), 79)\n",
      "(('mutations', 'strongly'), 79)\n",
      "(('plays', 'critical'), 79)\n",
      "(('juxtamembrane', 'domain'), 79)\n",
      "(('kit', 'mutations'), 79)\n",
      "(('involved', 'cell'), 79)\n",
      "(('among', 'patients'), 78)\n",
      "(('nude', 'mice'), 78)\n",
      "(('functional', 'impact'), 78)\n",
      "(('provide', 'evidence'), 78)\n",
      "(('two', 'cases'), 78)\n",
      "(('that,', 'like'), 78)\n",
      "(('hydrogen', 'bond'), 78)\n",
      "(('also', 'tested'), 78)\n",
      "(('(figure', '3b)'), 78)\n",
      "(('breast', 'tumors'), 78)\n",
      "(('oncogenic', 'potential'), 78)\n",
      "(('polyclonal', 'antibody'), 78)\n",
      "(('two', 'groups'), 78)\n",
      "(('three', 'cell'), 78)\n",
      "(('two', 'independent'), 78)\n",
      "(('repair', 'proteins'), 78)\n",
      "(('protein', 'stability'), 78)\n",
      "(('vitro', 'kinase'), 78)\n",
      "(('identified', 'one'), 78)\n",
      "(('four', 'patients'), 78)\n",
      "(('revealed', 'presence'), 78)\n",
      "(('brct', 'variants'), 78)\n",
      "(('cancer', 'information'), 78)\n",
      "(('brca1', 'brct'), 78)\n",
      "(('table', 's1)'), 77)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('increased', 'sensitivity'), 77)\n",
      "(('mutant', 'alleles'), 77)\n",
      "(('affect', 'protein'), 77)\n",
      "(('survival', 'compared'), 77)\n",
      "(('increased', 'levels'), 77)\n",
      "(('monoclonal', 'antibody'), 77)\n",
      "(('splice', 'site'), 77)\n",
      "(('protein', 'dna'), 77)\n",
      "(('might', 'also'), 77)\n",
      "(('associated', 'poor'), 77)\n",
      "(('poor', 'prognosis'), 77)\n",
      "(('mutation', 'showed'), 77)\n",
      "(('akt', 'phosphorylation'), 77)\n",
      "(('ba/f3', 'cell'), 77)\n",
      "(('primary', 'tumor'), 77)\n",
      "(('kinase', 'activation'), 77)\n",
      "(('domain', 'mutant'), 77)\n",
      "(('information', 'core'), 77)\n",
      "(('using', 'following'), 76)\n",
      "(('hela', 'cells'), 76)\n",
      "(('sequencing', 'analysis'), 76)\n",
      "(('obtained', 'using'), 76)\n",
      "(('point', 'mutations,'), 76)\n",
      "(('exon', '12'), 76)\n",
      "(('human', 'tumors'), 76)\n",
      "(('domain', '(fig'), 76)\n",
      "(('sequence', 'variants'), 76)\n",
      "(('3', 'table'), 76)\n",
      "(('kinase', 'assays'), 76)\n",
      "(('nsclc', 'patients'), 76)\n",
      "(('egfr', 'tyrosine'), 76)\n",
      "(('kinase', 'mutations'), 76)\n",
      "(('(santa', 'cruz'), 76)\n",
      "(('acute', 'lymphoblastic'), 76)\n",
      "(('melanoma', 'cell'), 76)\n",
      "(('exon', '17'), 76)\n",
      "(('brct', 'domain'), 76)\n",
      "(('domain', 'brca1'), 76)\n",
      "((\"manufacturer's\", 'instructions'), 75)\n",
      "(('fluorescent', 'protein'), 75)\n",
      "(('identified', 'mutations'), 75)\n",
      "(('(table', '1'), 75)\n",
      "(('activity', 'vitro'), 75)\n",
      "(('found', 'two'), 75)\n",
      "(('lung', 'cancer,'), 75)\n",
      "(('low', 'level'), 75)\n",
      "(('mutation', 'associated'), 75)\n",
      "(('brca1', 'protein'), 75)\n",
      "(('egfr', 'gene'), 75)\n",
      "(('wild-type', 'egfr'), 75)\n",
      "(('confers', 'resistance'), 75)\n",
      "(('mutations', 'study'), 75)\n",
      "(('vitro', 'studies'), 75)\n",
      "(('rabbit', 'polyclonal'), 75)\n",
      "(('resistance', 'imatinib'), 75)\n",
      "(('used', 'positive'), 75)\n",
      "(('kit', 'exon'), 75)\n",
      "(('variants', 'also'), 75)\n",
      "(('mutations', 'destabilize'), 75)\n",
      "(('brca1', 'tumor'), 75)\n",
      "(('established', 'human'), 75)\n",
      "(('figure', '7'), 74)\n",
      "(('(fig', 'data'), 74)\n",
      "(('construct', 'containing'), 74)\n",
      "(('cells', 'mutant'), 74)\n",
      "(('ectopically', 'expressed'), 74)\n",
      "(('higher', 'frequency'), 74)\n",
      "(('recently', 'shown'), 74)\n",
      "(('role', 'response'), 74)\n",
      "(('ability', 'activate'), 74)\n",
      "(('interaction', 'surfaces'), 74)\n",
      "(('multiple', 'sequence'), 74)\n",
      "(('exon', '8'), 73)\n",
      "(('powerpoint', 'slide'), 73)\n",
      "(('cells', 'led'), 73)\n",
      "(('possible', 'explanation'), 73)\n",
      "(('mutations', 'tend'), 73)\n",
      "(('cells', 'containing'), 73)\n",
      "(('activation', 'mapk'), 73)\n",
      "(('exon', '3'), 73)\n",
      "(('cycle', 'regulation'), 73)\n",
      "(('colon', 'cancer'), 73)\n",
      "(('wild', 'type,'), 73)\n",
      "(('mutant', 'cell'), 73)\n",
      "(('gene', 'associated'), 73)\n",
      "(('gefitinib', 'erlotinib'), 73)\n",
      "(('egfr', 'kinase'), 73)\n",
      "(('examined', 'effect'), 73)\n",
      "(('variants', 'showed'), 73)\n",
      "(('genetic', 'data'), 73)\n",
      "(('receptor', 'activation'), 73)\n",
      "(('activity', 'compared'), 73)\n",
      "(('found', 'human'), 73)\n",
      "(('negative', 'controls'), 73)\n",
      "(('sequencing', 'efforts'), 73)\n",
      "(('activation', 'activity'), 73)\n",
      "(('another', 'signaling'), 73)\n",
      "(('significant', 'reduction'), 72)\n",
      "(('deletion', 'mutants'), 72)\n",
      "(('compared', 'wild'), 72)\n",
      "(('green', 'fluorescent'), 72)\n",
      "(('could', 'also'), 72)\n",
      "(('previous', 'report'), 72)\n",
      "(('erk', 'phosphorylation'), 72)\n",
      "(('germline', 'mutation'), 72)\n",
      "(('survival', 'patients'), 72)\n",
      "(('gene', 'copy'), 72)\n",
      "(('similar', 'wild-type'), 72)\n",
      "(('used', 'generate'), 72)\n",
      "(('several', 'proteins'), 72)\n",
      "(('two', 'novel'), 72)\n",
      "(('c', 'terminus'), 72)\n",
      "(('6,', '7,'), 72)\n",
      "(('results', 'generation'), 72)\n",
      "(('targeted', 'therapy'), 72)\n",
      "(('phosphorylation', 'status'), 72)\n",
      "(('mutants', 'bind'), 72)\n",
      "(('susceptibility', 'gene'), 72)\n",
      "(('sequence-specific', 'dna'), 72)\n",
      "(('whereas', 'wild-type'), 72)\n",
      "(('domain', 'linked'), 72)\n",
      "(('domains', 'brca1'), 72)\n",
      "(('kit', '(stratagene,'), 71)\n",
      "(('previously', 'observed'), 71)\n",
      "(('expressing', 'cells'), 71)\n",
      "(('suppressor', 'gene'), 71)\n",
      "(('mrna', 'levels'), 71)\n",
      "(('genes', 'involved'), 71)\n",
      "(('4', '5'), 71)\n",
      "(('plasma', 'membrane'), 71)\n",
      "(('frequently', 'observed'), 71)\n",
      "(('examine', 'whether'), 71)\n",
      "(('cellular', 'proliferation'), 71)\n",
      "(('although', 'cannot'), 71)\n",
      "(('domain', 'human'), 71)\n",
      "(('activity', 'transcription'), 71)\n",
      "(('brct', 'domains'), 71)\n",
      "(('lymphoblastic', 'leukemia'), 70)\n",
      "(('constitutive', 'phosphorylation'), 70)\n",
      "(('receptor', '(egfr)'), 70)\n",
      "(('identified', 'human'), 70)\n",
      "(('wild-type', 'protein'), 70)\n",
      "(('relatively', 'high'), 70)\n",
      "(('carcinoma', 'cell'), 70)\n",
      "(('recently', 'published'), 70)\n",
      "(('cells', '(a)'), 70)\n",
      "(('drug', 'sensitivity'), 70)\n",
      "(('functional', 'importance'), 70)\n",
      "(('in-frame', 'deletions'), 70)\n",
      "(('increasing', 'concentrations'), 70)\n",
      "(('mass', 'spectrometry'), 70)\n",
      "(('mutations', 'here,'), 70)\n",
      "(('here,', 'present'), 70)\n",
      "(('next', 'generation'), 70)\n",
      "(('empty', 'vector,'), 70)\n",
      "(('pull-down', 'assay'), 70)\n",
      "(('studies', 'attempted'), 70)\n",
      "(('c-terminal', 'region'), 69)\n",
      "(('showed', 'reduced'), 69)\n",
      "(('full', 'table'), 69)\n",
      "(('functional', 'effects'), 69)\n",
      "(('present', 'first'), 69)\n",
      "(('previously', 'published'), 69)\n",
      "(('united', 'states'), 69)\n",
      "(('region', 'also'), 69)\n",
      "(('according', 'manufacturers'), 69)\n",
      "(('results', 'showed'), 69)\n",
      "(('mutant', 'cells'), 69)\n",
      "(('cancer', 'risks'), 69)\n",
      "(('exon', '1'), 69)\n",
      "(('expression', 'plasmids'), 69)\n",
      "(('(a)', 'schematic'), 69)\n",
      "(('clinical', 'responses'), 69)\n",
      "(('egfr', 'mutations,'), 69)\n",
      "(('mutations', 'lung'), 69)\n",
      "(('(cell', 'signaling'), 69)\n",
      "(('mutations', 'tyrosine'), 69)\n",
      "(('cos-7', 'cells'), 69)\n",
      "(('sensitive', 'mutations'), 69)\n",
      "(('imatinib', 'resistance'), 69)\n",
      "(('number', 'variants'), 69)\n",
      "(('consistent', 'results'), 69)\n",
      "(('linked', 'disease'), 69)\n",
      "(('suppression', 'brca1'), 69)\n",
      "(('similar', 'observed'), 68)\n",
      "(('19,', '20,'), 68)\n",
      "(('full', 'size'), 68)\n",
      "(('ba/f3', 'cells,'), 68)\n",
      "(('tumor', 'specimens'), 68)\n",
      "(('gene', 'encoding'), 68)\n",
      "(('potently', 'inhibited'), 68)\n",
      "(('expression', 'wild-type'), 68)\n",
      "(('significantly', 'decreased'), 68)\n",
      "(('mutations', 'kinase'), 68)\n",
      "(('allowed', 'us'), 68)\n",
      "(('mutations', 'using'), 68)\n",
      "(('active', 'site'), 68)\n",
      "(('mutations', 'show'), 68)\n",
      "(('mutations', 'egfr'), 68)\n",
      "(('kinase', 'inhibitor,'), 68)\n",
      "(('also', 'induced'), 68)\n",
      "(('phosphorylated', 'protein'), 68)\n",
      "(('structural', 'modeling'), 68)\n",
      "(('exon', '13'), 68)\n",
      "(('genotoxic', 'stress'), 68)\n",
      "(('domain', '(residues'), 68)\n",
      "(('mutations', 'activate'), 68)\n",
      "(('variants', 'within'), 68)\n",
      "(('peptide', 'containing'), 68)\n",
      "(('cancer', 'using'), 68)\n",
      "(('domain', 'variants'), 68)\n",
      "(('9', 'mutations'), 68)\n",
      "(('brct', 'repeats'), 68)\n",
      "(('mutations', 'deletions'), 67)\n",
      "(('ligand-independent', 'activation'), 67)\n",
      "(('play', 'important'), 67)\n",
      "(('mutants,', 'wild-type'), 67)\n",
      "(('studies', 'reported'), 67)\n",
      "(('hek293t', 'cells'), 67)\n",
      "(('acid', 'change'), 67)\n",
      "(('functional', 'studies'), 67)\n",
      "(('gel', 'electrophoresis'), 67)\n",
      "(('atp', 'binding'), 67)\n",
      "(('(see', 'supplementary'), 67)\n",
      "(('associated', 'cancer'), 67)\n",
      "(('phosphorylation', 'downstream'), 67)\n",
      "(('reaction', '(pcr)'), 67)\n",
      "(('patients', 'lung'), 67)\n",
      "(('resistance', 'mutation'), 67)\n",
      "(('particular', 'interest'), 67)\n",
      "(('disease', 'using'), 67)\n",
      "(('activity', 'observed'), 67)\n",
      "(('vector', 'used'), 67)\n",
      "(('protein', 'folding'), 67)\n",
      "(('baf3', 'cells'), 66)\n",
      "(('observed', 'cells'), 66)\n",
      "(('size', 'image'), 66)\n",
      "(('human', 'wild-type'), 66)\n",
      "(('mutants', '(figure'), 66)\n",
      "(('view', 'inlineview'), 66)\n",
      "(('mutation', 'located'), 66)\n",
      "(('mutations', 'either'), 66)\n",
      "(('tumor', 'formation'), 66)\n",
      "(('phosphorylation', 'levels'), 66)\n",
      "(('studies', 'suggest'), 66)\n",
      "(('study', 'analyzed'), 66)\n",
      "(('cells', 'resulted'), 66)\n",
      "(('also', 'associated'), 66)\n",
      "(('found', 'patients'), 66)\n",
      "(('resistance', 'egfr'), 66)\n",
      "(('progression-free', 'survival'), 66)\n",
      "(('used', 'template'), 66)\n",
      "(('mutation', 'occurred'), 66)\n",
      "(('functionally', 'characterized'), 66)\n",
      "(('studies', 'involving'), 66)\n",
      "(('partner', 'proteins'), 66)\n",
      "(('24', 'hours'), 66)\n",
      "(('early-onset', 'breast'), 66)\n",
      "(('variants', 'analyzed'), 66)\n",
      "(('stability', 'protein'), 66)\n",
      "(('tested', 'using'), 66)\n",
      "(('time', 'diagnosis'), 66)\n",
      "(('cell', 'systems'), 66)\n",
      "(('(figure', '3)'), 65)\n",
      "(('associated', 'increased'), 65)\n",
      "(('whether', 'mutations'), 65)\n",
      "(('also', 'detected'), 65)\n",
      "(('studies', 'show'), 65)\n",
      "(('genome', 'atlas'), 65)\n",
      "(('associated', 'mutations'), 65)\n",
      "(('inhibitory', 'effect'), 65)\n",
      "(('increased', 'susceptibility'), 65)\n",
      "(('observed', 'patients'), 65)\n",
      "(('mutations', 'response'), 65)\n",
      "(('effort', 'made'), 65)\n",
      "(('mutations', 'tested'), 65)\n",
      "(('variants', 'using'), 65)\n",
      "(('(figure', '5)'), 65)\n",
      "(('given', 'supplementary'), 65)\n",
      "(('bioinformatics', 'analysis'), 65)\n",
      "(('subcellular', 'localization'), 65)\n",
      "(('regulate', 'expression'), 65)\n",
      "(('activity', 'several'), 65)\n",
      "(('variants', 'used'), 65)\n",
      "(('c2', 'domain'), 65)\n",
      "(('indicate', 'that,'), 65)\n",
      "(('fold', 'stability'), 65)\n",
      "(('region', 'protein'), 65)\n",
      "(('brct', 'domains,'), 65)\n",
      "(('brct', 'domain,'), 65)\n",
      "(('cells', '(supplementary'), 64)\n",
      "(('domain', 'interacts'), 64)\n",
      "(('b', 'cell'), 64)\n",
      "(('activity', 'wild-type'), 64)\n",
      "(('broad', 'spectrum'), 64)\n",
      "(('mutations', 'gene'), 64)\n",
      "(('relative', 'wild-type'), 64)\n",
      "(('wild-type', 'control'), 64)\n",
      "(('gastric', 'cancer'), 64)\n",
      "(('associated', 'large'), 64)\n",
      "(('mutant', 'also'), 64)\n",
      "(('also', 'examined'), 64)\n",
      "(('statistical', 'significance'), 64)\n",
      "(('cells', 'used'), 64)\n",
      "(('selective', 'advantage'), 64)\n",
      "(('taken', 'together'), 64)\n",
      "(('homologous', 'recombination'), 64)\n",
      "(('variants', 'uncertain'), 64)\n",
      "(('primer', 'pairs'), 64)\n",
      "(('four', 'cell'), 64)\n",
      "(('mutations', 'involving'), 64)\n",
      "(('approximately', '50%'), 64)\n",
      "(('cancer', '(nsclc)'), 64)\n",
      "(('patients', 'egfr'), 64)\n",
      "(('hybridization', '(fish)'), 64)\n",
      "(('lines', 'stably'), 64)\n",
      "(('colleagues', '(35)'), 64)\n",
      "(('thus', 'far'), 64)\n",
      "(('large', 'set'), 64)\n",
      "(('assays', 'assess'), 64)\n",
      "(('santa', 'cruz,'), 64)\n",
      "(('assess', 'ability'), 64)\n",
      "(('proteins', 'dna'), 64)\n",
      "(('functional', 'defects'), 64)\n",
      "(('conserved', 'regions'), 64)\n",
      "(('salt', 'bridge'), 63)\n",
      "(('novel', 'mutations'), 63)\n",
      "(('comparative', 'genomic'), 63)\n",
      "(('mutants', 'also'), 63)\n",
      "(('functional', 'role'), 63)\n",
      "(('similar', 'levels'), 63)\n",
      "(('mutations', 'alter'), 63)\n",
      "(('binding', 'affinity'), 63)\n",
      "(('protein', '(fig'), 63)\n",
      "(('used', 'study'), 63)\n",
      "(('exon', '2'), 63)\n",
      "(('risks', 'associated'), 63)\n",
      "(('pcr', 'amplification'), 63)\n",
      "(('flow', 'cytometry'), 63)\n",
      "(('naturally', 'occurring'), 63)\n",
      "(('4', 'patients'), 63)\n",
      "(('recent', 'report'), 63)\n",
      "(('phase', 'iii'), 63)\n",
      "(('patients', 'tumors'), 63)\n",
      "(('mutations', 'identified,'), 63)\n",
      "(('cells', 'cells'), 63)\n",
      "(('located', 'chromosome'), 63)\n",
      "(('wild-type', 'p53'), 63)\n",
      "(('al', 'ability'), 63)\n",
      "(('uncovered', 'large'), 63)\n",
      "(('unclassified', 'missense'), 63)\n",
      "(('coding', 'exons'), 62)\n",
      "(('point', 'mutants'), 62)\n",
      "(('mutations', 'showed'), 62)\n",
      "(('growth', 'survival'), 62)\n",
      "(('transforming', 'ability'), 62)\n",
      "(('treatment', 'patients'), 62)\n",
      "(('tyrosine', 'kinases,'), 62)\n",
      "(('tumors', 'analyzed'), 62)\n",
      "(('gene', 'amplification,'), 62)\n",
      "(('domain', 'structure'), 62)\n",
      "(('may', 'reflect'), 62)\n",
      "(('mapk', 'signaling'), 62)\n",
      "(('cells,', 'suggesting'), 62)\n",
      "(('germ', 'line'), 62)\n",
      "(('normal', 'cells'), 62)\n",
      "(('factors', 'involved'), 62)\n",
      "(('(figure', '3c)'), 62)\n",
      "(('screening', 'breast'), 62)\n",
      "(('table', '4'), 62)\n",
      "(('ii', 'trial'), 62)\n",
      "(('reported', 'mutations'), 62)\n",
      "(('absence', 'ligand'), 62)\n",
      "(('cells', 'expression'), 62)\n",
      "(('mutation', 'present'), 62)\n",
      "(('fail', 'bind'), 62)\n",
      "(('cell', 'transformation'), 62)\n",
      "(('defects', 'associated'), 62)\n",
      "(('binding', 'pocket'), 62)\n",
      "(('strongly', 'linked'), 62)\n",
      "(('see', 'ref'), 62)\n",
      "(('tested', 'binding'), 62)\n",
      "(('brca1', 'thought'), 62)\n",
      "(('regions', 'brca1,'), 62)\n",
      "(('wt', 'brca1'), 62)\n",
      "(('mutation', '(figure'), 61)\n",
      "(('exclude', 'possibility'), 61)\n",
      "(('cells', 'without'), 61)\n",
      "(('loss', 'heterozygosity'), 61)\n",
      "(('signaling', 'pathways,'), 61)\n",
      "(('cells', 'vitro'), 61)\n",
      "(('target', 'gene'), 61)\n",
      "(('expression', 'analysis'), 61)\n",
      "(('activity', 'mutant'), 61)\n",
      "(('activity', '(figure'), 61)\n",
      "(('compared', 'normal'), 61)\n",
      "(('one', 'family'), 61)\n",
      "(('vector', 'containing'), 61)\n",
      "(('may', 'lead'), 61)\n",
      "(('stable', 'expression'), 61)\n",
      "(('top', 'page'), 61)\n",
      "(('studies', 'showed'), 61)\n",
      "(('assess', 'whether'), 61)\n",
      "(('entire', 'coding'), 61)\n",
      "(('mutation', 'carriers'), 61)\n",
      "(('wild-type', 'receptor'), 61)\n",
      "(('exome', 'sequencing'), 61)\n",
      "(('(n', '5)'), 61)\n",
      "(('showed', 'strong'), 61)\n",
      "(('germ-line', 'mutations'), 61)\n",
      "(('deletion', 'mutation'), 61)\n",
      "(('accession', 'number'), 61)\n",
      "(('2', 'patients'), 61)\n",
      "(('analysis', 'egfr'), 61)\n",
      "(('mutations', 'primary'), 61)\n",
      "(('residues', 'conserved'), 61)\n",
      "(('partial', 'response'), 61)\n",
      "(('transduction', 'pathways'), 61)\n",
      "(('transfected', 'empty'), 61)\n",
      "(('resulted', 'constitutive'), 61)\n",
      "(('(fig', 'although'), 61)\n",
      "(('vectors', 'encoding'), 61)\n",
      "(('response', 'cells'), 61)\n",
      "(('variants', 'strong'), 61)\n",
      "(('mutation', 'disease'), 61)\n",
      "(('protein', 'multiple'), 61)\n",
      "(('brct', 'repeat'), 61)\n",
      "(('expression', 'reporter'), 61)\n",
      "(('gene', 'brca1'), 61)\n",
      "(('phosphopeptide', 'recognition'), 61)\n",
      "(('containing', 'brca1'), 61)\n",
      "(('supplementary', 'fig'), 60)\n",
      "(('elevated', 'levels'), 60)\n",
      "(('open', 'reading'), 60)\n",
      "(('ca,', 'usa)'), 60)\n",
      "(('significantly', 'associated'), 60)\n",
      "(('patients', 'advanced'), 60)\n",
      "(('mutations', 'occurred'), 60)\n",
      "(('recurrent', 'mutations'), 60)\n",
      "(('homozygous', 'deletion'), 60)\n",
      "(('transiently', 'expressed'), 60)\n",
      "(('dna', 'polymerase'), 60)\n",
      "(('mechanisms', 'underlying'), 60)\n",
      "(('transmembrane', 'domain'), 60)\n",
      "(('5,', '6,'), 60)\n",
      "(('(fig', 'interestingly,'), 60)\n",
      "(('genes,', 'including'), 60)\n",
      "(('may', 'important'), 60)\n",
      "(('egfr', 'mutant'), 60)\n",
      "(('phosphorylation', 'akt'), 60)\n",
      "(('egfr', 'exon'), 60)\n",
      "(('tumor', 'regression'), 60)\n",
      "(('could', 'affect'), 60)\n",
      "(('kinase', 'inhibitors,'), 60)\n",
      "(('analysis', 'based'), 60)\n",
      "(('transient', 'expression'), 60)\n",
      "(('quantitative', 'pcr'), 60)\n",
      "(('genetic', 'screening'), 60)\n",
      "(('p53', 'dna'), 60)\n",
      "(('likely', 'affect'), 60)\n",
      "(('fig', 'hand,'), 60)\n",
      "(('binding', 'activities'), 60)\n",
      "(('seven', 'mutants,'), 60)\n",
      "(('sensitive', 'imatinib'), 60)\n",
      "(('lee', 'et'), 60)\n",
      "(('protein', 'targets'), 60)\n",
      "(('results', 'compared'), 60)\n",
      "(('activation', 'signaling'), 60)\n",
      "(('compared', 'experiments'), 60)\n",
      "(('pilot', 'study'), 60)\n",
      "(('protein', 'documented'), 60)\n",
      "(('peptide', 'motif'), 60)\n",
      "(('ability', 'brca1'), 60)\n",
      "(('brca1', 'amino'), 60)\n",
      "(('used', 'limited'), 60)\n",
      "(('brct', 'missense'), 60)\n",
      "(('domain', 'introduction'), 60)\n",
      "(('core', 'database'), 60)\n",
      "(('sequence', 'brca1'), 60)\n",
      "(('also', 'functionally'), 60)\n",
      "(('one', 'mutation'), 59)\n",
      "(('mutation', 'results'), 59)\n",
      "(('functional', 'characterization'), 59)\n",
      "(('(table', '3)'), 59)\n",
      "(('mutations', 'common'), 59)\n",
      "(('heterozygous', 'mutations'), 59)\n",
      "(('protein', 'fold'), 59)\n",
      "(('significantly', 'less'), 59)\n",
      "(('also', 'seen'), 59)\n",
      "(('1', '1'), 59)\n",
      "(('cells', 'grown'), 59)\n",
      "(('high', 'expression'), 59)\n",
      "(('much', 'lower'), 59)\n",
      "(('2,', '3,'), 59)\n",
      "(('studies', 'also'), 59)\n",
      "(('using', 'antibody'), 59)\n",
      "(('previous', 'study'), 59)\n",
      "(('gene', 'found'), 59)\n",
      "(('western', 'blots'), 59)\n",
      "(('commonly', 'found'), 59)\n",
      "(('biochemical', 'properties'), 59)\n",
      "(('significantly', 'longer'), 59)\n",
      "(('disease,', 'whereas'), 59)\n",
      "(('much', 'less'), 59)\n",
      "(('genes', 'p21'), 59)\n",
      "(('brca1', 'brca2'), 59)\n",
      "(('functional', 'assays'), 59)\n",
      "(('relatively', 'unstable'), 59)\n",
      "(('detected', 'date'), 59)\n",
      "(('variants', 'specifically'), 59)\n",
      "(('acid', 'nuclear'), 59)\n",
      "(('control', 'results'), 59)\n",
      "(('specifically', 'interacts'), 59)\n",
      "(('modulate', 'activity'), 59)\n",
      "(('tend', 'cluster'), 59)\n",
      "(('variants', '(14)'), 59)\n",
      "(('controls', 'examined'), 59)\n",
      "(('study', 'involving'), 59)\n",
      "(('brca1', 'activity'), 59)\n",
      "(('protein', 'folding,'), 59)\n",
      "(('phosphopeptide', 'binding'), 59)\n",
      "(('risk', 'assessment'), 59)\n",
      "(('number', 'brca1'), 59)\n",
      "(('distinct', 'brca1'), 59)\n",
      "(('based', 'multiple'), 59)\n",
      "(('bind', 'specifically'), 59)\n",
      "(('human', 'population'), 59)\n",
      "(('deletion', 'mutant'), 58)\n",
      "(('result', 'loss'), 58)\n",
      "(('(fig', 'expression'), 58)\n",
      "(('quikchange', 'ii'), 58)\n",
      "(('also', 'inhibited'), 58)\n",
      "(('(figure', '2c)'), 58)\n",
      "(('braf', 'mutations'), 58)\n",
      "(('would', 'expected'), 58)\n",
      "(('sought', 'determine'), 58)\n",
      "(('significant', 'association'), 58)\n",
      "(('(see', 'fig'), 58)\n",
      "(('results', 'support'), 58)\n",
      "(('showed', 'significantly'), 58)\n",
      "(('cells', 'per'), 58)\n",
      "(('methods', 'results'), 58)\n",
      "(('sufficient', 'data'), 58)\n",
      "(('line', 'expressing'), 58)\n",
      "(('lines', 'also'), 58)\n",
      "(('studies', 'performed'), 58)\n",
      "(('7,', '8'), 58)\n",
      "(('provides', 'useful'), 58)\n",
      "(('shown', 'table'), 58)\n",
      "(('inhibited', 'phosphorylation'), 58)\n",
      "(('tumors', 'harbored'), 58)\n",
      "(('egfr', 'mutants'), 58)\n",
      "(('patients', 'received'), 58)\n",
      "(('inhibitor', '(tki)'), 58)\n",
      "(('identified', 'study'), 58)\n",
      "(('nonsmall', 'cell'), 58)\n",
      "(('stable', 'cell'), 58)\n",
      "(('uncertain', 'clinical'), 58)\n",
      "(('kit', '(stratagene)'), 58)\n",
      "(('genetic', 'diversity'), 58)\n",
      "(('therefore', 'included'), 58)\n",
      "(('secondary', 'mutations'), 58)\n",
      "(('here,', 'use'), 58)\n",
      "(('inhibitory', 'concentration'), 58)\n",
      "(('genes', 'encoding'), 58)\n",
      "(('cannot', 'rule'), 58)\n",
      "(('results', 'agreement'), 58)\n",
      "(('exon', '11,'), 58)\n",
      "(('transcriptional', 'assays'), 58)\n",
      "(('binding', 'phosphorylated'), 58)\n",
      "(('melanoma', 'cells'), 58)\n",
      "(('c-terminal', 'domain'), 58)\n",
      "(('mutations', 'kit'), 58)\n",
      "(('variants', 'predicted'), 58)\n",
      "(('distinct', 'missense'), 58)\n",
      "(('classes', 'mutations'), 58)\n",
      "(('cells', 'genotoxic'), 58)\n",
      "(('recognition', 'critical'), 58)\n",
      "(('complete', 'set'), 58)\n",
      "(('phosphorylated', 'version'), 58)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('data', 'linking'), 58)\n",
      "(('risk', 'estimates'), 58)\n",
      "(('individuals', 'family'), 58)\n",
      "(('showed', 'certain'), 58)\n",
      "(('limited', 'proteolysis'), 58)\n",
      "(('relay', 'signals'), 58)\n",
      "(('target', 'sequence'), 58)\n",
      "(('analysis', 'transcriptional'), 58)\n",
      "(('10', 'variants'), 58)\n",
      "(('protein', 'plays'), 58)\n",
      "(('g129e,', 'r130g,'), 58)\n",
      "(('detailed', 'study'), 58)\n",
      "(('determination', 'protein'), 58)\n",
      "(('m1775r', 'a1708e'), 58)\n",
      "(('family-based', 'data,'), 58)\n",
      "(('25', 'brca1'), 58)\n",
      "(('analyzed', 'peptide'), 58)\n",
      "(('well', 'peptide'), 58)\n",
      "(('brca1-brct', '(1,6461,858)'), 58)\n",
      "(('examined', 'time'), 58)\n",
      "(('peptides', 'shown'), 58)\n",
      "(('brct', 'well'), 58)\n",
      "(('specifically', 'pser-x-x-phecontaining'), 58)\n",
      "(('brct', 'mutants'), 58)\n",
      "(('using', 'family-based'), 58)\n",
      "(('m1775r', 'carried'), 58)\n",
      "(('peptide', 'bindingdeficient'), 58)\n",
      "(('ring', 'brct'), 58)\n",
      "(('(1,6461,858)', 'positive'), 58)\n",
      "(('m1775r', 'y1853x'), 58)\n",
      "(('control,', 'm1775r'), 58)\n",
      "(('(1,35),', 'fail'), 58)\n",
      "(('shown', 'pilot'), 58)\n",
      "(('wild-type', 'brca1-brct'), 58)\n",
      "(('many', 'brca1'), 58)\n",
      "(('assays', 'brca1'), 58)\n",
      "(('classification', 'brca1'), 58)\n",
      "(('experiments', 'wild-type'), 58)\n",
      "(('supplementary', 'materials'), 58)\n",
      "(('m1775r', '(1,35),'), 58)\n",
      "(('control', 'variants'), 58)\n",
      "(('carried', 'parallel'), 58)\n",
      "(('activities', '25'), 58)\n",
      "(('pser-x-x-phecontaining', 'peptides'), 58)\n",
      "(('variants', 'a1708e'), 58)\n",
      "(('a1708e', 'm1775r'), 58)\n",
      "(('target', 'brca1'), 58)\n",
      "(('a1708e', 'negative'), 58)\n",
      "(('data,', 'a1708e'), 58)\n",
      "(('bindingdeficient', 'variants'), 58)\n",
      "(('mutation', 'screening'), 57)\n",
      "(('mutations', 'would'), 57)\n",
      "(('transfected', 'plasmids'), 57)\n",
      "(('cells', 'wild-type'), 57)\n",
      "(('non-small', 'cell'), 57)\n",
      "(('erk', 'activation'), 57)\n",
      "(('exon', '10'), 57)\n",
      "(('gain', 'function'), 57)\n",
      "(('(table', 's1)'), 57)\n",
      "(('indicates', 'functional'), 57)\n",
      "(('also', 'show'), 57)\n",
      "(('vector', 'alone'), 57)\n",
      "(('investigate', 'role'), 57)\n",
      "(('based', 'findings,'), 57)\n",
      "(('results', 'consistent'), 57)\n",
      "(('study,', 'investigated'), 57)\n",
      "(('stability', 'p53'), 57)\n",
      "(('able', 'show'), 57)\n",
      "(('whereas', 'expression'), 57)\n",
      "(('induce', 'apoptosis'), 57)\n",
      "(('mutation', 'egfr'), 57)\n",
      "(('various', 'concentrations'), 57)\n",
      "(('response', 'patients'), 57)\n",
      "(('cells', 'bearing'), 57)\n",
      "(('table', 's2)'), 57)\n",
      "(('phosphorylation', '(figure'), 57)\n",
      "(('provides', 'important'), 57)\n",
      "(('domain', 'first'), 57)\n",
      "(('assays', 'using'), 57)\n",
      "(('based', 'genetic'), 57)\n",
      "(('data', 'large'), 57)\n",
      "(('variants', 'many'), 57)\n",
      "(('one', 'mechanism'), 57)\n",
      "(('population', 'risk'), 57)\n",
      "(('control', 't7'), 57)\n",
      "(('gain-of-function', 'mutation'), 57)\n",
      "(('destabilize', 'protein'), 57)\n",
      "(('dna', 'helicase,'), 57)\n",
      "(('pi3k', 'pathway'), 57)\n",
      "(('pdgfra', 'mutations'), 57)\n",
      "(('whereas', '10'), 57)\n",
      "(('study', 'variants'), 57)\n",
      "(('vitro', 'protein'), 57)\n",
      "(('contribute', 'ability'), 57)\n",
      "(('strong', 'functional'), 57)\n",
      "(('repeat', 'regions,'), 57)\n",
      "(('yeast', 'cell'), 57)\n",
      "(('human', 'yeast'), 57)\n",
      "(('controls', 'activity'), 57)\n",
      "(('domain', 'indicate'), 57)\n",
      "(('variant', 'set'), 57)\n",
      "(('used', 'alignment'), 57)\n",
      "(('first', 'comprehensive'), 57)\n",
      "(('variants', 'corresponding'), 57)\n",
      "(('signals', 'generated'), 57)\n",
      "(('also', 'act'), 57)\n",
      "(('domain,', 'relatively'), 57)\n",
      "(('(wt)', 'used'), 57)\n",
      "(('1,859', '(wt)'), 57)\n",
      "(('brca1', 'genes'), 57)\n",
      "(('use', 'biochemical'), 57)\n",
      "(('y1853x', 'negative'), 57)\n",
      "(('activities', 'brca1'), 57)\n",
      "(('thought', 'act'), 57)\n",
      "(('time', 'transcriptional'), 57)\n",
      "(('interacts', 'phosphorylated'), 57)\n",
      "(('brca1', 'regulate'), 57)\n",
      "(('additional', 'sequence'), 57)\n",
      "(('motif,', 'known'), 57)\n",
      "(('ability', 'variants'), 57)\n",
      "(('disease', 'risk'), 57)\n",
      "(('brct', 'variant'), 57)\n",
      "(('assays', 'construct'), 57)\n",
      "(('within', 'breast'), 57)\n",
      "(('signaling', 'protein,'), 57)\n",
      "(('variants', 'm1775r'), 57)\n",
      "(('assays,', 'previously'), 57)\n",
      "(('acids', '1,646'), 57)\n",
      "(('nine', 'mammalian'), 57)\n",
      "(('1,646', '1,859'), 57)\n",
      "(('increased', 'cell'), 56)\n",
      "(('mutations', 'predicted'), 56)\n",
      "(('deletions', 'found'), 56)\n",
      "(('found', 'one'), 56)\n",
      "(('decreased', 'expression'), 56)\n",
      "(('(fig', 'notably,'), 56)\n",
      "(('plasmids', 'encoding'), 56)\n",
      "(('genomic', 'hybridization'), 56)\n",
      "(('3', '4'), 56)\n",
      "(('mutation', 'rate'), 56)\n",
      "(('bladder', 'cancer'), 56)\n",
      "(('cellular', 'transformation'), 56)\n",
      "(('table', 'options'), 56)\n",
      "(('clear', 'cell'), 56)\n",
      "(('limited', 'number'), 56)\n",
      "(('cells', 'figure'), 56)\n",
      "(('(not', 'shown)'), 56)\n",
      "(('play', 'critical'), 56)\n",
      "(('asked', 'whether'), 56)\n",
      "(('central', 'role'), 56)\n",
      "(('lower', 'levels'), 56)\n",
      "(('mutagenesis', 'using'), 56)\n",
      "(('luciferase', 'activity'), 56)\n",
      "(('studies', 'demonstrated'), 56)\n",
      "(('similar', 'seen'), 56)\n",
      "(('mutation', 'might'), 56)\n",
      "(('reading', 'frame'), 56)\n",
      "(('suppressor', 'function'), 56)\n",
      "(('response', 'gefitinib'), 56)\n",
      "(('tk', 'domain'), 56)\n",
      "(('domain', 'egfr'), 56)\n",
      "(('exons', '18'), 56)\n",
      "(('subgroup', 'patients'), 56)\n",
      "(('ability', 'induce'), 56)\n",
      "(('response', 'rates'), 56)\n",
      "(('prompted', 'us'), 56)\n",
      "(('targeted', 'therapies'), 56)\n",
      "(('downstream', 'effectors'), 56)\n",
      "(('focus', 'formation'), 56)\n",
      "(('2', '2'), 56)\n",
      "(('analyzed', 'western'), 56)\n",
      "(('transcriptional', 'activity,'), 56)\n",
      "(('kit', 'pdgfra'), 56)\n",
      "(('kit', 'mutation'), 56)\n",
      "(('mutations', 'involve'), 56)\n",
      "(('factors', 'p53'), 56)\n",
      "(('interact', 'peptide'), 56)\n",
      "(('domain', 'sensitive'), 56)\n",
      "(('worldwide', 'sequencing'), 56)\n",
      "(('included', 'variants'), 56)\n",
      "(('vector', 'places'), 56)\n",
      "(('(511)', 'interactions'), 56)\n",
      "(('recognition', 'peptide'), 56)\n",
      "(('targets', 'dna'), 56)\n",
      "(('like', 'brca1'), 56)\n",
      "(('cancer', 'second'), 56)\n",
      "(('associated', 'unclassified'), 56)\n",
      "(('large,', '1,863amino'), 56)\n",
      "(('act', 'transcriptional'), 56)\n",
      "(('proteolysis', 'determine'), 56)\n",
      "(('number', 'fall'), 56)\n",
      "(('classes', 'yet'), 56)\n",
      "(('selective', 'recognition'), 56)\n",
      "(('abraxas,', 'relay'), 56)\n",
      "(('set', 'brca1'), 56)\n",
      "(('analyses', 'clinical'), 56)\n",
      "(('protein,', 'abraxas,'), 56)\n",
      "(('nuclear', 'signaling'), 56)\n",
      "(('effect', 'classphp)'), 56)\n",
      "(('essential', 'brct'), 56)\n",
      "(('activity', 'plotted'), 56)\n",
      "(('(1)', 'unfortunately,'), 56)\n",
      "(('variants', 'particular'), 56)\n",
      "(('involving', 'subsets'), 56)\n",
      "(('(17,18)', 'cancer-associated'), 56)\n",
      "(('useful', 'set'), 56)\n",
      "(('yet', 'done,'), 56)\n",
      "(('clinical', 'family'), 56)\n",
      "(('likely', 'contain'), 56)\n",
      "(('set', '117'), 56)\n",
      "(('second', 'method,'), 56)\n",
      "(('alignment', 'data'), 56)\n",
      "(('data', 'protein'), 56)\n",
      "(('consequences', 'set'), 56)\n",
      "(('history', 'data'), 56)\n",
      "(('since', 'cloning'), 56)\n",
      "(('residues', '1,6461,859)'), 56)\n",
      "(('1994,', 'large'), 56)\n",
      "(('fold', 'brca1'), 56)\n",
      "(('essential', 'mediator'), 56)\n",
      "(('done,', 'expected'), 56)\n",
      "(('117', 'distinct'), 56)\n",
      "(('surrounding', 'damage'), 56)\n",
      "(('activity', 'region'), 56)\n",
      "(('population', 'stability'), 56)\n",
      "(('dissection', 'tumor-suppressive'), 56)\n",
      "(('iarc', 'classes'), 56)\n",
      "(('diversity', 'exists'), 56)\n",
      "(('repeats', '(1214)'), 56)\n",
      "(('(53)', 'although'), 56)\n",
      "(('data', 'addition,'), 56)\n",
      "(('generation', 'large'), 56)\n",
      "(('introduction', 'since'), 56)\n",
      "(('stress', '(24)'), 56)\n",
      "(('critical', 'tumorsuppressive'), 56)\n",
      "(('specifically', 'interact'), 56)\n",
      "(('fish,', 'sea'), 56)\n",
      "(('mediated', 'tandem'), 56)\n",
      "(('linking', 'mutation'), 56)\n",
      "(('corresponding', 'complete'), 56)\n",
      "(('brca1,', 'likely'), 56)\n",
      "(('set', 'brct'), 56)\n",
      "(('provide', 'cancer'), 56)\n",
      "(('folding,', 'phosphopeptide-binding,'), 56)\n",
      "(('description', 'brca1'), 56)\n",
      "(('involving', '25'), 56)\n",
      "(('addition', 'classes'), 56)\n",
      "(('trypsin-mediated', 'proteolysis'), 56)\n",
      "(('unstable', 'domain'), 56)\n",
      "(('(15,16)', 'assays'), 56)\n",
      "(('activity', 'brct'), 56)\n",
      "(('phosphopeptide', 'pull-down'), 56)\n",
      "(('phosphopeptidebinding', 'module'), 56)\n",
      "(('(http://research', 'coli'), 56)\n",
      "(('expected', 'large'), 56)\n",
      "(('brca1', 'target'), 56)\n",
      "(('functionally', 'structurally'), 56)\n",
      "(('termed', 'brct'), 56)\n",
      "(('brct', 'coding'), 56)\n",
      "(('database', 'brca1/brca2'), 56)\n",
      "(('(14,48),', 'therefore'), 56)\n",
      "(('er', '(17,18)'), 56)\n",
      "(('classphp)', 'uses'), 56)\n",
      "(('modify', 'dna-binding'), 56)\n",
      "(('reported', '(14)'), 56)\n",
      "(('cloning', 'brca1'), 56)\n",
      "(('module,', 'activity'), 56)\n",
      "(('proteolysis', '(29)'), 56)\n",
      "(('stability', 'instead'), 56)\n",
      "(('align-gvgd', 'algorithm'), 56)\n",
      "(('brca1', 'sequences,'), 56)\n",
      "(('mammalian', 'brca1'), 56)\n",
      "(('bach1', 'several'), 56)\n",
      "(('number', 'rare'), 56)\n",
      "(('p21', 'gadd45'), 56)\n",
      "(('functional', 'work'), 56)\n",
      "(('regulation', '(511)'), 56)\n",
      "(('population', 'detailed'), 56)\n",
      "(('alignment', 'nine'), 56)\n",
      "(('peptides', 'tested'), 56)\n",
      "(('assign', '14'), 56)\n",
      "(('deposited', 'within'), 56)\n",
      "(('3', '(uncertain),'), 56)\n",
      "(('although', 'systematic'), 56)\n",
      "(('date', 'human'), 56)\n",
      "(('(29)', 'analysis'), 56)\n",
      "(('signaling', 'events,'), 56)\n",
      "(('(corresponding', 'residues'), 56)\n",
      "(('suppression', 'tested'), 56)\n",
      "(('surfaces', '(53)'), 56)\n",
      "(('proteins', 'mediated'), 56)\n",
      "(('obtained', 'given'), 56)\n",
      "(('three', 'depths'), 56)\n",
      "(('1,6461,859)', 'control'), 56)\n",
      "(('structure,', 'leading'), 56)\n",
      "(('available', 'mutations'), 56)\n",
      "(('first', 'look'), 56)\n",
      "(('set', 'reagents'), 56)\n",
      "(('destabilize', 'structure,'), 56)\n",
      "(('domain', 'vector'), 56)\n",
      "(('predicted', 'neutral'), 56)\n",
      "(('align-gvgd', 'classification'), 56)\n",
      "(('well', 'analyses'), 56)\n",
      "(('interest', 'database'), 56)\n",
      "(('(1214)', 'brct'), 56)\n",
      "(('al', 'description'), 56)\n",
      "(('population', '(100%'), 56)\n",
      "(('act', 'essential'), 56)\n",
      "(('set', 'obtained'), 56)\n",
      "(('made', 'sequence'), 56)\n",
      "(('indicate', 'ring'), 56)\n",
      "(('assigned', 'risks'), 56)\n",
      "(('117', 'brca1'), 56)\n",
      "(('lack', 'informative'), 56)\n",
      "(('brca1/brca2', 'mutations'), 56)\n",
      "(('p53', 'er'), 56)\n",
      "(('version', 'peptide'), 56)\n",
      "(('majority', 'cancer-associated'), 56)\n",
      "(('brca1', 'uncovered'), 56)\n",
      "(('neutral', 'lee'), 56)\n",
      "(('domain', 'phosphopeptidebinding'), 56)\n",
      "(('attempted', 'predict'), 56)\n",
      "(('within', 'essential'), 56)\n",
      "(('sequences,', 'chicken'), 56)\n",
      "(('mutations', '(33)'), 56)\n",
      "(('brct', 'transcriptional'), 56)\n",
      "(('(14)', 'seven'), 56)\n",
      "(('cancers', '(1)'), 56)\n",
      "(('mutations', '(1,2428)'), 56)\n",
      "(('coverage)', 'brca1'), 56)\n",
      "(('depths', 'alignment'), 56)\n",
      "(('biochemical', 'cellbased'), 56)\n",
      "(('addition,', 'fact'), 56)\n",
      "(('database', '(http://research'), 56)\n",
      "(('women', 'increased'), 56)\n",
      "(('provide', 'impulse'), 56)\n",
      "(('significance', 'here,'), 56)\n",
      "(('method,', 'used'), 56)\n",
      "(('work', 'brca1'), 56)\n",
      "(('(45)', 'pilot'), 56)\n",
      "(('proteinprotein', 'interaction'), 56)\n",
      "(('damageinduced', 'nuclear'), 56)\n",
      "(('(for', 'detail,'), 56)\n",
      "(('analysis', 'phosphopeptide'), 56)\n",
      "(('nuclease', 'ctip,'), 56)\n",
      "(('repeats', 'cooh'), 56)\n",
      "(('damage', 'downstream'), 56)\n",
      "(('motif', 'pser-x-x-phe'), 56)\n",
      "(('detailed', 'studies'), 56)\n",
      "(('structure', 'prediction,'), 56)\n",
      "(('fold', 'set'), 56)\n",
      "(('(100%', 'coverage)'), 56)\n",
      "(('functional', 'dissection'), 56)\n",
      "(('genetic-based', 'data'), 56)\n",
      "(('mediator', 'protein'), 56)\n",
      "(('t7', 'promotor'), 56)\n",
      "(('binding', 'module,'), 56)\n",
      "(('promotor', 'determination'), 56)\n",
      "(('problematic,', 'due'), 56)\n",
      "(('helicase,', 'bach1,'), 56)\n",
      "(('alignment', '(for'), 56)\n",
      "(('117', 'brct'), 56)\n",
      "(('documented', 'individuals'), 56)\n",
      "(('pser-containing', 'peptides'), 56)\n",
      "(('14', 'brca1'), 56)\n",
      "(('risks', 'based'), 56)\n",
      "(('identified', 'activity'), 56)\n",
      "(('structurally', 'rationalized'), 56)\n",
      "(('alignment', '(pmsa)'), 56)\n",
      "(('phosphorylated', 'partner'), 56)\n",
      "(('brca1', 'encodes'), 56)\n",
      "(('tandem', 'pair'), 56)\n",
      "(('analyzed', 'align-gvgd'), 56)\n",
      "(('studies', 'stability'), 56)\n",
      "(('gadd45', '(15,16)'), 56)\n",
      "(('fold', 'addition'), 56)\n",
      "(('associated', 'disease,'), 56)\n",
      "(('set', 'provides'), 56)\n",
      "(('percentage', 'wt'), 56)\n",
      "(('using', 'lee'), 56)\n",
      "(('risk', 'early-onset'), 56)\n",
      "(('substitutions', 'brct'), 56)\n",
      "(('large', 'effort'), 56)\n",
      "(('brca1,', 'termed'), 56)\n",
      "(('exists', 'human'), 56)\n",
      "(('data', 'abrogate,'), 56)\n",
      "(('interactions', 'phosphorylated'), 56)\n",
      "(('predict', 'cancer'), 56)\n",
      "(('sequence', '(either'), 56)\n",
      "(('lack', 'genetic-based'), 56)\n",
      "(('phosphopeptide-binding,', 'cell-based'), 56)\n",
      "(('coli', 'expression'), 56)\n",
      "(('protein', '(15,16)'), 56)\n",
      "(('contain', 'vast'), 56)\n",
      "(('use', 'vitro'), 56)\n",
      "(('version', 'control'), 56)\n",
      "(('assess', 'structural'), 56)\n",
      "(('efforts', 'deposited'), 56)\n",
      "(('dna-binding', 'proteinprotein'), 56)\n",
      "(('chromatin', 'surrounding'), 56)\n",
      "(('important', 'first'), 56)\n",
      "(('(31,32,46)', 'easton'), 56)\n",
      "(('(uncertain),', 'due'), 56)\n",
      "(('look', 'kind'), 56)\n",
      "(('consequences', '117'), 56)\n",
      "(('terminus', 'brca1,'), 56)\n",
      "(('(1014,44)', 'mutations'), 56)\n",
      "(('pser-x-x-phe', 'motif,'), 56)\n",
      "(('algorithm', 'three'), 56)\n",
      "(('material', '(supplementary'), 56)\n",
      "(('pser-pro-thr-phe', 'unphosphorylated'), 56)\n",
      "(('(24)', 'brca1'), 56)\n",
      "(('showing', 'critical'), 56)\n",
      "(('genes', 'women'), 56)\n",
      "(('containing', 'motif'), 56)\n",
      "(('brca1', 'destabilize'), 56)\n",
      "(('module', 'tumor'), 56)\n",
      "(('(14)', 'indicates'), 56)\n",
      "(('tumor-suppressive', 'activities'), 56)\n",
      "(('encodes', 'large,'), 56)\n",
      "(('places', 'brca1'), 56)\n",
      "(('easton', 'colleagues'), 56)\n",
      "(('containing', 'pser-x-x-phe'), 56)\n",
      "(('targets', 'containing'), 56)\n",
      "(('(35)', 'assigned'), 56)\n",
      "(('estimates', 'brca1'), 56)\n",
      "(('role', 'domains'), 56)\n",
      "(('(33)', 'discussion'), 56)\n",
      "(('informative', 'pedigree'), 56)\n",
      "(('parallel', 'analysis'), 56)\n",
      "(('variants', 'phosphopeptide'), 56)\n",
      "(('brca1', 'bioinformatics'), 56)\n",
      "(('rare', 'brca1'), 56)\n",
      "(('ref', 'sufficient'), 56)\n",
      "(('susceptibility', 'trypsin-mediated'), 56)\n",
      "(('unfortunately,', 'large'), 56)\n",
      "(('brca1', '(14,48),'), 56)\n",
      "(('activity', 'align-gvgd'), 56)\n",
      "(('systems', '(45)'), 56)\n",
      "(('rationalized', 'provide'), 56)\n",
      "(('bach1,', 'nuclease'), 56)\n",
      "(('plotted', 'percentage'), 56)\n",
      "(('gene', '(31,32,46)'), 56)\n",
      "(('pair', 'repeats'), 56)\n",
      "(('variants', 'uncovered'), 56)\n",
      "(('domains,', 'highly'), 56)\n",
      "(('kind', 'genetic'), 56)\n",
      "(('linked', 'sequence-specific'), 56)\n",
      "(('previously', 'structural'), 56)\n",
      "(('unphosphorylated', 'version'), 56)\n",
      "(('certain', 'single'), 56)\n",
      "(('variants', 'iarc'), 56)\n",
      "(('puffer', 'fish,'), 56)\n",
      "(('detail,', 'see'), 56)\n",
      "(('pser-x-x-phe', '(1014,44)'), 56)\n",
      "(('fall', 'class'), 56)\n",
      "(('importance', 'brca1'), 56)\n",
      "(('fact', 'variants'), 56)\n",
      "(('prediction,', 'well'), 56)\n",
      "(('assessment', 'problematic,'), 56)\n",
      "(('pser-x-x-phe', 'several'), 56)\n",
      "(('subsets', 'brca1'), 56)\n",
      "(('associated', 'brct'), 56)\n",
      "(('mutations', 'provides'), 56)\n",
      "(('regions,', 'showing'), 56)\n",
      "(('dna', 'damageinduced'), 56)\n",
      "(('events,', 'interacts'), 56)\n",
      "(('variants', '117'), 56)\n",
      "(('large-scale', 'classification'), 56)\n",
      "(('sea', 'urchin),'), 56)\n",
      "(('(either', 'frog,'), 56)\n",
      "(('variants', 'transcriptional'), 56)\n",
      "(('markedly', 'diminish,'), 56)\n",
      "(('frog,', 'puffer'), 56)\n",
      "(('urchin),', 'used'), 56)\n",
      "(('bind', 'pser-containing'), 56)\n",
      "(('determine', 'stability'), 56)\n",
      "(('cellbased', 'transcriptional'), 56)\n",
      "(('risk', 'brca1'), 56)\n",
      "(('tumorsuppressive', 'function'), 56)\n",
      "(('vitro', 'transcribed/translated'), 56)\n",
      "(('domain', 'specifically'), 56)\n",
      "(('comprehensive', 'assessment'), 56)\n",
      "(('uncovered', 'worldwide'), 56)\n",
      "(('(15,16)', 'modulate'), 56)\n",
      "(('available', 'assign'), 56)\n",
      "(('reagents', 'detailed'), 56)\n",
      "(('sequence', 'pser-pro-thr-phe'), 56)\n",
      "(('chicken', 'one'), 56)\n",
      "(('impulse', 'functional'), 56)\n",
      "(('sequence', '(corresponding'), 56)\n",
      "(('used', 'phosphopeptide'), 56)\n",
      "(('pedigree', 'data'), 56)\n",
      "(('peptide-binding', 'assays,'), 56)\n",
      "(('brca1', '1994,'), 56)\n",
      "(('uncovered', 'human'), 56)\n",
      "(('systematic', 'large-scale'), 56)\n",
      "(('uncovered', 'brca1'), 56)\n",
      "(('assessment', 'structural'), 56)\n",
      "(('cell-based', 'transcriptional'), 56)\n",
      "(('pilot', 'studies'), 56)\n",
      "(('transcribed/translated', 'material'), 56)\n",
      "(('abrogate,', 'markedly'), 56)\n",
      "(('cluster', 'ring'), 56)\n",
      "(('generated', 'chromatin'), 56)\n",
      "(('diminish,', 'expression'), 56)\n",
      "(('interacts', 'bach1'), 56)\n",
      "(('activation', 'established'), 56)\n",
      "(('(1,2428)', 'studies'), 56)\n",
      "(('1,863amino', 'acid'), 56)\n",
      "(('(pmsa)', 'data'), 56)\n",
      "(('hand,', 'pilot'), 56)\n",
      "(('uses', 'protein'), 56)\n",
      "(('instead', 'modify'), 56)\n",
      "(('folding', 'defects'), 56)\n",
      "(('proteins', 'selective'), 56)\n",
      "(('family-based', 'data'), 56)\n",
      "(('using', 'peptide-binding'), 56)\n",
      "(('ctip,', 'another'), 56)\n",
      "(('proteins', 'factors'), 56)\n",
      "(('functional', 'target'), 56)\n",
      "(('mutations', 'lead'), 55)\n",
      "(('even', 'absence'), 55)\n",
      "(('ring', 'domain'), 55)\n",
      "(('observed', 'mutations'), 55)\n",
      "(('missense', 'changes'), 55)\n",
      "(('cells', 'carrying'), 55)\n",
      "(('closely', 'related'), 55)\n",
      "(('mutants', 'could'), 55)\n",
      "(('four', 'mutations'), 55)\n",
      "(('mutations,', 'whereas'), 55)\n",
      "(('myelogenous', 'leukemia'), 55)\n",
      "(('cell', 'growth,'), 55)\n",
      "(('may', 'affect'), 55)\n",
      "(('increased', 'phosphorylation'), 55)\n",
      "(('also', 'suggest'), 55)\n",
      "(('patients', 'carrying'), 55)\n",
      "(('wide', 'range'), 55)\n",
      "(('current', 'study,'), 55)\n",
      "(('lines', 'primary'), 55)\n",
      "(('1', 'mutations'), 55)\n",
      "(('ovarian', 'cancer,'), 55)\n",
      "(('atp-binding', 'site'), 55)\n",
      "(('first', 'time,'), 55)\n",
      "(('two', 'mutants'), 55)\n",
      "(('strongly', 'associated'), 55)\n",
      "(('(table', '1,'), 55)\n",
      "(('mutations', 'induce'), 55)\n",
      "(('here,', 'show'), 55)\n",
      "(('downstream', 'phosphorylation'), 55)\n",
      "(('egfr', 'inhibitors'), 55)\n",
      "(('wild-type', 'egfr,'), 55)\n",
      "(('also', 'analyzed'), 55)\n",
      "(('cancer', 'specimens'), 55)\n",
      "(('clinical', 'benefit'), 55)\n",
      "(('partial', 'response;'), 55)\n",
      "(('kinase', 'inhibition'), 55)\n",
      "(('vitro', 'binding'), 55)\n",
      "(('deleterious', 'mutations'), 55)\n",
      "(('mutant', 'constructs'), 54)\n",
      "(('acid', 'changes'), 54)\n",
      "(('activity', 'cells'), 54)\n",
      "(('100', 'nm'), 54)\n",
      "(('mutations', '(p'), 54)\n",
      "(('mutant', 'showed'), 54)\n",
      "(('also', 'known'), 54)\n",
      "(('mutations', 'resulted'), 54)\n",
      "(('popup', 'table'), 54)\n",
      "(('mice', 'results'), 54)\n",
      "(('al', 'thus,'), 54)\n",
      "(('formalin-fixed,', 'paraffin-embedded'), 54)\n",
      "(('growth', 'factors'), 54)\n",
      "(('n-terminal', 'region'), 54)\n",
      "(('acid', 'sequence'), 54)\n",
      "(('potential', 'therapeutic'), 54)\n",
      "(('expression', 'cell'), 54)\n",
      "(('frameshift', 'mutations'), 54)\n",
      "(('associated', 'decreased'), 54)\n",
      "(('without', 'egfr'), 54)\n",
      "(('egfr', 'tkis'), 54)\n",
      "(('characteristics', 'patients'), 54)\n",
      "(('introduced', 'full-length'), 54)\n",
      "(('egfr', 'signaling'), 54)\n",
      "(('substitution', 'mutations'), 54)\n",
      "(('gene', 'amplified'), 54)\n",
      "(('clinically', 'relevant'), 54)\n",
      "(('response', 'treatment'), 54)\n",
      "(('clinical', 'activity'), 54)\n",
      "(('proliferation', 'ba/f3'), 54)\n",
      "(('biotechnology,', 'santa'), 54)\n",
      "(('blotting', 'using'), 54)\n",
      "(('pdgfra', 'mutation'), 54)\n",
      "(('full-length', 'human'), 54)\n",
      "(('(genbank', 'accession'), 54)\n",
      "(('stably', 'expressed'), 53)\n",
      "(('short', 'hairpin'), 53)\n",
      "(('nonsense', 'mutation'), 53)\n",
      "(('gene', 'expression,'), 53)\n",
      "(('sequencing', 'data'), 53)\n",
      "(('fish', 'analysis'), 53)\n",
      "(('clinical', 'relevance'), 53)\n",
      "(('examined', 'whether'), 53)\n",
      "(('results', 'provide'), 53)\n",
      "(('rna', 'polymerase'), 53)\n",
      "(('tumor', 'suppressors'), 53)\n",
      "(('mutations', 'two'), 53)\n",
      "(('based', 'crystal'), 53)\n",
      "(('along', 'increased'), 53)\n",
      "(('inactivating', 'mutations'), 53)\n",
      "(('active', 'form'), 53)\n",
      "(('reduced', 'levels'), 53)\n",
      "(('function', 'may'), 53)\n",
      "(('cycle', 'progression'), 53)\n",
      "(('cancer', 'cells,'), 53)\n",
      "(('results', 'indicated'), 53)\n",
      "(('substantially', 'increased'), 53)\n",
      "(('mutations', 'genes'), 53)\n",
      "(('deletion', 'mutations'), 53)\n",
      "(('human', 'lung'), 53)\n",
      "(('al', 'however,'), 53)\n",
      "(('cycle', 'arrest'), 53)\n",
      "(('48', 'h'), 53)\n",
      "(('sufficient', 'induce'), 53)\n",
      "(('expressing', 'various'), 53)\n",
      "(('mutation', 'sensitive'), 53)\n",
      "(('lung', 'adenocarcinomas,'), 53)\n",
      "(('activity,', 'suggested'), 53)\n",
      "(('mutations', 'epidermal'), 53)\n",
      "(('kinase', '(tk)'), 53)\n",
      "(('interest', 'determine'), 53)\n",
      "(('clustered', 'region'), 53)\n",
      "(('treatment', 'egfr'), 53)\n",
      "(('18,', '19,'), 53)\n",
      "(('large', 'cell'), 53)\n",
      "(('inhibitors', '(tkis)'), 53)\n",
      "(('cancer', 'patient'), 53)\n",
      "(('patients', 'nsclc'), 53)\n",
      "(('(amino', 'acid'), 53)\n",
      "(('analyzed', 'study'), 53)\n",
      "(('additional', 'studies'), 53)\n",
      "(('inhibitor', 'therapy'), 53)\n",
      "(('mutations', 'sensitive'), 53)\n",
      "(('assay', 'using'), 53)\n",
      "(('per', 'well'), 53)\n",
      "(('foci', 'formation'), 53)\n",
      "(('cell', 'factor'), 53)\n",
      "(('24', 'h'), 52)\n",
      "(('low', 'frequency'), 52)\n",
      "(('inhibitors', 'might'), 52)\n",
      "(('failed', 'induce'), 52)\n",
      "(('reduced', 'expression'), 52)\n",
      "(('(tables', '1'), 52)\n",
      "(('formation', 'soft'), 52)\n",
      "(('without', 'mutations'), 52)\n",
      "(('activation', 'mutant'), 52)\n",
      "(('identified', 'four'), 52)\n",
      "(('published', 'data'), 52)\n",
      "(('several', 'different'), 52)\n",
      "(('data', 'strongly'), 52)\n",
      "(('30', 'min'), 52)\n",
      "(('al', 'figure'), 52)\n",
      "(('remains', 'unknown'), 52)\n",
      "(('mechanism', 'resistance'), 52)\n",
      "(('reverse', 'primer'), 52)\n",
      "(('tumors', 'without'), 52)\n",
      "(('pcr', 'using'), 52)\n",
      "(('tumor', 'suppressive'), 52)\n",
      "(('tumor', 'tissues'), 52)\n",
      "(('3,', '4,'), 52)\n",
      "(('endometrial', 'cancers,'), 52)\n",
      "(('endometrial', 'cancer'), 52)\n",
      "(('pcr', 'product'), 52)\n",
      "(('also', 'investigated'), 52)\n",
      "(('hot', 'spot'), 52)\n",
      "(('likely', 'due'), 52)\n",
      "(('total', 'number'), 52)\n",
      "(('mutants', 'compared'), 52)\n",
      "(('published', 'reports'), 52)\n",
      "(('imatinib', 'treatment'), 52)\n",
      "(('within', 'kinase'), 52)\n",
      "(('seven', 'patients'), 52)\n",
      "(('molecular', 'analysis'), 52)\n",
      "(('currently', 'available'), 52)\n",
      "(('growth', 'ba/f3'), 52)\n",
      "(('comparable', 'levels'), 52)\n",
      "(('1', '5'), 52)\n",
      "(('dependent', 'upon'), 52)\n",
      "(('kinase', '(rtk)'), 52)\n",
      "(('also', 'noted'), 52)\n",
      "(('primary', 'resistance'), 52)\n",
      "(('two', 'recent'), 52)\n",
      "(('mutants', 'found'), 52)\n",
      "(('chromosomal', 'translocation'), 52)\n",
      "(('therapeutic', 'targeting'), 52)\n",
      "(('5', '7'), 52)\n",
      "(('brca1', 'sequence'), 52)\n",
      "(('cannot', 'exclude'), 51)\n",
      "(('gene', 'product'), 51)\n",
      "(('absence', 'il-3'), 51)\n",
      "(('proliferation', 'assays'), 51)\n",
      "(('led', 'identification'), 51)\n",
      "(('myeloid', 'malignancies'), 51)\n",
      "(('negative', 'regulatory'), 51)\n",
      "(('experiments', 'using'), 51)\n",
      "(('growth', 'cells'), 51)\n",
      "(('common', 'mutations'), 51)\n",
      "(('(figure', '4c)'), 51)\n",
      "(('inlineview', 'popup'), 51)\n",
      "(('observations', 'indicate'), 51)\n",
      "(('therapeutic', 'strategy'), 51)\n",
      "(('inhibited', 'growth'), 51)\n",
      "(('absence', 'serum'), 51)\n",
      "(('also', 'reduced'), 51)\n",
      "(('therapy', 'patients'), 51)\n",
      "(('lung', 'cancers,'), 51)\n",
      "(('found', 'significant'), 51)\n",
      "(('al', 'finally,'), 51)\n",
      "(('cells', 'cultured'), 51)\n",
      "(('protein', 'sequence'), 51)\n",
      "(('hematopoietic', 'stem'), 51)\n",
      "(('could', 'detected'), 51)\n",
      "(('line', '(fig'), 51)\n",
      "(('gefitinib', 'treatment'), 51)\n",
      "(('observed', 'wild-type'), 51)\n",
      "(('growth', 'factors,'), 51)\n",
      "(('egfr', 'mutation,'), 51)\n",
      "(('mutations', 'selected'), 51)\n",
      "(('patients', 'wild-type'), 51)\n",
      "(('formed', 'colonies'), 51)\n",
      "(('also', 'sensitive'), 51)\n",
      "(('crystallographic', 'structure'), 51)\n",
      "(('protein', 'data'), 51)\n",
      "(('primary', 'mutation'), 51)\n",
      "(('mutations', 'compared'), 51)\n",
      "(('aspartic', 'acid'), 51)\n",
      "(('high', 'risk'), 51)\n",
      "(('(fig', 'addition,'), 51)\n",
      "(('transform', 'cells'), 51)\n",
      "(('ability', 'bind'), 51)\n",
      "(('gene', 'fusion'), 51)\n",
      "(('constitutive', 'tyrosine'), 51)\n",
      "(('variants', 'unknown'), 51)\n",
      "(('activation', 'function'), 51)\n",
      "(('deletion', 'mutant,'), 50)\n",
      "(('hematopoietic', 'cells'), 50)\n",
      "(('genetic', 'events'), 50)\n",
      "(('mutations', 'different'), 50)\n",
      "(('(fig', 'observations'), 50)\n",
      "(('mutations', 'well'), 50)\n",
      "(('support', 'hypothesis'), 50)\n",
      "(('lines', 'express'), 50)\n",
      "(('proteins', 'containing'), 50)\n",
      "(('dna', 'isolated'), 50)\n",
      "(('mutations', 'commonly'), 50)\n",
      "(('293t', 'cells,'), 50)\n",
      "(('small', 'subset'), 50)\n",
      "(('full-length', 'cdna'), 50)\n",
      "(('recently', 'described'), 50)\n",
      "(('tissue', 'samples'), 50)\n",
      "(('statistical', 'analysis'), 50)\n",
      "(('question', 'whether'), 50)\n",
      "(('pathogenic', 'variants'), 50)\n",
      "(('data', 'obtained'), 50)\n",
      "(('conformational', 'changes'), 50)\n",
      "(('exon', '4'), 50)\n",
      "(('may', 'influence'), 50)\n",
      "(('(fig', 'moreover,'), 50)\n",
      "(('lesser', 'extent,'), 50)\n",
      "(('results', 'strongly'), 50)\n",
      "(('key', 'role'), 50)\n",
      "(('cells', 'sensitive'), 50)\n",
      "(('activation', 'wild-type'), 50)\n",
      "(('potent', 'inhibitors'), 50)\n",
      "(('detected', 'using'), 50)\n",
      "(('conserved', 'among'), 50)\n",
      "(('may', 'effective'), 50)\n",
      "(('exons', '18,'), 50)\n",
      "(('(wt)', 'mutant'), 50)\n",
      "(('1', 'm'), 50)\n",
      "(('clinical', 'experience'), 50)\n",
      "(('highly', 'expressed'), 50)\n",
      "(('studies', 'revealed'), 50)\n",
      "(('protein', 'tyrosine'), 50)\n",
      "(('50%', 'inhibitory'), 50)\n",
      "(('oligonucleotide', 'primers'), 50)\n",
      "(('secondary', 'resistance'), 50)\n",
      "(('molecular', 'mechanism'), 50)\n",
      "(('reverse', 'transcription'), 50)\n",
      "(('normal', 'tissue'), 50)\n",
      "(('prognostic', 'significance'), 50)\n",
      "(('complete', 'inhibition'), 50)\n",
      "(('activate', 'signaling'), 50)\n",
      "(('braf', 'mutation'), 50)\n",
      "(('exon', '15'), 50)\n",
      "(('similar', 'mutations'), 50)\n",
      "(('mek', 'inhibitors'), 50)\n",
      "(('exposed', 'mutations'), 50)\n",
      "(('manner', 'similar'), 49)\n",
      "(('large', 'cohort'), 49)\n",
      "(('cells', 'revealed'), 49)\n",
      "(('independent', 'growth'), 49)\n",
      "(('analyses', 'performed'), 49)\n",
      "(('findings', 'indicate'), 49)\n",
      "(('lines', 'derived'), 49)\n",
      "(('cell', 'types'), 49)\n",
      "(('cdna', 'sequence'), 49)\n",
      "(('figure', 'viewer'), 49)\n",
      "(('polymerase', 'ii'), 49)\n",
      "(('previously', 'characterized'), 49)\n",
      "(('greatly', 'reduced'), 49)\n",
      "(('10', 'patients'), 49)\n",
      "(('first', 'report'), 49)\n",
      "(('cancer', 'associated'), 49)\n",
      "(('cells', 'low'), 49)\n",
      "(('primary', 'antibodies'), 49)\n",
      "(('analyzed', 'using'), 49)\n",
      "(('identified', 'three'), 49)\n",
      "(('cells', 'shown'), 49)\n",
      "(('tested', 'effect'), 49)\n",
      "(('transfected', 'either'), 49)\n",
      "(('repair', '(mmr)'), 49)\n",
      "(('colorectal', 'endometrial'), 49)\n",
      "(('mmr', 'genes'), 49)\n",
      "(('mlh1,', 'msh2,'), 49)\n",
      "(('msh6', 'pms2'), 49)\n",
      "(('prior', 'studies'), 49)\n",
      "(('give', 'rise'), 49)\n",
      "(('patients', 'study'), 49)\n",
      "(('tumor', 'cells,'), 49)\n",
      "(('well', 'wild-type'), 49)\n",
      "(('gain', 'insight'), 49)\n",
      "(('within', 'tumor'), 49)\n",
      "(('nonsmall-cell', 'lung'), 49)\n",
      "(('(tk)', 'domain'), 49)\n",
      "(('mutation', 'confers'), 49)\n",
      "(('signaling', 'technology)'), 49)\n",
      "(('coding', 'sequence,'), 49)\n",
      "(('chronic', 'myelogenous'), 49)\n",
      "(('best', 'response'), 49)\n",
      "(('study,', 'used'), 49)\n",
      "(('progressive', 'disease'), 49)\n",
      "(('egfr', 'tki'), 49)\n",
      "(('5,', '6)'), 49)\n",
      "(('antibodies', 'used'), 49)\n",
      "(('levels', 'wild-type'), 49)\n",
      "(('gene', 'fusions'), 49)\n",
      "(('dna', 'sequence'), 49)\n",
      "(('one', 'following'), 49)\n",
      "(('damage', 'repair'), 49)\n",
      "(('endothelial', 'growth'), 49)\n",
      "(('11', 'mutation'), 49)\n",
      "(('mutations', 'abolish'), 49)\n",
      "(('eight', 'mutations'), 49)\n",
      "(('vus', 'neutral'), 49)\n",
      "(('showed', 'similar'), 48)\n",
      "(('share', 'common'), 48)\n",
      "(('group', 'patients'), 48)\n",
      "(('also', 'described'), 48)\n",
      "(('(table', '1),'), 48)\n",
      "(('(also', 'known'), 48)\n",
      "(('indicated', 'concentrations'), 48)\n",
      "(('structure', 'human'), 48)\n",
      "(('lung', 'tumors'), 48)\n",
      "(('central', 'nervous'), 48)\n",
      "(('may', 'useful'), 48)\n",
      "(('lines', 'showed'), 48)\n",
      "(('mutation', 'affects'), 48)\n",
      "(('cancer', 'cases'), 48)\n",
      "(('test', 'hypothesis,'), 48)\n",
      "(('expression', 'human'), 48)\n",
      "(('associated', 'loss'), 48)\n",
      "(('unclear', 'whether'), 48)\n",
      "(('rule', 'possibility'), 48)\n",
      "(('using', 'specific'), 48)\n",
      "(('sequence', 'identity'), 48)\n",
      "(('little', 'effect'), 48)\n",
      "(('tumors', 'contained'), 48)\n",
      "(('markedly', 'reduced'), 48)\n",
      "(('mrna', 'protein'), 48)\n",
      "(('kinase', 'assay'), 48)\n",
      "(('pathway', 'activated'), 48)\n",
      "(('study', 'identified'), 48)\n",
      "(('gene', 'encodes'), 48)\n",
      "(('showed', 'higher'), 48)\n",
      "(('chronic', 'myeloid'), 48)\n",
      "(('tumors', 'egfr'), 48)\n",
      "(('recent', 'work'), 48)\n",
      "(('harbor', 'mutations'), 48)\n",
      "(('egfr', 'coding'), 48)\n",
      "(('tumors', 'similar'), 48)\n",
      "(('mutations', 'results'), 48)\n",
      "(('mutations', 'nsclc'), 48)\n",
      "(('egfr', 'phosphorylation'), 48)\n",
      "(('inactive', 'conformation'), 48)\n",
      "(('3', 'months'), 48)\n",
      "(('examined', 'using'), 48)\n",
      "(('oncogenic', 'role'), 48)\n",
      "(('signaling', 'also'), 48)\n",
      "(('72', 'hours'), 48)\n",
      "(('confirmed', 'presence'), 48)\n",
      "(('better', 'understand'), 48)\n",
      "(('variants', 'may'), 48)\n",
      "(('less', 'common'), 48)\n",
      "(('solid', 'tumors,'), 48)\n",
      "(('mechanisms', 'resistance'), 48)\n",
      "(('structural', 'studies'), 48)\n",
      "(('increased', 'activity'), 48)\n",
      "(('screened', 'presence'), 48)\n",
      "(('genetic', 'testing'), 48)\n",
      "(('remaining', 'three'), 48)\n",
      "(('primary', 'human'), 48)\n",
      "(('activation', 'receptor'), 48)\n",
      "(('phosphatase', 'domain'), 48)\n",
      "(('brca1', 'gene'), 48)\n",
      "(('development', 'provides'), 48)\n",
      "(('nonconservative', 'changes'), 48)\n",
      "(('methods', 'supplementary'), 47)\n",
      "(('lung', 'tumor'), 47)\n",
      "(('containing', 'either'), 47)\n",
      "(('marrow', 'cells'), 47)\n",
      "(('mutations', 'involved'), 47)\n",
      "(('index', 'next'), 47)\n",
      "(('mutations', 'acquired'), 47)\n",
      "(('lymph', 'node'), 47)\n",
      "(('types,', 'including'), 47)\n",
      "(('mutations', 'tumors'), 47)\n",
      "(('mutations', 'clinical'), 47)\n",
      "(('patient', 'survival'), 47)\n",
      "(('table', '3)'), 47)\n",
      "(('may', 'help'), 47)\n",
      "(('hotspot', 'mutations'), 47)\n",
      "(('(figure', '5c)'), 47)\n",
      "(('colorectal', 'cancers'), 47)\n",
      "(('secondary', 'structure'), 47)\n",
      "(('kinase', '(mapk)'), 47)\n",
      "(('functional', 'significance'), 47)\n",
      "(('293', 'cells'), 47)\n",
      "(('proteins', 'detected'), 47)\n",
      "(('table', 's3)'), 47)\n",
      "(('msh2,', 'msh6'), 47)\n",
      "(('table', '7)'), 47)\n",
      "(('mutations', 'among'), 47)\n",
      "(('mice', 'injected'), 47)\n",
      "(('low', 'expression'), 47)\n",
      "(('mutation', 'case'), 47)\n",
      "(('sensitivity', 'egfr'), 47)\n",
      "(('responses', 'patients'), 47)\n",
      "(('despite', 'presence'), 47)\n",
      "(('sensitive', 'inhibition'), 47)\n",
      "(('expression', 'construct'), 47)\n",
      "(('previously', 'reported,'), 47)\n",
      "(('among', 'them,'), 47)\n",
      "(('clinical', 'characteristics'), 47)\n",
      "(('kinase', '(mapk),'), 47)\n",
      "(('subset', 'patients'), 47)\n",
      "(('expression', 'also'), 47)\n",
      "(('gatekeeper', 'mutation'), 47)\n",
      "(('constitutive', 'kinase'), 47)\n",
      "(('mutant', '(figure'), 47)\n",
      "(('(fig', 'similarly,'), 47)\n",
      "(('mutants', 'study'), 47)\n",
      "(('(supplemental', 'figure'), 47)\n",
      "(('inhibitory', 'activity'), 47)\n",
      "(('chromosomal', 'translocations'), 47)\n",
      "(('total', 'rna'), 47)\n",
      "(('mutant', 'p53'), 47)\n",
      "(('10,', '11,'), 47)\n",
      "(('recently', 'demonstrated'), 47)\n",
      "(('form', 'receptor'), 47)\n",
      "(('resulted', 'significant'), 46)\n",
      "(('driver', 'mutations'), 46)\n",
      "(('mutants', '(fig'), 46)\n",
      "(('vectors', 'expressing'), 46)\n",
      "(('mutation', 'previously'), 46)\n",
      "(('expression', 'either'), 46)\n",
      "(('oncogenic', 'activity'), 46)\n",
      "(('suppressor', 'genes'), 46)\n",
      "(('mutation', 'rates'), 46)\n",
      "(('(1,', '2)'), 46)\n",
      "(('mutations', 'rare'), 46)\n",
      "(('expression', '(fig'), 46)\n",
      "(('bona', 'fide'), 46)\n",
      "(('collectively,', 'data'), 46)\n",
      "(('al', 'although'), 46)\n",
      "(('inhibited', 'cell'), 46)\n",
      "(('mutation', 'within'), 46)\n",
      "(('ii', 'xl'), 46)\n",
      "(('study', 'also'), 46)\n",
      "(('renal', 'cell'), 46)\n",
      "(('also', 'demonstrated'), 46)\n",
      "(('control', 'cells,'), 46)\n",
      "(('cancer', 'cases,'), 46)\n",
      "(('premature', 'stop'), 46)\n",
      "(('four', 'different'), 46)\n",
      "(('tumors', 'may'), 46)\n",
      "(('transfected', 'vector'), 46)\n",
      "(('frameshift', 'mutation'), 46)\n",
      "(('expressed', 'cells'), 46)\n",
      "(('genes', 'mlh1,'), 46)\n",
      "(('carrier', 'status'), 46)\n",
      "(('(figure', '1b),'), 46)\n",
      "(('summarized', 'table'), 46)\n",
      "(('effect', 'mutation'), 46)\n",
      "(('mlh1', 'msh2'), 46)\n",
      "(('growth', 'rate'), 46)\n",
      "(('fig', 'contrast,'), 46)\n",
      "(('19', 'deletion'), 46)\n",
      "(('used', 'identify'), 46)\n",
      "(('18', '21'), 46)\n",
      "(('mutation', 'patient'), 46)\n",
      "(('8,', '9,'), 46)\n",
      "(('mutations', 'however,'), 46)\n",
      "(('advanced', 'nsclc'), 46)\n",
      "(('missense', 'mutants'), 46)\n",
      "(('preclinical', 'models'), 46)\n",
      "(('domain', 'mutation'), 46)\n",
      "(('cells', 'found'), 46)\n",
      "(('11,', '12)'), 46)\n",
      "(('1', '3'), 46)\n",
      "(('next', 'section'), 46)\n",
      "(('cancer', 'phenotype'), 46)\n",
      "(('erk', 'signaling'), 46)\n",
      "(('mutations', 'highly'), 46)\n",
      "(('mapped', 'onto'), 46)\n",
      "(('activation', 'either'), 46)\n",
      "(('fused', 'frame'), 46)\n",
      "(('pdgfra', 'd842v'), 46)\n",
      "(('better', 'outcome'), 46)\n",
      "(('inhibitor', 'treatment'), 45)\n",
      "(('type', 'mutant'), 45)\n",
      "(('constitutively', 'activate'), 45)\n",
      "(('also', 'included'), 45)\n",
      "(('b', 'cells'), 45)\n",
      "(('patients', '(table'), 45)\n",
      "(('mapk', 'activation'), 45)\n",
      "(('mutations', 'mainly'), 45)\n",
      "(('amplification', 'may'), 45)\n",
      "(('analysis', 'confirmed'), 45)\n",
      "(('study', 'provides'), 45)\n",
      "(('figure', '2)'), 45)\n",
      "(('n-', 'c-terminal'), 45)\n",
      "(('mutants', 'exhibited'), 45)\n",
      "(('nih-3t3', 'cells'), 45)\n",
      "(('additional', 'mutations'), 45)\n",
      "(('small', 'interfering'), 45)\n",
      "(('protein', 'kinases'), 45)\n",
      "(('also', 'increased'), 45)\n",
      "(('receptor', 'gene'), 45)\n",
      "(('gene', 'activation'), 45)\n",
      "(('cells', 'results'), 45)\n",
      "(('however,', 'unlike'), 45)\n",
      "(('forward', 'primer'), 45)\n",
      "(('also', 'performed'), 45)\n",
      "(('insertions', 'deletions'), 45)\n",
      "(('prior', 'probabilities'), 45)\n",
      "(('real-time', 'pcr'), 45)\n",
      "(('breast', 'tumor'), 45)\n",
      "(('may', 'account'), 45)\n",
      "(('next-generation', 'sequencing'), 45)\n",
      "(('support', 'notion'), 45)\n",
      "(('genomic', 'alterations'), 45)\n",
      "(('findings', 'demonstrate'), 45)\n",
      "(('sensitive', 'gefitinib'), 45)\n",
      "(('total', 'protein'), 45)\n",
      "(('treatment', 'advanced'), 45)\n",
      "(('within', 'exons'), 45)\n",
      "(('studies', 'required'), 45)\n",
      "(('mutation', 'type'), 45)\n",
      "(('rare', 'mutations'), 45)\n",
      "(('abi', 'prism'), 45)\n",
      "(('treatment', 'response'), 45)\n",
      "(('also', 'evaluated'), 45)\n",
      "(('activating', 'mutations,'), 45)\n",
      "(('response', 'egfr'), 45)\n",
      "(('19,', '20'), 45)\n",
      "(('best', 'knowledge,'), 45)\n",
      "(('amplified', 'using'), 45)\n",
      "(('one', 'tumor'), 45)\n",
      "(('mutants', 'identified'), 45)\n",
      "(('1', 'g'), 45)\n",
      "(('pivotal', 'role'), 45)\n",
      "(('(protein', 'data'), 45)\n",
      "(('mutations', 'activation'), 45)\n",
      "(('section', 'next'), 45)\n",
      "(('previous', 'section'), 45)\n",
      "(('cancer', 'table'), 45)\n",
      "(('mutations', '(figure'), 45)\n",
      "(('mutant', '(fig'), 45)\n",
      "(('vascular', 'endothelial'), 45)\n",
      "(('mutations', 'disrupt'), 45)\n",
      "(('growth', 'rates'), 45)\n",
      "(('mutations', 'causing'), 45)\n",
      "(('impact', 'missense'), 45)\n",
      "(('wild-type', 'kit'), 45)\n",
      "(('mutations', 'braf'), 45)\n",
      "(('map', 'kinase'), 45)\n",
      "(('functions', 'brca1'), 45)\n",
      "(('pten', 'cdna'), 45)\n",
      "(('human', 'brca1'), 45)\n",
      "(('following', 'primers:'), 44)\n",
      "(('deletions', 'exon'), 44)\n",
      "(('one', 'allele'), 44)\n",
      "(('active', 'state'), 44)\n",
      "(('mast', 'cell'), 44)\n",
      "(('detected', 'two'), 44)\n",
      "(('functional', 'analyses'), 44)\n",
      "(('tumour', 'suppressor'), 44)\n",
      "(('cancer', 'mutations'), 44)\n",
      "(('promoter', 'region'), 44)\n",
      "(('mutations', 'localized'), 44)\n",
      "(('human', 'cancers,'), 44)\n",
      "(('remains', 'determined'), 44)\n",
      "(('suggest', 'mutations'), 44)\n",
      "(('complementary', 'dna'), 44)\n",
      "(('expression', 'profiles'), 44)\n",
      "(('(fig', 'taken'), 44)\n",
      "(('significant', 'effect'), 44)\n",
      "(('(fig', 'similar'), 44)\n",
      "(('(table', '2),'), 44)\n",
      "(('mutation', 'could'), 44)\n",
      "(('gtpase', 'activity'), 44)\n",
      "(('1', 'h'), 44)\n",
      "(('(fig', 'also'), 44)\n",
      "(('growth', '(fig'), 44)\n",
      "(('would', 'also'), 44)\n",
      "(('identified', 'patient'), 44)\n",
      "(('role', 'pathogenesis'), 44)\n",
      "(('remains', 'unclear'), 44)\n",
      "(('affected', 'individuals'), 44)\n",
      "(('kinase', 'complex'), 44)\n",
      "(('mutant', 'receptor'), 44)\n",
      "(('lung', 'carcinoma'), 44)\n",
      "(('formation', 'assay'), 44)\n",
      "(('fusion', 'transcript'), 44)\n",
      "(('validation', 'set'), 44)\n",
      "(('(mmr)', 'genes'), 44)\n",
      "(('prostate', 'breast'), 44)\n",
      "(('exon', '7'), 44)\n",
      "(('also', 'suggested'), 44)\n",
      "(('mice', 'developed'), 44)\n",
      "(('colony', 'growth'), 44)\n",
      "(('one', 'possible'), 44)\n",
      "(('used', 'wild-type'), 44)\n",
      "(('association', 'egfr'), 44)\n",
      "(('levels', 'protein'), 44)\n",
      "(('wild-type', 'protein,'), 44)\n",
      "(('patients', 'harbored'), 44)\n",
      "(('phosphorylation', 'egfr'), 44)\n",
      "(('inhibitors', 'targeting'), 44)\n",
      "(('measured', 'levels'), 44)\n",
      "(('tyrosine', 'residues,'), 44)\n",
      "(('phosphatidylinositol', '3-kinase'), 44)\n",
      "(('clinical', 'samples'), 44)\n",
      "(('data', 'showed'), 44)\n",
      "(('single', 'agent'), 44)\n",
      "(('may', 'even'), 44)\n",
      "(('however,', 'two'), 44)\n",
      "(('tumors,', 'including'), 44)\n",
      "(('stable', 'disease'), 44)\n",
      "(('whose', 'tumor'), 44)\n",
      "(('1', 'patient'), 44)\n",
      "(('(13,', '14)'), 44)\n",
      "(('lung', 'adenocarcinoma,'), 44)\n",
      "(('pr,', 'partial'), 44)\n",
      "(('molecular', 'target'), 44)\n",
      "(('useful', 'predicting'), 44)\n",
      "(('showed', 'constitutive'), 44)\n",
      "(('cells', 'expressed'), 44)\n",
      "(('provide', 'first'), 44)\n",
      "(('potent', 'inhibitor'), 44)\n",
      "(('primary', 'secondary'), 44)\n",
      "(('levels', 'expression'), 44)\n",
      "(('cells', 'cell'), 44)\n",
      "(('exhibited', 'increased'), 44)\n",
      "(('less', 'potent'), 44)\n",
      "(('al', 'recently,'), 44)\n",
      "(('imatinib', 'therapy'), 44)\n",
      "(('pdgfra', 'exon'), 44)\n",
      "(('secondary', 'kit'), 44)\n",
      "(('transcription', 'activation'), 44)\n",
      "(('equal', 'higher'), 44)\n",
      "(('yeast', 'two-hybrid'), 43)\n",
      "(('significant', '(p'), 43)\n",
      "(('aml', 'patients'), 43)\n",
      "(('may', 'prevent'), 43)\n",
      "(('mutants', 'displayed'), 43)\n",
      "(('figure', 'viewerdownload'), 43)\n",
      "(('viewerdownload', 'powerpoint'), 43)\n",
      "(('similar', 'effects'), 43)\n",
      "(('tandem', 'duplication'), 43)\n",
      "(('proteins', '(fig'), 43)\n",
      "(('cell', 'membrane'), 43)\n",
      "(('mutations', 'various'), 43)\n",
      "(('(table', 's2)'), 43)\n",
      "(('exon', '5'), 43)\n",
      "(('southern', 'blot'), 43)\n",
      "(('amplification', 'observed'), 43)\n",
      "(('prognostic', 'factor'), 43)\n",
      "(('mutations', 'caused'), 43)\n",
      "(('complex', 'formation'), 43)\n",
      "(('slightly', 'lower'), 43)\n",
      "(('inhibitors', 'may'), 43)\n",
      "(('patients', 'mutation'), 43)\n",
      "(('consistent', 'observation'), 43)\n",
      "(('page', 'results'), 43)\n",
      "(('mutation', 'observed'), 43)\n",
      "(('patients', 'screened'), 43)\n",
      "(('coding', 'regions'), 43)\n",
      "(('48', 'hours'), 43)\n",
      "(('amplification', 'overexpression'), 43)\n",
      "(('three', 'four'), 43)\n",
      "(('provide', 'insight'), 43)\n",
      "(('revealed', 'mutations'), 43)\n",
      "(('mutations', 'mismatch'), 43)\n",
      "(('causing', 'lynch'), 43)\n",
      "(('four', 'mmr'), 43)\n",
      "(('risk', 'colorectal'), 43)\n",
      "(('dna', 'samples'), 43)\n",
      "(('also', 'failed'), 43)\n",
      "(('activity', 'cell'), 43)\n",
      "(('mutation', 'patients'), 43)\n",
      "(('primer', 'sequences'), 43)\n",
      "(('3', 'cases'), 43)\n",
      "(('proteins,', 'including'), 43)\n",
      "(('cells', 'may'), 43)\n",
      "(('protein', 'lysates'), 43)\n",
      "(('report', 'identification'), 43)\n",
      "(('cells', 'patients'), 43)\n",
      "(('recurrently', 'mutated'), 43)\n",
      "(('associated', 'sensitivity'), 43)\n",
      "(('egfr', 'activation'), 43)\n",
      "(('egfr', 'expression'), 43)\n",
      "(('tumor', 'responses'), 43)\n",
      "(('absence', 'ligand,'), 43)\n",
      "(('study,', 'found'), 43)\n",
      "(('ability', 'predict'), 43)\n",
      "(('leading', 'constitutive'), 43)\n",
      "(('show', 'increased'), 43)\n",
      "(('made', 'possible'), 43)\n",
      "(('1', 'fig'), 43)\n",
      "(('still', 'unclear'), 43)\n",
      "(('additional', 'genetic'), 43)\n",
      "(('cells', 'analyzed'), 43)\n",
      "(('embryonic', 'kidney'), 43)\n",
      "(('cells', 'show'), 43)\n",
      "(('chromosomal', 'rearrangements'), 43)\n",
      "(('p53', 'mutants'), 43)\n",
      "(('mutation', 'codon'), 43)\n",
      "(('recently', 'determined'), 43)\n",
      "(('gave', 'rise'), 43)\n",
      "(('mutations', 'characterized'), 43)\n",
      "(('mutant', 'kit'), 43)\n",
      "(('complexes', 'containing'), 43)\n",
      "(('leukemia', 'progression'), 43)\n",
      "(('xenopus', 'laevis'), 43)\n",
      "(('within', 'nh2-terminal'), 43)\n",
      "(('(residues', '186351),'), 43)\n",
      "(('cdna', 'missense'), 43)\n",
      "(('double', 'mutation'), 42)\n",
      "(('lesser', 'extent'), 42)\n",
      "(('supplemental', 'figure'), 42)\n",
      "(('activation', 'akt'), 42)\n",
      "(('determined', 'whether'), 42)\n",
      "(('located', 'exon'), 42)\n",
      "(('intracellular', 'signaling'), 42)\n",
      "(('cells', 'significantly'), 42)\n",
      "(('autosomal', 'dominant'), 42)\n",
      "(('performed', 'described'), 42)\n",
      "(('mutations,', 'also'), 42)\n",
      "(('cancer', 'types,'), 42)\n",
      "(('prognostic', 'value'), 42)\n",
      "(('tumors', 'showed'), 42)\n",
      "(('small', 'molecules'), 42)\n",
      "(('commonly', 'mutated'), 42)\n",
      "(('mutations', 'respect'), 42)\n",
      "(('various', 'types'), 42)\n",
      "(('interfering', 'rna'), 42)\n",
      "(('(supplemental', 'table'), 42)\n",
      "(('catalytic', 'subunit'), 42)\n",
      "(('using', 'site-directed'), 42)\n",
      "(('downstream', 'target'), 42)\n",
      "(('predisposition', 'cancer'), 42)\n",
      "(('genomic', 'instability'), 42)\n",
      "(('index', 'cases'), 42)\n",
      "(('expression', 'levels,'), 42)\n",
      "(('mutants', 'generated'), 42)\n",
      "(('(fig', 'therefore,'), 42)\n",
      "(('6', '7'), 42)\n",
      "(('heterozygous', 'missense'), 42)\n",
      "(('ligand-binding', 'domain'), 42)\n",
      "(('mutation', 'alters'), 42)\n",
      "(('sample', 'size'), 42)\n",
      "(('retroviral', 'vector'), 42)\n",
      "(('levels', '(fig'), 42)\n",
      "(('(online', 'methods'), 42)\n",
      "(('(d)', 'four'), 42)\n",
      "(('carriers', 'mutations'), 42)\n",
      "(('class', '5'), 42)\n",
      "(('binding', 'studies'), 42)\n",
      "(('control', 'cell'), 42)\n",
      "(('mice', 'expressing'), 42)\n",
      "(('mechanism', 'underlying'), 42)\n",
      "(('alone', 'combination'), 42)\n",
      "(('preclinical', 'studies'), 42)\n",
      "(('deletion', 'exons'), 42)\n",
      "(('mutation', 'significantly'), 42)\n",
      "(('receptor', 'mutation'), 42)\n",
      "(('never', 'smokers,'), 42)\n",
      "(('types', 'cancer'), 42)\n",
      "(('egfr', 'proteins'), 42)\n",
      "(('patient', 'characteristics'), 42)\n",
      "(('5', 'patients'), 42)\n",
      "(('egfr', '(table'), 42)\n",
      "(('sequencing', 'egfr'), 42)\n",
      "(('11', 'patients'), 42)\n",
      "(('10-fold', 'lower'), 42)\n",
      "(('33', 'patients'), 42)\n",
      "(('exon', '18,'), 42)\n",
      "(('nsclc', 'patient'), 42)\n",
      "(('patients', 'clinical'), 42)\n",
      "(('11', 'exon'), 42)\n",
      "(('ligand', 'stimulation'), 42)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('three', 'distinct'), 42)\n",
      "(('cruz', 'biotechnology'), 42)\n",
      "(('may', 'benefit'), 42)\n",
      "(('20,', '21)'), 42)\n",
      "(('assay', '(figure'), 42)\n",
      "(('to:', 'materials'), 42)\n",
      "(('investigated', 'effect'), 42)\n",
      "(('whole', 'cell'), 42)\n",
      "(('levels', 'similar'), 42)\n",
      "(('distinct', 'mechanisms'), 42)\n",
      "(('therapeutic', 'targets'), 42)\n",
      "(('protein', 'function'), 42)\n",
      "(('nonsense', 'mutations'), 42)\n",
      "(('independent', 'prognostic'), 42)\n",
      "(('mutations', 'occurring'), 42)\n",
      "(('brca2', 'mutations'), 42)\n",
      "(('next', 'tested'), 42)\n",
      "(('cell', 'lymphoma'), 42)\n",
      "(('cell', 'leukemia'), 42)\n",
      "(('biotechnology', 'information'), 42)\n",
      "(('center', 'biotechnology'), 42)\n",
      "(('pcr', '(29)'), 42)\n",
      "(('v133i,', 'm134l,'), 42)\n",
      "(('(residues', '7185),'), 42)\n",
      "(('7185),', '9'), 42)\n",
      "(('within', 'coohterminal'), 42)\n",
      "(('v369g,', 't401i)'), 42)\n",
      "(('g20e,', 'y27s,'), 42)\n",
      "(('31', 'mutations'), 42)\n",
      "(('f347l,', 'v369g,'), 42)\n",
      "(('l112p,', 'l112r,'), 42)\n",
      "(('c136y,', 'y155c,'), 42)\n",
      "(('l42r,', 'h61r,'), 42)\n",
      "(('k342n,', 'v343e,'), 42)\n",
      "(('d331g,', 'f341v,'), 42)\n",
      "(('c71y,', 'h93y,'), 42)\n",
      "(('h93y,', 'c105f,'), 42)\n",
      "(('mutations', '(s227f,'), 42)\n",
      "(('y16c,', 'g20e,'), 42)\n",
      "(('s227f,', 'g251c,'), 42)\n",
      "(('186351),', '2'), 42)\n",
      "(('mutations', '(v369g'), 42)\n",
      "(('y27s,', 'l42r,'), 42)\n",
      "(('y155c,', 'g165r,'), 42)\n",
      "(('v343e,', 'l345q,'), 42)\n",
      "(('t401i)', 'close'), 42)\n",
      "(('t401i)', 'constructed'), 42)\n",
      "(('r173p,', 'y174n)'), 42)\n",
      "(('m134l,', 'c136y,'), 42)\n",
      "(('r130l,', 'r130q,'), 42)\n",
      "(('(s10n,', 'y16c,'), 42)\n",
      "(('c124r,', 'g129r,'), 42)\n",
      "(('f341v,', 'k342n,'), 42)\n",
      "(('g251c,', 'k289e,'), 42)\n",
      "(('coohterminal', 'c2'), 42)\n",
      "(('r173c,', 'r173h,'), 42)\n",
      "(('g129r,', 'g129e,'), 42)\n",
      "(('megaprimer', 'pcr'), 42)\n",
      "(('close', 'cooh-terminal'), 42)\n",
      "(('f347l)', 'within'), 42)\n",
      "(('nh2-terminal', 'phosphatase'), 42)\n",
      "(('g165r,', 's170n,'), 42)\n",
      "(('y174n)', 'within'), 42)\n",
      "(('s170n,', 's170r,'), 42)\n",
      "(('r173p,', 'y174n,'), 42)\n",
      "(('d107y,', 'l112p,'), 42)\n",
      "(('c105f,', 'd107y,'), 42)\n",
      "(('l345q,', 'f347l,'), 42)\n",
      "(('constructed', 'megaprimer'), 42)\n",
      "(('l112r,', 'a121p,'), 42)\n",
      "(('l345q,', 'f347l)'), 42)\n",
      "(('a121p,', 'c124r,'), 42)\n",
      "(('mutations', '(s10n,'), 42)\n",
      "(('r130g,', 'r130l,'), 42)\n",
      "(('(v369g', 't401i)'), 42)\n",
      "(('r173h,', 'r173p,'), 42)\n",
      "(('s170r,', 'r173c,'), 42)\n",
      "(('y68h,', 'c71y,'), 42)\n",
      "(('y174n,', 's227f,'), 42)\n",
      "(('k289e,', 'd331g,'), 42)\n",
      "(('h61r,', 'y68h,'), 42)\n",
      "(('r130q,', 'v133i,'), 42)\n",
      "(('(s227f,', 'g251c,'), 42)\n",
      "(('mutations', 'brca1,'), 42)\n",
      "(('resulted', 'increased'), 41)\n",
      "(('cells', '(data'), 41)\n",
      "(('cells', 'presence'), 41)\n",
      "(('(table', '1;'), 41)\n",
      "(('cell', 'receptor'), 41)\n",
      "(('7', '8'), 41)\n",
      "(('functional', 'properties'), 41)\n",
      "(('(see', 'figure'), 41)\n",
      "(('thyroid', 'cancer'), 41)\n",
      "(('relatively', 'low'), 41)\n",
      "(('found', 'many'), 41)\n",
      "(('amplification', 'found'), 41)\n",
      "(('led', 'increased'), 41)\n",
      "(('embryonic', 'stem'), 41)\n",
      "(('may', 'sufficient'), 41)\n",
      "(('al', 'based'), 41)\n",
      "(('mutations', '2'), 41)\n",
      "(('similar', 'reported'), 41)\n",
      "(('found', 'within'), 41)\n",
      "(('xenograft', 'model'), 41)\n",
      "(('tumor', 'growth,'), 41)\n",
      "(('exon', '6'), 41)\n",
      "(('mouse', 'monoclonal'), 41)\n",
      "(('assessed', 'whether'), 41)\n",
      "(('non-small-cell', 'lung'), 41)\n",
      "(('identified', 'several'), 41)\n",
      "(('variant', 'carrier'), 41)\n",
      "(('activates', 'transcription'), 41)\n",
      "(('alone', 'sufficient'), 41)\n",
      "(('type', 'ii'), 41)\n",
      "(('3', 'end'), 41)\n",
      "(('san', 'diego,'), 41)\n",
      "(('relevant', 'information'), 41)\n",
      "(('causality', 'segregation'), 41)\n",
      "(('probabilities', 'pathogenicity'), 41)\n",
      "(('variants', 'included'), 41)\n",
      "(('criteria', 'required'), 41)\n",
      "(('cancers1,', '2,'), 41)\n",
      "(('small', 'bowel,'), 41)\n",
      "(('information', 'cancer'), 41)\n",
      "(('published', 'unpublished'), 41)\n",
      "(('algorithm', 'based'), 41)\n",
      "(('three', 'five'), 41)\n",
      "(('target', 'genes,'), 41)\n",
      "(('two', 'cell'), 41)\n",
      "(('identity', 'human'), 41)\n",
      "(('phosphorylation', 'sites'), 41)\n",
      "(('shown', 'mutations'), 41)\n",
      "(('functional', 'activity'), 41)\n",
      "(('cell', 'populations'), 41)\n",
      "(('four', 'amino'), 41)\n",
      "(('adenocarcinoma', 'cell'), 41)\n",
      "(('mutations', 'account'), 41)\n",
      "(('nsclc', 'egfr'), 41)\n",
      "(('presence', 'egfr'), 41)\n",
      "(('erbb', 'family'), 41)\n",
      "(('screened', 'egfr'), 41)\n",
      "(('lysates', 'cells'), 41)\n",
      "(('three', 'tumors'), 41)\n",
      "(('experimental', 'design:'), 41)\n",
      "(('egfr', 'protein'), 41)\n",
      "(('oncology', 'group'), 41)\n",
      "(('might', 'important'), 41)\n",
      "(('clinical', 'trial,'), 41)\n",
      "(('1', '10'), 41)\n",
      "(('patients', 'exon'), 41)\n",
      "(('16', 'patients'), 41)\n",
      "(('found', 'harbor'), 41)\n",
      "(('binding', 'sites'), 41)\n",
      "(('different', 'concentrations'), 41)\n",
      "(('using', 'dna'), 41)\n",
      "(('recent', 'data'), 41)\n",
      "(('site', 'mutation'), 41)\n",
      "(('phosphorylated', 'tyrosine'), 41)\n",
      "(('clinical', 'management'), 41)\n",
      "(('data', 'consistent'), 41)\n",
      "(('intracellular', 'domain'), 41)\n",
      "(('second', 'mutation'), 41)\n",
      "(('human', 'embryonic'), 41)\n",
      "(('section', 'results'), 41)\n",
      "(('gene', 'rearrangements'), 41)\n",
      "(('molecular', 'basis'), 41)\n",
      "(('results', 'identification'), 41)\n",
      "(('missense', 'substitutions'), 41)\n",
      "(('mek', 'inhibition'), 41)\n",
      "(('fig', 'expression'), 41)\n",
      "(('reported', 'recently'), 41)\n",
      "(('5', '3'), 41)\n",
      "(('lysates', 'prepared'), 41)\n",
      "(('leads', 'tumor'), 41)\n",
      "(('national', 'center'), 41)\n",
      "(('cooh-terminal', 'end'), 41)\n",
      "(('cyclin', 'd1'), 40)\n",
      "(('two', 'common'), 40)\n",
      "(('cells', 'increased'), 40)\n",
      "(('mutants', 'may'), 40)\n",
      "(('activation', 'erk'), 40)\n",
      "(('inhibitory', 'effects'), 40)\n",
      "(('provide', 'strong'), 40)\n",
      "(('associated', 'higher'), 40)\n",
      "(('b', 'c'), 40)\n",
      "(('braf', 'v600e'), 40)\n",
      "(('lung', 'carcinomas'), 40)\n",
      "(('mrna', 'level'), 40)\n",
      "(('copy', 'numbers'), 40)\n",
      "(('full', 'length'), 40)\n",
      "(('fig', 'results'), 40)\n",
      "(('majority', 'patients'), 40)\n",
      "(('distribution', 'mutations'), 40)\n",
      "(('transfected', 'wt'), 40)\n",
      "(('may', 'disrupt'), 40)\n",
      "(('familial', 'breast'), 40)\n",
      "(('may', 'indicate'), 40)\n",
      "(('cultured', 'cells'), 40)\n",
      "(('p53', 'protein'), 40)\n",
      "(('small', 'cell'), 40)\n",
      "(('used', 'amplify'), 40)\n",
      "(('mutated', 'genes'), 40)\n",
      "(('showed', 'high'), 40)\n",
      "(('13', '14'), 40)\n",
      "(('study,', 'identified'), 40)\n",
      "(('robustness', 'qualitative'), 40)\n",
      "(('variants', 'class'), 40)\n",
      "(('(ae)', 'distribution'), 40)\n",
      "(('assess', 'mlh1,'), 40)\n",
      "(('(6%)', 'variants'), 40)\n",
      "(('scores34', 'mlh1,'), 40)\n",
      "(('msh6,', 'n'), 40)\n",
      "(('segregation', 'data16,'), 40)\n",
      "(('classification', 'criteria,'), 40)\n",
      "(('stratification', 'variants'), 40)\n",
      "(('prioritize', 'variants'), 40)\n",
      "(('169', '(b),'), 40)\n",
      "(('80%', 'prioritize'), 40)\n",
      "(('lynch', 'syndrome1'), 40)\n",
      "(('criteria,', '170'), 40)\n",
      "(('qualitative', 'classification'), 40)\n",
      "(('5', '(pathogenic)'), 40)\n",
      "(('(39%),', '842'), 40)\n",
      "(('(e),', 'showing'), 40)\n",
      "(('genes', '(e),'), 40)\n",
      "(('pms2', 'causing'), 40)\n",
      "(('(68', 'mlh1,'), 40)\n",
      "(('mlh1,', 'n'), 40)\n",
      "(('bowel,', 'urothelial,'), 40)\n",
      "(('unpublished', 'pedigrees'), 40)\n",
      "(('cancer', 'variant'), 40)\n",
      "(('required', 'assignment'), 40)\n",
      "(('hepatobiliary,', 'pancreatic,'), 40)\n",
      "(('n', '24'), 40)\n",
      "(('demonstrate', 'robustness'), 40)\n",
      "(('(c),', 'pms2,'), 40)\n",
      "(('breast', 'cancers1,'), 40)\n",
      "(('13', 'msh6'), 40)\n",
      "(('8', '2,360'), 40)\n",
      "(('syndrome1', 'substantially'), 40)\n",
      "(('assumed', 'pathogenic'), 40)\n",
      "(('investigation', 'classification'), 40)\n",
      "(('msh2,', '13'), 40)\n",
      "(('distribution', 'probabilities'), 40)\n",
      "(('probabilities', '20%'), 40)\n",
      "(('risk', 'ovarian,'), 40)\n",
      "(('(19%)', '137'), 40)\n",
      "(('data16,', '51'), 40)\n",
      "(('932', 'mlh1'), 40)\n",
      "(('n', '169'), 40)\n",
      "(('msh2', '(36%),'), 40)\n",
      "(('n', '145'), 40)\n",
      "(('variants', 'investigation'), 40)\n",
      "(('(a),', 'msh2,'), 40)\n",
      "(('bladder,', 'kidney,'), 40)\n",
      "(('mlh1', '(39%),'), 40)\n",
      "(('urothelial,', 'brain,'), 40)\n",
      "(('pancreatic,', 'bladder,'), 40)\n",
      "(('class', '5b'), 40)\n",
      "(('ovarian,', 'gastric,'), 40)\n",
      "(('pms2,', 'n'), 40)\n",
      "(('n', '186'), 40)\n",
      "(('cancers,', 'along'), 40)\n",
      "(('variant', 'causality'), 40)\n",
      "(('based', 'customized'), 40)\n",
      "(('(pathogenic)', 'qualitative'), 40)\n",
      "(('brain,', 'hepatobiliary,'), 40)\n",
      "(('pms2', '(6%)'), 40)\n",
      "(('customized', 'mapp'), 40)\n",
      "(('75', 'msh2,'), 40)\n",
      "(('170', 'assumed'), 40)\n",
      "(('msh2,', 'n'), 40)\n",
      "(('msh6', '14'), 40)\n",
      "(('sufficient', 'relevant'), 40)\n",
      "(('msh6', '(19%)'), 40)\n",
      "(('145', '(c),'), 40)\n",
      "(('186', '(a),'), 40)\n",
      "(('137', 'pms2'), 40)\n",
      "(('estimated', 'calibrated'), 40)\n",
      "(('qualitative', 'criteria'), 40)\n",
      "(('pms2)', 'reviewed'), 40)\n",
      "(('variants', '(68'), 40)\n",
      "(('(36%),', '449'), 40)\n",
      "(('2,360', 'constitutional'), 40)\n",
      "(('51', 'published'), 40)\n",
      "(('constitutional', 'variants'), 40)\n",
      "(('842', 'msh2'), 40)\n",
      "(('set', 'class'), 40)\n",
      "(('20%', '80%'), 40)\n",
      "(('mapp', 'polyphen-2'), 40)\n",
      "(('kidney,', 'prostate'), 40)\n",
      "(('assignment', 'variants'), 40)\n",
      "(('(b),', 'msh6,'), 40)\n",
      "(('pathogenicity', 'estimated'), 40)\n",
      "(('included', '932'), 40)\n",
      "(('mlh1,', '75'), 40)\n",
      "(('polyphen-2', 'scores34'), 40)\n",
      "(('pedigrees', 'sufficient'), 40)\n",
      "(('variants', 'prior'), 40)\n",
      "(('showing', 'stratification'), 40)\n",
      "(('calibrated', 'algorithm'), 40)\n",
      "(('reviewed', 'validation'), 40)\n",
      "(('5b', '(online'), 40)\n",
      "(('pms2', 'variant'), 40)\n",
      "(('449', 'msh6'), 40)\n",
      "(('14', 'pms2)'), 40)\n",
      "(('gastric,', 'small'), 40)\n",
      "(('24', '(d)'), 40)\n",
      "(('slightly', 'higher'), 40)\n",
      "(('two', 'three'), 40)\n",
      "(('may', 'alter'), 40)\n",
      "(('mutations', 'generally'), 40)\n",
      "(('median', 'survival'), 40)\n",
      "(('phosphorylation', '(fig'), 40)\n",
      "(('normal', 'cell'), 40)\n",
      "(('egfr', 'mutational'), 40)\n",
      "(('differential', 'sensitivity'), 40)\n",
      "(('within', 'extracellular'), 40)\n",
      "(('sensitivity', 'mutant'), 40)\n",
      "(('cancer', 'center'), 40)\n",
      "(('found', 'egfr'), 40)\n",
      "(('never', 'smokers'), 40)\n",
      "(('wild-type', 'mutated'), 40)\n",
      "(('basis', 'data'), 40)\n",
      "(('adenosine', 'triphosphate'), 40)\n",
      "(('well', 'characterized'), 40)\n",
      "(('assay', 'used'), 40)\n",
      "(('cells', 'transformed'), 40)\n",
      "(('activation', 'mechanism'), 40)\n",
      "(('tki', 'treatment'), 40)\n",
      "(('three', 'mutants'), 40)\n",
      "(('cancer', 'association'), 40)\n",
      "(('sd,', 'stable'), 40)\n",
      "(('noted', 'above,'), 40)\n",
      "(('cells', 'harbor'), 40)\n",
      "(('case', 'report'), 40)\n",
      "(('highest', 'level'), 40)\n",
      "(('fusions', 'involving'), 40)\n",
      "(('data', 'also'), 40)\n",
      "(('determined', 'structures'), 40)\n",
      "(('alternative', 'mechanism'), 40)\n",
      "(('ligand-independent', 'manner'), 40)\n",
      "(('drug', 'binding'), 40)\n",
      "(('expression', 'plasmid'), 40)\n",
      "(('genomic', 'amplification'), 40)\n",
      "(('fusion', 'transcripts'), 40)\n",
      "(('fusion', 'partner'), 40)\n",
      "(('samples', 'patients'), 40)\n",
      "(('fusion', 'genes'), 40)\n",
      "(('(fig', '(a)'), 40)\n",
      "(('two', 'human'), 40)\n",
      "(('potential', 'role'), 40)\n",
      "(('cdna', 'cloned'), 40)\n",
      "(('cytoplasmic', 'localization'), 40)\n",
      "(('failed', 'detect'), 40)\n",
      "(('manufacturers', 'instructions'), 40)\n",
      "(('either', 'one'), 40)\n",
      "(('notch', 'signaling'), 40)\n",
      "(('data', 'presented'), 40)\n",
      "(('disease-free', 'survival'), 40)\n",
      "(('hinge', 'region'), 40)\n",
      "(('(life', 'technologies)'), 40)\n",
      "(('response', 'imatinib'), 40)\n",
      "(('biological', 'significance'), 40)\n",
      "(('jm', 'domain'), 40)\n",
      "(('11', 'del'), 40)\n",
      "(('laboratory', 'others'), 40)\n",
      "(('frequency', 'control'), 40)\n",
      "(('classified', 'deleterious'), 40)\n",
      "(('predicting', 'cancer'), 40)\n",
      "(('metastatic', 'melanoma'), 40)\n",
      "(('brca1', 'mutants'), 40)\n",
      "(('brca1', 'constructs'), 40)\n",
      "(('m1775r,', 'y1853x,'), 40)\n",
      "(('11', 'v560d'), 40)\n",
      "(('kit', 'kinase'), 40)\n",
      "(('viable', 'cells'), 39)\n",
      "(('ubiquitin', 'ligase'), 39)\n",
      "(('stably', 'transduced'), 39)\n",
      "(('status', 'clinical'), 39)\n",
      "(('xl', 'site-directed'), 39)\n",
      "(('first', 'reported'), 39)\n",
      "(('family', 'member'), 39)\n",
      "(('domain', 'also'), 39)\n",
      "(('three', 'cases'), 39)\n",
      "(('table', '5'), 39)\n",
      "(('mutations', 'melanoma'), 39)\n",
      "(('age', 'diagnosis'), 39)\n",
      "(('data', 'suggested'), 39)\n",
      "(('differentially', 'expressed'), 39)\n",
      "(('expression', 'profiling'), 39)\n",
      "(('side', 'chains'), 39)\n",
      "(('hydrogen', 'bonds'), 39)\n",
      "(('led', 'us'), 39)\n",
      "(('also', 'expressed'), 39)\n",
      "(('tumors', 'harbor'), 39)\n",
      "(('tumor', 'types,'), 39)\n",
      "(('primary', 'lung'), 39)\n",
      "(('transgenic', 'mice'), 39)\n",
      "(('2', '1'), 39)\n",
      "(('c', 'g'), 39)\n",
      "(('found', 'mutations'), 39)\n",
      "(('0', 'thus,'), 39)\n",
      "(('tested', 'ability'), 39)\n",
      "(('also', 'confirmed'), 39)\n",
      "(('pancreatic', 'cancer'), 39)\n",
      "(('mutations', 'similar'), 39)\n",
      "(('expressed', 'cell'), 39)\n",
      "(('strongly', 'suggests'), 39)\n",
      "(('uncertain', 'significance'), 39)\n",
      "(('mutations', 'outside'), 39)\n",
      "(('brca1,', 'brca2,'), 39)\n",
      "(('expression', 'wt'), 39)\n",
      "(('levels', 'endogenous'), 39)\n",
      "(('two', 'genes'), 39)\n",
      "(('cell', 'differentiation'), 39)\n",
      "(('selectively', 'inhibits'), 39)\n",
      "(('egfr', '(p'), 39)\n",
      "(('normal', 'lung'), 39)\n",
      "(('21', 'egfr'), 39)\n",
      "(('also', 'demonstrate'), 39)\n",
      "(('essential', 'tumor'), 39)\n",
      "(('completely', 'inhibited'), 39)\n",
      "(('frequency', 'egfr'), 39)\n",
      "(('mutants', 'vitro'), 39)\n",
      "(('mutation', 'mutation'), 39)\n",
      "(('nsclc', 'samples'), 39)\n",
      "(('expression', 'egfr'), 39)\n",
      "(('responses', 'gefitinib'), 39)\n",
      "(('(17,', '18)'), 39)\n",
      "(('structure', 'egfr'), 39)\n",
      "(('patients', 'demonstrated'), 39)\n",
      "(('cells', 'egfr'), 39)\n",
      "(('particularly', 'sensitive'), 39)\n",
      "(('amplification', '(fig'), 39)\n",
      "(('3', 'patients'), 39)\n",
      "(('binding', 'mode'), 39)\n",
      "(('clinical', 'studies'), 39)\n",
      "(('egfr', 'inhibitor'), 39)\n",
      "(('table', '4)'), 39)\n",
      "(('gene', 'chromosome'), 39)\n",
      "(('exon', '19,'), 39)\n",
      "(('effectively', 'inhibited'), 39)\n",
      "(('constructs', 'encoding'), 39)\n",
      "(('overall', 'survival,'), 39)\n",
      "(('complex', 'mutation'), 39)\n",
      "(('hydrophobic', 'core'), 39)\n",
      "(('using', 'two'), 39)\n",
      "(('occur', 'frequently'), 39)\n",
      "(('constitutive', 'activity'), 39)\n",
      "(('five', 'mutations'), 39)\n",
      "(('could', 'induce'), 39)\n",
      "(('control', 'vector'), 39)\n",
      "(('time', 'point'), 39)\n",
      "(('tumors', 'expressing'), 39)\n",
      "(('cells', 'resistant'), 39)\n",
      "(('one', 'cell'), 39)\n",
      "(('originally', 'identified'), 39)\n",
      "(('family', 'members,'), 39)\n",
      "(('firefly', 'luciferase'), 39)\n",
      "(('mutations', 'alk'), 39)\n",
      "(('al', 'hand,'), 39)\n",
      "(('showed', 'clear'), 39)\n",
      "(('(3,', '4,'), 39)\n",
      "(('phosphorylation', 'erk'), 39)\n",
      "(('increase', 'ability'), 39)\n",
      "(('slightly', 'increased'), 39)\n",
      "(('alk', 'kinase'), 39)\n",
      "(('t-cell', 'acute'), 39)\n",
      "(('least', 'partially,'), 39)\n",
      "(('(appendix', 'table'), 39)\n",
      "(('wt', 'cells'), 39)\n",
      "(('different', 'cell'), 39)\n",
      "(('transcription', 'reporter'), 39)\n",
      "(('differences', 'among'), 39)\n",
      "(('cells', 'stimulated'), 39)\n",
      "(('may', 'therefore'), 39)\n",
      "(('ask', 'whether'), 39)\n",
      "(('domains', 'seven'), 39)\n",
      "(('functional', 'domains'), 39)\n",
      "(('patients', 'secondary'), 39)\n",
      "(('progression', 'disease'), 39)\n",
      "(('2', ''), 39)\n",
      "(('(1', '2)'), 39)\n",
      "(('impact', 'transcriptional'), 39)\n",
      "(('brca1', 'nuclear'), 39)\n",
      "(('somatic', 'mutations,'), 38)\n",
      "(('express', 'wild-type'), 38)\n",
      "(('well', 'established'), 38)\n",
      "(('cells,', 'even'), 38)\n",
      "(('negative', 'regulator'), 38)\n",
      "(('engineered', 'express'), 38)\n",
      "(('kb)', 'previous'), 38)\n",
      "(('(n', '3)'), 38)\n",
      "(('providing', 'evidence'), 38)\n",
      "(('two', 'studies'), 38)\n",
      "(('transcriptional', 'regulation'), 38)\n",
      "(('ovarian', 'tumors'), 38)\n",
      "(('(fig', 'go'), 38)\n",
      "(('cells', 'seeded'), 38)\n",
      "(('hotspot', 'mutation'), 38)\n",
      "(('heterozygous', 'deletion'), 38)\n",
      "(('progenitor', 'cells'), 38)\n",
      "(('cells', 'vivo'), 38)\n",
      "(('surface', 'expression'), 38)\n",
      "(('close', 'proximity'), 38)\n",
      "(('oncogenic', 'transformation'), 38)\n",
      "(('structural', 'basis'), 38)\n",
      "(('coiled-coil', 'domain'), 38)\n",
      "(('mutations', 'figure'), 38)\n",
      "(('increase', 'number'), 38)\n",
      "(('future', 'studies'), 38)\n",
      "(('shown', 'previously'), 38)\n",
      "(('found', 'approximately'), 38)\n",
      "(('(figure', '4d)'), 38)\n",
      "(('kindly', 'provided'), 38)\n",
      "(('gene', 'fused'), 38)\n",
      "(('identification', 'novel'), 38)\n",
      "(('homozygous', 'deletions'), 38)\n",
      "(('1', 'clinical'), 38)\n",
      "(('two', 'proteins'), 38)\n",
      "(('mutational', 'analyses'), 38)\n",
      "(('insertion', 'mutation'), 38)\n",
      "(('consistent', 'this,'), 38)\n",
      "(('pathways', 'activated'), 38)\n",
      "(('transfected', 'expression'), 38)\n",
      "(('hairpin', 'rna'), 38)\n",
      "(('glutamic', 'acid'), 38)\n",
      "(('significance', 'mutations'), 38)\n",
      "(('deletions', 'also'), 38)\n",
      "(('atp-binding', 'pocket'), 38)\n",
      "(('exons', '2'), 38)\n",
      "(('patients', 'responded'), 38)\n",
      "(('egfr', 'patients'), 38)\n",
      "(('study', 'functional'), 38)\n",
      "(('contained', 'mutations'), 38)\n",
      "(('mutations,', 'determined'), 38)\n",
      "(('however,', 'data'), 38)\n",
      "(('tk', 'inhibitor'), 38)\n",
      "(('within', 'atp-binding'), 38)\n",
      "(('treatment', 'nsclc'), 38)\n",
      "(('clinical', 'use'), 38)\n",
      "(('one', 'hand,'), 38)\n",
      "(('inhibition', 'downstream'), 38)\n",
      "(('catalogue', 'somatic'), 38)\n",
      "(('median', 'pfs'), 38)\n",
      "(('lines', 'mouse'), 38)\n",
      "(('japanese', 'patients'), 38)\n",
      "(('mutations', 'wild-type'), 38)\n",
      "(('objective', 'response'), 38)\n",
      "(('one', 'three'), 38)\n",
      "(('monoclonal', 'antibodies'), 38)\n",
      "(('disease', 'control'), 38)\n",
      "(('different', 'egfr'), 38)\n",
      "(('(15,', '16)'), 38)\n",
      "(('(11,', '12)'), 38)\n",
      "(('could', 'used'), 38)\n",
      "(('functional', 'relevance'), 38)\n",
      "(('mutations', 'included'), 38)\n",
      "(('expression', 'may'), 38)\n",
      "(('reported', 'associated'), 38)\n",
      "(('studies', 'using'), 38)\n",
      "(('expression', 'mutants'), 38)\n",
      "(('receptors', '(figure'), 38)\n",
      "(('lines,', 'including'), 38)\n",
      "(('molecular', 'modeling'), 38)\n",
      "(('likely', 'respond'), 38)\n",
      "(('phosphorylated', 'akt'), 38)\n",
      "(('high', 'resolution'), 38)\n",
      "(('higher', 'cancer'), 38)\n",
      "(('functions,', 'including'), 38)\n",
      "(('shown', 'induce'), 38)\n",
      "(('change', 'structure'), 38)\n",
      "(('described', 'above,'), 38)\n",
      "(('protein', 'partners'), 38)\n",
      "(('structure', 'functional'), 38)\n",
      "(('primary', 'kit'), 38)\n",
      "(('kinase', '3'), 38)\n",
      "(('well', 'correlation'), 38)\n",
      "(('braf', 'kinase'), 38)\n",
      "(('observed', 'loss'), 38)\n",
      "(('gal4', 'dna-binding'), 38)\n",
      "(('different', 'mutants'), 38)\n",
      "(('domain', 'used'), 38)\n",
      "(('tandem', 'brct'), 38)\n",
      "(('13', 'v654a'), 38)\n",
      "(('patients', 'different'), 37)\n",
      "(('cells', 'relative'), 37)\n",
      "(('32d', 'cells'), 37)\n",
      "(('even', 'presence'), 37)\n",
      "(('cells', 'absence'), 37)\n",
      "(('determine', 'frequency'), 37)\n",
      "(('high', 'incidence'), 37)\n",
      "(('signaling', 'molecules'), 37)\n",
      "(('data', 'showing'), 37)\n",
      "(('function', 'mutations'), 37)\n",
      "(('number', 'cells'), 37)\n",
      "(('mutants', 'wild-type'), 37)\n",
      "(('breast', 'carcinoma'), 37)\n",
      "(('primary', 'melanoma'), 37)\n",
      "(('nervous', 'system'), 37)\n",
      "(('cancer', 'development'), 37)\n",
      "(('cancer', 'type'), 37)\n",
      "(('pediatric', 'patients'), 37)\n",
      "(('large', 'series'), 37)\n",
      "(('prognosis', 'patients'), 37)\n",
      "(('gene', 'dosage'), 37)\n",
      "(('following', 'treatment'), 37)\n",
      "(('cells', 'two'), 37)\n",
      "(('supporting', 'notion'), 37)\n",
      "(('loss', 'one'), 37)\n",
      "(('transcript', 'levels'), 37)\n",
      "(('sufficient', 'initiate'), 37)\n",
      "(('growth', 'soft'), 37)\n",
      "(('signal-regulated', 'kinase'), 37)\n",
      "(('also', 'present'), 37)\n",
      "(('heterozygous', 'mutation'), 37)\n",
      "(('cancer', 'predisposition'), 37)\n",
      "(('lower', 'level'), 37)\n",
      "(('tumors', 'patients'), 37)\n",
      "(('tissue', 'culture'), 37)\n",
      "(('recently', 'found'), 37)\n",
      "(('adjacent', 'normal'), 37)\n",
      "(('specific', 'mutations'), 37)\n",
      "(('number', 'studies'), 37)\n",
      "(('significant', 'correlation'), 37)\n",
      "(('pcr', 'primers'), 37)\n",
      "(('would', 'predicted'), 37)\n",
      "(('alternative', 'splicing'), 37)\n",
      "(('figure', '8'), 37)\n",
      "(('deletions', 'within'), 37)\n",
      "(('patients', 'methods'), 37)\n",
      "(('genetic', 'analysis'), 37)\n",
      "(('three', 'independent'), 37)\n",
      "(('al', 'briefly,'), 37)\n",
      "(('dna', 'synthesis'), 37)\n",
      "(('associated', 'inherited'), 37)\n",
      "(('cells', '(see'), 37)\n",
      "(('mutant', 'compared'), 37)\n",
      "(('samples', 'analyzed'), 37)\n",
      "(('mutations', 'less'), 37)\n",
      "(('tissue', 'patients'), 37)\n",
      "(('analysis', 'tumor'), 37)\n",
      "(('tumors', '(p'), 37)\n",
      "(('type', 'egfr'), 37)\n",
      "(('point', 'mutant'), 37)\n",
      "(('inhibits', 'kinase'), 37)\n",
      "(('egfr', 'exons'), 37)\n",
      "(('leads', 'activation'), 37)\n",
      "(('egfr', '(fig'), 37)\n",
      "(('mutant', 'egfr,'), 37)\n",
      "(('l858r', 'mutant'), 37)\n",
      "(('tumors', 'exon'), 37)\n",
      "(('subset', 'lung'), 37)\n",
      "(('survival', 'benefit'), 37)\n",
      "(('commonly', 'used'), 37)\n",
      "(('clinically', 'significant'), 37)\n",
      "(('analysis', 'mutations'), 37)\n",
      "(('bearing', 'mutations'), 37)\n",
      "(('mutations', 'examined'), 37)\n",
      "(('clinical', 'outcomes'), 37)\n",
      "(('analysis,', 'identified'), 37)\n",
      "(('kras', 'mutation'), 37)\n",
      "(('studies', 'focused'), 37)\n",
      "(('mutation', '(n'), 37)\n",
      "(('comprehensive', 'analysis'), 37)\n",
      "(('pcr', 'sequencing'), 37)\n",
      "(('l858r', 'exon'), 37)\n",
      "(('however,', 'studies'), 37)\n",
      "(('another', 'category'), 37)\n",
      "(('primary', 'metastatic'), 37)\n",
      "(('reduced', 'ability'), 37)\n",
      "(('parental', 'cells'), 37)\n",
      "(('cells', 'grow'), 37)\n",
      "(('conferred', 'resistance'), 37)\n",
      "(('drugs', 'target'), 37)\n",
      "(('less', 'active'), 37)\n",
      "(('suggesting', 'mutations'), 37)\n",
      "(('signal', 'transduction,'), 37)\n",
      "(('activate', 'downstream'), 37)\n",
      "(('significantly', 'greater'), 37)\n",
      "(('al,', '2006),'), 37)\n",
      "(('(9,', '10,'), 37)\n",
      "(('identified', 'patients'), 37)\n",
      "(('bank', 'code'), 37)\n",
      "(('common', 'mechanism'), 37)\n",
      "(('using', 'program'), 37)\n",
      "(('demonstrated', 'using'), 37)\n",
      "(('strongly', 'support'), 37)\n",
      "(('renilla', 'luciferase'), 37)\n",
      "(('reporter', 'plasmid'), 37)\n",
      "(('mutation', 'brca1'), 37)\n",
      "(('3', '2'), 37)\n",
      "(('activating', 'transcription'), 37)\n",
      "(('contains', 'one'), 37)\n",
      "(('domain', '(amino'), 37)\n",
      "(('levels', 'active'), 37)\n",
      "(('may', 'form'), 37)\n",
      "(('cellular', 'proliferation,'), 37)\n",
      "(('induce', 'transformation'), 37)\n",
      "(('decreased', 'tumor'), 37)\n",
      "(('regulatory', 'region'), 37)\n",
      "(('al', 'importantly,'), 37)\n",
      "(('mutant', 'kinases'), 37)\n",
      "(('order', 'assess'), 37)\n",
      "(('13,', '14,'), 37)\n",
      "(('sequence', 'database'), 37)\n",
      "(('concentrations', 'imatinib'), 37)\n",
      "(('kit', 'mutations,'), 37)\n",
      "(('resistance', 'patient'), 37)\n",
      "(('9,', '11,'), 37)\n",
      "(('imatinib', 'mesylate'), 37)\n",
      "(('receptor', 'expression'), 37)\n",
      "(('assess', 'transcriptional'), 37)\n",
      "(('expression', 'function'), 37)\n",
      "(('pathway', 'alterations'), 37)\n",
      "(('correlates', 'ability'), 37)\n",
      "(('studies', 'determined'), 37)\n",
      "(('residues', 'mutations'), 37)\n",
      "(('1', ''), 37)\n",
      "(('classified', 'pathogenic'), 37)\n",
      "(('indicate', 'tumor'), 37)\n",
      "(('brca2', 'mutation'), 37)\n",
      "(('including', 'brca1'), 37)\n",
      "(('overlapping', 'extension'), 37)\n",
      "(('mutations', 'm1775r,'), 37)\n",
      "(('template', 'modeling'), 37)\n",
      "(('mutagenesis', 'mammalian'), 37)\n",
      "(('fused', 'heterologous'), 37)\n",
      "(('w1718c', 'site-directed'), 37)\n",
      "(('sequence', '(residues'), 37)\n",
      "(('lexa', 'dna-binding'), 37)\n",
      "(('folding', 'brct'), 37)\n",
      "(('introducing', 'mutations'), 37)\n",
      "(('13961863)', 'fused'), 37)\n",
      "(('(residues', '13961863)'), 37)\n",
      "(('control', 'introducing'), 37)\n",
      "(('frame', 'lexa'), 37)\n",
      "(('y1853x,', 'm1689r,'), 37)\n",
      "(('m1689r,', 'w1718c'), 37)\n",
      "(('kit', 'activation'), 37)\n",
      "(('kit', 'mutants'), 37)\n",
      "(('deletions', 'involving'), 36)\n",
      "(('showed', 'increase'), 36)\n",
      "(('cell', 'motility'), 36)\n",
      "(('ca)', 'according'), 36)\n",
      "(('would', 'result'), 36)\n",
      "(('stably', 'express'), 36)\n",
      "(('experiments', 'showed'), 36)\n",
      "(('mutation', 'introduced'), 36)\n",
      "(('al', 'first'), 36)\n",
      "(('proteins', 'involved'), 36)\n",
      "(('mutations', 'increased'), 36)\n",
      "(('negative', 'effect'), 36)\n",
      "(('19,', '21'), 36)\n",
      "(('actually', 'observed'), 36)\n",
      "(('mice', 'compared'), 36)\n",
      "(('mutants', 'induced'), 36)\n",
      "(('novel', 'mutation'), 36)\n",
      "(('important', 'implications'), 36)\n",
      "(('mutations', 'prevalent'), 36)\n",
      "(('indicating', 'mutations'), 36)\n",
      "(('tumor', 'samples,'), 36)\n",
      "(('(fig', 'contrast'), 36)\n",
      "(('(figure', '5a)'), 36)\n",
      "(('rna', 'sequencing'), 36)\n",
      "(('cells', 'confirmed'), 36)\n",
      "(('two', 'variants'), 36)\n",
      "(('complex', '2'), 36)\n",
      "(('nih3t3', 'cell'), 36)\n",
      "(('cells', 'addition,'), 36)\n",
      "(('also', 'displayed'), 36)\n",
      "(('studies', 'needed'), 36)\n",
      "(('normal', 'human'), 36)\n",
      "(('human', 'breast'), 36)\n",
      "(('mutation', 'leads'), 36)\n",
      "(('immunohistochemical', 'analysis'), 36)\n",
      "(('high', 'degree'), 36)\n",
      "(('embryonic', 'fibroblasts'), 36)\n",
      "(('mutations', 'clustered'), 36)\n",
      "(('0', '0'), 36)\n",
      "(('malignant', 'melanoma'), 36)\n",
      "(('melanoma', 'patients'), 36)\n",
      "(('whereas', 'two'), 36)\n",
      "(('two', 'four'), 36)\n",
      "(('11', 'mutations'), 36)\n",
      "(('mutation', 'predicted'), 36)\n",
      "(('relatively', 'small'), 36)\n",
      "(('study', 'effect'), 36)\n",
      "(('found', 'four'), 36)\n",
      "(('mutation', 'frequencies'), 36)\n",
      "(('proteins', 'expressed'), 36)\n",
      "(('receptors', 'expressed'), 36)\n",
      "(('frequently', 'associated'), 36)\n",
      "(('following', 'primer'), 36)\n",
      "(('detailed', 'analysis'), 36)\n",
      "(('overall', 'structure'), 36)\n",
      "(('deleterious', 'variants'), 36)\n",
      "(('pathogenic', 'mutations'), 36)\n",
      "(('mutant', 'found'), 36)\n",
      "(('repair', 'gene'), 36)\n",
      "(('25', 'patients'), 36)\n",
      "(('activity', 'patients'), 36)\n",
      "(('loss', 'function,'), 36)\n",
      "(('inhibits', 'growth'), 36)\n",
      "(('two', 'major'), 36)\n",
      "(('(figure', '6b)'), 36)\n",
      "(('functional', 'sites'), 36)\n",
      "(('splice', 'sites'), 36)\n",
      "(('(fig', 'mutations'), 36)\n",
      "(('mutations', 'particular'), 36)\n",
      "(('first', 'study'), 36)\n",
      "(('four', 'additional'), 36)\n",
      "(('domains', '(residues'), 36)\n",
      "(('predictive', 'value'), 36)\n",
      "(('assessed', 'sensitivity'), 36)\n",
      "(('however,', 'patients'), 36)\n",
      "(('activity', 'egfr'), 36)\n",
      "(('sensitivity', 'gefitinib'), 36)\n",
      "(('role', 'egfr'), 36)\n",
      "(('mutations', 'data'), 36)\n",
      "(('indistinguishable', 'wild-type'), 36)\n",
      "(('inhibitor', 'egfr'), 36)\n",
      "(('lines', 'tumor'), 36)\n",
      "(('continued', 'activation'), 36)\n",
      "(('adenocarcinoma', 'histology'), 36)\n",
      "(('exon', '20,'), 36)\n",
      "(('mechanism', 'activation'), 36)\n",
      "(('phase', '2'), 36)\n",
      "(('within', '2'), 36)\n",
      "(('mutations', '(data'), 36)\n",
      "(('(fig', 'discussion'), 36)\n",
      "(('objective', 'responses'), 36)\n",
      "(('tumor', 'response'), 36)\n",
      "(('cancer', '(cosmic)'), 36)\n",
      "(('(32,', '33)'), 36)\n",
      "(('mutation', 'detection'), 36)\n",
      "(('patients', 'developed'), 36)\n",
      "(('transformation', 'potential'), 36)\n",
      "(('downstream', 'effector'), 36)\n",
      "(('compared', 'parental'), 36)\n",
      "(('translational', 'relevance'), 36)\n",
      "(('predict', 'mutations'), 36)\n",
      "(('significant', 'clinical'), 36)\n",
      "(('juxtamembrane', 'region'), 36)\n",
      "(('samples', 'showed'), 36)\n",
      "(('nsclc', 'cells'), 36)\n",
      "(('expression', 'phosphorylation'), 36)\n",
      "(('comparable', 'wt'), 36)\n",
      "(('less', 'frequently'), 36)\n",
      "(('gene', 'identified'), 36)\n",
      "(('reported', 'first'), 36)\n",
      "(('pathway', 'activation,'), 36)\n",
      "(('approximately', '7%'), 36)\n",
      "(('nuclear', 'staining'), 36)\n",
      "(('expression', 'observed'), 36)\n",
      "(('six', 'mutations'), 36)\n",
      "(('three-dimensional', 'structures'), 36)\n",
      "(('cdna', 'encoding'), 36)\n",
      "(('mutations', 'reduce'), 36)\n",
      "(('levels', 'compared'), 36)\n",
      "(('determined', 'binding'), 36)\n",
      "(('transfected', 'wild'), 36)\n",
      "(('study', 'examined'), 36)\n",
      "(('found', 'exon'), 36)\n",
      "(('predicted', 'impact'), 36)\n",
      "(('known', 'deleterious'), 36)\n",
      "(('provide', 'new'), 36)\n",
      "(('al', 'contrast'), 36)\n",
      "(('mek', 'inhibitor'), 36)\n",
      "(('cdna', 'amplified'), 36)\n",
      "(('constitutively', 'phosphorylated'), 36)\n",
      "(('ii', 'study'), 36)\n",
      "(('secondary', 'imatinib'), 36)\n",
      "(('mutational', 'activation'), 36)\n",
      "(('mutation', 'kit'), 36)\n",
      "(('predict', 'functional'), 36)\n",
      "(('several', 'transcription'), 36)\n",
      "(('involved', 'several'), 36)\n",
      "(('potential', 'importance'), 36)\n",
      "(('cruz,', 'ca)'), 36)\n",
      "(('provided', 'dr'), 36)\n",
      "(('likely', 'disrupt'), 36)\n",
      "(('gene', 'individuals'), 36)\n",
      "(('may', 'classified'), 36)\n",
      "(('six', 'sequences'), 36)\n",
      "(('given', 'mutation'), 36)\n",
      "(('mutations', 'intact'), 36)\n",
      "(('mutations', 'surface'), 36)\n",
      "(('outside', 'putative'), 36)\n",
      "(('affect', 'transcriptional'), 36)\n",
      "(('species,', 'including'), 36)\n",
      "(('transcription', 'assays,'), 36)\n",
      "(('disrupt', 'interactions'), 36)\n",
      "(('(fig', 'remaining'), 36)\n",
      "(('involved', 'transcriptional'), 36)\n",
      "(('splicing', 'overlapping'), 36)\n",
      "(('brca1', 'involved'), 36)\n",
      "(('structure', 'brca1'), 36)\n",
      "(('brca1', 'correlates'), 36)\n",
      "(('cancers', '(1'), 36)\n",
      "(('inherited', 'breast'), 36)\n",
      "(('ii', 'holoenzyme'), 36)\n",
      "(('protein', 'activates'), 36)\n",
      "(('among', 'brca1'), 36)\n",
      "(('brca1', 'likely'), 36)\n",
      "(('deleterious', 'mutation'), 36)\n",
      "(('found', 'mutated'), 35)\n",
      "(('using', 'lipofectamine'), 35)\n",
      "(('transfection', 'expression'), 35)\n",
      "(('chronic', 'myelomonocytic'), 35)\n",
      "(('hematologic', 'malignancies'), 35)\n",
      "(('effect', 'cell'), 35)\n",
      "(('high', 'concentrations'), 35)\n",
      "(('fig', 'according'), 35)\n",
      "(('figure', '2:'), 35)\n",
      "(('high', 'prevalence'), 35)\n",
      "(('(14,', '15)'), 35)\n",
      "(('may', 'serve'), 35)\n",
      "(('known', 'impact'), 35)\n",
      "(('independent', 'predictor'), 35)\n",
      "(('10', '5'), 35)\n",
      "(('amplification', 'also'), 35)\n",
      "(('mammary', 'epithelial'), 35)\n",
      "(('patients', 'compared'), 35)\n",
      "(('7', 'figure'), 35)\n",
      "(('nucleotide', 'sequence'), 35)\n",
      "(('six', 'cases'), 35)\n",
      "(('wt', '(figure'), 35)\n",
      "(('identified', 'recurrent'), 35)\n",
      "(('analysis', 'indicated'), 35)\n",
      "(('(figure', '5b)'), 35)\n",
      "(('lesser', 'degree'), 35)\n",
      "(('shed', 'light'), 35)\n",
      "(('mutations', 'generated'), 35)\n",
      "(('(figure', '2d)'), 35)\n",
      "(('results', 'suggested'), 35)\n",
      "(('vivo', 'tumor'), 35)\n",
      "(('cancer', 'brca1'), 35)\n",
      "(('mutation', 'analyses'), 35)\n",
      "(('provides', 'evidence'), 35)\n",
      "(('novel', 'missense'), 35)\n",
      "(('two', 'mutant'), 35)\n",
      "(('steady', 'state'), 35)\n",
      "(('al', 'another'), 35)\n",
      "(('three', 'times'), 35)\n",
      "(('plasmids', 'expressing'), 35)\n",
      "(('5', '6'), 35)\n",
      "(('extracellular', 'domain,'), 35)\n",
      "(('receptor', 'protein'), 35)\n",
      "(('prostate', 'cancer,'), 35)\n",
      "(('signaling', 'plays'), 35)\n",
      "(('provide', 'rationale'), 35)\n",
      "(('may', 'confer'), 35)\n",
      "(('reporter', 'activity'), 35)\n",
      "(('fig', 'sequence'), 35)\n",
      "(('activity', 'vivo'), 35)\n",
      "(('lines', 'wild-type'), 35)\n",
      "(('identified', 'eight'), 35)\n",
      "(('discussion', 'study'), 35)\n",
      "(('performed', 'previously'), 35)\n",
      "(('conserved', 'sequence'), 35)\n",
      "(('study', 'made'), 35)\n",
      "(('cells', 'following'), 35)\n",
      "(('least', 'three'), 35)\n",
      "(('mutations', 'expected'), 35)\n",
      "(('first', 'described'), 35)\n",
      "(('recent', 'reports'), 35)\n",
      "(('genetic', 'epigenetic'), 35)\n",
      "(('kinase', 'signaling'), 35)\n",
      "(('total', 'cell'), 35)\n",
      "(('mutations', '(supplementary'), 35)\n",
      "(('study', 'showed'), 35)\n",
      "(('number', 'alterations'), 35)\n",
      "(('another', 'study'), 35)\n",
      "(('phosphorylation', 'tyrosine'), 35)\n",
      "(('distinct', 'subset'), 35)\n",
      "(('inhibition', 'egfr'), 35)\n",
      "(('sites', 'within'), 35)\n",
      "(('may', 'require'), 35)\n",
      "(('inhibitors', 'important'), 35)\n",
      "(('egfr', '(n'), 35)\n",
      "(('reduced', 'kinase'), 35)\n",
      "(('found', 'eight'), 35)\n",
      "(('relative', 'levels'), 35)\n",
      "(('concentrations', 'gefitinib'), 35)\n",
      "(('treated', 'gefitinib'), 35)\n",
      "(('mutations', 'tk'), 35)\n",
      "(('cohort', 'patients'), 35)\n",
      "(('(n', '11)'), 35)\n",
      "(('crucial', 'role'), 35)\n",
      "(('homology', 'modeling'), 35)\n",
      "(('mutation', 'types'), 35)\n",
      "(('clinically', 'effective'), 35)\n",
      "(('ongoing', 'investigations'), 35)\n",
      "(('also', 'provide'), 35)\n",
      "(('tyrosine', 'kinase,'), 35)\n",
      "(('paraffin-embedded', 'tumor'), 35)\n",
      "(('tumor', 'sample'), 35)\n",
      "(('mutations', 'codon'), 35)\n",
      "(('mutations', 'exhibited'), 35)\n",
      "(('c-terminal', 'tail'), 35)\n",
      "(('shown', 'activate'), 35)\n",
      "(('stable', 'disease;'), 35)\n",
      "(('single', 'mutation'), 35)\n",
      "(('years', 'ago,'), 35)\n",
      "(('15', 'patients'), 35)\n",
      "(('models', '(28)'), 35)\n",
      "(('del,', 'deletion;'), 35)\n",
      "(('test', 'possibility,'), 35)\n",
      "(('murine', 'hematopoietic'), 35)\n",
      "(('cells', 'displayed'), 35)\n",
      "(('amplified', 'polymerase'), 35)\n",
      "(('signaling', 'technology,'), 35)\n",
      "(('molecular', 'dynamics'), 35)\n",
      "(('however,', 'results'), 35)\n",
      "(('strongly', 'activated'), 35)\n",
      "(('presence', 'il-3'), 35)\n",
      "(('section', 'discussion'), 35)\n",
      "(('tumors', '(figure'), 35)\n",
      "(('(wu', 'et'), 35)\n",
      "(('wild-type', 'human'), 35)\n",
      "(('also', 'revealed'), 35)\n",
      "(('deletion', 'may'), 35)\n",
      "(('mutants,', 'including'), 35)\n",
      "(('sporadic', 'tumors'), 35)\n",
      "(('lines', 'containing'), 35)\n",
      "(('cells', 'transient'), 35)\n",
      "(('cells', 'fig'), 35)\n",
      "(('structures', 'two'), 35)\n",
      "(('individuals', 'risk'), 35)\n",
      "(('proliferation', 'cells'), 35)\n",
      "(('activation', 'also'), 35)\n",
      "(('agreement', 'vitro'), 35)\n",
      "(('confirmed', 'sequencing'), 35)\n",
      "(('two', 'ways:'), 35)\n",
      "(('rattus', 'norvegicus'), 35)\n",
      "(('homo', 'sapiens'), 35)\n",
      "(('dimer', 'formation'), 35)\n",
      "(('thus,', 'one'), 35)\n",
      "(('following', 'two'), 35)\n",
      "(('receptor', ''), 35)\n",
      "(('mutations', 'pdgfra'), 35)\n",
      "(('in-frame', 'fusion'), 35)\n",
      "(('anti-phosphotyrosine', 'antibody'), 35)\n",
      "(('patients', 'kit'), 35)\n",
      "(('also', 'strongly'), 35)\n",
      "(('family', '(fig'), 35)\n",
      "(('mutations', 'constitutively'), 35)\n",
      "(('mutations', 'region'), 35)\n",
      "(('domains', '(amino'), 35)\n",
      "(('structure', 'among'), 35)\n",
      "(('role', 'activated'), 35)\n",
      "(('braf', 'gene'), 35)\n",
      "(('mapk', 'akt'), 35)\n",
      "(('constructs', 'subcloned'), 35)\n",
      "(('including', 'homo'), 35)\n",
      "(('control', 'population'), 35)\n",
      "(('determined', 'sequences'), 35)\n",
      "(('(18', '19)'), 35)\n",
      "(('(15', '16)'), 35)\n",
      "(('mutation', 'set'), 35)\n",
      "(('less', 'likely,'), 35)\n",
      "(('50%', 'cases'), 35)\n",
      "(('or,', 'less'), 35)\n",
      "(('residue', 'substitution'), 35)\n",
      "(('may', 'substantially'), 35)\n",
      "(('site', 'next,'), 35)\n",
      "(('', 'fig'), 35)\n",
      "(('nonredundant', 'protein'), 35)\n",
      "(('various', 'proteins'), 35)\n",
      "(('brca1', 'domains'), 35)\n",
      "(('brca1', 'leads'), 35)\n",
      "(('substitution', 'given'), 35)\n",
      "(('given', 'position'), 35)\n",
      "(('brca1', 'c-terminus'), 35)\n",
      "(('brca1', 'homologs'), 35)\n",
      "(('pcdna3', 'results'), 35)\n",
      "(('assays,', 'brca1'), 35)\n",
      "(('predicting', 'functional'), 35)\n",
      "(('putative', 'binding'), 35)\n",
      "(('rat', 'brca1'), 35)\n",
      "(('orthologs', '(supplementary'), 35)\n",
      "(('predicting', 'whether'), 35)\n",
      "(('previously', '(18,'), 35)\n",
      "(('cancers', 'screened'), 35)\n",
      "(('whether', 'given'), 35)\n",
      "(('site', 'brca1'), 35)\n",
      "(('protein', '(figure'), 34)\n",
      "(('8', '9'), 34)\n",
      "(('72', 'h'), 34)\n",
      "(('order', 'determine'), 34)\n",
      "(('enhanced', 'expression'), 34)\n",
      "(('expressing', 'wild'), 34)\n",
      "(('myeloid', 'cells'), 34)\n",
      "(('cells', 'observed'), 34)\n",
      "(('protein', 'truncation'), 34)\n",
      "(('strong', 'evidence'), 34)\n",
      "(('also', 'contained'), 34)\n",
      "(('cell', 'type'), 34)\n",
      "(('quantitative', 'rt-pcr'), 34)\n",
      "(('thyroid', 'carcinoma'), 34)\n",
      "(('papillary', 'thyroid'), 34)\n",
      "(('dose-dependent', 'manner'), 34)\n",
      "(('0', '1'), 34)\n",
      "(('gene', 'deletion'), 34)\n",
      "(('patients', 'reported'), 34)\n",
      "(('mice', 'showed'), 34)\n",
      "(('consistent', 'vitro'), 34)\n",
      "(('(figure', '1d)'), 34)\n",
      "(('increased', 'binding'), 34)\n",
      "(('extracellular', 'signal-regulated'), 34)\n",
      "(('focal', 'amplification'), 34)\n",
      "(('hereditary', 'breast'), 34)\n",
      "(('essential', 'role'), 34)\n",
      "(('dimerization', 'domain'), 34)\n",
      "(('sequence', 'alterations'), 34)\n",
      "(('showed', 'evidence'), 34)\n",
      "(('however,', 'also'), 34)\n",
      "(('two', 'separate'), 34)\n",
      "(('(see', 'table'), 34)\n",
      "(('cell-cycle', 'arrest'), 34)\n",
      "(('cells', 'co-transfected'), 34)\n",
      "(('u2os', 'cells'), 34)\n",
      "(('might', 'contribute'), 34)\n",
      "(('author', 'full'), 34)\n",
      "(('studies', 'including'), 34)\n",
      "(('to:', 'conclusions'), 34)\n",
      "(('levels', 'comparable'), 34)\n",
      "(('(in', 'absence'), 34)\n",
      "(('different', 'wild-type'), 34)\n",
      "(('cells', 'cotransfected'), 34)\n",
      "(('selective', 'pressure'), 34)\n",
      "(('tumors', 'compared'), 34)\n",
      "(('assess', 'effect'), 34)\n",
      "(('observed', 'significantly'), 34)\n",
      "(('used', 'studies'), 34)\n",
      "(('many', 'studies'), 34)\n",
      "(('may', 'related'), 34)\n",
      "(('age', 'onset'), 34)\n",
      "(('results', 'presented'), 34)\n",
      "(('using', 'standard'), 34)\n",
      "(('cell', 'extracts'), 34)\n",
      "(('mutations', 'analyzed'), 34)\n",
      "(('17', 'patients'), 34)\n",
      "(('protein', 'expressed'), 34)\n",
      "(('(figure', '6a)'), 34)\n",
      "(('and,', 'lesser'), 34)\n",
      "(('rt-pcr', 'using'), 34)\n",
      "(('longer', 'survival'), 34)\n",
      "(('novel', 'therapeutic'), 34)\n",
      "(('analysis', 'also'), 34)\n",
      "(('(egfr)', 'tyrosine'), 34)\n",
      "(('difference', 'egfr'), 34)\n",
      "(('gefitinib', 'patients'), 34)\n",
      "(('five', 'tumors'), 34)\n",
      "(('plasma', 'concentrations'), 34)\n",
      "(('egfr', 'tk'), 34)\n",
      "(('associated', 'gefitinib'), 34)\n",
      "(('compared', 'primary'), 34)\n",
      "(('signaling', 'inhibition'), 34)\n",
      "(('tk', 'domain,'), 34)\n",
      "(('types', 'egfr'), 34)\n",
      "(('substitution', 'within'), 34)\n",
      "(('lung', 'tissue'), 34)\n",
      "(('specific', 'egfr'), 34)\n",
      "(('relationship', 'egfr'), 34)\n",
      "(('role', 'tumor'), 34)\n",
      "(('containing', 'mutant'), 34)\n",
      "(('current', 'smokers'), 34)\n",
      "(('truncated', 'egfr'), 34)\n",
      "(('whereas', 'overexpression'), 34)\n",
      "(('better', 'response'), 34)\n",
      "(('resistance', 'gefitinib'), 34)\n",
      "(('mutations', 'discovered'), 34)\n",
      "(('20,', '21,'), 34)\n",
      "(('cancers', 'harboring'), 34)\n",
      "(('conformation', 'kinase'), 34)\n",
      "(('preclinical', 'data'), 34)\n",
      "(('initiation', 'progression'), 34)\n",
      "(('outcomes', 'patients'), 34)\n",
      "(('mutations', 'treated'), 34)\n",
      "(('colonies', 'soft'), 34)\n",
      "(('100', 'nmol/l'), 34)\n",
      "(('express', 'either'), 34)\n",
      "(('neck', 'squamous'), 34)\n",
      "(('experiments', 'performed'), 34)\n",
      "(('lines', 'tested'), 34)\n",
      "(('tempting', 'speculate'), 34)\n",
      "(('amplified', 'reverse'), 34)\n",
      "(('immediately', 'following'), 34)\n",
      "(('pik3ca', 'mutations'), 34)\n",
      "(('expressed', 'mutant'), 34)\n",
      "(('cruz', 'biotechnology)'), 34)\n",
      "(('may', 'respond'), 34)\n",
      "(('(n', '4),'), 34)\n",
      "(('type', '(wt)'), 34)\n",
      "(('acid', 'residue'), 34)\n",
      "(('residues', 'involved'), 34)\n",
      "(('ectopically', 'expressing'), 34)\n",
      "(('positive', 'controls'), 34)\n",
      "(('raising', 'possibility'), 34)\n",
      "(('2', '0'), 34)\n",
      "(('p53', 'mutations'), 34)\n",
      "(('similar', 'manner'), 34)\n",
      "(('amplification', 'detected'), 34)\n",
      "(('internal', 'control'), 34)\n",
      "(('tp53', 'mutations'), 34)\n",
      "(('cell', 'migration'), 34)\n",
      "(('fusion', 'partners'), 34)\n",
      "(('kit', 'gene'), 34)\n",
      "(('determine', 'functional'), 34)\n",
      "(('others', 'recently'), 34)\n",
      "(('targeted', 'sequencing'), 34)\n",
      "(('epithelial', 'cell'), 34)\n",
      "(('reduced', 'activity'), 34)\n",
      "(('predicted', 'effect'), 34)\n",
      "(('genetic', 'inactivation'), 34)\n",
      "(('using', 'antibodies'), 34)\n",
      "(('protein', 'complexes'), 34)\n",
      "(('one', 'mutation,'), 34)\n",
      "(('t-all', 'cell'), 34)\n",
      "(('also', 'occur'), 34)\n",
      "(('located', 'near'), 34)\n",
      "(('substitution', 'likely'), 34)\n",
      "(('(fig', 'one'), 34)\n",
      "(('mutations', 'receptor'), 34)\n",
      "(('ligand-dependent', 'activation'), 34)\n",
      "(('identified', 'activating'), 34)\n",
      "(('sensitivity', 'imatinib'), 34)\n",
      "(('pdgfra', 'mutants'), 34)\n",
      "(('strong', 'constitutive'), 34)\n",
      "(('antibody', '(santa'), 34)\n",
      "(('lysates', 'immunoprecipitated'), 34)\n",
      "(('(santa', 'cruz,'), 34)\n",
      "(('appears', 'promote'), 34)\n",
      "(('relatively', 'weak'), 34)\n",
      "(('mutations', 'c-kit'), 34)\n",
      "(('increase', 'proportion'), 34)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('might', 'related'), 34)\n",
      "(('al,', '2003),'), 34)\n",
      "(('treatment', 'human'), 34)\n",
      "(('(a)', 'exposed'), 34)\n",
      "(('gallus', '(af355273),'), 34)\n",
      "(('domains', '(ref'), 34)\n",
      "(('brca1', 'ligand(s)'), 34)\n",
      "(('therefore', 'correlates'), 34)\n",
      "(('domains', 'assess'), 34)\n",
      "(('models', 'available'), 34)\n",
      "(('suggest', 'clustering'), 34)\n",
      "(('activation', 'thirty-one'), 34)\n",
      "(('integrity', 'necessary'), 34)\n",
      "(('cancer', 'classification'), 34)\n",
      "(('(7', '8)'), 34)\n",
      "(('applied', '57'), 34)\n",
      "(('domains', 'eight'), 34)\n",
      "(('prediction', 'cancer'), 34)\n",
      "(('mus', 'musculus'), 34)\n",
      "(('37', 'mutations,'), 34)\n",
      "(('annotating', 'given'), 34)\n",
      "(('(18,', '19,'), 34)\n",
      "(('principles', 'protein'), 34)\n",
      "(('(23)', 'latter'), 34)\n",
      "(('(residues', '16491859)'), 34)\n",
      "(('mutation', 'applied'), 34)\n",
      "(('activation', 'function,'), 34)\n",
      "(('classification', 'tree'), 34)\n",
      "(('interactions', 'brca1'), 34)\n",
      "(('1)', 'without'), 34)\n",
      "(('37', 'missense'), 34)\n",
      "(('function,', 'evidence'), 34)\n",
      "(('contoured', 'red'), 34)\n",
      "(('sequence', 'structure'), 34)\n",
      "(('without', 'transcriptional'), 34)\n",
      "(('mutations,', 'derived'), 34)\n",
      "(('sites', '1'), 34)\n",
      "(('residues', '16521855)'), 34)\n",
      "(('sites', 'bind'), 34)\n",
      "(('16491859)', 'included'), 34)\n",
      "(('derived', 'set'), 34)\n",
      "(('likelihood', 'residue'), 34)\n",
      "(('moreover,', 'putative'), 34)\n",
      "(('groove', 'presumably'), 34)\n",
      "(('affect', 'affect'), 34)\n",
      "(('activation', 'functions'), 34)\n",
      "(('sapiens', '(genbank'), 34)\n",
      "(('fig', 'wild-type'), 34)\n",
      "(('rules', 'predicting'), 34)\n",
      "(('forms', 'binding'), 34)\n",
      "(('domain', 'pcdna3'), 34)\n",
      "(('changes', 'core'), 34)\n",
      "(('domain', '(18'), 34)\n",
      "(('experimentally', 'determined'), 34)\n",
      "(('repair', '(15'), 34)\n",
      "(('study', 'view'), 34)\n",
      "(('variability', 'defined'), 34)\n",
      "(('ligand(s)', 'suggest'), 34)\n",
      "(('partially,', 'transcriptional'), 34)\n",
      "(('sequence', 'variability'), 34)\n",
      "(('(u50709),', 'xenopus'), 34)\n",
      "(('(28)', 'mutation'), 34)\n",
      "(('fold', 'or,'), 34)\n",
      "(('proteins', '(9,'), 34)\n",
      "(('2', 'prediction'), 34)\n",
      "(('90%', 'sequence'), 34)\n",
      "(('6)', 'three-dimensional'), 34)\n",
      "(('comparative', 'protein'), 34)\n",
      "(('highlights', 'binding'), 34)\n",
      "(('characterization', 'brca1'), 34)\n",
      "(('native', 'fold'), 34)\n",
      "(('germ-line', 'brca1'), 34)\n",
      "(('population', 'equal'), 34)\n",
      "(('(1l0b;', 'ref'), 34)\n",
      "(('every', 'column'), 34)\n",
      "(('57', 'germ-line'), 34)\n",
      "(('14)', 'brca1'), 34)\n",
      "(('16)', 'tandem'), 34)\n",
      "(('defined', 'every'), 34)\n",
      "(('troglodytes', '(af207822),'), 34)\n",
      "(('norvegicus', '(af036760),'), 34)\n",
      "(('therefore,', 'ability'), 34)\n",
      "(('problem,', 'use'), 34)\n",
      "(('(msa)', 'orthologous'), 34)\n",
      "(('30%', '90%'), 34)\n",
      "(('molecules,', '(b)'), 34)\n",
      "(('surface', 'outside'), 34)\n",
      "(('chemical', 'nature'), 34)\n",
      "(('thirty-one', '37'), 34)\n",
      "(('database', 'comparative'), 34)\n",
      "(('evidence', 'cancer-association;'), 34)\n",
      "(('set', 'deposited'), 34)\n",
      "(('red', 'yellow,'), 34)\n",
      "(('likely', 'depend,'), 34)\n",
      "(('sequences', 'sequences'), 34)\n",
      "(('genetic', 'framework'), 34)\n",
      "(('frame', 'gal4'), 34)\n",
      "(('(15)', 'binding'), 34)\n",
      "(('psi-blast', '(23)'), 34)\n",
      "(('21)', 'supported'), 34)\n",
      "(('residues', 'single'), 34)\n",
      "(('code', '1jnx;'), 34)\n",
      "(('variants', 'materials'), 34)\n",
      "(('domains', 'native'), 34)\n",
      "(('position', 'reflects'), 34)\n",
      "(('holoenzyme', 'via'), 34)\n",
      "(('(13', '14)'), 34)\n",
      "(('pan', 'troglodytes'), 34)\n",
      "(('next,', 'structure-based'), 34)\n",
      "(('domains', '(1l0b;'), 34)\n",
      "(('yellow,', 'respectively'), 34)\n",
      "(('attempted', 'rationalize'), 34)\n",
      "(('well', 'actually'), 34)\n",
      "(('fold', 'localized'), 34)\n",
      "(('background', 'substitution'), 34)\n",
      "(('surface', 'human'), 34)\n",
      "(('facilitates', 'dna'), 34)\n",
      "(('8', '24'), 34)\n",
      "(('association', 'introduction'), 34)\n",
      "(('according', 'psi-blast'), 34)\n",
      "(('transcription', 'facilitates'), 34)\n",
      "(('model', 'set'), 34)\n",
      "(('clustering', 'residues'), 34)\n",
      "(('msa', 'brca1'), 34)\n",
      "(('activation', 'brca1;'), 34)\n",
      "(('domains', 'tumor'), 34)\n",
      "(('19)', 'observations'), 34)\n",
      "(('core,', 'may'), 34)\n",
      "(('brca1', 'model'), 34)\n",
      "(('canis', 'familiaris'), 34)\n",
      "(('phenotype', '57'), 34)\n",
      "(('helicase', '(17)'), 34)\n",
      "(('involve', 'nonconservative'), 34)\n",
      "(('several', 'functions,'), 34)\n",
      "(('three', 'exposed'), 34)\n",
      "(('ways:', '(a)'), 34)\n",
      "(('unknown', 'cancer'), 34)\n",
      "(('single', 'patch'), 34)\n",
      "(('absence', 'co-segregation'), 34)\n",
      "(('modbase', 'database'), 34)\n",
      "(('brca1', 'rna'), 34)\n",
      "(('(u68174),', 'rattus'), 34)\n",
      "(('provides', 'genetic'), 34)\n",
      "(('exposed', 'solvent,'), 34)\n",
      "(('reflects', 'background'), 34)\n",
      "(('(af416868),', 'obtained'), 34)\n",
      "(('2)', 'address'), 34)\n",
      "(('cases,', 'absence'), 34)\n",
      "(('protein', 'three-dimensional'), 34)\n",
      "(('abolish', 'transcription,'), 34)\n",
      "(('transcription', 'missense'), 34)\n",
      "(('function,', 'respectively,'), 34)\n",
      "(('rationalize', 'known'), 34)\n",
      "(('brca1;', 'mutations'), 34)\n",
      "(('12)', 'including'), 34)\n",
      "(('one', 'gapped'), 34)\n",
      "(('substantially', 'change'), 34)\n",
      "(('uncharacterized', 'missense'), 34)\n",
      "(('inactivates', 'transcriptional'), 34)\n",
      "(('site', 'involved'), 34)\n",
      "(('ability', 'identify'), 34)\n",
      "(('probabilities', 'blosum62'), 34)\n",
      "(('domains', 'used'), 34)\n",
      "(('binding', 'rna'), 34)\n",
      "(('affect', 'structure'), 34)\n",
      "(('via', 'rna'), 34)\n",
      "(('framework', 'characterization'), 34)\n",
      "(('cancer-association;', 'frequency'), 34)\n",
      "(('classification', '94'), 34)\n",
      "(('transcription,', 'propose'), 34)\n",
      "(('straightforward', 'use'), 34)\n",
      "(('co-segregation', 'least'), 34)\n",
      "(('number', 'u14680),'), 34)\n",
      "(('blosum62', 'matrix'), 34)\n",
      "(('likely,', 'affect'), 34)\n",
      "(('readily', 'rationalized'), 34)\n",
      "(('benefited', 'prior'), 34)\n",
      "(('relatively', 'conserved'), 34)\n",
      "(('observed', 'substitutions'), 34)\n",
      "(('set', 'table'), 34)\n",
      "(('prevent', 'folding'), 34)\n",
      "(('rationalized', 'structural'), 34)\n",
      "(('structure-based', 'rules'), 34)\n",
      "(('site', 'relatively'), 34)\n",
      "(('(17)', 'straightforward'), 34)\n",
      "(('brca1', 'highlights'), 34)\n",
      "(('(22)', 'contains'), 34)\n",
      "(('respectively,', 'mapped'), 34)\n",
      "(('structures', 'brca1'), 34)\n",
      "(('included', 'experimentally'), 34)\n",
      "(('brct', 'orthologs'), 34)\n",
      "(('domains', '(tables'), 34)\n",
      "(('mutation', 'likelihood'), 34)\n",
      "(('activation', 'therefore'), 34)\n",
      "(('u14680),', 'pan'), 34)\n",
      "(('set', 'rules'), 34)\n",
      "(('available', 'brca1'), 34)\n",
      "(('structure', 'chemical'), 34)\n",
      "(('nature', 'functional'), 34)\n",
      "(('column', 'msa'), 34)\n",
      "(('address', 'problem,'), 34)\n",
      "(('impact', '32'), 34)\n",
      "(('given', 'missense'), 34)\n",
      "(('tree', 'annotating'), 34)\n",
      "(('program', 'clustalw'), 34)\n",
      "(('consequences', 'given'), 34)\n",
      "(('introduction', 'many'), 34)\n",
      "(('possible', 'recently'), 34)\n",
      "(('alignment', '(msa)'), 34)\n",
      "(('including', 'modulation'), 34)\n",
      "(('modulation', 'activity'), 34)\n",
      "(('1', 'functionally'), 34)\n",
      "(('information', '30%'), 34)\n",
      "(('total', '94'), 34)\n",
      "(('ref', 'crystallographic'), 34)\n",
      "(('row', '1)'), 34)\n",
      "(('terms', 'loss-of-function'), 34)\n",
      "(('laevis', '(af416868),'), 34)\n",
      "(('discussion', 'tabulated'), 34)\n",
      "(('(b)', 'buried'), 34)\n",
      "(('supported', 'mutation'), 34)\n",
      "(('use', 'three-dimensional'), 34)\n",
      "(('tabulated', 'total'), 34)\n",
      "(('familiaris', '(u50709),'), 34)\n",
      "(('surface', 'brca1'), 34)\n",
      "(('presumably', 'forms'), 34)\n",
      "(('suggested', 'previously'), 34)\n",
      "(('addition,', 'benefited'), 34)\n",
      "(('mutation', 'inactivates'), 34)\n",
      "(('conservative', 'changes'), 34)\n",
      "(('24', '),'), 34)\n",
      "(('methods', 'multiple'), 34)\n",
      "(('domains', 'cooh-terminus'), 34)\n",
      "(('cooh-terminus', 'brca1'), 34)\n",
      "(('domains', '(protein'), 34)\n",
      "(('use', 'principles'), 34)\n",
      "(('structure', 'models'), 34)\n",
      "(('localized', 'groove'), 34)\n",
      "(('substitutions', 'given'), 34)\n",
      "(('factors', '(15)'), 34)\n",
      "(('refs', '8'), 34)\n",
      "(('', '2'), 34)\n",
      "(('transcription', 'moreover,'), 34)\n",
      "(('16521855)', 'view'), 34)\n",
      "(('onto', 'crystallographic'), 34)\n",
      "(('(brct)', 'fold'), 34)\n",
      "(('correlation', 'cancer-associated'), 34)\n",
      "(('brca1,', 'set'), 34)\n",
      "(('rationalize', 'predict'), 34)\n",
      "(('94', 'missense'), 34)\n",
      "(('brca1', '(13'), 34)\n",
      "(('solvent-exposed', 'residues'), 34)\n",
      "(('position', '(supplementary'), 34)\n",
      "(('deposited', 'modbase'), 34)\n",
      "(('', 'row'), 34)\n",
      "(('known', 'affect'), 34)\n",
      "(('identify', 'cancer-associated'), 34)\n",
      "(('database', 'national'), 34)\n",
      "(('mammalian', 'transcription'), 34)\n",
      "(('heterologous', 'dna-binding'), 34)\n",
      "(('homologs', '(7'), 34)\n",
      "(('gallus', 'gallus'), 34)\n",
      "(('structure', 'well'), 34)\n",
      "(('(17)', 'activating'), 34)\n",
      "(('32', '37'), 34)\n",
      "(('set', 'use'), 34)\n",
      "(('bind', 'molecules,'), 34)\n",
      "(('brca1', 'useful'), 34)\n",
      "(('characterized', 'missense'), 34)\n",
      "(('8)', 'addition,'), 34)\n",
      "(('either', 'conservative'), 34)\n",
      "(('necessary', 'transcription'), 34)\n",
      "(('(af207822),', 'mus'), 34)\n",
      "(('c-terminus', '(brct)'), 34)\n",
      "(('intact', 'transcriptional'), 34)\n",
      "(('gapped', 'position'), 34)\n",
      "(('domains', 'rationalize'), 34)\n",
      "(('abolishing', 'transcriptional'), 34)\n",
      "(('suggest', 'abolishing'), 34)\n",
      "(('abolish', 'transcriptional'), 34)\n",
      "(('core', 'surface'), 34)\n",
      "(('msa', 'position'), 34)\n",
      "(('holoenzyme', '(17)'), 34)\n",
      "(('musculus', '(u68174),'), 34)\n",
      "(('sequences', '(3,'), 34)\n",
      "(('orthologous', 'brca1'), 34)\n",
      "(('functionally', 'uncharacterized'), 34)\n",
      "(('solvent,', 'may'), 34)\n",
      "(('position', 'msa'), 34)\n",
      "(('propose', 'disrupt'), 34)\n",
      "(('structural', 'terms'), 34)\n",
      "(('brca1', 'readily'), 34)\n",
      "(('site', 'integrity'), 34)\n",
      "(('polymerase', 'holoenzyme'), 34)\n",
      "(('seven', 'species,'), 34)\n",
      "(('brca1', 'contoured'), 34)\n",
      "(('latter', 'six'), 34)\n",
      "(('rna', 'helicase'), 34)\n",
      "(('fig', 'classification'), 34)\n",
      "(('core', 'brca1'), 34)\n",
      "(('modeling', '(8)'), 34)\n",
      "(('depend,', 'least'), 34)\n",
      "(('many', 'germ-line'), 34)\n",
      "(('domains', 'solvent-exposed'), 34)\n",
      "(('(ref', 'missense'), 34)\n",
      "(('matrix', 'well'), 34)\n",
      "(('(af036760),', 'gallus'), 34)\n",
      "(('1jnx;', 'refs'), 34)\n",
      "(('buried', 'core,'), 34)\n",
      "(('patch', 'surface'), 34)\n",
      "(('(af355273),', 'canis'), 34)\n",
      "(('homologs', '(fig'), 34)\n",
      "(('correlates', 'predisposition'), 34)\n",
      "(('', 'therefore,'), 34)\n",
      "(('),', 'rat'), 34)\n",
      "(('domains', 'analyzed'), 34)\n",
      "(('abstract', 'brca1'), 34)\n",
      "(('sequences', 'nonredundant'), 34)\n",
      "(('given', 'msa'), 34)\n",
      "(('substitution', 'probabilities'), 34)\n",
      "(('clustalw', '(22)'), 34)\n",
      "(('57', 'functionally'), 34)\n",
      "(('subcloned', 'frame'), 34)\n",
      "(('(8)', 'models'), 34)\n",
      "(('genes', 'found'), 33)\n",
      "(('xl', 'kit'), 33)\n",
      "(('per', 'cell'), 33)\n",
      "(('large', 'deletions'), 33)\n",
      "(('probably', 'due'), 33)\n",
      "(('decreased', 'overall'), 33)\n",
      "(('completely', 'abolished'), 33)\n",
      "(('human', 'disease'), 33)\n",
      "(('overexpression', 'wild-type'), 33)\n",
      "(('table', '1:'), 33)\n",
      "(('leukemia', 'cell'), 33)\n",
      "(('exon', '12,'), 33)\n",
      "(('observed', 'two'), 33)\n",
      "(('transfected', 'various'), 33)\n",
      "(('acid', 'positions'), 33)\n",
      "(('common', 'feature'), 33)\n",
      "(('tumor', 'type'), 33)\n",
      "(('two', 'tumors'), 33)\n",
      "(('mutated', 'gene'), 33)\n",
      "(('11', '12'), 33)\n",
      "(('5', '10'), 33)\n",
      "(('', ''), 33)\n",
      "(('(figure', '6)'), 33)\n",
      "(('missense', 'variant'), 33)\n",
      "(('myelodysplastic', 'syndrome'), 33)\n",
      "(('cells', 'derived'), 33)\n",
      "(('introduction', 'mutations'), 33)\n",
      "(('catalytic', 'site'), 33)\n",
      "(('previous', 'work'), 33)\n",
      "(('signaling', 'growth'), 33)\n",
      "(('al', 'here,'), 33)\n",
      "(('reduced', 'cell'), 33)\n",
      "(('levels', '(figure'), 33)\n",
      "(('consistent', 'prior'), 33)\n",
      "(('focal', 'amplifications'), 33)\n",
      "(('sufficient', 'cause'), 33)\n",
      "(('variants', 'found'), 33)\n",
      "(('breast', 'carcinomas'), 33)\n",
      "(('tested', '(fig'), 33)\n",
      "(('mouse', 'embryonic'), 33)\n",
      "(('mutant', 'form'), 33)\n",
      "(('analysis', 'identified'), 33)\n",
      "(('studies', 'suggested'), 33)\n",
      "(('reporter', 'construct'), 33)\n",
      "(('40', '60'), 33)\n",
      "(('may', 'due'), 33)\n",
      "(('signaling', 'may'), 33)\n",
      "(('small', 'sample'), 33)\n",
      "(('receptor', 'kinase'), 33)\n",
      "(('signaling', 'pathway,'), 33)\n",
      "(('active', 'kinase'), 33)\n",
      "(('reporter', 'assay'), 33)\n",
      "(('lines', 'used'), 33)\n",
      "(('extracts', 'cells'), 33)\n",
      "(('exact', 'test'), 33)\n",
      "(('observed', 'tumors'), 33)\n",
      "(('experimental', 'data'), 33)\n",
      "(('introduced', 'using'), 33)\n",
      "(('direct', 'evidence'), 33)\n",
      "(('primary', 'tumors,'), 33)\n",
      "(('1', 'summary'), 33)\n",
      "(('1', 'case'), 33)\n",
      "(('that,', 'compared'), 33)\n",
      "(('mutations', 'suggests'), 33)\n",
      "(('form', 'colonies'), 33)\n",
      "(('exons', 'encoding'), 33)\n",
      "(('tumors,', 'whereas'), 33)\n",
      "(('identified', 'five'), 33)\n",
      "(('high', 'proportion'), 33)\n",
      "(('download', 'powerpoint'), 33)\n",
      "(('disease', 'associated'), 33)\n",
      "(('exon', '9,'), 33)\n",
      "(('myeloid', 'cell'), 33)\n",
      "(('patients', 'mutant'), 33)\n",
      "(('next', 'performed'), 33)\n",
      "(('although', 'statistically'), 33)\n",
      "(('total', 'egfr'), 33)\n",
      "(('19', '21'), 33)\n",
      "(('patients', 'adenocarcinoma'), 33)\n",
      "(('harboring', 'egfr'), 33)\n",
      "(('(egfr)', 'gene'), 33)\n",
      "(('nine', 'patients'), 33)\n",
      "(('mutants', 'egfr'), 33)\n",
      "(('egfr', 'constructs'), 33)\n",
      "(('treatment', 'gefitinib'), 33)\n",
      "(('egfr', 'could'), 33)\n",
      "(('(fig', 'sequence'), 33)\n",
      "(('gefitinib-treated', 'nsclc'), 33)\n",
      "(('overexpression', 'egfr'), 33)\n",
      "(('pharmacologic', 'inhibition'), 33)\n",
      "(('analysis', 'wild-type'), 33)\n",
      "(('among', 'vertebrate'), 33)\n",
      "(('domain', 'encoded'), 33)\n",
      "(('mutation,', 'found'), 33)\n",
      "(('however,', 'egfr'), 33)\n",
      "(('protein', 'absence'), 33)\n",
      "(('lung', 'squamous'), 33)\n",
      "(('(9,', '10)'), 33)\n",
      "(('also', 'determined'), 33)\n",
      "(('treated', '1'), 33)\n",
      "(('egfr', 'status'), 33)\n",
      "(('sensitivity', 'various'), 33)\n",
      "(('performance', 'status'), 33)\n",
      "(('data', 'confirm'), 33)\n",
      "(('either', 'wt'), 33)\n",
      "(('study', 'included'), 33)\n",
      "(('signaling', 'technology'), 33)\n",
      "(('survival', '(os)'), 33)\n",
      "(('increased', 'proliferation'), 33)\n",
      "(('study,', 'show'), 33)\n",
      "(('amplified', 'pcr'), 33)\n",
      "(('reported', 'patients'), 33)\n",
      "(('tm,', 'transmembrane'), 33)\n",
      "(('inhibition', 'observed'), 33)\n",
      "(('significant', 'proportion'), 33)\n",
      "(('mutations', 'accompanied'), 33)\n",
      "(('transformed', 'cells'), 33)\n",
      "(('mutation', 'activates'), 33)\n",
      "(('stat5', 'activation'), 33)\n",
      "(('favorable', 'prognosis'), 33)\n",
      "(('mutation', 'cell'), 33)\n",
      "(('observed', 'several'), 33)\n",
      "(('genetic', 'aberrations'), 33)\n",
      "(('ras', 'pathway'), 33)\n",
      "(('mutations', 'tp53'), 33)\n",
      "(('reaction', '(rt-pcr)'), 33)\n",
      "(('expressed', 'ba/f3'), 33)\n",
      "(('cancers,', 'including'), 33)\n",
      "(('ongoing', 'clinical'), 33)\n",
      "(('hamster', 'ovary'), 33)\n",
      "(('chinese', 'hamster'), 33)\n",
      "(('different', 'types'), 33)\n",
      "(('develop', 'resistance'), 33)\n",
      "(('mutation', 'homozygous'), 33)\n",
      "(('respectively', '(figure'), 33)\n",
      "(('deleterious', 'missense'), 33)\n",
      "(('increased', 'basal'), 33)\n",
      "(('human', 'cell'), 33)\n",
      "(('might', 'induce'), 33)\n",
      "(('suppressive', 'role'), 33)\n",
      "(('extension', 'pcr'), 33)\n",
      "(('leukemia', '(t-all)'), 33)\n",
      "(('including', 'one'), 33)\n",
      "(('binding', 'protein'), 33)\n",
      "(('transmembrane', 'receptor'), 33)\n",
      "(('specific', 'ligand'), 33)\n",
      "(('cells', 'lysed'), 33)\n",
      "(('fms-like', 'tyrosine'), 33)\n",
      "(('receptor', 'beta'), 33)\n",
      "(('fragments', 'containing'), 33)\n",
      "(('al', 'addition,'), 33)\n",
      "(('another', 'mechanism'), 33)\n",
      "(('involved', 'either'), 33)\n",
      "(('residual', 'disease'), 33)\n",
      "(('minimal', 'residual'), 33)\n",
      "(('molecular', 'targeted'), 33)\n",
      "(('alterations', 'cancer'), 33)\n",
      "(('pathway', 'signaling'), 33)\n",
      "(('13', 'mutations'), 33)\n",
      "(('s2', 'site'), 33)\n",
      "(('braf', 'mutant'), 33)\n",
      "(('cells', 'specific'), 33)\n",
      "(('also', 'disrupt'), 33)\n",
      "(('target', 'promoters'), 33)\n",
      "(('basis', 'observation'), 33)\n",
      "(('activation', 'notch1'), 33)\n",
      "(('v560d', 'exon'), 33)\n",
      "(('kit', 'mutant'), 33)\n",
      "(('notch1', 'notch2'), 33)\n",
      "(('clinical', 'phenotype'), 32)\n",
      "(('a549', 'cells'), 32)\n",
      "(('deletion', 'region'), 32)\n",
      "(('compared', 'vector'), 32)\n",
      "(('ligase', 'activity'), 32)\n",
      "(('displayed', 'enhanced'), 32)\n",
      "(('iii', 'receptor'), 32)\n",
      "(('previously', 'found'), 32)\n",
      "(('several', 'groups'), 32)\n",
      "(('well', 'known'), 32)\n",
      "(('show', 'loss'), 32)\n",
      "(('evolutionarily', 'conserved'), 32)\n",
      "(('truncated', 'protein'), 32)\n",
      "(('acid', 'sequences'), 32)\n",
      "(('soft', 'tissue'), 32)\n",
      "(('raised', 'possibility'), 32)\n",
      "(('mutations', 'frequent'), 32)\n",
      "(('subset', 'human'), 32)\n",
      "(('egfr', 'amplification'), 32)\n",
      "(('adult', 'patients'), 32)\n",
      "(('95%', 'confidence'), 32)\n",
      "(('cells', 'exhibited'), 32)\n",
      "(('rna', '(sirna)'), 32)\n",
      "(('mutation', 'site'), 32)\n",
      "(('2', 'g'), 32)\n",
      "(('sequencing', 'revealed'), 32)\n",
      "(('different', 'tumor'), 32)\n",
      "(('supplemental', 'table'), 32)\n",
      "(('mutation', 'would'), 32)\n",
      "(('displayed', 'reduced'), 32)\n",
      "(('experimental', 'procedures'), 32)\n",
      "(('could', 'due'), 32)\n",
      "(('promote', 'tumor'), 32)\n",
      "(('tumors', 'also'), 32)\n",
      "(('domain', 'complex'), 32)\n",
      "(('different', 'patients'), 32)\n",
      "(('index', 'patient'), 32)\n",
      "(('rna', 'interference'), 32)\n",
      "(('one', 'possibility'), 32)\n",
      "(('activity', 'also'), 32)\n",
      "(('studies', 'indicated'), 32)\n",
      "(('common', 'mutation'), 32)\n",
      "(('associated', 'high'), 32)\n",
      "(('confirmed', 'using'), 32)\n",
      "(('proteins', 'compared'), 32)\n",
      "(('schematic', 'diagram'), 32)\n",
      "(('expression', 'detected'), 32)\n",
      "(('examine', 'role'), 32)\n",
      "(('could', 'show'), 32)\n",
      "(('expression', 'could'), 32)\n",
      "(('intermediate', 'levels'), 32)\n",
      "(('may', 'associated'), 32)\n",
      "(('5', 'days'), 32)\n",
      "(('model', 'human'), 32)\n",
      "(('colony', 'formation,'), 32)\n",
      "(('cells', 'wild'), 32)\n",
      "(('4', 'weeks'), 32)\n",
      "(('mutations', 'four'), 32)\n",
      "(('interactions', 'within'), 32)\n",
      "(('patients', 'analyzed'), 32)\n",
      "(('mutant', 'versus'), 32)\n",
      "(('others', 'shown'), 32)\n",
      "(('genes', 'identified'), 32)\n",
      "(('lower', 'concentrations'), 32)\n",
      "(('activation', 'egfr'), 32)\n",
      "(('within', 'specific'), 32)\n",
      "(('endogenous', 'egfr'), 32)\n",
      "(('sensitive', 'erlotinib'), 32)\n",
      "(('(roche', 'applied'), 32)\n",
      "(('inhibition', 'mutant'), 32)\n",
      "(('0', 'egfr'), 32)\n",
      "(('activity', 'drug'), 32)\n",
      "(('mutations', 'egfr,'), 32)\n",
      "(('identified', 'specific'), 32)\n",
      "(('following', 'antibodies:'), 32)\n",
      "(('concentrations', 'egfr'), 32)\n",
      "(('atp-competitive', 'inhibitor'), 32)\n",
      "(('l858r', 'mutations'), 32)\n",
      "(('inhibition', 'gefitinib'), 32)\n",
      "(('current', 'data'), 32)\n",
      "(('patients', 'nsclc,'), 32)\n",
      "(('mutant', 'inhibited'), 32)\n",
      "(('egfr', 'dna'), 32)\n",
      "(('activation', 'three'), 32)\n",
      "(('one', 'nine'), 32)\n",
      "(('(17,', '18),'), 32)\n",
      "(('correlate', 'clinical'), 32)\n",
      "(('cancer', 'institute'), 32)\n",
      "(('h3255', 'cells'), 32)\n",
      "(('mutations', 'detected,'), 32)\n",
      "(('distinct', 'mutations'), 32)\n",
      "(('domain', '(exons'), 32)\n",
      "(('within', 'entire'), 32)\n",
      "(('vitro', 'analysis'), 32)\n",
      "(('responsiveness', 'gefitinib'), 32)\n",
      "(('surgically', 'resected'), 32)\n",
      "(('versus', 'exon'), 32)\n",
      "(('nearly', 'identical'), 32)\n",
      "(('treatment', 'vitro'), 32)\n",
      "(('mutated', 'egfr'), 32)\n",
      "(('survival', 'analysis'), 32)\n",
      "(('tkis', 'compared'), 32)\n",
      "(('cases', 'harbored'), 32)\n",
      "(('patients', 'receiving'), 32)\n",
      "(('novel', 'egfr'), 32)\n",
      "(('several', 'patients'), 32)\n",
      "(('ic50', 'value'), 32)\n",
      "(('downstream', 'signaling,'), 32)\n",
      "(('recently,', 'demonstrated'), 32)\n",
      "(('expressed', 'high'), 32)\n",
      "(('3', 'gene'), 32)\n",
      "(('single', 'mutant'), 32)\n",
      "(('generated', 'site-directed'), 32)\n",
      "(('shown', 'confer'), 32)\n",
      "(('could', 'lead'), 32)\n",
      "(('lie', 'within'), 32)\n",
      "(('high-performance', 'liquid'), 32)\n",
      "(('several', 'mutations'), 32)\n",
      "(('show', 'mutations'), 32)\n",
      "(('plays', 'role'), 32)\n",
      "(('residues', 'within'), 32)\n",
      "(('mutation', 'two'), 32)\n",
      "(('mutations', 'expressed'), 32)\n",
      "(('using', 'primer'), 32)\n",
      "(('restriction', 'enzyme'), 32)\n",
      "(('examined', 'ability'), 32)\n",
      "(('almost', 'exclusively'), 32)\n",
      "(('(this', 'study)'), 32)\n",
      "(('splice', 'variant'), 32)\n",
      "(('predicted', 'result'), 32)\n",
      "(('sequencing', 'entire'), 32)\n",
      "(('study,', 'evaluated'), 32)\n",
      "(('0', 'figure'), 32)\n",
      "(('another', 'report'), 32)\n",
      "(('domain', '1'), 32)\n",
      "(('led', 'similar'), 32)\n",
      "(('h1299', 'cells'), 32)\n",
      "(('hcc', 'cells'), 32)\n",
      "(('mutations', 'reduced'), 32)\n",
      "(('alk', 'mutations'), 32)\n",
      "(('study', 'demonstrates'), 32)\n",
      "(('activity', 'pten'), 32)\n",
      "(('notch1', 'activation'), 32)\n",
      "(('myeloid', 'leukemia,'), 32)\n",
      "(('vitro', 'functional'), 32)\n",
      "(('inhibit', 'kinase'), 32)\n",
      "(('mutated', 'human'), 32)\n",
      "(('cells', 'example,'), 32)\n",
      "(('higher', 'dose'), 32)\n",
      "(('breast', 'tumors,'), 32)\n",
      "(('inhibitor', 'imatinib'), 32)\n",
      "(('prognostic', 'marker'), 32)\n",
      "(('minor', 'role'), 32)\n",
      "(('mutation', 'common'), 32)\n",
      "(('molecule', 'inhibitors'), 32)\n",
      "(('pdgfra', 'cdna'), 32)\n",
      "(('expected', 'result'), 32)\n",
      "(('phase', 'i/ii'), 32)\n",
      "(('vitro', 'assays'), 32)\n",
      "(('inhibition', 'kit'), 32)\n",
      "(('lines', '(figure'), 32)\n",
      "(('pathway', 'tumors'), 32)\n",
      "(('mutations,', 'small'), 32)\n",
      "(('specific', 'expression'), 32)\n",
      "(('al,', '2013)'), 32)\n",
      "(('3', 'variants'), 32)\n",
      "(('likely', 'also'), 32)\n",
      "(('fgfr', 'inhibitors'), 32)\n",
      "(('mutation', 'involving'), 32)\n",
      "(('pre-clinical', 'models'), 32)\n",
      "(('adverse', 'prognostic'), 32)\n",
      "(('hematopoietic', 'progenitors'), 32)\n",
      "(('helical', 'bundle'), 32)\n",
      "(('genes', 'important'), 32)\n",
      "(('could', 'occur'), 32)\n",
      "(('burkitt', 'lymphoma,'), 32)\n",
      "(('thus', 'potential'), 32)\n",
      "(('tumorigenic', 'role'), 32)\n",
      "(('ten', 'years'), 32)\n",
      "(('proteolytic', 'cleavage'), 32)\n",
      "(('11', 'w557g'), 32)\n",
      "(('17', 'n822k'), 32)\n",
      "(('kit', 'phosphorylation'), 32)\n",
      "(('mutations', 'notch1'), 32)\n",
      "(('supporting', 'hypothesis'), 31)\n",
      "(('line', 'derived'), 31)\n",
      "(('increase', 'cell'), 31)\n",
      "(('figure', 's1)'), 31)\n",
      "(('cell', 'survival,'), 31)\n",
      "(('(fig', 'determine'), 31)\n",
      "(('contrast,', 'cells'), 31)\n",
      "(('kit', 'expression'), 31)\n",
      "(('leukemia', '(aml)'), 31)\n",
      "(('clinical', 'course'), 31)\n",
      "(('frequently', 'found'), 31)\n",
      "(('jolla,', 'ca,'), 31)\n",
      "(('figure', '1:'), 31)\n",
      "(('detected', 'three'), 31)\n",
      "(('studies', 'identified'), 31)\n",
      "(('activation', 'stat5'), 31)\n",
      "(('agreement', 'previous'), 31)\n",
      "(('lines', 'using'), 31)\n",
      "(('containing', 'mutations'), 31)\n",
      "(('braf', 'mutations,'), 31)\n",
      "(('15', 'tumors'), 31)\n",
      "(('table', 's4)'), 31)\n",
      "(('type', 'mutation'), 31)\n",
      "(('together,', 'findings'), 31)\n",
      "(('(for', 'example,'), 31)\n",
      "(('mutations', 'confirmed'), 31)\n",
      "(('test', 'hypothesis'), 31)\n",
      "(('findings', 'consistent'), 31)\n",
      "(('shown)', 'although'), 31)\n",
      "(('pcr', 'analysis'), 31)\n",
      "(('human', 'tumors,'), 31)\n",
      "(('cancers', 'go'), 31)\n",
      "(('matched', 'normal'), 31)\n",
      "(('colorectal', 'cancer,'), 31)\n",
      "(('suggesting', 'mutation'), 31)\n",
      "(('(figure', '1a),'), 31)\n",
      "(('interesting', 'note'), 31)\n",
      "(('expressed', 'wild-type'), 31)\n",
      "(('evaluate', 'role'), 31)\n",
      "(('previous', 'data'), 31)\n",
      "(('germline', 'dna'), 31)\n",
      "(('tumour', 'cells'), 31)\n",
      "(('mammalian', 'cells,'), 31)\n",
      "(('signaling', 'activity'), 31)\n",
      "(('results', 'mutations'), 31)\n",
      "(('colon', 'cancers'), 31)\n",
      "(('produce', 'truncated'), 31)\n",
      "(('rt-pcr', 'analysis'), 31)\n",
      "(('8', 'figure'), 31)\n",
      "(('atpase', 'activity'), 31)\n",
      "(('mutations', 'series'), 31)\n",
      "(('mutations', 'three'), 31)\n",
      "(('sequencing', 'pcr'), 31)\n",
      "(('caused', 'mutations'), 31)\n",
      "(('unclassified', 'variants'), 31)\n",
      "(('examined', 'mutations'), 31)\n",
      "(('markedly', 'decreased'), 31)\n",
      "(('promotes', 'cell'), 31)\n",
      "(('mice', 'develop'), 31)\n",
      "(('thus,', 'although'), 31)\n",
      "(('growth', 'arrest'), 31)\n",
      "(('protein', 'synthesis'), 31)\n",
      "(('normal', 'tumor'), 31)\n",
      "(('lymphocytic', 'leukemia'), 31)\n",
      "(('human', 'cells'), 31)\n",
      "(('founder', 'mutations'), 31)\n",
      "(('determined', 'using'), 31)\n",
      "(('patients', 'known'), 31)\n",
      "(('figure', '3,'), 31)\n",
      "(('mutations', 'contribute'), 31)\n",
      "(('cells', 'however,'), 31)\n",
      "(('tp53', 'mutation'), 31)\n",
      "(('expression', 'associated'), 31)\n",
      "(('line', 'egfr'), 31)\n",
      "(('bound', 'gefitinib'), 31)\n",
      "(('ability', 'phosphorylate'), 31)\n",
      "(('19', 'egfr'), 31)\n",
      "(('type', '(fig'), 31)\n",
      "(('supporting', 'information'), 31)\n",
      "(('sloan-kettering', 'cancer'), 31)\n",
      "(('transfection', 'assays'), 31)\n",
      "(('25', 'egfr'), 31)\n",
      "(('cells', 'serum-starved'), 31)\n",
      "(('gefitinib', 'lung'), 31)\n",
      "(('full-length', 'egfr'), 31)\n",
      "(('identified', 'presence'), 31)\n",
      "(('results,', 'together'), 31)\n",
      "(('mutations', 'used'), 31)\n",
      "(('within', 'tyrosine'), 31)\n",
      "(('sufficient', 'suppress'), 31)\n",
      "(('particular', 'interest,'), 31)\n",
      "(('gefitinib', 'wild-type'), 31)\n",
      "(('various', 'egfr'), 31)\n",
      "(('important', 'determine'), 31)\n",
      "(('entire', 'egfr'), 31)\n",
      "(('previously', 'detected'), 31)\n",
      "(('whether', 'egfr'), 31)\n",
      "(('domain', 'egfr,'), 31)\n",
      "(('egfr', 'kinases'), 31)\n",
      "(('receptor', 'may'), 31)\n",
      "(('analysis', 'mutant'), 31)\n",
      "(('addition', 'egf'), 31)\n",
      "(('immunoblotting', 'using'), 31)\n",
      "(('effect', 'drug'), 31)\n",
      "(('may', 'activate'), 31)\n",
      "(('19', 'deletions'), 31)\n",
      "(('showed', 'egfr'), 31)\n",
      "(('chromosome', '7'), 31)\n",
      "(('observed', 'egfr'), 31)\n",
      "(('similar', 'patients'), 31)\n",
      "(('patients', 'responses'), 31)\n",
      "(('iii', 'clinical'), 31)\n",
      "(('kinase', 'b'), 31)\n",
      "(('mutation', 'lung'), 31)\n",
      "(('n-terminal', 'lobe'), 31)\n",
      "(('(fig', 'expected,'), 31)\n",
      "(('relatively', 'higher'), 31)\n",
      "(('combination', 'chemotherapy,'), 31)\n",
      "(('similar', 'previous'), 31)\n",
      "(('amplification', 'reported'), 31)\n",
      "(('intracellular', 'tyrosine'), 31)\n",
      "(('able', 'transform'), 31)\n",
      "(('phosphorylation', 'cells'), 31)\n",
      "(('expressed', 'comparable'), 31)\n",
      "(('tki', 'therapy'), 31)\n",
      "(('hydrophobic', 'residues'), 31)\n",
      "(('approximately', '20%'), 31)\n",
      "(('mice', '(fig'), 31)\n",
      "(('erk1/2', 'akt'), 31)\n",
      "(('genes', 'previously'), 31)\n",
      "(('able', 'activate'), 31)\n",
      "(('room', 'temperature'), 31)\n",
      "(('low', 'nanomolar'), 31)\n",
      "(('able', 'induce'), 31)\n",
      "(('assay', '(fig'), 31)\n",
      "(('report', 'first'), 31)\n",
      "(('mutation', 'corresponding'), 31)\n",
      "(('transformed', 'phenotype'), 31)\n",
      "(('cells', 'activated'), 31)\n",
      "(('tumor', 'dna'), 31)\n",
      "(('mutant', 'tumors'), 31)\n",
      "(('3', '3'), 31)\n",
      "(('cells', 'induced'), 31)\n",
      "(('activity', 'wt'), 31)\n",
      "(('mutation', 'either'), 31)\n",
      "(('directed', 'mutagenesis'), 31)\n",
      "(('fig', 'interestingly,'), 31)\n",
      "(('chromatin', 'immunoprecipitation'), 31)\n",
      "(('alk', 'inhibitors'), 31)\n",
      "(('characterized', 'activating'), 31)\n",
      "(('exons', '11'), 31)\n",
      "(('included', 'study'), 31)\n",
      "(('table', '1,'), 31)\n",
      "(('plasmid', 'containing'), 31)\n",
      "(('endothelial', 'cells'), 31)\n",
      "(('mutant', 'p53s'), 31)\n",
      "(('fig', 'also'), 31)\n",
      "(('packing', 'interactions'), 31)\n",
      "(('report', 'showing'), 31)\n",
      "(('type', '1'), 31)\n",
      "(('among', 'mutations'), 31)\n",
      "(('suggesting', 'tumor'), 31)\n",
      "(('form', 'part'), 31)\n",
      "(('domain', 'notch1'), 31)\n",
      "(('fbxw7', 'mutations'), 31)\n",
      "(('whether', 'notch1'), 31)\n",
      "(('patients', 'primary'), 31)\n",
      "(('generation', 'sequencing'), 31)\n",
      "(('mutant', 'isoforms'), 31)\n",
      "(('fig', '2'), 31)\n",
      "(('epitope', 'tag'), 31)\n",
      "(('mutation', 'secondary'), 31)\n",
      "(('twice', 'daily'), 31)\n",
      "(('found', 'single'), 31)\n",
      "(('cos', 'cells'), 31)\n",
      "(('leukemia', '(cml)'), 31)\n",
      "(('disrupt', 'interaction'), 31)\n",
      "(('nonsense', 'mutations,'), 31)\n",
      "(('mutations', 'notch1,'), 31)\n",
      "(('inhibits', 'tumor'), 31)\n",
      "(('attractive', 'target'), 31)\n",
      "(('stimulatory', 'effects'), 31)\n",
      "(('presence', 'additional'), 31)\n",
      "(('highly', 'resistant'), 31)\n",
      "(('furthermore,', 'using'), 31)\n",
      "(('flt3', 'mutations'), 31)\n",
      "(('buried', 'hydrophobic'), 31)\n",
      "(('mtor', 'mutations'), 31)\n",
      "(('key', 'genes'), 31)\n",
      "(('t-all', 'cells'), 31)\n",
      "(('lost', 'tumors'), 31)\n",
      "(('upregulation', 'myc'), 31)\n",
      "(('proline', 'substitution'), 31)\n",
      "(('human', 't-all'), 31)\n",
      "(('notch', 'pathway'), 31)\n",
      "(('17', 'y823d'), 31)\n",
      "(('17', 'd820g'), 31)\n",
      "(('treatment', 'disease'), 31)\n",
      "(('notch1', '(fig'), 31)\n",
      "(('chemotherapeutic', 'response'), 31)\n",
      "(('could', 'result'), 30)\n",
      "(('using', 'quickchange'), 30)\n",
      "(('comparable', 'wild-type'), 30)\n",
      "(('patient', '1'), 30)\n",
      "(('controls', '(fig'), 30)\n",
      "(('ubiquitination', 'degradation'), 30)\n",
      "(('lipofectamine', '2000'), 30)\n",
      "(('phosphorylation', 'observed'), 30)\n",
      "(('cells', 'well'), 30)\n",
      "(('growth', '(figure'), 30)\n",
      "(('cells', 'similar'), 30)\n",
      "(('mutations', 'seem'), 30)\n",
      "(('1', '2)'), 30)\n",
      "(('normal', 'tissues'), 30)\n",
      "(('conserved', 'amino'), 30)\n",
      "(('receptor', '(egfr),'), 30)\n",
      "(('variant', 'allele'), 30)\n",
      "(('prostate', 'cancers'), 30)\n",
      "(('reporter', 'assays'), 30)\n",
      "(('first', 'evidence'), 30)\n",
      "(('may', 'function'), 30)\n",
      "(('gene', 'amplifications'), 30)\n",
      "(('arrows', 'indicate'), 30)\n",
      "(('biochemical', 'data'), 30)\n",
      "(('activity', 'using'), 30)\n",
      "(('(fig', 'deletion'), 30)\n",
      "(('vitro', 'data,'), 30)\n",
      "(('mice', '(figure'), 30)\n",
      "(('bovine', 'serum'), 30)\n",
      "(('significantly', 'inhibited'), 30)\n",
      "(('damage', 'response'), 30)\n",
      "(('0', 'however,'), 30)\n",
      "(('dominant-negative', 'effect'), 30)\n",
      "(('showed', 'decreased'), 30)\n",
      "(('protein', 'detected'), 30)\n",
      "(('domain', 'two'), 30)\n",
      "(('interestingly,', 'one'), 30)\n",
      "(('embryonic', 'lethality'), 30)\n",
      "(('assess', 'functional'), 30)\n",
      "(('binding', '(fig'), 30)\n",
      "(('al', 'contrast,'), 30)\n",
      "(('evaluated', 'using'), 30)\n",
      "(('(fig', '5b)'), 30)\n",
      "(('growth', 'inhibitory'), 30)\n",
      "(('may', 'necessary'), 30)\n",
      "(('lead', 'development'), 30)\n",
      "(('detected', 'western'), 30)\n",
      "(('mutations', 'genomic'), 30)\n",
      "(('al', 'furthermore,'), 30)\n",
      "(('(b)', 'expression'), 30)\n",
      "(('oncogenic', 'activation'), 30)\n",
      "(('affect', 'affinity'), 30)\n",
      "(('results', 'could'), 30)\n",
      "(('model', 'system'), 30)\n",
      "(('heterodimer', 'formation'), 30)\n",
      "(('phenotype', 'observed'), 30)\n",
      "(('mutations', 'tumor'), 30)\n",
      "(('role', 'played'), 30)\n",
      "(('next', 'used'), 30)\n",
      "(('cells', 'examined'), 30)\n",
      "(('time', 'points'), 30)\n",
      "(('biological', 'functions'), 30)\n",
      "(('(applied', 'biosystems,'), 30)\n",
      "(('pathways,', 'including'), 30)\n",
      "(('mutation,', 'whereas'), 30)\n",
      "(('structural', 'information'), 30)\n",
      "(('expression', 'pattern'), 30)\n",
      "(('6', '9'), 30)\n",
      "(('blood', 'cell'), 30)\n",
      "(('equal', 'numbers'), 30)\n",
      "(('higher', 'percentage'), 30)\n",
      "(('however,', 'observed'), 30)\n",
      "(('might', 'represent'), 30)\n",
      "(('regulatory', 'approval'), 30)\n",
      "(('seven', 'tumors'), 30)\n",
      "(('treatment', '100'), 30)\n",
      "(('dose-dependent', 'inhibition'), 30)\n",
      "(('clustered', 'near'), 30)\n",
      "(('patients', 'response'), 30)\n",
      "(('also', 'harbor'), 30)\n",
      "(('expression', 'total'), 30)\n",
      "(('revealed', 'missense'), 30)\n",
      "(('account', 'majority'), 30)\n",
      "(('benefit', 'treatment'), 30)\n",
      "(('contain', 'mutations'), 30)\n",
      "(('consistent', 'results,'), 30)\n",
      "(('important', 'questions'), 30)\n",
      "(('levels', 'tumor'), 30)\n",
      "(('two', 'overlapping'), 30)\n",
      "(('l858r', 'mutation,'), 30)\n",
      "(('mutations,', '11'), 30)\n",
      "(('treatment', 'strategy'), 30)\n",
      "(('duration', 'response'), 30)\n",
      "(('could', 'confer'), 30)\n",
      "(('patients', 'bearing'), 30)\n",
      "(('genetically', 'engineered'), 30)\n",
      "(('rare', 'egfr'), 30)\n",
      "(('study,', 'showed'), 30)\n",
      "(('18', '(4'), 30)\n",
      "(('pathogenesis', 'lung'), 30)\n",
      "(('used', 'predict'), 30)\n",
      "(('progression', 'within'), 30)\n",
      "(('one', 'four'), 30)\n",
      "(('substitution', 'mutation'), 30)\n",
      "(('mutations', 'studied'), 30)\n",
      "(('higher', 'rate'), 30)\n",
      "(('amplification', 'mutations'), 30)\n",
      "(('increased', 'affinity'), 30)\n",
      "(('kinase', 'binding'), 30)\n",
      "(('mutations', 'distributed'), 30)\n",
      "(('exon', '21,'), 30)\n",
      "(('insertion', 'mutations'), 30)\n",
      "(('patients', 'tested'), 30)\n",
      "(('mutation', 'show'), 30)\n",
      "(('tki', 'resistance'), 30)\n",
      "(('study', 'patients'), 30)\n",
      "(('6', 'patients'), 30)\n",
      "(('study', 'suggests'), 30)\n",
      "(('several', 'reports'), 30)\n",
      "(('phosphorylation', 'activation'), 30)\n",
      "(('activating', 'role'), 30)\n",
      "(('shown', 'several'), 30)\n",
      "(('(applied', 'biosystems)'), 30)\n",
      "(('patients', 'de'), 30)\n",
      "(('mutations', 'displayed'), 30)\n",
      "(('mutation', 'described'), 30)\n",
      "(('relatively', 'less'), 30)\n",
      "(('shown', 'supplementary'), 30)\n",
      "(('96-well', 'plates'), 30)\n",
      "(('growth', 'factorindependent'), 30)\n",
      "(('view', 'largedownload'), 30)\n",
      "(('largedownload', 'slide'), 30)\n",
      "(('genetic', 'alteration'), 30)\n",
      "(('carcinoma', 'cells'), 30)\n",
      "(('il-3independent', 'growth'), 30)\n",
      "(('another', 'study,'), 30)\n",
      "(('could', 'overcome'), 30)\n",
      "(('substrate', 'phosphorylation'), 30)\n",
      "(('2', '15'), 30)\n",
      "(('obtained', 'patients'), 30)\n",
      "(('summary,', 'identified'), 30)\n",
      "(('proliferation', 'rate'), 30)\n",
      "(('first', 'used'), 30)\n",
      "(('therapeutic', 'strategies'), 30)\n",
      "(('activity', 'due'), 30)\n",
      "(('exhibited', 'constitutive'), 30)\n",
      "(('differential', 'effects'), 30)\n",
      "(('cells', 'assessed'), 30)\n",
      "(('role', 'development'), 30)\n",
      "(('partner', 'gene'), 30)\n",
      "(('exclusive', 'tp53'), 30)\n",
      "(('fusions', 'identified'), 30)\n",
      "(('mutation', 'tumors'), 30)\n",
      "(('figure', '1a)'), 30)\n",
      "(('cells', 'lacking'), 30)\n",
      "(('(fig', 'whereas'), 30)\n",
      "(('aggressive', 'phenotype'), 30)\n",
      "(('inhibited', 'proliferation'), 30)\n",
      "(('protein', 'interaction'), 30)\n",
      "(('indicating', 'increased'), 30)\n",
      "(('test', 'this,'), 30)\n",
      "(('strong', 'activation'), 30)\n",
      "(('expressing', 'mutated'), 30)\n",
      "(('translocation', 'immunoglobulin'), 30)\n",
      "(('expression', 'activated'), 30)\n",
      "(('progression', 'towards'), 30)\n",
      "(('2', 'fig'), 30)\n",
      "(('examined', 'notch1'), 30)\n",
      "(('fbxw7', 'mutational'), 30)\n",
      "(('primary', 't-all'), 30)\n",
      "(('patients', 'aml'), 30)\n",
      "(('al', 'genetic'), 30)\n",
      "(('fibroblast', 'growth'), 30)\n",
      "(('lymphoma', 'patients'), 30)\n",
      "(('resistance', 'associated'), 30)\n",
      "(('factor', 'stimulation'), 30)\n",
      "(('17', 'mutations'), 30)\n",
      "(('pathogenic', 'mutation'), 30)\n",
      "(('activity', 'despite'), 30)\n",
      "(('development', 'mice'), 30)\n",
      "(('(see', 'below),'), 30)\n",
      "(('gene', 'located'), 30)\n",
      "(('poor', 'clinical'), 30)\n",
      "(('san', 'francisco,'), 30)\n",
      "(('likely', 'contribute'), 30)\n",
      "(('shown', 'expression'), 30)\n",
      "(('pdgfra', 'v561d'), 30)\n",
      "(('11,', '13,'), 30)\n",
      "(('gists', 'kit'), 30)\n",
      "(('three', 'variants'), 30)\n",
      "(('weak', 'strong'), 30)\n",
      "(('dead', 'disease;'), 30)\n",
      "(('dod,', 'dead'), 30)\n",
      "(('loading', 'control'), 30)\n",
      "(('ligand-activated', 'wild-type'), 30)\n",
      "(('sufficient', 'confer'), 30)\n",
      "(('chose', '15'), 30)\n",
      "(('mutation', 'catalytic'), 30)\n",
      "(('oncogenic', 'tumor'), 30)\n",
      "(('al', 'authors'), 30)\n",
      "(('inhibition', 'cancer'), 30)\n",
      "(('pathogenic', 'investigation'), 30)\n",
      "(('probabilities', 'mlh1'), 30)\n",
      "(('deleterious', '(fig'), 30)\n",
      "(('bimodal,', 'suggesting'), 30)\n",
      "(('distribution', 'prior'), 30)\n",
      "(('clearly', 'bimodal,'), 30)\n",
      "(('classification', 'assess'), 30)\n",
      "(('msh2', 'missense'), 30)\n",
      "(('~50%', 'mlh1'), 30)\n",
      "(('suggesting', '~50%'), 30)\n",
      "(('7)', 'distribution'), 30)\n",
      "(('variants', 'clearly'), 30)\n",
      "(('msh2', 'class'), 30)\n",
      "(('investigation', '(ae)'), 30)\n",
      "(('significance', '(vus)'), 30)\n",
      "(('therapy', 'go'), 30)\n",
      "(('lower', 'frequencies'), 30)\n",
      "(('promoters', 'enhancers'), 30)\n",
      "(('leukemia', 'cells'), 30)\n",
      "(('ankyrin', 'repeat'), 30)\n",
      "(('leading', 'high'), 30)\n",
      "(('group', 'mutations,'), 30)\n",
      "(('therapy', 'treatment'), 30)\n",
      "(('mek', 'erk'), 30)\n",
      "(('myc', 'promoters'), 30)\n",
      "(('cll', 'cases'), 30)\n",
      "(('observed', 'frequency'), 30)\n",
      "(('ring', 'domain)'), 30)\n",
      "(('domain', 'folding'), 30)\n",
      "(('show', 'drugs'), 30)\n",
      "(('table', 'a1)'), 30)\n",
      "(('authors', 'suggested'), 30)\n",
      "(('binding', 'patterns'), 30)\n",
      "(('17', 'd820a'), 30)\n",
      "(('notch2', 'mutations'), 30)\n",
      "(('heterodimerization', 'domain'), 30)\n",
      "(('pest', 'domain'), 30)\n",
      "(('notch1', 'mutations'), 30)\n",
      "(('domain', 'found'), 29)\n",
      "(('cell', 'numbers'), 29)\n",
      "(('enhanced', 'cell'), 29)\n",
      "(('showed', 'enhanced'), 29)\n",
      "(('comparable', 'wild'), 29)\n",
      "(('survival', 'advantage'), 29)\n",
      "(('proliferation', 'survival'), 29)\n",
      "(('e2', 'ubiquitin'), 29)\n",
      "(('ubiquitin', 'conjugating'), 29)\n",
      "(('figures/tables', 'index'), 29)\n",
      "(('nucleotide', 'polymorphisms'), 29)\n",
      "(('revealed', 'two'), 29)\n",
      "(('vivo', 'studies'), 29)\n",
      "(('myelomonocytic', 'leukemia'), 29)\n",
      "(('would', 'interest'), 29)\n",
      "(('examined', 'effects'), 29)\n",
      "(('fig', 'however,'), 29)\n",
      "(('strong', 'association'), 29)\n",
      "(('investigate', 'function'), 29)\n",
      "(('(fig', 'treatment'), 29)\n",
      "(('results', 'reported'), 29)\n",
      "(('frequent', 'mutations'), 29)\n",
      "(('mutations,', 'respectively'), 29)\n",
      "(('found', 'among'), 29)\n",
      "(('evaluate', 'whether'), 29)\n",
      "(('exhibited', 'significant'), 29)\n",
      "(('indicated', 'mutations'), 29)\n",
      "(('truncating', 'mutations,'), 29)\n",
      "(('together,', 'studies'), 29)\n",
      "(('stromal', 'cells'), 29)\n",
      "(('(fig', 'table'), 29)\n",
      "(('including', 'three'), 29)\n",
      "(('(wang', 'et'), 29)\n",
      "(('mutant', 'could'), 29)\n",
      "(('basal', 'levels'), 29)\n",
      "(('us', 'identify'), 29)\n",
      "(('similar', 'wt'), 29)\n",
      "(('generous', 'gift'), 29)\n",
      "(('data', 'collection'), 29)\n",
      "(('pathway', 'genes'), 29)\n",
      "(('could', 'potentially'), 29)\n",
      "(('cells', '(b)'), 29)\n",
      "(('cells', 'either'), 29)\n",
      "(('evaluated', 'whether'), 29)\n",
      "(('table', 'ii'), 29)\n",
      "(('cells', 'engineered'), 29)\n",
      "(('hodgkin', 'lymphoma'), 29)\n",
      "(('cancer', 'tissues'), 29)\n",
      "(('development', 'progression'), 29)\n",
      "(('cells', 'transfection'), 29)\n",
      "(('seems', 'likely'), 29)\n",
      "(('identified', 'mutation'), 29)\n",
      "(('kinase', 'receptor'), 29)\n",
      "(('also', 'significantly'), 29)\n",
      "(('kinase', 'activities'), 29)\n",
      "(('transfection', 'mutant'), 29)\n",
      "(('occur', 'simply'), 29)\n",
      "(('showed', 'deletion'), 29)\n",
      "(('stem', 'cells,'), 29)\n",
      "(('molecular', 'level,'), 29)\n",
      "(('could', 'obtained'), 29)\n",
      "(('structural', 'rearrangements'), 29)\n",
      "(('alleles', 'expressed'), 29)\n",
      "(('(table', '4)'), 29)\n",
      "(('early', 'b'), 29)\n",
      "(('leads', 'increased'), 29)\n",
      "(('nuclear', 'translocation'), 29)\n",
      "(('quantitative', 'real-time'), 29)\n",
      "(('protein', 'may'), 29)\n",
      "(('conserved', 'domains'), 29)\n",
      "(('complex', '(figure'), 29)\n",
      "(('relative', 'expression'), 29)\n",
      "(('significant', 'decrease'), 29)\n",
      "(('expression', 'normal'), 29)\n",
      "(('homology', 'model'), 29)\n",
      "(('splicing', 'defects'), 29)\n",
      "(('observed', 'human'), 29)\n",
      "(('(a', 'b)'), 29)\n",
      "(('white', 'blood'), 29)\n",
      "(('notch1', 'sf3b1'), 29)\n",
      "(('mutation', 'occurs'), 29)\n",
      "(('assessed', 'effect'), 29)\n",
      "(('lrea', 'motif'), 29)\n",
      "(('associated', 'egfr'), 29)\n",
      "(('identify', 'patients'), 29)\n",
      "(('correlation', 'patient'), 29)\n",
      "(('similar', 'sensitivity'), 29)\n",
      "(('prospective', 'clinical'), 29)\n",
      "(('7', '10'), 29)\n",
      "(('mutant', 'egfrs'), 29)\n",
      "(('egfr', 'l858r'), 29)\n",
      "(('trials', 'egfr'), 29)\n",
      "(('gene', 'egfr'), 29)\n",
      "(('surgical', 'resection'), 29)\n",
      "(('mutant', 'similar'), 29)\n",
      "(('demonstrated', 'that,'), 29)\n",
      "(('inhibition', 'patients'), 29)\n",
      "(('advantage', 'cells'), 29)\n",
      "(('response', 'phase'), 29)\n",
      "(('time', 'progression'), 29)\n",
      "(('harbor', 'egfr'), 29)\n",
      "(('memorial', 'sloan-kettering'), 29)\n",
      "(('tkis,', 'gefitinib'), 29)\n",
      "(('domain', 'epidermal'), 29)\n",
      "(('genes', 'described'), 29)\n",
      "(('mutations', 'gefitinib'), 29)\n",
      "(('10%', '20%'), 29)\n",
      "(('resistance', 'tkis'), 29)\n",
      "(('cooperative', 'oncology'), 29)\n",
      "(('pleural', 'effusion'), 29)\n",
      "(('egfr', 'wild-type'), 29)\n",
      "(('pcr', 'amplifications'), 29)\n",
      "(('10', 'exon'), 29)\n",
      "(('response', 'patient'), 29)\n",
      "(('treated', 'indicated'), 29)\n",
      "(('stark', 'contrast,'), 29)\n",
      "(('correlated', 'clinical'), 29)\n",
      "(('mammalian', 'expression'), 29)\n",
      "(('resulted', 'increase'), 29)\n",
      "(('vitro', 'data'), 29)\n",
      "(('mutation', 'shown'), 29)\n",
      "(('showed', 'elevated'), 29)\n",
      "(('promotes', 'tumor'), 29)\n",
      "(('evidence', 'suggests'), 29)\n",
      "(('6', 'months'), 29)\n",
      "(('cdna', 'constructs'), 29)\n",
      "(('less', 'effective'), 29)\n",
      "(('mutations', 'mutations'), 29)\n",
      "(('clinical', 'efficacy'), 29)\n",
      "(('supplemented', '10%'), 29)\n",
      "(('served', 'positive'), 29)\n",
      "(('findings', 'provide'), 29)\n",
      "(('varying', 'degrees'), 29)\n",
      "(('wild-type', 'cells'), 29)\n",
      "(('denaturing', 'high-performance'), 29)\n",
      "(('informed', 'consent'), 29)\n",
      "(('also', 'indicated'), 29)\n",
      "(('clinical', 'resistance'), 29)\n",
      "(('retained', 'independent'), 29)\n",
      "(('kinase', 'pathway'), 29)\n",
      "(('showed', 'expression'), 29)\n",
      "(('9', 'patients'), 29)\n",
      "(('cytoplasmic', 'domain'), 29)\n",
      "(('large', 'panel'), 29)\n",
      "(('domain', 'activate'), 29)\n",
      "(('one', 'major'), 29)\n",
      "(('resistance', 'treatment'), 29)\n",
      "(('recent', 'study,'), 29)\n",
      "(('mutation', 'increases'), 29)\n",
      "(('greater', 'extent'), 29)\n",
      "(('increased', 'kinase'), 29)\n",
      "(('concentration', '(ic50)'), 29)\n",
      "(('intracellular', 'part'), 29)\n",
      "(('accompanied', 'mutations'), 29)\n",
      "(('also', 'consistent'), 29)\n",
      "(('signaling', 'human'), 29)\n",
      "(('amino', 'acids)'), 29)\n",
      "(('(see', 'below)'), 29)\n",
      "(('resulted', 'loss'), 29)\n",
      "(('alk', 'inhibitor'), 29)\n",
      "(('tumor', 'characteristics'), 29)\n",
      "(('new', 'therapeutic'), 29)\n",
      "(('cell', 'adhesion'), 29)\n",
      "(('hereditary', 'nonpolyposis'), 29)\n",
      "(('plasmids', 'containing'), 29)\n",
      "(('cells', 'determine'), 29)\n",
      "(('lifetime', 'risk'), 29)\n",
      "(('failed', 'reveal'), 29)\n",
      "(('structures', 'provide'), 29)\n",
      "(('(supplemental', 'fig'), 29)\n",
      "(('pathways', 'involved'), 29)\n",
      "(('m,', 'male;'), 29)\n",
      "(('exhibit', 'gain-of-function'), 29)\n",
      "(('and,', 'therefore,'), 29)\n",
      "(('cells', '(figures'), 29)\n",
      "(('activity', 'mutants'), 29)\n",
      "(('assay', 'performed'), 29)\n",
      "(('tumor', 'invasion'), 29)\n",
      "(('allele', 'frequency'), 29)\n",
      "(('2,', '3'), 29)\n",
      "(('compound', 'deletion'), 29)\n",
      "(('mcf10a', 'cells'), 29)\n",
      "(('activity', 'comparable'), 29)\n",
      "(('subjected', 'immunoprecipitation'), 29)\n",
      "(('al', 'mutational'), 29)\n",
      "(('resulted', 'strong'), 29)\n",
      "(('pdgfra', 'gene'), 29)\n",
      "(('9', 'mutation'), 29)\n",
      "(('gist', 'patients'), 29)\n",
      "(('treated', 'imatinib'), 29)\n",
      "(('tumors', 'primary'), 29)\n",
      "(('et', 'al9'), 29)\n",
      "(('demonstrated', 'antagonism'), 29)\n",
      "(('resulted', 'reduced'), 29)\n",
      "(('(dan', 'et'), 29)\n",
      "(('greatly', 'increased'), 29)\n",
      "(('residues', '(fig'), 29)\n",
      "(('gene', 'one'), 29)\n",
      "(('role', 'controlling'), 29)\n",
      "(('recurrent', 'gain-of-function'), 29)\n",
      "(('identified', 'cancer'), 29)\n",
      "(('progression', 'primary'), 29)\n",
      "(('cause', 'constitutive'), 29)\n",
      "(('antibody', 'purchased'), 29)\n",
      "(('using', 'next'), 29)\n",
      "(('significance', 'survival'), 29)\n",
      "(('passenger', 'mutations'), 29)\n",
      "(('domain', 'interface'), 29)\n",
      "(('cases', 'exhibit'), 29)\n",
      "(('pool', 'cells'), 29)\n",
      "(('lymphoblastic', 'lymphoma'), 29)\n",
      "(('truncated', 'receptors'), 29)\n",
      "(('prognostic', 'parameter'), 29)\n",
      "(('hek-293', 'cells'), 29)\n",
      "(('group', 'mutations'), 29)\n",
      "(('also,', 'high'), 29)\n",
      "(('disease', '(mrd)'), 29)\n",
      "(('response', 'lower'), 29)\n",
      "(('(avila', 'et'), 29)\n",
      "(('tp53', 'abnormalities'), 29)\n",
      "(('demonstrated', 'vivo'), 29)\n",
      "(('mutations', 't-all'), 29)\n",
      "(('potentially', 'important'), 29)\n",
      "(('leading', 'complete'), 29)\n",
      "(('survival', 'outcomes'), 29)\n",
      "(('smzl', 'cases'), 29)\n",
      "(('locus,', 'leading'), 29)\n",
      "(('activated', 'notch1'), 29)\n",
      "(('association', 'childhood'), 29)\n",
      "(('may', 'inhibit'), 29)\n",
      "(('e2', 'interface'), 29)\n",
      "(('polar', 'group'), 29)\n",
      "(('group:', 'likely'), 29)\n",
      "(('hydrophobic', 'group:'), 29)\n",
      "(('substitution', 'buried'), 29)\n",
      "(('al', 'fact,'), 29)\n",
      "(('strongly', 'related'), 29)\n",
      "(('binding', 'loss'), 29)\n",
      "(('13', 'k642e'), 29)\n",
      "(('frequency', 'disease'), 29)\n",
      "(('might', 'attractive'), 29)\n",
      "(('complete', 'disease'), 29)\n",
      "(('notch', 'receptors'), 29)\n",
      "(('two', 'mechanisms:'), 29)\n",
      "(('egf-like', 'repeat'), 29)\n",
      "(('al', 'treatment'), 29)\n",
      "(('human', 't-all,'), 29)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('form', 'notch1'), 29)\n",
      "(('2', '(prc2)'), 29)\n",
      "(('function', 'specific'), 29)\n",
      "(('study', 'transcriptional'), 29)\n",
      "(('loss', 'h3k27me3'), 29)\n",
      "(('human', 'notch1'), 29)\n",
      "(('notch1', '(icn1)'), 29)\n",
      "(('recently', 'characterized'), 29)\n",
      "(('none', 'mutations'), 28)\n",
      "(('increased', 'number'), 28)\n",
      "(('one', 'copy'), 28)\n",
      "(('transformed', 'ba/f3'), 28)\n",
      "(('mutants', 'expressed'), 28)\n",
      "(('proteasomal', 'degradation'), 28)\n",
      "(('e3', 'ligase'), 28)\n",
      "(('newly', 'identified'), 28)\n",
      "(('myeloproliferative', 'disease'), 28)\n",
      "(('whether', 'mutant'), 28)\n",
      "(('conserved', 'residues'), 28)\n",
      "(('cellular', 'functions'), 28)\n",
      "(('two', 'missense'), 28)\n",
      "(('exons', '(exons'), 28)\n",
      "(('previous', 'studies,'), 28)\n",
      "(('lines', '(fig'), 28)\n",
      "(('figure', '3:'), 28)\n",
      "(('enhanced', 'sensitivity'), 28)\n",
      "(('2', '6'), 28)\n",
      "(('activation', 'tyrosine'), 28)\n",
      "(('consistent', 'findings'), 28)\n",
      "(('2', '3)'), 28)\n",
      "(('(fig', '1'), 28)\n",
      "(('mutations', 'represent'), 28)\n",
      "(('appear', 'associated'), 28)\n",
      "(('poor', 'outcome'), 28)\n",
      "(('region', 'spanning'), 28)\n",
      "(('assess', 'effects'), 28)\n",
      "(('may', 'occur'), 28)\n",
      "(('conditional', 'deletion'), 28)\n",
      "(('atlas', '(tcga)'), 28)\n",
      "(('variety', 'human'), 28)\n",
      "(('comparison', 'wild-type'), 28)\n",
      "(('growth', 'human'), 28)\n",
      "(('result', 'suggests'), 28)\n",
      "(('also', 'mutated'), 28)\n",
      "(('exhibited', 'similar'), 28)\n",
      "(('due', 'presence'), 28)\n",
      "(('mutation', 'residue'), 28)\n",
      "(('kinase', 'inactive'), 28)\n",
      "(('hek', '293t'), 28)\n",
      "(('mapk', 'activity'), 28)\n",
      "(('expression', 'data'), 28)\n",
      "(('relatively', 'common'), 28)\n",
      "(('mean', 'age'), 28)\n",
      "(('genetic', 'instability'), 28)\n",
      "(('cancer', 'age'), 28)\n",
      "(('two', 'regions'), 28)\n",
      "(('phosphorylation', 'site'), 28)\n",
      "(('', '105'), 28)\n",
      "(('na', 'na'), 28)\n",
      "(('genotype-phenotype', 'correlation'), 28)\n",
      "(('probe', 'amplification'), 28)\n",
      "(('also', 'sequenced'), 28)\n",
      "(('knowledge,', 'first'), 28)\n",
      "(('mutation', 'seems'), 28)\n",
      "(('n', 'terminal'), 28)\n",
      "(('(fig', 'using'), 28)\n",
      "(('activity', 'receptor'), 28)\n",
      "(('mutations,', 'suggesting'), 28)\n",
      "(('approximately', 'half'), 28)\n",
      "(('expression', 'found'), 28)\n",
      "(('retroviral', 'expression'), 28)\n",
      "(('p', 'value'), 28)\n",
      "(('levels', 'observed'), 28)\n",
      "(('assessed', 'immunoblotting'), 28)\n",
      "(('catalytically', 'inactive'), 28)\n",
      "(('sequenced', 'using'), 28)\n",
      "(('partial', 'loss'), 28)\n",
      "(('also', 'required'), 28)\n",
      "(('biochemical', 'analysis'), 28)\n",
      "(('encoding', 'wild-type'), 28)\n",
      "(('patient', 'population'), 28)\n",
      "(('endometrial', 'cancers'), 28)\n",
      "(('absence', 'mutations'), 28)\n",
      "(('nuclear', 'accumulation'), 28)\n",
      "(('(fig', 'cells'), 28)\n",
      "(('mutations', 'thus,'), 28)\n",
      "(('personal', 'history'), 28)\n",
      "(('gene', 'involved'), 28)\n",
      "(('alternative', 'splice'), 28)\n",
      "(('data', 'previous'), 28)\n",
      "(('found', 'high'), 28)\n",
      "(('clustered', 'within'), 28)\n",
      "(('splice', 'variants'), 28)\n",
      "(('acid', 'position'), 28)\n",
      "(('0', '20'), 28)\n",
      "(('also', 'able'), 28)\n",
      "(('likely', 'caused'), 28)\n",
      "(('viral', 'oncogene'), 28)\n",
      "(('may', 'particularly'), 28)\n",
      "(('mutation', 'frequently'), 28)\n",
      "(('showed', 'heterozygous'), 28)\n",
      "(('none', 'detected'), 28)\n",
      "(('mutation', 'three'), 28)\n",
      "(('partial', 'responses'), 28)\n",
      "(('examined', 'status'), 28)\n",
      "(('presence', 'heterozygous'), 28)\n",
      "(('whereas', 'phosphorylation'), 28)\n",
      "(('15', 'minutes,'), 28)\n",
      "(('sensitivity', 'tkis,'), 28)\n",
      "(('confer', 'susceptibility'), 28)\n",
      "(('egfr', 'compared'), 28)\n",
      "(('somatic', 'egfr'), 28)\n",
      "(('egfr', '(figs'), 28)\n",
      "(('molecule', 'inhibitors,'), 28)\n",
      "(('approved', 'japan'), 28)\n",
      "(('shown', 'promising'), 28)\n",
      "(('effects', 'egfr'), 28)\n",
      "(('showed', 'much'), 28)\n",
      "(('vivo', 'vitro'), 28)\n",
      "(('mutation', 'patterns'), 28)\n",
      "(('conditions', 'reported'), 28)\n",
      "(('six', 'patients'), 28)\n",
      "(('phosphorylation', 'events'), 28)\n",
      "(('common', 'egfr'), 28)\n",
      "(('mutations', 'independent'), 28)\n",
      "(('phosphorylation', 'presence'), 28)\n",
      "(('higher', 'concentrations'), 28)\n",
      "(('test,', 'p'), 28)\n",
      "(('mutations', '(a)'), 28)\n",
      "(('analysis,', 'presence'), 28)\n",
      "(('cause', 'resistance'), 28)\n",
      "(('mutations', 'primarily'), 28)\n",
      "(('pocket', 'kinase'), 28)\n",
      "(('ability', 'form'), 28)\n",
      "(('expression', 'oncogenic'), 28)\n",
      "(('tki', 'treatment,'), 28)\n",
      "(('due', 'high'), 28)\n",
      "(('also', 'compared'), 28)\n",
      "(('mutations', 'table'), 28)\n",
      "(('20', 'patients'), 28)\n",
      "(('c-terminal', 'end'), 28)\n",
      "(('suggests', 'mutations'), 28)\n",
      "(('one', 'third'), 28)\n",
      "(('phosphorylation', 'mutant'), 28)\n",
      "(('somatic', 'missense'), 28)\n",
      "(('parental', 'ba/f3'), 28)\n",
      "(('showed', 'little'), 28)\n",
      "(('1', 'exon'), 28)\n",
      "(('time', 'treatment'), 28)\n",
      "(('thus,', 'results'), 28)\n",
      "(('rna', 'extracted'), 28)\n",
      "(('substitution', 'predicted'), 28)\n",
      "(('n', '3)'), 28)\n",
      "(('kinase', 'activity;'), 28)\n",
      "(('signals', 'several'), 28)\n",
      "(('mechanism', 'action'), 28)\n",
      "(('displayed', 'increased'), 28)\n",
      "(('previously', 'demonstrated'), 28)\n",
      "(('major', 'role'), 28)\n",
      "(('results', 'confirmed'), 28)\n",
      "(('al', 'therefore,'), 28)\n",
      "(('protein', 'overexpression'), 28)\n",
      "(('previously', 'known'), 28)\n",
      "(('substantially', 'reduced'), 28)\n",
      "(('showed', 'wild-type'), 28)\n",
      "(('regulates', 'cell'), 28)\n",
      "(('therefore', 'likely'), 28)\n",
      "(('site', 'directed'), 28)\n",
      "(('rearrangements', 'involving'), 28)\n",
      "(('using', 'forward'), 28)\n",
      "(('allele', 'frequencies'), 28)\n",
      "(('expression', 'constructs,'), 28)\n",
      "(('regulatory', 'regions'), 28)\n",
      "(('high', 'potency'), 28)\n",
      "(('surface', 'representation'), 28)\n",
      "(('fusion', 'gene,'), 28)\n",
      "(('kinase', 'fusions'), 28)\n",
      "(('cells', '(table'), 28)\n",
      "(('patients', 'included'), 28)\n",
      "(('acute', 'leukemia'), 28)\n",
      "(('e2', 'binding'), 28)\n",
      "(('presence', 'serum'), 28)\n",
      "(('variants', 'would'), 28)\n",
      "(('eight', 'different'), 28)\n",
      "(('cells', 'since'), 28)\n",
      "(('staining', '(fig'), 28)\n",
      "(('mutants', 'wt'), 28)\n",
      "(('overall', 'fold'), 28)\n",
      "(('f,', 'female;'), 28)\n",
      "(('proportion', 'cells'), 28)\n",
      "(('regulatory', 'domain'), 28)\n",
      "(('reporter', 'gene,'), 28)\n",
      "(('mutation', 'likely'), 28)\n",
      "(('using', 'total'), 28)\n",
      "(('first', 'direct'), 28)\n",
      "(('multiple', 'myeloma'), 28)\n",
      "(('analyses', 'also'), 28)\n",
      "(('activate', 'receptor'), 28)\n",
      "(('mutations', 'cluster'), 28)\n",
      "(('additional', 'variants'), 28)\n",
      "(('conclusion,', 'provide'), 28)\n",
      "(('amplification', 'kit'), 28)\n",
      "(('cell', 'lysates,'), 28)\n",
      "(('imatinib', 'sensitivity'), 28)\n",
      "(('results', 'increased'), 28)\n",
      "(('cruz,', 'ca,'), 28)\n",
      "(('imatinib', 'inhibiting'), 28)\n",
      "(('1,000', 'nmol/l'), 28)\n",
      "(('different', 'signaling'), 28)\n",
      "(('proliferation,', 'survival'), 28)\n",
      "(('without', 'affecting'), 28)\n",
      "(('three', 'amino'), 28)\n",
      "(('pathway', 'activity'), 28)\n",
      "(('pten', 'mutations'), 28)\n",
      "(('end', 'exon'), 28)\n",
      "(('targeted', 'inhibition'), 28)\n",
      "(('would', 'useful'), 28)\n",
      "(('16', 'tumors'), 28)\n",
      "(('biotechnology', '(santa'), 28)\n",
      "(('tumors', 'tumors'), 28)\n",
      "(('kinases', 'including'), 28)\n",
      "(('lung', 'metastasis'), 28)\n",
      "(('tumor', 'models'), 28)\n",
      "(('endoplasmic', 'reticulum'), 28)\n",
      "(('activity', 'full-length'), 28)\n",
      "(('wild-type', 'alk'), 28)\n",
      "(('disrupts', 'hydrogen'), 28)\n",
      "(('c-kit', 'mutations'), 28)\n",
      "(('hydrophobic', 'contact'), 28)\n",
      "(('cells', 'risk'), 28)\n",
      "(('translocation', 'found'), 28)\n",
      "(('non-conservative', 'substitution'), 28)\n",
      "(('representation', 'human'), 28)\n",
      "(('lies', 'close'), 28)\n",
      "(('region', 'brca1'), 28)\n",
      "(('largely', 'buried'), 28)\n",
      "(('brca1', 'ring'), 28)\n",
      "(('brct', 'region'), 28)\n",
      "(('response', 'time'), 28)\n",
      "(('lower', 'minimal'), 28)\n",
      "(('(moellering', 'et'), 28)\n",
      "(('activating', 'mutations),'), 28)\n",
      "(('cell', 'case),'), 28)\n",
      "(('protocols', 'concluded'), 28)\n",
      "(('differentiation', 'status'), 28)\n",
      "(('loss', 'notch1'), 28)\n",
      "(('notch1', 'activity'), 28)\n",
      "(('augments', 'notch1'), 28)\n",
      "(('al', 'various'), 28)\n",
      "(('(or', 'cll),'), 28)\n",
      "(('notch1', 'myc'), 28)\n",
      "(('thus', 'inhibiting'), 28)\n",
      "(('notch', 'leukemia'), 28)\n",
      "(('concluded', 'notch1'), 28)\n",
      "(('(haines', 'irvine,'), 28)\n",
      "(('targeting', 'notch'), 28)\n",
      "(('laboratories', 'reported'), 28)\n",
      "(('roles', 'notch'), 28)\n",
      "(('adaptive', 'immunity'), 28)\n",
      "(('ccc', 'development'), 28)\n",
      "(('(t-all)', 'notch1'), 28)\n",
      "(('levels', '(clappier'), 28)\n",
      "(('lymphoma', '(schwarzer'), 28)\n",
      "(('locking', 'cells'), 28)\n",
      "(('(rossi', 'et'), 28)\n",
      "(('hand,', 'intriguing'), 28)\n",
      "(('mainly', 'cluster'), 28)\n",
      "(('study', '(breit'), 28)\n",
      "(('occurring', 'approximately'), 28)\n",
      "(('including', 'all-berlin-frankfurt-munster'), 28)\n",
      "(('category', 'notch1'), 28)\n",
      "(('to:', 'brief'), 28)\n",
      "(('example,', 'targeting'), 28)\n",
      "(('liver', 'mice'), 28)\n",
      "(('activated', 'upregulation'), 28)\n",
      "(('complex', 'prerequisite'), 28)\n",
      "(('irvine,', '2002;'), 28)\n",
      "(('mutations,', 'juxtamembrane'), 28)\n",
      "(('ligands', 'within'), 28)\n",
      "(('cases', '(rossi'), 28)\n",
      "(('exhibit', 'activating'), 28)\n",
      "(('stabilization', 'negative'), 28)\n",
      "(('region', '(nrr)'), 28)\n",
      "(('suppressive', 'roles'), 28)\n",
      "(('attractive', 'therapy'), 28)\n",
      "(('disrupt', 'ligand'), 28)\n",
      "(('survival,', 'retained'), 28)\n",
      "(('sf3b1', '(a'), 28)\n",
      "(('(weng', 'et'), 28)\n",
      "(('mechanisms:', 'inhibition'), 28)\n",
      "(('developed', 'study'), 28)\n",
      "(('notch1-expressing', 't-all'), 28)\n",
      "(('notch2', 'act'), 28)\n",
      "(('leukemia', '(ellisen'), 28)\n",
      "(('2003;', 'okajima'), 28)\n",
      "(('receptors', '(named'), 28)\n",
      "(('cell', 'identity,'), 28)\n",
      "(('gene', 'array,'), 28)\n",
      "(('jundt', 'colleagues'), 28)\n",
      "(('mutations', 'adverse'), 28)\n",
      "(('truncated,', 'constitutively-active'), 28)\n",
      "(('prerequisite', 'activation'), 28)\n",
      "(('eortc-clg', '58881'), 28)\n",
      "(('observation', 'remains'), 28)\n",
      "(('2002;', 'rebay'), 28)\n",
      "(('simply', 'define'), 28)\n",
      "(('role', 'notch1'), 28)\n",
      "(('immunity', 'plays'), 28)\n",
      "(('mainly', 'characterized'), 28)\n",
      "(('disease', '(aster'), 28)\n",
      "(('genome/transcriptome', '(rna-seq,'), 28)\n",
      "(('factor)', 'mutations'), 28)\n",
      "(('finally,', 'constitutive'), 28)\n",
      "(('receptors,', 'protection'), 28)\n",
      "(('map', 'notch1'), 28)\n",
      "(('proposed', 'molecular'), 28)\n",
      "(('sahm1', 'resulted'), 28)\n",
      "(('vivo', 'led'), 28)\n",
      "(('pathway', 'myeloid'), 28)\n",
      "(('underlining', 'significance'), 28)\n",
      "(('-secretase', 'inhibitor'), 28)\n",
      "(('proteolytic', 'cleavage,'), 28)\n",
      "(('approaches', '1012%'), 28)\n",
      "(('development', 'towards'), 28)\n",
      "(('notch1', 'activating'), 28)\n",
      "(('induction', 'progression'), 28)\n",
      "(('(clappier', 'et'), 28)\n",
      "(('al,', '2012b),'), 28)\n",
      "(('normal', 'notch1'), 28)\n",
      "(('notch1', 'decoy'), 28)\n",
      "(('colleagues', 'characterized'), 28)\n",
      "(('which-in', 'turn-is'), 28)\n",
      "(('classic', 'hodgkin'), 28)\n",
      "(('absence', 'downstream'), 28)\n",
      "(('(named', 'notch14)'), 28)\n",
      "(('specific', 'differentiation'), 28)\n",
      "(('intriguing', 'ask'), 28)\n",
      "(('promote', 'disease'), 28)\n",
      "(('chip-seq', 'notch1'), 28)\n",
      "(('34', '29'), 28)\n",
      "(('progression', '(rangarajan'), 28)\n",
      "(('mutations', '2025%'), 28)\n",
      "(('notch1', 'hes1)'), 28)\n",
      "(('look', 'laboratories'), 28)\n",
      "(('encompass', 'another'), 28)\n",
      "(('okajima', 'irvine,'), 28)\n",
      "(('(schwarzer', 'et'), 28)\n",
      "(('notch1', 'mutually'), 28)\n",
      "(('role', 'notch'), 28)\n",
      "(('downstream', 'activating'), 28)\n",
      "(('due', 'translocation'), 28)\n",
      "(('subsequently', 'groups,'), 28)\n",
      "(('(a', 'splicing'), 28)\n",
      "(('factor', '(ebf)'), 28)\n",
      "(('notch-specific', 'inhibition'), 28)\n",
      "(('genome', '(castel'), 28)\n",
      "(('myc', 'due'), 28)\n",
      "(('t-all,', 'laboratory'), 28)\n",
      "(('remission', 'vivo'), 28)\n",
      "(('notch2,', 'notch1'), 28)\n",
      "(('transcriptional', 'co-activators'), 28)\n",
      "(('act', 'mainly'), 28)\n",
      "(('outcomes', '(oscier'), 28)\n",
      "(('promote', 'pdac'), 28)\n",
      "(('unknown', '(hanlon'), 28)\n",
      "(('another', 'rare'), 28)\n",
      "(('deletion', 'notch1'), 28)\n",
      "(('ligands', '(haines'), 28)\n",
      "(('vivo', 't-all'), 28)\n",
      "(('progression', '(moellering'), 28)\n",
      "(('t-all', 'oncogenes'), 28)\n",
      "(('spen', 'dtx1'), 28)\n",
      "(('brd', 'proteins'), 28)\n",
      "(('study', 'japan'), 28)\n",
      "(('inactivation', 'notch2,'), 28)\n",
      "(('(aster', 'blacklow,'), 28)\n",
      "(('to:', 'therapeutic'), 28)\n",
      "(('t-all,', 'occurring'), 28)\n",
      "(('work', 'ellisen'), 28)\n",
      "(('(trb)', 'locus,'), 28)\n",
      "(('controlling', 'nf-kb'), 28)\n",
      "(('suggested', 'notch1'), 28)\n",
      "(('companies', 'generated'), 28)\n",
      "(('report', '134'), 28)\n",
      "(('viability', 'xenograft'), 28)\n",
      "(('active', 'notch1'), 28)\n",
      "(('aster', 'look'), 28)\n",
      "(('target', 'notch1-'), 28)\n",
      "(('parameter', 'disease'), 28)\n",
      "(('patients', '(love'), 28)\n",
      "(('leukemia', '(t-all),'), 28)\n",
      "(('case),', 'truly'), 28)\n",
      "(('expression', 'truncated,'), 28)\n",
      "(('(ellisen', 'et'), 28)\n",
      "(('comes', 'work'), 28)\n",
      "(('juxtamembrane', 'expansion'), 28)\n",
      "(('irvine,', '2003;'), 28)\n",
      "(('presenilin/-secretase', 'complex'), 28)\n",
      "(('notch1', '(avila'), 28)\n",
      "(('notch', 'hcc'), 28)\n",
      "(('notch1-', 'myc-regulated'), 28)\n",
      "(('blocking', 'notch1'), 28)\n",
      "(('notch1', 'liver'), 28)\n",
      "(('finally,', 'another'), 28)\n",
      "(('2012)', 'genome/transcriptome'), 28)\n",
      "(('status', '(t'), 28)\n",
      "(('beta', '(trb)'), 28)\n",
      "(('pathway,', 'chip-seq'), 28)\n",
      "(('mutations', 'associate'), 28)\n",
      "(('hepatocytes,', 'suggesting'), 28)\n",
      "(('transcription', 't-all,'), 28)\n",
      "(('deregulation', 'angiogenesis'), 28)\n",
      "(('poor', 'cases'), 28)\n",
      "(('inhibit', 'expansion'), 28)\n",
      "(('(funahashi', 'et'), 28)\n",
      "(('notch1', 't-all,'), 28)\n",
      "(('(ebf)', '(jundt'), 28)\n",
      "(('affect', 'notch1-expressing'), 28)\n",
      "(('confirmed', 'notch1'), 28)\n",
      "(('hyperproliferative', 'hepatocytes,'), 28)\n",
      "(('hes1)', '(ntziachristos'), 28)\n",
      "(('disease', 'burkitt'), 28)\n",
      "(('first', 'activating'), 28)\n",
      "(('(sulis', 'et'), 28)\n",
      "(('notch1', 'cdna'), 28)\n",
      "(('al', 'brd'), 28)\n",
      "(('notch4', '34'), 28)\n",
      "(('(rna-seq,', 'gene'), 28)\n",
      "(('al', 'notch1'), 28)\n",
      "(('gain-of-function', 'notch1'), 28)\n",
      "(('master', 'transcription'), 28)\n",
      "(('20%', 'smzl'), 28)\n",
      "(('option', 'notch1-induced'), 28)\n",
      "(('(aifantis', 'et'), 28)\n",
      "(('deletion', 'notch1/2'), 28)\n",
      "(('4', 'simplified'), 28)\n",
      "(('cll', '(20%)'), 28)\n",
      "(('high-risk', 'richter'), 28)\n",
      "(('89%', 'patients'), 28)\n",
      "(('mutations', '89%'), 28)\n",
      "(('4)', '(dan'), 28)\n",
      "(('display', 'recurrent'), 28)\n",
      "(('notch1', 'leukemia'), 28)\n",
      "(('therapy', 'option'), 28)\n",
      "(('binding', 'produce'), 28)\n",
      "(('characterized', 'tumor'), 28)\n",
      "(('remission', 'target:'), 28)\n",
      "(('recently', 'connected'), 28)\n",
      "(('important', 'b'), 28)\n",
      "(('patients', 'eortc-clg'), 28)\n",
      "(('repeats,', 'notch3'), 28)\n",
      "(('factor', 'controls'), 28)\n",
      "(('complex', 'pivotal'), 28)\n",
      "(('mutants', 'encompass'), 28)\n",
      "(('(jundt', 'et'), 28)\n",
      "(('myc-regulated', 'transcription'), 28)\n",
      "(('pharmaceutical', 'companies'), 28)\n",
      "(('shown', 'favorable'), 28)\n",
      "(('notch1', 'thus'), 28)\n",
      "(('notch1', 'appears'), 28)\n",
      "(('briefly,', 'recently'), 28)\n",
      "(('description', 'notch'), 28)\n",
      "(('h3k27me3', 'demethylation'), 28)\n",
      "(('cells', '(qi'), 28)\n",
      "(('58951', 'protocols'), 28)\n",
      "(('niche', 'suppresses'), 28)\n",
      "(('even', 'promote'), 28)\n",
      "(('tumors', 'proteolytic'), 28)\n",
      "(('29', 'repeats,'), 28)\n",
      "(('(rangarajan', 'et'), 28)\n",
      "(('notch1', 'polypeptides'), 28)\n",
      "(('phage', 'display'), 28)\n",
      "(('scheme', 'notch'), 28)\n",
      "(('1)', 'also,'), 28)\n",
      "(('common', 'binding'), 28)\n",
      "(('selective', 'blocking'), 28)\n",
      "(('upregulation', 'ligands'), 28)\n",
      "(('simplified', 'scheme'), 28)\n",
      "(('effect,', 'compound'), 28)\n",
      "(('status', 't-all'), 28)\n",
      "(('initiate', 'ccc'), 28)\n",
      "(('ellisen', 'colleagues,'), 28)\n",
      "(('(park', 'et'), 28)\n",
      "(('target:', 'a)'), 28)\n",
      "(('h3k27me3', 'target'), 28)\n",
      "(('rearrangement', 'intracellular'), 28)\n",
      "(('(20%)', '(fabbri'), 28)\n",
      "(('t-alls', 'interestingly,'), 28)\n",
      "(('gsi', 'efficiently'), 28)\n",
      "(('colleagues,', 'confirmed'), 28)\n",
      "(('notch', 'interactions'), 28)\n",
      "(('notch1-induced', 't-all'), 28)\n",
      "(('blocks', 'notch1'), 28)\n",
      "(('constitutively-active', 'notch1'), 28)\n",
      "(('ago,', 'aster'), 28)\n",
      "(('corresponding', 'ligands'), 28)\n",
      "(('2009)', 'furthermore,'), 28)\n",
      "(('mutants', '(jme)'), 28)\n",
      "(('towards', 'cells'), 28)\n",
      "(('interestingly,', 'notch1'), 28)\n",
      "(('results', 'hyperproliferative'), 28)\n",
      "(('(mrd)', 'levels'), 28)\n",
      "(('splicing', 'factor)'), 28)\n",
      "(('outcome', 'post-treatment'), 28)\n",
      "(('t-all', 'remission'), 28)\n",
      "(('hematopoietic', 'development'), 28)\n",
      "(('pathways', 'cancer'), 28)\n",
      "(('genes', 'induction'), 28)\n",
      "(('also', 'augments'), 28)\n",
      "(('t-all', 'sahm1'), 28)\n",
      "(('(fabbri', 'et'), 28)\n",
      "(('(t', 'cell'), 28)\n",
      "(('abnormalities', 'survival'), 28)\n",
      "(('co-activators', 'share'), 28)\n",
      "(('cll),', '2012)'), 28)\n",
      "(('disease', 'able'), 28)\n",
      "(('expansion', 'mutants'), 28)\n",
      "(('cleavage,', 'thus'), 28)\n",
      "(('2013)', 'confirmed'), 28)\n",
      "(('notch-mediated', 'effect,'), 28)\n",
      "(('production', 'icn1/2'), 28)\n",
      "(('innate', 'adaptive'), 28)\n",
      "(('mainly', 'stabilization'), 28)\n",
      "(('(breit', 'et'), 28)\n",
      "(('pdac', 'development,'), 28)\n",
      "(('(30%)', 'chemo-refractory'), 28)\n",
      "(('models', 'least'), 28)\n",
      "(('repeats,', 'respectively,'), 28)\n",
      "(('notch1,', 'underlining'), 28)\n",
      "(('findings,', 'h3k27me3'), 28)\n",
      "(('notch1', 'fbxw7'), 28)\n",
      "(('controls', 'innate'), 28)\n",
      "(('finding', 'notch'), 28)\n",
      "(('connected', 'notch1'), 28)\n",
      "(('palomero', 'ferrando,'), 28)\n",
      "(('associate', 'improved'), 28)\n",
      "(('binding', 'genome'), 28)\n",
      "(('hcc', '(croquelois'), 28)\n",
      "(('confirmed', 'subsequently'), 28)\n",
      "(('japan', 'association'), 28)\n",
      "(('rare', 'group'), 28)\n",
      "(('t-all', 'homing'), 28)\n",
      "(('affinity', 'corresponding'), 28)\n",
      "(('groups,', 'revealed'), 28)\n",
      "(('expansion', 'hcc'), 28)\n",
      "(('(prc2)', 'recruitment'), 28)\n",
      "(('receptors', 'presenilin/-secretase'), 28)\n",
      "(('36', 'egf-like'), 28)\n",
      "(('contrast', 'tumorigenic'), 28)\n",
      "(('notch1', 'inhibits'), 28)\n",
      "(('notch1/2', 'vivo'), 28)\n",
      "(('activating', 'notch1'), 28)\n",
      "(('pathway', 'which-in'), 28)\n",
      "(('blacklow,', '2012;'), 28)\n",
      "(('cells', 'proposed'), 28)\n",
      "(('interactions', 'signaling'), 28)\n",
      "(('58881', '58951'), 28)\n",
      "(('2000', 'study'), 28)\n",
      "(('array,', 'chip-seq'), 28)\n",
      "(('characterized', 'upregulation'), 28)\n",
      "(('cancer', 'comes'), 28)\n",
      "(('early', 'chemotherapeutic'), 28)\n",
      "(('part', 'notch1'), 28)\n",
      "(('suppresses', 'genes'), 28)\n",
      "(('notch1', 'signaling'), 28)\n",
      "(('various', 'tools'), 28)\n",
      "(('nf-kb', 'pathway'), 28)\n",
      "(('patterns', 't-all'), 28)\n",
      "(('models', '(funahashi'), 28)\n",
      "(('to:', 't-cell'), 28)\n",
      "(('lineage,', 'locking'), 28)\n",
      "(('binding', 'polycomb'), 28)\n",
      "(('vivo', 'recently'), 28)\n",
      "(('development,', 'although'), 28)\n",
      "(('egf-like', 'repeats,'), 28)\n",
      "(('2003),', 'deletion'), 28)\n",
      "(('gain-of-function', 'notch2'), 28)\n",
      "(('antibodies', 'notch1'), 28)\n",
      "(('1012%', 'cll'), 28)\n",
      "(('related', 'notch'), 28)\n",
      "(('post-treatment', 'number'), 28)\n",
      "(('activity', 't-all'), 28)\n",
      "(('respectively,', 'affect'), 28)\n",
      "(('all-berlin-frankfurt-munster', '2000'), 28)\n",
      "(('disease', 'remission'), 28)\n",
      "(('notch1', 'classic'), 28)\n",
      "(('strong,', 'notch-specific'), 28)\n",
      "(('t-cell', 'lymphoblastic'), 28)\n",
      "(('fact,', 'loss'), 28)\n",
      "(('notch1', 'sufficient'), 28)\n",
      "(('notch1', 'binding'), 28)\n",
      "(('notch2', '36'), 28)\n",
      "(('(zender', 'et'), 28)\n",
      "(('inhibiting', 'production'), 28)\n",
      "(('region', 'notch1'), 28)\n",
      "(('notch1', 'function'), 28)\n",
      "(('generation', 'sequencing-based'), 28)\n",
      "(('2012b),', '20%'), 28)\n",
      "(('mutations', 'prognostic'), 28)\n",
      "(('improved', 'early'), 28)\n",
      "(('prognosis', 'better'), 28)\n",
      "(('although', 'basis'), 28)\n",
      "(('potential', 'disrupt'), 28)\n",
      "(('technology,', 'pharmaceutical'), 28)\n",
      "(('first', 'finding'), 28)\n",
      "(('t-all', 'occur'), 28)\n",
      "(('notch1-induced', 'ikk'), 28)\n",
      "(('mammals', 'notch1'), 28)\n",
      "(('cluster', 'egf-like'), 28)\n",
      "(('e12/e47', 'early'), 28)\n",
      "(('chip-chip', 'map'), 28)\n",
      "(('(ntziachristos', 'et'), 28)\n",
      "(('decoy', 'decreased'), 28)\n",
      "(('sequencing-based', 'approaches'), 28)\n",
      "(('marker', 'mutations'), 28)\n",
      "(('oncogenes', 'notch1'), 28)\n",
      "(('pathway', 'four'), 28)\n",
      "(('malignancies', 'notch-mediated'), 28)\n",
      "(('define', 'lineage,'), 28)\n",
      "(('notch1', 'activated'), 28)\n",
      "(('cells', '(aifantis'), 28)\n",
      "(('demethylation', 'inhibitors'), 28)\n",
      "(('antagonism', 'notch1'), 28)\n",
      "(('ferrando,', '2009)'), 28)\n",
      "(('tumor', 'niche'), 28)\n",
      "(('role', 'directing'), 28)\n",
      "(('al', 'selective'), 28)\n",
      "(('notch14)', 'mammals'), 28)\n",
      "(('(croquelois', 'et'), 28)\n",
      "(('proliferation,', 'leukemia'), 28)\n",
      "(('leukemia', 'study'), 28)\n",
      "(('t-all,', 'leading'), 28)\n",
      "(('transformation', '(30%)'), 28)\n",
      "(('finally,', 'jundt'), 28)\n",
      "(('tools', 'developed'), 28)\n",
      "(('t-all', 't-cell'), 28)\n",
      "(('locus,', 'display'), 28)\n",
      "(('(nrr)', 'receptors,'), 28)\n",
      "(('134', 'pediatric'), 28)\n",
      "(('chemo-refractory', 'cll'), 28)\n",
      "(('metastasis', 'go'), 28)\n",
      "(('importantly,', 'ten'), 28)\n",
      "(('rebay', 'et'), 28)\n",
      "(('specialized', 'antibodies'), 28)\n",
      "(('immunoglobulin', 'locus,'), 28)\n",
      "(('finally,', 'pest'), 28)\n",
      "(('efficiently', 'blocks'), 28)\n",
      "(('hd', 'domain'), 28)\n",
      "(('richter', 'transformation'), 28)\n",
      "(('enhancers', 'key'), 28)\n",
      "(('mice', '(zender'), 28)\n",
      "(('(hanlon', 'et'), 28)\n",
      "(('cases', '(weng'), 28)\n",
      "(('notch1,', 'spen'), 28)\n",
      "(('activity', 'pathway,'), 28)\n",
      "(('identity,', 'e12/e47'), 28)\n",
      "(('brief', 'description'), 28)\n",
      "(('molecule', 'gsi'), 28)\n",
      "(('resulted', 'strong,'), 28)\n",
      "(('(love', 'et'), 28)\n",
      "(('dtx1', '(rossi'), 28)\n",
      "(('four', 'notch'), 28)\n",
      "(('icn1/2', '(wu'), 28)\n",
      "(('(jme)', 'also'), 28)\n",
      "(('promoters', 'demonstrated'), 28)\n",
      "(('(icn1)', 'cell'), 28)\n",
      "(('targeting', 'expression'), 28)\n",
      "(('growth', 'deregulation'), 28)\n",
      "(('notch1', 'may'), 28)\n",
      "(('(oscier', 'et'), 28)\n",
      "(('notch1', 'signals'), 28)\n",
      "(('(castel', 'et'), 28)\n",
      "(('t-all', '(or'), 28)\n",
      "(('leukemia', '(figure'), 28)\n",
      "(('signaling', '(in'), 28)\n",
      "(('proteins', 'transcriptional'), 28)\n",
      "(('cleavage', 'notch'), 28)\n",
      "(('activity', '(ntziachristos'), 28)\n",
      "(('patients', '(park'), 28)\n",
      "(('growth', 'pre-clinical'), 28)\n",
      "(('recruitment', 'activity'), 28)\n",
      "(('directing', 'hematopoietic'), 28)\n",
      "(('notch1', 'master'), 28)\n",
      "(('polycomb', 'complex'), 28)\n",
      "(('mutations),', 'small'), 28)\n",
      "(('chip-seq', 'chip-chip'), 28)\n",
      "(('model', 'notch1-driven'), 28)\n",
      "(('notch1-driven', 't-all'), 28)\n",
      "(('2006),', 'study'), 28)\n",
      "(('(qi', 'et'), 28)\n",
      "(('highly', 'specialized'), 28)\n",
      "(('lymphoma,', 'mainly'), 28)\n",
      "(('1', 'oncogenic'), 28)\n",
      "(('ligand', 'could'), 28)\n",
      "(('turn-is', 'strongly'), 28)\n",
      "(('display', 'technology,'), 28)\n",
      "(('2012;', 'palomero'), 28)\n",
      "(('angiogenesis', 'notch1'), 28)\n",
      "(('t-all', 'shown'), 28)\n",
      "(('outcomes', 'poor'), 28)\n",
      "(('ikk', 'kinase'), 28)\n",
      "(('homing', 'metastasis'), 28)\n",
      "(('2025%', 't-alls'), 28)\n",
      "(('truly', 'transform'), 28)\n",
      "(('notch1', 'observed'), 28)\n",
      "(('generated', 'highly'), 28)\n",
      "(('childhood', 'leukemia'), 28)\n",
      "(('towards', 'high-risk'), 28)\n",
      "(('using', 'phage'), 28)\n",
      "(('revealed', 'rearrangement'), 28)\n",
      "(('a)', 'notch1-induced'), 28)\n",
      "(('notch3', 'notch4'), 28)\n",
      "(('membrane', '(sulis'), 28)\n",
      "(('notch1', 'extracellular'), 28)\n",
      "(('protection', 'proteolytic'), 28)\n",
      "(('mutations', 'seen'), 27)\n",
      "(('non-synonymous', 'mutations'), 27)\n",
      "(('sequence', 'variant'), 27)\n",
      "(('deletions', 'mutations'), 27)\n",
      "(('erk1/2', 'phosphorylation'), 27)\n",
      "(('mutant', 'induced'), 27)\n",
      "(('500', 'nm'), 27)\n",
      "(('mutations', 'induced'), 27)\n",
      "(('liquid', 'chromatography'), 27)\n",
      "(('assessed', 'ability'), 27)\n",
      "(('new', 'mutations'), 27)\n",
      "(('proliferation', 'rates'), 27)\n",
      "(('binding', 'e2'), 27)\n",
      "(('fig', 'figure'), 27)\n",
      "(('found', 'cancer'), 27)\n",
      "(('primers', 'designed'), 27)\n",
      "(('difference', 'prevalence'), 27)\n",
      "(('one', 'frequent'), 27)\n",
      "(('likely', 'represent'), 27)\n",
      "(('1', 'shows'), 27)\n",
      "(('high-level', 'amplification'), 27)\n",
      "(('mutation', 'figure'), 27)\n",
      "(('mutant', 'may'), 27)\n",
      "(('genetic', 'abnormalities'), 27)\n",
      "(('data', 'sets'), 27)\n",
      "(('patients', 'diagnosed'), 27)\n",
      "(('data', 'indicated'), 27)\n",
      "(('al', 'consistent'), 27)\n",
      "(('negatively', 'charged'), 27)\n",
      "(('n', 'terminus'), 27)\n",
      "(('tensin', 'homolog'), 27)\n",
      "(('treatment', '(figure'), 27)\n",
      "(('study', 'reported'), 27)\n",
      "(('summary,', 'results'), 27)\n",
      "(('cells', 'incubated'), 27)\n",
      "(('collectively,', 'results'), 27)\n",
      "(('growth', 'signaling'), 27)\n",
      "(('activity', 'necessary'), 27)\n",
      "(('serum', 'starvation'), 27)\n",
      "(('highly', 'associated'), 27)\n",
      "(('within', 'region'), 27)\n",
      "(('immunohistochemical', 'staining'), 27)\n",
      "(('truncating', 'mutation'), 27)\n",
      "(('selective', 'inhibitor'), 27)\n",
      "(('transfectants', 'expressing'), 27)\n",
      "(('possible', 'mechanism'), 27)\n",
      "(('using', 'quantitative'), 27)\n",
      "(('although', 'several'), 27)\n",
      "(('immunoprecipitation', 'using'), 27)\n",
      "(('20', '40'), 27)\n",
      "(('catalytic', 'core'), 27)\n",
      "(('might', 'associated'), 27)\n",
      "(('growth', 'absence'), 27)\n",
      "(('cells', 'contrast,'), 27)\n",
      "(('receptor', 'cell'), 27)\n",
      "(('(unpublished', 'data)'), 27)\n",
      "(('cells', 'showing'), 27)\n",
      "(('mutation', 'position'), 27)\n",
      "(('show', 'significant'), 27)\n",
      "(('either', 'wild'), 27)\n",
      "(('expressed', 'low'), 27)\n",
      "(('small', 'deletions'), 27)\n",
      "(('reduced', 'binding'), 27)\n",
      "(('mutations', 'first'), 27)\n",
      "(('mutations', 'dna'), 27)\n",
      "(('found', 'sequencing'), 27)\n",
      "(('behaved', 'like'), 27)\n",
      "(('two', 'mutations,'), 27)\n",
      "(('analysis', 'shown'), 27)\n",
      "(('highly', 'homologous'), 27)\n",
      "(('fig', '3'), 27)\n",
      "(('strongly', 'reduced'), 27)\n",
      "(('important', 'note'), 27)\n",
      "(('germ-line', 'mutation'), 27)\n",
      "(('sequencing', 'genomic'), 27)\n",
      "(('higher', 'incidence'), 27)\n",
      "(('cell', 'development'), 27)\n",
      "(('diminished', 'levels'), 27)\n",
      "(('treatment', '(fig'), 27)\n",
      "(('inhibits', 'cell'), 27)\n",
      "(('mcf7', 'cells'), 27)\n",
      "(('protein', 'stability,'), 27)\n",
      "(('al', 'studies'), 27)\n",
      "(('cell', 'adhesion,'), 27)\n",
      "(('negative', 'regulation'), 27)\n",
      "(('shown)', 'results'), 27)\n",
      "(('candidate', 'gene'), 27)\n",
      "(('conditions', '(figure'), 27)\n",
      "(('foster', 'city,'), 27)\n",
      "(('knockdown', 'cells'), 27)\n",
      "(('patients', 'previous'), 27)\n",
      "(('concentrations', 'drug'), 27)\n",
      "(('per', 'se'), 27)\n",
      "(('effect', 'notch1'), 27)\n",
      "(('wild-type', 'enzyme'), 27)\n",
      "(('demonstrated', 'significant'), 27)\n",
      "(('altered', 'sensitivity'), 27)\n",
      "(('egfr', 'autophosphorylation'), 27)\n",
      "(('reported', 'heterozygous'), 27)\n",
      "(('cells,', 'low'), 27)\n",
      "(('amplification', 'mutant'), 27)\n",
      "(('dana-farber', 'cancer'), 27)\n",
      "(('used', 'immunoblotting'), 27)\n",
      "(('found', 'lung'), 27)\n",
      "(('2', 'somatic'), 27)\n",
      "(('gene', 'responsible'), 27)\n",
      "(('l858r', 'egfr'), 27)\n",
      "(('randomly', 'selected'), 27)\n",
      "(('frequent', 'amplification'), 27)\n",
      "(('numbers', 'cells'), 27)\n",
      "(('sequencing', 'exons'), 27)\n",
      "(('akt', 'erk1/2'), 27)\n",
      "(('despite', 'high'), 27)\n",
      "(('disease', 'refractory'), 27)\n",
      "(('hours', '(figure'), 27)\n",
      "(('egfr', 'inhibition'), 27)\n",
      "(('response', 'gefitinib,'), 27)\n",
      "(('confirm', 'extend'), 27)\n",
      "(('autophosphorylation', 'mutant'), 27)\n",
      "(('clinical', 'criteria'), 27)\n",
      "(('frequency', 'amplification'), 27)\n",
      "(('east', 'asian'), 27)\n",
      "(('mutations', 'changed'), 27)\n",
      "(('mutations', 'made'), 27)\n",
      "(('trials', 'patients'), 27)\n",
      "(('c-terminal', 'lobe'), 27)\n",
      "(('oncogenic', 'cell'), 27)\n",
      "(('patients', 'examined'), 27)\n",
      "(('involved', 'pathogenesis'), 27)\n",
      "(('univariate', 'analysis,'), 27)\n",
      "(('patients', 'furthermore,'), 27)\n",
      "(('significantly', 'correlated'), 27)\n",
      "(('partial', 'response,'), 27)\n",
      "(('common', 'exon'), 27)\n",
      "(('indicate', 'presence'), 27)\n",
      "(('inhibitors', '(tki)'), 27)\n",
      "(('specimens', 'patients'), 27)\n",
      "(('sequences', 'also'), 27)\n",
      "(('independent', 'variables'), 27)\n",
      "(('2', 'trial'), 27)\n",
      "(('pfs', 'os'), 27)\n",
      "(('gefitinib', 'treatment,'), 27)\n",
      "(('somatic', 'alterations'), 27)\n",
      "(('results', 'seen'), 27)\n",
      "(('insertion', 'may'), 27)\n",
      "(('mutants', 'relative'), 27)\n",
      "(('inhibit', 'egfr'), 27)\n",
      "(('cell', 'type,'), 27)\n",
      "(('pleural', 'effusions'), 27)\n",
      "(('tumors', 'bearing'), 27)\n",
      "(('18', '21,'), 27)\n",
      "(('gefitinib', 'therapy'), 27)\n",
      "(('two', 'previous'), 27)\n",
      "(('detect', 'mutations'), 27)\n",
      "(('mutation', 't790m'), 27)\n",
      "(('clinical', 'significance,'), 27)\n",
      "(('amplification', 'egfr'), 27)\n",
      "(('effect', 'growth'), 27)\n",
      "(('high', 'egfr'), 27)\n",
      "(('survival', '(pfs)'), 27)\n",
      "(('mutants', 'able'), 27)\n",
      "(('levels', 'wt'), 27)\n",
      "(('gene', 'also'), 27)\n",
      "(('corresponding', 'region'), 27)\n",
      "(('patients', 'either'), 27)\n",
      "(('one', 'study'), 27)\n",
      "(('prognostic', 'factors'), 27)\n",
      "(('relatively', 'resistant'), 27)\n",
      "(('results', 'using'), 27)\n",
      "(('showed', 'marked'), 27)\n",
      "(('proportion', 'total'), 27)\n",
      "(('reduced', 'sensitivity'), 27)\n",
      "(('thus', 'confirming'), 27)\n",
      "(('overcome', 'resistance'), 27)\n",
      "(('transforming', 'properties'), 27)\n",
      "(('patients', 'achieved'), 27)\n",
      "(('mutation', 'kinase'), 27)\n",
      "(('single', 'mutations'), 27)\n",
      "(('mutations', 'enhance'), 27)\n",
      "(('samples', '5'), 27)\n",
      "(('constructs', 'treated'), 27)\n",
      "(('done', 'using'), 27)\n",
      "(('several', 'novel'), 27)\n",
      "(('phosphorylation', 'stat5'), 27)\n",
      "(('expression', '(figure'), 27)\n",
      "(('worse', 'prognosis'), 27)\n",
      "(('os', 'patients'), 27)\n",
      "(('mutation', 'allelic'), 27)\n",
      "(('patients', 'acquired'), 27)\n",
      "(('in-frame', 'deletions,'), 27)\n",
      "(('responses', 'observed'), 27)\n",
      "(('mutants', 'using'), 27)\n",
      "(('cells', 'ectopically'), 27)\n",
      "(('relative', 'wt'), 27)\n",
      "(('alternative', 'mechanisms'), 27)\n",
      "(('anchorage', 'independent'), 27)\n",
      "(('mutants', 'analyzed'), 27)\n",
      "(('using', 'purified'), 27)\n",
      "(('oncogenic', 'driver'), 27)\n",
      "(('first', 'examined'), 27)\n",
      "(('possibility', 'may'), 27)\n",
      "(('wide', 'variety'), 27)\n",
      "(('dna', 'copy'), 27)\n",
      "(('pathway', 'inhibition'), 27)\n",
      "(('confirm', 'presence'), 27)\n",
      "(('intron', '11'), 27)\n",
      "(('cancer-associated', 'mutants'), 27)\n",
      "(('malignant', 'transformation'), 27)\n",
      "(('carried', 'using'), 27)\n",
      "(('cell', 'types,'), 27)\n",
      "(('mutations', 'including'), 27)\n",
      "(('exon', '14,'), 27)\n",
      "(('small', 'insertions'), 27)\n",
      "(('results', 'confirm'), 27)\n",
      "(('proliferation', 'differentiation'), 27)\n",
      "(('(figure', '2a,'), 27)\n",
      "(('kb)', 'figures'), 27)\n",
      "(('core', 'domain'), 27)\n",
      "(('growth', 'suppression'), 27)\n",
      "(('wild-type', 'p53,'), 27)\n",
      "(('figure', '4a)'), 27)\n",
      "(('change', 'conformation'), 27)\n",
      "(('mutations', 'missense'), 27)\n",
      "(('functional', 'data'), 27)\n",
      "(('decreased', 'activity'), 27)\n",
      "(('activation', 'however,'), 27)\n",
      "(('luciferase', 'assays'), 27)\n",
      "(('murine', 'models'), 27)\n",
      "(('leukemic', 'cells'), 27)\n",
      "(('pathogenesis', 'human'), 27)\n",
      "(('exogenously', 'expressed'), 27)\n",
      "(('identified', 'somatic'), 27)\n",
      "(('protein', 'interactions'), 27)\n",
      "(('approximately', '5%'), 27)\n",
      "(('identified', 'previously'), 27)\n",
      "(('1),', 'including'), 27)\n",
      "(('samples', 'harboring'), 27)\n",
      "(('probed', 'antibodies'), 27)\n",
      "(('whole', 'exome'), 27)\n",
      "(('15', '16'), 27)\n",
      "(('mutations', 'corresponding'), 27)\n",
      "(('substitution', 'disrupts'), 27)\n",
      "(('mutation', 'using'), 27)\n",
      "(('mutation', 'activity'), 27)\n",
      "(('described', 'earlier'), 27)\n",
      "(('5', 'end'), 27)\n",
      "(('protein', 'a/g'), 27)\n",
      "(('type', 'iii'), 27)\n",
      "(('shown', 'mutation'), 27)\n",
      "(('tumors', '(gist)'), 27)\n",
      "(('tumors', '(gists)'), 27)\n",
      "(('mutations', 'gists'), 27)\n",
      "(('7', 'exon'), 27)\n",
      "(('results', 'constitutive'), 27)\n",
      "(('alterations', 'identified'), 27)\n",
      "(('variant', 'wild-type'), 27)\n",
      "(('prognostic', 'impact'), 27)\n",
      "(('therapeutic', 'opportunities'), 27)\n",
      "(('resulting', 'amino'), 27)\n",
      "(('mutations', 'pten'), 27)\n",
      "(('ovarian', 'tumors,'), 27)\n",
      "(('another', 'possibility'), 27)\n",
      "(('proliferation', 'survival,'), 27)\n",
      "(('new', 'england'), 27)\n",
      "(('good', 'agreement'), 27)\n",
      "(('tyrosine', 'autophosphorylation'), 27)\n",
      "(('plasmids', 'introduced'), 27)\n",
      "(('interaction', 'protein'), 27)\n",
      "(('braf', 'mutation,'), 27)\n",
      "(('figure', '1b'), 27)\n",
      "(('braf', 'inhibitors'), 27)\n",
      "(('last', '11'), 27)\n",
      "(('likely', 'weaken'), 27)\n",
      "(('activation', 'map'), 27)\n",
      "(('expected', 'disrupt'), 27)\n",
      "(('alk', 'mutants'), 27)\n",
      "(('interaction', 'two'), 27)\n",
      "(('binding', 'site;'), 27)\n",
      "(('mutations', 'juxtamembrane'), 27)\n",
      "(('flt3', 'kinase'), 27)\n",
      "(('study', '(table'), 27)\n",
      "(('resistance', 'go'), 27)\n",
      "(('mutations', 'flt3'), 27)\n",
      "(('expression', 'flt3'), 27)\n",
      "(('tumor', 'one'), 27)\n",
      "(('trex', 'cells'), 27)\n",
      "(('t-all', 'cases'), 27)\n",
      "(('using', 'plasmid'), 27)\n",
      "(('pip3', 'phosphatase'), 27)\n",
      "(('classified', 'six'), 27)\n",
      "(('weaken', 'binding'), 27)\n",
      "(('interface', 'may'), 27)\n",
      "(('effect', 'substitution'), 27)\n",
      "(('substitutions', 'brca1'), 27)\n",
      "(('conjugating', 'enzyme'), 27)\n",
      "(('brca1', 'prevent'), 27)\n",
      "(('prevent', 'interaction'), 27)\n",
      "(('interaction', 'e2'), 27)\n",
      "(('residue', 'predicted'), 27)\n",
      "(('substitution', 'hydrophobic'), 27)\n",
      "(('small', 'colony'), 27)\n",
      "(('11', 'residue'), 27)\n",
      "(('brca1', 'one'), 27)\n",
      "(('h1966y,', 'i2285v,'), 27)\n",
      "(('i2285v,', 'm784v,'), 27)\n",
      "(('ivs2-14', 'c>t,'), 27)\n",
      "(('brca1', 'ivs2-14'), 27)\n",
      "(('n1878k,', 'y42c'), 27)\n",
      "(('s127n,', 'v1804d'), 27)\n",
      "(('a1170v,', 'e2856a,'), 27)\n",
      "(('tumors', 'providing'), 27)\n",
      "(('brca2', 'a1170v,'), 27)\n",
      "(('y42c', 'lost'), 27)\n",
      "(('e2856a,', 'h1966y,'), 27)\n",
      "(('v1804d', 'brca2'), 27)\n",
      "(('c>t,', 's127n,'), 27)\n",
      "(('m784v,', 'n1878k,'), 27)\n",
      "(('17', 'd816h'), 27)\n",
      "(('kit', 'protein'), 27)\n",
      "(('11', 'l576p'), 27)\n",
      "(('activation', 'kit'), 27)\n",
      "(('notch1', 'receptors'), 27)\n",
      "(('transmembrane', '(ntm)'), 27)\n",
      "(('(ntm)', 'subunits'), 27)\n",
      "(('nec,', 'notch1'), 27)\n",
      "(('ram,', 'ram'), 27)\n",
      "(('ank,', 'ankyrin'), 27)\n",
      "(('full-length', 'notch1'), 27)\n",
      "(('several', 'stages'), 27)\n",
      "(('(nec)', 'transmembrane'), 27)\n",
      "(('notch1', 'hd'), 27)\n",
      "(('domain;', 'ank,'), 27)\n",
      "(('ram', 'domain;'), 27)\n",
      "(('activate', 'notch1'), 27)\n",
      "(('transmembrane', 'subunit;'), 27)\n",
      "(('induction', 't-all'), 27)\n",
      "(('notch1', 'transmembrane'), 27)\n",
      "(('repeat', 'domain;'), 27)\n",
      "(('noncovalently', 'associated'), 27)\n",
      "(('associated', 'extracellular'), 27)\n",
      "(('domain;', 'tad,'), 27)\n",
      "(('heterodimerization', 'domain;'), 27)\n",
      "(('notch1', 'heterodimer'), 27)\n",
      "(('extracellular', '(nec)'), 27)\n",
      "(('proliferation', 'assay'), 26)\n",
      "(('role', 'cell'), 26)\n",
      "(('associated', 'clinical'), 26)\n",
      "(('exons', '1'), 26)\n",
      "(('known', 'oncogenic'), 26)\n",
      "(('(fig', 'a,'), 26)\n",
      "(('mutations,', 'expressed'), 26)\n",
      "(('e3', 'ubiquitin'), 26)\n",
      "(('found', 'mutant'), 26)\n",
      "(('flt3', 'mutation'), 26)\n",
      "(('residue', 'position'), 26)\n",
      "(('table', 'table'), 26)\n",
      "(('found', 'carry'), 26)\n",
      "(('screened', 'mutations'), 26)\n",
      "(('mononuclear', 'cells'), 26)\n",
      "(('previous', 'figures/tables'), 26)\n",
      "(('growth', 'control'), 26)\n",
      "(('activity', 'might'), 26)\n",
      "(('induced', 'clear'), 26)\n",
      "(('base', 'pairs'), 26)\n",
      "(('one', 'mechanisms'), 26)\n",
      "(('genome', 'sequencing'), 26)\n",
      "(('2', '3,'), 26)\n",
      "(('mutations', 'several'), 26)\n",
      "(('number', 'cases'), 26)\n",
      "(('several', 'types'), 26)\n",
      "(('malignant', 'pleural'), 26)\n",
      "(('may', 'identify'), 26)\n",
      "(('case', 'showed'), 26)\n",
      "(('expression', 'significantly'), 26)\n",
      "(('cell', 'carcinoma,'), 26)\n",
      "(('expression', 'compared'), 26)\n",
      "(('binding', 'dna'), 26)\n",
      "(('expression', 'increased'), 26)\n",
      "(('identified', 'six'), 26)\n",
      "(('study', 'show'), 26)\n",
      "(('compared', 'controls'), 26)\n",
      "(('mutant', 'sequences'), 26)\n",
      "(('tcga', 'data'), 26)\n",
      "(('public', 'databases'), 26)\n",
      "(('effect', 'mutations'), 26)\n",
      "(('percentage', 'cells'), 26)\n",
      "(('wt', 'protein'), 26)\n",
      "(('gtp', 'hydrolysis'), 26)\n",
      "(('(figure', '6c)'), 26)\n",
      "(('ras', 'mutations'), 26)\n",
      "(('signaling', 'properties'), 26)\n",
      "(('proteins', '(figure'), 26)\n",
      "(('tumors', 'cell'), 26)\n",
      "(('normal', 'control'), 26)\n",
      "(('possible', 'role'), 26)\n",
      "(('chromosomal', 'aberrations'), 26)\n",
      "(('cells', 'normal'), 26)\n",
      "(('double', 'mutants'), 26)\n",
      "(('likely', 'accounts'), 26)\n",
      "(('basal', 'level'), 26)\n",
      "(('here,', 'identified'), 26)\n",
      "(('mcf-7', 'cells'), 26)\n",
      "(('point', 'mutation,'), 26)\n",
      "(('(fig', 'importantly,'), 26)\n",
      "(('among', 'three'), 26)\n",
      "(('essential', 'component'), 26)\n",
      "(('may', 'involved'), 26)\n",
      "(('transfected', 'constructs'), 26)\n",
      "(('activation', 'ras'), 26)\n",
      "(('domain', 'compared'), 26)\n",
      "(('carry', 'mutation'), 26)\n",
      "(('commercially', 'available'), 26)\n",
      "(('suggest', 'that,'), 26)\n",
      "(('involved', 'dna'), 26)\n",
      "(('less', 'likely'), 26)\n",
      "(('reported', 'several'), 26)\n",
      "(('even', 'higher'), 26)\n",
      "(('human', 'tumor'), 26)\n",
      "(('growth', 'curves'), 26)\n",
      "(('finding', 'suggests'), 26)\n",
      "(('clinical', 'implications'), 26)\n",
      "(('chronic', 'lymphocytic'), 26)\n",
      "(('data', 'two'), 26)\n",
      "(('gene', 'set'), 26)\n",
      "(('molecular', 'pathogenesis'), 26)\n",
      "(('biosystems,', 'foster'), 26)\n",
      "(('number', 'samples'), 26)\n",
      "(('one', 'mutant'), 26)\n",
      "(('showed', 'mutant'), 26)\n",
      "(('(n', '16)'), 26)\n",
      "(('mutations', 'based'), 26)\n",
      "(('identification', 'somatic'), 26)\n",
      "(('analysis', 'found'), 26)\n",
      "(('array-based', 'comparative'), 26)\n",
      "(('may', 'predict'), 26)\n",
      "(('effects', 'gefitinib'), 26)\n",
      "(('mechanisms', 'regulate'), 26)\n",
      "(('detected', 'multiple'), 26)\n",
      "(('mutation,', 'previously'), 26)\n",
      "(('well', 'correlated'), 26)\n",
      "(('oncogene', 'homolog'), 26)\n",
      "(('patients', 'sensitive'), 26)\n",
      "(('cancer', 'figure'), 26)\n",
      "(('dna', 'revealed'), 26)\n",
      "(('cells', 'pretreated'), 26)\n",
      "(('mechanisms', 'involving'), 26)\n",
      "(('3-fold', 'higher'), 26)\n",
      "(('kinase', 'bound'), 26)\n",
      "(('using', 'transient'), 26)\n",
      "(('egfr', 'confer'), 26)\n",
      "(('factor', '(egf)'), 26)\n",
      "(('(8,', '9)'), 26)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('greater', 'sensitivity'), 26)\n",
      "(('selective', 'inhibitors'), 26)\n",
      "(('incidence', 'egfr'), 26)\n",
      "(('whether', 'alterations'), 26)\n",
      "(('response', 'rate,'), 26)\n",
      "(('sensitive', 'tkis,'), 26)\n",
      "(('together', 'finding'), 26)\n",
      "(('another', 'tumor'), 26)\n",
      "(('rearrangements', 'within'), 26)\n",
      "(('lines', 'contain'), 26)\n",
      "(('tyrosine', 'phosphorylations'), 26)\n",
      "(('analyses', 'egfr'), 26)\n",
      "(('status', 'response'), 26)\n",
      "(('emerging', 'paradigm'), 26)\n",
      "(('although', 'effect'), 26)\n",
      "(('presence', 'gene'), 26)\n",
      "(('immunoblotting', 'lysates'), 26)\n",
      "(('significant', 'responses'), 26)\n",
      "(('associated', 'growth'), 26)\n",
      "(('gefitinib', 'compared'), 26)\n",
      "(('mutants', 'enhanced'), 26)\n",
      "(('patients', 'progressed'), 26)\n",
      "(('low', 'concentrations'), 26)\n",
      "(('ascertain', 'whether'), 26)\n",
      "(('tumors', 'sensitive'), 26)\n",
      "(('antibodies', 'directed'), 26)\n",
      "(('predictor', 'response'), 26)\n",
      "(('interestingly,', 'mutant'), 26)\n",
      "(('mutations', 'nonsmall-cell'), 26)\n",
      "(('tumors', 'activating'), 26)\n",
      "(('show', 'differential'), 26)\n",
      "(('groups', 'shown'), 26)\n",
      "(('proteins', 'induced'), 26)\n",
      "(('2', 'major'), 26)\n",
      "(('mutations', 'correlated'), 26)\n",
      "(('mutation', 'l858r'), 26)\n",
      "(('reported', 'egfr'), 26)\n",
      "(('antibodies', 'small'), 26)\n",
      "(('resistance', 'due'), 26)\n",
      "(('family', 'tyrosine'), 26)\n",
      "(('included', 'one'), 26)\n",
      "(('vitro', 'analysis,'), 26)\n",
      "(('furthermore,', 'patients'), 26)\n",
      "(('4', 'shows'), 26)\n",
      "(('identification', 'activating'), 26)\n",
      "(('figure', '2),'), 26)\n",
      "(('inactive', 'active'), 26)\n",
      "(('adenocarcinoma', 'patients'), 26)\n",
      "(('status', 'egfr'), 26)\n",
      "(('widely', 'used'), 26)\n",
      "(('found', 'cells'), 26)\n",
      "(('independent', 'pcr'), 26)\n",
      "(('induced', 'apoptosis'), 26)\n",
      "(('resistance', 'apoptosis'), 26)\n",
      "(('significant', 'inhibition'), 26)\n",
      "(('none', 'seven'), 26)\n",
      "(('control', 'group'), 26)\n",
      "(('29', 'patients'), 26)\n",
      "(('common', 'type'), 26)\n",
      "(('mutations', 'structure'), 26)\n",
      "(('kinase', 'active'), 26)\n",
      "(('figure', '1,'), 26)\n",
      "(('whereas', 'patients'), 26)\n",
      "(('could', 'associated'), 26)\n",
      "(('responses', 'egfr'), 26)\n",
      "(('mutations', 'effect'), 26)\n",
      "(('although', 'frequency'), 26)\n",
      "(('chemotherapy', 'regimen'), 26)\n",
      "(('cancer', 'treatment'), 26)\n",
      "(('one', 'sample'), 26)\n",
      "(('transducer', 'activator'), 26)\n",
      "(('vitro', 'sensitivity'), 26)\n",
      "(('respond', 'well'), 26)\n",
      "(('patients', 'relapsed'), 26)\n",
      "(('11', 'insertion'), 26)\n",
      "(('il-3', 'independent'), 26)\n",
      "(('mutants', 'sensitive'), 26)\n",
      "(('receptor', 'phosphorylation'), 26)\n",
      "(('retroviral', 'transduction'), 26)\n",
      "(('require', 'additional'), 26)\n",
      "(('g1', 'phase'), 26)\n",
      "(('following', 'transfection'), 26)\n",
      "(('despite', 'low'), 26)\n",
      "(('nucleotide', 'exchange'), 26)\n",
      "(('functionally', 'relevant'), 26)\n",
      "(('downstream', 'pathways'), 26)\n",
      "(('proportion', 'patients'), 26)\n",
      "(('mutations', 'addition'), 26)\n",
      "(('showed', 'highest'), 26)\n",
      "(('may', 'cause'), 26)\n",
      "(('able', 'inhibit'), 26)\n",
      "(('conferring', 'resistance'), 26)\n",
      "(('mutations', 'multiple'), 26)\n",
      "(('protein', 'domain'), 26)\n",
      "(('structural', 'model'), 26)\n",
      "(('1', '11'), 26)\n",
      "(('harbored', 'mutations'), 26)\n",
      "(('amplification', 'and/or'), 26)\n",
      "(('mutation', 'still'), 26)\n",
      "(('discussion', 'report,'), 26)\n",
      "(('30', 'minutes'), 26)\n",
      "(('cells', 'might'), 26)\n",
      "(('positive', 'negative'), 26)\n",
      "(('vector', '(figure'), 26)\n",
      "(('contribute', 'tumor'), 26)\n",
      "(('mutations', 'tested,'), 26)\n",
      "(('(table', '2'), 26)\n",
      "(('mutations', 'resistant'), 26)\n",
      "(('missense', 'mutant'), 26)\n",
      "(('2', '10'), 26)\n",
      "(('data', 'set'), 26)\n",
      "(('activating', 'mutation,'), 26)\n",
      "(('basal', 'activity'), 26)\n",
      "(('readily', 'detectable'), 26)\n",
      "(('receptor', '(fig'), 26)\n",
      "(('activation', 'likely'), 26)\n",
      "(('pcr', 'performed'), 26)\n",
      "(('dna', 'fragment'), 26)\n",
      "(('determined', 'effect'), 26)\n",
      "(('fold', 'increase'), 26)\n",
      "(('3-kinase', '(pi3k)'), 26)\n",
      "(('inhibit', 'growth'), 26)\n",
      "(('patients', 'likely'), 26)\n",
      "(('using', 'small'), 26)\n",
      "(('mutation', 'amplification'), 26)\n",
      "(('methods', 'cell'), 26)\n",
      "(('structural', 'changes'), 26)\n",
      "(('analysis', 'demonstrated'), 26)\n",
      "(('whereas', 'none'), 26)\n",
      "(('less', 'stable'), 26)\n",
      "(('fusions', 'found'), 26)\n",
      "(('common', 'genetic'), 26)\n",
      "(('cruz,', 'ca),'), 26)\n",
      "(('biological', 'relevance'), 26)\n",
      "(('low', 'incidence'), 26)\n",
      "(('allele', 'one'), 26)\n",
      "(('protein', '1'), 26)\n",
      "(('active', 'mutant'), 26)\n",
      "(('line', '(figure'), 26)\n",
      "(('genomic', 'analysis'), 26)\n",
      "(('cells', 'versus'), 26)\n",
      "(('vector', 'control,'), 26)\n",
      "(('one', 'wild-type'), 26)\n",
      "(('pi3k', 'signaling'), 26)\n",
      "(('interaction', 'smad4'), 26)\n",
      "(('increase', 'rate'), 26)\n",
      "(('whether', 'changes'), 26)\n",
      "(('mice', 'express'), 26)\n",
      "(('steady-state', 'levels'), 26)\n",
      "(('characterized', 'study'), 26)\n",
      "(('several', 'mutations,'), 26)\n",
      "(('activity', 'relative'), 26)\n",
      "(('cells', 'a,'), 26)\n",
      "(('described', 'recently'), 26)\n",
      "(('ic50', '(nm)'), 26)\n",
      "(('activity', 'imatinib'), 26)\n",
      "(('sequence', 'alignments'), 26)\n",
      "(('normal', 'samples'), 26)\n",
      "(('kinase', 'insert'), 26)\n",
      "(('well', 'studied'), 26)\n",
      "(('immunoprecipitation', 'western'), 26)\n",
      "(('nras', 'mutations'), 26)\n",
      "(('imatinib', 'sunitinib'), 26)\n",
      "(('resistant', 'imatinib'), 26)\n",
      "(('determine', 'potential'), 26)\n",
      "(('factor', '1'), 26)\n",
      "(('tumors', 'associated'), 26)\n",
      "(('(exon', '14)'), 26)\n",
      "(('also', 'support'), 26)\n",
      "(('10', 'nm'), 26)\n",
      "(('showed', 'interaction'), 26)\n",
      "(('number', 'individuals'), 26)\n",
      "(('transactivation', 'domain'), 26)\n",
      "(('cells', 'even'), 26)\n",
      "(('met', 'receptor'), 26)\n",
      "(('lines', 'tested,'), 26)\n",
      "(('cancer', 'biology'), 26)\n",
      "(('present', 'studies'), 26)\n",
      "(('results', 'activation'), 26)\n",
      "(('signaling', 'required'), 26)\n",
      "(('supplementary', 'information'), 26)\n",
      "(('pten', 'mutant'), 26)\n",
      "(('disrupts', 'interaction'), 26)\n",
      "(('antibody', 'specific'), 26)\n",
      "(('il-3-independent', 'growth'), 26)\n",
      "(('mutations', 'stabilize'), 26)\n",
      "(('acquisition', 'additional'), 26)\n",
      "(('francisco,', 'ca),'), 26)\n",
      "(('primary', 'aml'), 26)\n",
      "(('janus', 'kinase'), 26)\n",
      "(('fully', 'abrogated'), 26)\n",
      "(('g165e,', 'g165v,'), 26)\n",
      "(('(2)', 'ubch5b'), 26)\n",
      "(('glu75:', 'likely'), 26)\n",
      "(('fold', 't77m'), 26)\n",
      "(('group', 'likely'), 26)\n",
      "(('class', 'substitution'), 26)\n",
      "(('t69:', 'predicted'), 26)\n",
      "(('e2', 'k50l/e'), 26)\n",
      "(('site;', 'substitution'), 26)\n",
      "(('group', 'buried'), 26)\n",
      "(('disrupt', 'backbone'), 26)\n",
      "(('charged', 'substitution'), 26)\n",
      "(('e2', 'k56n'), 26)\n",
      "(('k56n', 'may'), 26)\n",
      "(('structure', 'cbl:ubch7'), 26)\n",
      "(('loss', 'hydrophobic'), 26)\n",
      "(('weaken', 'folding'), 26)\n",
      "(('h41l/y/r', 'c44f/r'), 26)\n",
      "(('ubch5b', '(31)'), 26)\n",
      "(('f79s', 'f79'), 26)\n",
      "(('orientation', 'helical'), 26)\n",
      "(('folding', 'c39s/r'), 26)\n",
      "(('brca1', '(2)'), 26)\n",
      "(('bond', 't69:'), 26)\n",
      "(('hydrophobic', 'surface'), 26)\n",
      "(('alter', 'packing'), 26)\n",
      "(('cbl:ubch7', '(29))'), 26)\n",
      "(('folding', 'removal'), 26)\n",
      "(('k65m', 'may'), 26)\n",
      "(('weaken', 'ring'), 26)\n",
      "(('c47g/t', 'c61g'), 26)\n",
      "(('structural', 'disruption'), 26)\n",
      "(('g57r', 'brca1:cbl'), 26)\n",
      "(('ring', 'orientation'), 26)\n",
      "(('part', 'e2'), 26)\n",
      "(('bridge', 'k65m'), 26)\n",
      "(('disrupt', 'ring'), 26)\n",
      "(('conformations;', 'substitution'), 26)\n",
      "(('buried', 'residue:'), 26)\n",
      "(('fold', 'l52f/h'), 26)\n",
      "(('ligand', 'c24r'), 26)\n",
      "(('domain', 's72r'), 26)\n",
      "(('(to', 'k4'), 26)\n",
      "(('bonds', 'salt'), 26)\n",
      "(('ring', 'f79s'), 26)\n",
      "(('interface', 'ring'), 26)\n",
      "(('enzyme,', 'proline'), 26)\n",
      "(('q92)', 'i26n'), 26)\n",
      "(('forms', 'packing'), 26)\n",
      "(('h', 'bonds'), 26)\n",
      "(('disrupts', 'salt'), 26)\n",
      "(('t37r', 'non-conservative'), 26)\n",
      "(('may', 'structural'), 26)\n",
      "(('brca1:cbl', 'backbones'), 26)\n",
      "(('ubch5a', 'e29v'), 26)\n",
      "(('ubch5a', 'class'), 26)\n",
      "(('h', 'bonding'), 26)\n",
      "(('structures', 'ring'), 26)\n",
      "(('domain', 'k45e/t/n'), 26)\n",
      "(('domain', 't37r'), 26)\n",
      "(('packing', 'e2'), 26)\n",
      "(('point,', 'may'), 26)\n",
      "(('form', 'h-bond'), 26)\n",
      "(('s72r', 'disrupts'), 26)\n",
      "(('domain)', 'l28p'), 26)\n",
      "(('e2', 'enzyme,'), 26)\n",
      "(('disruption', '(i26'), 26)\n",
      "(('(29))', 'zn2'), 26)\n",
      "(('ligand', 'likely'), 26)\n",
      "(('substitution', 'largely'), 26)\n",
      "(('preventing', 'folding'), 26)\n",
      "(('interface', 'ubch5a'), 26)\n",
      "(('(to', 's91'), 26)\n",
      "(('buried', 'group'), 26)\n",
      "(('surface', 'residue'), 26)\n",
      "(('superimposition', 'brca1'), 26)\n",
      "(('ring', 'e3:e2'), 26)\n",
      "(('i26n', 'may'), 26)\n",
      "(('q60l', 'lies'), 26)\n",
      "(('removal', 'h-bond'), 26)\n",
      "(('c44f/r', 'c47g/t'), 26)\n",
      "(('introduction', 'larger'), 26)\n",
      "(('e2', 'q60l'), 26)\n",
      "(('k4', 'ubch5b)'), 26)\n",
      "(('disrupt', 'h'), 26)\n",
      "(('weaken', 'overall'), 26)\n",
      "(('loss', 'zn2'), 26)\n",
      "(('folding', 'g57r'), 26)\n",
      "(('k45e/t/n', 'substitution'), 26)\n",
      "(('e3:e2', 'structure'), 26)\n",
      "(('helical', 'domains'), 26)\n",
      "(('hydrophobic', 'polar:'), 26)\n",
      "(('bulky', 'polar'), 26)\n",
      "(('polar:', 'expected'), 26)\n",
      "(('c24r', 'loss'), 26)\n",
      "(('close', 'e2'), 26)\n",
      "(('ubch5b', 'q92)'), 26)\n",
      "(('larger', 'buried'), 26)\n",
      "(('(based', 'superimposition'), 26)\n",
      "(('within', 'ring'), 26)\n",
      "(('e2', 'within'), 26)\n",
      "(('backbone', 'preventing'), 26)\n",
      "(('zn2', 'ligand'), 26)\n",
      "(('(to', 'ubch5b'), 26)\n",
      "(('t77m', 'may'), 26)\n",
      "(('substitution', '(based'), 26)\n",
      "(('domain', 'i31t'), 26)\n",
      "(('ring', 'helical'), 26)\n",
      "(('(31)', 'structures'), 26)\n",
      "(('residue:', 'likely'), 26)\n",
      "(('k50l/e', 'may'), 26)\n",
      "(('differ', 'point,'), 26)\n",
      "(('n', 'hydrophobic'), 26)\n",
      "(('fold', 's59r'), 26)\n",
      "(('domain', 'i68k'), 26)\n",
      "(('s91', 'ubch5b)'), 26)\n",
      "(('c64g', 'ring'), 26)\n",
      "(('bridge', 'brca1'), 26)\n",
      "(('i68k', 'charged'), 26)\n",
      "(('enzyme', 'ubch5a'), 26)\n",
      "(('contact', 'e2'), 26)\n",
      "(('e29v', 'disrupts'), 26)\n",
      "(('s59r', 'charged'), 26)\n",
      "(('bridge', '(to'), 26)\n",
      "(('disrupt', 'h-bond'), 26)\n",
      "(('backbones', 'differ'), 26)\n",
      "(('c39s/r', 'h41l/y/r'), 26)\n",
      "(('lies', 'interface'), 26)\n",
      "(('s80', 'conformations;'), 26)\n",
      "(('h-bond', 's80'), 26)\n",
      "(('disrupt', 'salt'), 26)\n",
      "(('predicted', 'weaken'), 26)\n",
      "(('l63f', 'introduction'), 26)\n",
      "(('h-bond', '(to'), 26)\n",
      "(('introduction', 'bulky'), 26)\n",
      "(('i31t', 'n'), 26)\n",
      "(('(i26', 'forms'), 26)\n",
      "(('l52f/h', 'introduction'), 26)\n",
      "(('h-bond', 'l63f'), 26)\n",
      "(('bonding', 'salt'), 26)\n",
      "(('l28p', 'loss'), 26)\n",
      "(('bundle', 'ring'), 26)\n",
      "(('d59', 'ubch5b)'), 26)\n",
      "(('(to', 'd59'), 26)\n",
      "(('c61g', 'c64g'), 26)\n",
      "(('brca1', 'glu75:'), 26)\n",
      "(('ubch5b)', 'weaken'), 26)\n",
      "(('f79', 'lies'), 26)\n",
      "(('brca2', 'variant'), 26)\n",
      "(('activity', 'kit'), 26)\n",
      "(('initial', 'imatinib'), 26)\n",
      "(('receptors', '(notch1-4),'), 26)\n",
      "(('(dll1,', 'dll3,'), 26)\n",
      "(('notch1', 'tumor'), 26)\n",
      "(('cancer', 'patients,'), 25)\n",
      "(('respectively', 'using'), 25)\n",
      "(('2', 'cases'), 25)\n",
      "(('(figure', '7)'), 25)\n",
      "(('3', 'days'), 25)\n",
      "(('flt3', 'receptor'), 25)\n",
      "(('stimulation', '(fig'), 25)\n",
      "(('also', 'induce'), 25)\n",
      "(('strongly', 'inhibited'), 25)\n",
      "(('(figure', '5d)'), 25)\n",
      "(('(figure', '4e)'), 25)\n",
      "(('activation', 'flt3'), 25)\n",
      "(('also', 'considered'), 25)\n",
      "(('detected', 'heterozygous'), 25)\n",
      "(('9', '10'), 25)\n",
      "(('figure', '4:'), 25)\n",
      "(('effect', 'observed'), 25)\n",
      "(('table', 'figures/tables'), 25)\n",
      "(('table', '2:'), 25)\n",
      "(('promote', 'cell'), 25)\n",
      "(('kit', '(agilent'), 25)\n",
      "(('10', 'ng'), 25)\n",
      "(('survival', 'curves'), 25)\n",
      "(('nih3t3', 'cells,'), 25)\n",
      "(('found', 'tumors'), 25)\n",
      "(('potential', 'mechanism'), 25)\n",
      "(('mutations', 'yet'), 25)\n",
      "(('hepatocellular', 'carcinoma'), 25)\n",
      "(('see', 'whether'), 25)\n",
      "(('significant', 'role'), 25)\n",
      "(('primers', 'specific'), 25)\n",
      "(('mobility', 'shift'), 25)\n",
      "(('transcription', '(fig'), 25)\n",
      "(('provided', 'evidence'), 25)\n",
      "(('recently,', 'two'), 25)\n",
      "(('(fig', 'findings'), 25)\n",
      "(('followed', 'selection'), 25)\n",
      "(('genomic', 'pcr'), 25)\n",
      "(('region', 'deletion'), 25)\n",
      "(('published', 'previously'), 25)\n",
      "(('markedly', 'increased'), 25)\n",
      "(('gene', 'products'), 25)\n",
      "(('resulting', 'loss'), 25)\n",
      "(('consistent', 'finding'), 25)\n",
      "(('refinement', 'statistics'), 25)\n",
      "(('either', 'alone'), 25)\n",
      "(('figure', '4)'), 25)\n",
      "(('mice', 'treated'), 25)\n",
      "(('pathways,', 'leading'), 25)\n",
      "(('analysis', '(fig'), 25)\n",
      "(('least', 'partially'), 25)\n",
      "(('possible', 'functional'), 25)\n",
      "(('sporadic', 'cases'), 25)\n",
      "(('genetic', 'background'), 25)\n",
      "(('cannot', 'explained'), 25)\n",
      "(('cellular', 'responses'), 25)\n",
      "(('study,', 'examined'), 25)\n",
      "(('engineered', 'mouse'), 25)\n",
      "(('contains', 'two'), 25)\n",
      "(('exhibited', 'significantly'), 25)\n",
      "(('amplification', 'could'), 25)\n",
      "(('gastrointestinal', 'tract'), 25)\n",
      "(('may', 'responsible'), 25)\n",
      "(('manner', '(fig'), 25)\n",
      "(('also', 'resulted'), 25)\n",
      "(('control', '(p'), 25)\n",
      "(('forward', 'reverse'), 25)\n",
      "(('microsatellite', 'instability'), 25)\n",
      "(('address', 'whether'), 25)\n",
      "(('experimental', 'design'), 25)\n",
      "(('five', 'different'), 25)\n",
      "(('mutations', 'addition,'), 25)\n",
      "(('could', 'account'), 25)\n",
      "(('dna', 'mismatch'), 25)\n",
      "(('cells', 'mice'), 25)\n",
      "(('2', '13'), 25)\n",
      "(('proposed', 'mechanism'), 25)\n",
      "(('reported', 'far'), 25)\n",
      "(('none', 'none'), 25)\n",
      "(('1', 'mutation'), 25)\n",
      "(('multiplex', 'ligation-dependent'), 25)\n",
      "(('genetic', 'laboratories'), 25)\n",
      "(('myriad', 'genetic'), 25)\n",
      "(('like', 'wild-type'), 25)\n",
      "(('materials', 'methods)'), 25)\n",
      "(('described', 'materials'), 25)\n",
      "(('tumors', 'mutations'), 25)\n",
      "(('gene', 'detected'), 25)\n",
      "(('association', 'mutations'), 25)\n",
      "(('significantly', 'enriched'), 25)\n",
      "(('lines', 'carrying'), 25)\n",
      "(('injected', 'subcutaneously'), 25)\n",
      "(('significant', 'changes'), 25)\n",
      "(('microarray', 'analysis'), 25)\n",
      "(('cells,', 'although'), 25)\n",
      "(('protein', 'also'), 25)\n",
      "(('7,', '8,'), 25)\n",
      "(('breast', 'cancers,'), 25)\n",
      "(('four', 'cases'), 25)\n",
      "(('whereas', 'cells'), 25)\n",
      "(('expressed', 'various'), 25)\n",
      "(('expression', 'profile'), 25)\n",
      "(('mutation', 'compared'), 25)\n",
      "(('expressed', 'genes'), 25)\n",
      "(('cell', 'cycle,'), 25)\n",
      "(('nested', 'pcr'), 25)\n",
      "(('several', 'cancer'), 25)\n",
      "(('0', '2'), 25)\n",
      "(('antitumor', 'activity'), 25)\n",
      "(('statistically', 'significant,'), 25)\n",
      "(('20%', 'patients'), 25)\n",
      "(('3', 'different'), 25)\n",
      "(('mutated', 'patients'), 25)\n",
      "(('mutations', 'cohort'), 25)\n",
      "(('6', '2'), 25)\n",
      "(('showed', 'moderate'), 25)\n",
      "(('whereas', 'three'), 25)\n",
      "(('sensitive', 'growth'), 25)\n",
      "(('smoking', 'history,'), 25)\n",
      "(('particularly', 'effective'), 25)\n",
      "(('reported', 'clinical'), 25)\n",
      "(('sensitivity', 'recently,'), 25)\n",
      "(('egfr', 'cdna'), 25)\n",
      "(('performed', 'mutational'), 25)\n",
      "(('produced', 'transient'), 25)\n",
      "(('response', 'survival'), 25)\n",
      "(('egfr', 'found'), 25)\n",
      "(('tumors', 'determine'), 25)\n",
      "(('three', 'tumors,'), 25)\n",
      "(('still', 'activate'), 25)\n",
      "(('mutant', 'appeared'), 25)\n",
      "(('presence', 'second'), 25)\n",
      "(('mutation', 'particular'), 25)\n",
      "(('however,', 'addition'), 25)\n",
      "(('demonstrate', 'mutations'), 25)\n",
      "(('encoded', 'exons'), 25)\n",
      "(('clustering', 'mutations'), 25)\n",
      "(('tumors', 'carried'), 25)\n",
      "(('gefitinib', 'sensitivity'), 25)\n",
      "(('dna', 'normal'), 25)\n",
      "(('gene', 'would'), 25)\n",
      "(('concentration', '0'), 25)\n",
      "(('either', 'egfr'), 25)\n",
      "(('whereas', 'activation'), 25)\n",
      "(('seven', 'eight'), 25)\n",
      "(('analyzed', 'dna'), 25)\n",
      "(('measured', 'effect'), 25)\n",
      "(('guide', 'clinical'), 25)\n",
      "(('nsclc', 'tumors'), 25)\n",
      "(('analysis', 'entire'), 25)\n",
      "(('signaling', 'technology),'), 25)\n",
      "(('thus,', 'important'), 25)\n",
      "(('found', 'additional'), 25)\n",
      "(('structure', 'active'), 25)\n",
      "(('yet', 'determined,'), 25)\n",
      "(('gefitinib', 'responsiveness'), 25)\n",
      "(('11', 'tumors'), 25)\n",
      "(('domain', 'lung'), 25)\n",
      "(('including', 'lung'), 25)\n",
      "(('disease', 'contrast,'), 25)\n",
      "(('egfr', 'genes'), 25)\n",
      "(('substitution', 'exon'), 25)\n",
      "(('6', 'hours'), 25)\n",
      "(('erlotinib', 'gefitinib,'), 25)\n",
      "(('egfr', 'mutants,'), 25)\n",
      "(('status', 'lung'), 25)\n",
      "(('genbank', '(accession'), 25)\n",
      "(('suggest', 'presence'), 25)\n",
      "(('resistant', 'mutation'), 25)\n",
      "(('patients', 'patients'), 25)\n",
      "(('(b)', 'overall'), 25)\n",
      "(('using', 'fish'), 25)\n",
      "(('patients', 'experimental'), 25)\n",
      "(('described', 'methods'), 25)\n",
      "(('structural', 'alteration'), 25)\n",
      "(('patients', 'classified'), 25)\n",
      "(('wild-type', 'also'), 25)\n",
      "(('pcr', 'conditions'), 25)\n",
      "(('1', 'deletion'), 25)\n",
      "(('obtained', 'genbank'), 25)\n",
      "(('inhibitors', 'patients'), 25)\n",
      "(('amplification', 'met'), 25)\n",
      "(('rates', 'two'), 25)\n",
      "(('(n', '10)'), 25)\n",
      "(('stable', 'disease,'), 25)\n",
      "(('l858r', 'point'), 25)\n",
      "(('sensitive', 'egfr'), 25)\n",
      "(('b', '(akt)'), 25)\n",
      "(('dose', 'response'), 25)\n",
      "(('2', 'months'), 25)\n",
      "(('treatment', 'using'), 25)\n",
      "(('clinical', 'importance'), 25)\n",
      "(('analysis', 'clinical'), 25)\n",
      "(('status', '(p'), 25)\n",
      "(('patients', 'lacking'), 25)\n",
      "(('pretreatment', 'tumor'), 25)\n",
      "(('driver', 'mutations,'), 25)\n",
      "(('patients', 'disease'), 25)\n",
      "(('therapies', 'may'), 25)\n",
      "(('following', \"manufacturer's\"), 25)\n",
      "(('number', 'colonies'), 25)\n",
      "(('activator', 'transcription'), 25)\n",
      "(('colonies', 'formed'), 25)\n",
      "(('domain', 'mutations,'), 25)\n",
      "(('triphosphate', '(atp)'), 25)\n",
      "(('7,', '9,'), 25)\n",
      "(('pathways', 'downstream'), 25)\n",
      "(('mediate', 'resistance'), 25)\n",
      "(('hepatocyte', 'growth'), 25)\n",
      "(('showed', 'modest'), 25)\n",
      "(('resulted', 'decreased'), 25)\n",
      "(('activation', '(fig'), 25)\n",
      "(('extensively', 'studied'), 25)\n",
      "(('mutations', 'total'), 25)\n",
      "(('rate', 'higher'), 25)\n",
      "(('higher', 'level'), 25)\n",
      "(('met', 'amplification'), 25)\n",
      "(('increasing', 'doses'), 25)\n",
      "(('express', 'endogenous'), 25)\n",
      "(('within', 'c'), 25)\n",
      "(('different', 'variants'), 25)\n",
      "(('tumour', 'progression'), 25)\n",
      "(('results', 'study'), 25)\n",
      "(('giving', 'rise'), 25)\n",
      "(('risk', 'relapse'), 25)\n",
      "(('downstream', 'effectors,'), 25)\n",
      "(('development', 'secondary'), 25)\n",
      "(('acquisition', 'secondary'), 25)\n",
      "(('patient', 'tumors'), 25)\n",
      "(('next,', 'examined'), 25)\n",
      "(('shown)', 'data'), 25)\n",
      "(('effective', 'inhibiting'), 25)\n",
      "(('used', 'test'), 25)\n",
      "(('(fig', 'three'), 25)\n",
      "(('study,', 'however,'), 25)\n",
      "(('highly', 'active'), 25)\n",
      "(('encoded', 'exon'), 25)\n",
      "(('grown', 'presence'), 25)\n",
      "(('expressing', 'different'), 25)\n",
      "(('tumors', 'driven'), 25)\n",
      "(('receptor', 'dimerization'), 25)\n",
      "(('higher', 'wt'), 25)\n",
      "(('consistent', 'recent'), 25)\n",
      "(('potential', 'impact'), 25)\n",
      "(('mutated', 'receptor'), 25)\n",
      "(('neuroblastoma', 'cell'), 25)\n",
      "(('biochemical', 'characterization'), 25)\n",
      "(('expression', 'tumors'), 25)\n",
      "(('stat5', 'phosphorylation'), 25)\n",
      "(('transformation', 'assays'), 25)\n",
      "(('relative', 'resistance'), 25)\n",
      "(('may', 'additional'), 25)\n",
      "(('cytoplasmic', 'staining'), 25)\n",
      "(('surface', 'area'), 25)\n",
      "(('could', 'demonstrated'), 25)\n",
      "(('proteins', 'may'), 25)\n",
      "(('yes', 'yes'), 25)\n",
      "(('oncogenic', 'effect'), 25)\n",
      "(('statistical', 'methods'), 25)\n",
      "(('protein-protein', 'interaction'), 25)\n",
      "(('failed', 'complement'), 25)\n",
      "(('identified', 'additional'), 25)\n",
      "(('pcr', 'amplified'), 25)\n",
      "(('used', 'negative'), 25)\n",
      "(('variants', '(fig'), 25)\n",
      "(('transfected', '1'), 25)\n",
      "(('s6', 'phosphorylation'), 25)\n",
      "(('lysis', 'buffer'), 25)\n",
      "(('sensitive', 'crizotinib'), 25)\n",
      "(('found', 'tumor'), 25)\n",
      "(('plasma', 'levels'), 25)\n",
      "(('(rt-pcr)', 'using'), 25)\n",
      "(('revealed', 'significant'), 25)\n",
      "(('mutations', 'created'), 25)\n",
      "(('ovary', 'cells'), 25)\n",
      "(('nonpolyposis', 'colorectal'), 25)\n",
      "(('arm', 'chromosome'), 25)\n",
      "(('online', 'only)'), 25)\n",
      "(('vector', 'expressing'), 25)\n",
      "(('figure', '5)'), 25)\n",
      "(('potential', 'molecular'), 25)\n",
      "(('(supplementary', 'figures'), 25)\n",
      "(('proteins', 'confirmed'), 25)\n",
      "(('frequent', 'event'), 25)\n",
      "(('control', 'mice'), 25)\n",
      "(('various', 'cellular'), 25)\n",
      "(('met', 'gene'), 25)\n",
      "(('(fig', 'test'), 25)\n",
      "(('leading', 'premature'), 25)\n",
      "(('malignancies,', 'including'), 25)\n",
      "(('result', 'increased'), 25)\n",
      "(('due', 'differences'), 25)\n",
      "(('introduced', 'site-directed'), 25)\n",
      "(('predominantly', 'nuclear'), 25)\n",
      "(('mutants', 'tested,'), 25)\n",
      "(('tumor', 'dna,'), 25)\n",
      "(('cells', 'exhibit'), 25)\n",
      "(('mutations', 'except'), 25)\n",
      "(('increase', 'tyrosine'), 25)\n",
      "(('kras', 'mutations,'), 25)\n",
      "(('fails', 'rescue'), 25)\n",
      "(('families', 'breast'), 25)\n",
      "(('sequencing', 'additional'), 25)\n",
      "(('sodium', 'dodecyl'), 25)\n",
      "(('exons', '9,'), 25)\n",
      "(('leukemic', 'cell'), 25)\n",
      "(('complete', 'remission'), 25)\n",
      "(('phosphorylated', 'forms'), 25)\n",
      "(('approximately', 'one-third'), 25)\n",
      "(('treatment', 'imatinib'), 25)\n",
      "(('pdgfra', 'kinase'), 25)\n",
      "(('multicenter', 'phase'), 25)\n",
      "(('ex', 'vivo'), 25)\n",
      "(('atp', 'pocket'), 25)\n",
      "(('induced', 'strong'), 25)\n",
      "(('result', 'constitutive'), 25)\n",
      "(('broad', 'range'), 25)\n",
      "(('frequency', 'loh'), 25)\n",
      "(('features', 'including'), 25)\n",
      "(('studies', 'comparing'), 25)\n",
      "(('s6k1', 'phosphorylation'), 25)\n",
      "(('sequence', 'numbering'), 25)\n",
      "(('placid,', 'ny)'), 25)\n",
      "(('using', 'anti-phosphotyrosine'), 25)\n",
      "(('loss', 'mutant'), 25)\n",
      "(('results', 'contrast'), 25)\n",
      "(('aml', 'cases'), 25)\n",
      "(('identified', 'frequent'), 25)\n",
      "(('next', 'asked'), 25)\n",
      "(('study', 'selected'), 25)\n",
      "(('differ', 'significantly'), 25)\n",
      "(('', 'helix'), 25)\n",
      "(('overall', 'view'), 25)\n",
      "(('assess', 'pathogenicity'), 25)\n",
      "(('and/or', 'mutations'), 25)\n",
      "(('represent', 'neutral'), 25)\n",
      "(('carry', 'variant'), 25)\n",
      "(('fgfr', 'kinase'), 25)\n",
      "(('expressing', 'flag-tagged'), 25)\n",
      "(('resistance', '(fig'), 25)\n",
      "(('none', 'four'), 25)\n",
      "(('activation,', 'first'), 25)\n",
      "(('4', '(a)'), 25)\n",
      "(('primary', 'site'), 25)\n",
      "(('eligibility', 'criteria'), 25)\n",
      "(('region', 'reported'), 25)\n",
      "(('ic', '50'), 25)\n",
      "(('cell', 'autonomous'), 25)\n",
      "(('functional', 'assay'), 25)\n",
      "(('growth', 'using'), 25)\n",
      "(('variants', 'likely'), 25)\n",
      "(('rare', 'cases'), 25)\n",
      "(('affect', 'pten'), 25)\n",
      "(('information', 'mutations'), 25)\n",
      "(('affect', 'brca1'), 25)\n",
      "(('braf', 'mutants'), 25)\n",
      "(('signaling', 'promoting'), 25)\n",
      "(('loss', 'variant'), 25)\n",
      "(('brca2', 'variants'), 25)\n",
      "(('11', 'v559d'), 25)\n",
      "(('9', 'kit'), 25)\n",
      "(('phosphorylation', 'kit'), 25)\n",
      "(('difference', 'statistically'), 24)\n",
      "(('showed', 'loss'), 24)\n",
      "(('lacking', 'exon'), 24)\n",
      "(('cells', 'upon'), 24)\n",
      "(('expressing', 'cell'), 24)\n",
      "(('src', 'family'), 24)\n",
      "(('rtk', 'signaling'), 24)\n",
      "(('mutant', 'flt3'), 24)\n",
      "(('mutant', 'human'), 24)\n",
      "(('proteins', 'also'), 24)\n",
      "(('loss', 'activity'), 24)\n",
      "(('two', 'domains'), 24)\n",
      "(('cases', 'without'), 24)\n",
      "(('enhanced', 'activation'), 24)\n",
      "(('mice', 'carrying'), 24)\n",
      "(('precursor', 'cells'), 24)\n",
      "(('mutations', 'high'), 24)\n",
      "(('19', 'mutations'), 24)\n",
      "(('poorly', 'differentiated'), 24)\n",
      "(('samples', '(table'), 24)\n",
      "(('reverse', 'transcriptase'), 24)\n",
      "(('recent', 'findings'), 24)\n",
      "(('studies', 'published'), 24)\n",
      "(('present', 'two'), 24)\n",
      "(('fig', '1'), 24)\n",
      "(('fig', 'taken'), 24)\n",
      "(('cleavage', 'product'), 24)\n",
      "(('less', 'severe'), 24)\n",
      "(('increased', 'apoptosis'), 24)\n",
      "(('appeared', 'similar'), 24)\n",
      "(('results', 'disruption'), 24)\n",
      "(('role', 'growth'), 24)\n",
      "(('protein-protein', 'interactions'), 24)\n",
      "(('expression', 'two'), 24)\n",
      "(('active', 'inactive'), 24)\n",
      "(('wt', 'mutants'), 24)\n",
      "(('2', '21'), 24)\n",
      "(('two', 'activating'), 24)\n",
      "(('3a)', 'similar'), 24)\n",
      "(('cells,', 'including'), 24)\n",
      "(('xenograft', 'tumors'), 24)\n",
      "(('inhibition', 'erk'), 24)\n",
      "(('phosphatase', 'tensin'), 24)\n",
      "(('inactive', 'form'), 24)\n",
      "(('(figure', '3d)'), 24)\n",
      "(('cells', 'reduced'), 24)\n",
      "(('showed', 'normal'), 24)\n",
      "(('protein', 'product'), 24)\n",
      "(('significantly', 'elevated'), 24)\n",
      "(('addition,', 'two'), 24)\n",
      "(('line', 'homozygous'), 24)\n",
      "(('diverse', 'tumor'), 24)\n",
      "(('cells', 'growth'), 24)\n",
      "(('without', 'amplification'), 24)\n",
      "(('many', 'different'), 24)\n",
      "(('compared', 'corresponding'), 24)\n",
      "(('transfected', 'u2os'), 24)\n",
      "(('induces', 'apoptosis'), 24)\n",
      "(('nd', 'nd'), 24)\n",
      "(('domain', 'figure'), 24)\n",
      "(('60', '80'), 24)\n",
      "(('mutation', 'consistent'), 24)\n",
      "(('malignant', 'cells'), 24)\n",
      "(('growth', 'inhibition,'), 24)\n",
      "(('contrast,', 'expression'), 24)\n",
      "(('loss', 'expression'), 24)\n",
      "(('colon', 'cancer,'), 24)\n",
      "(('strong', 'expression'), 24)\n",
      "(('expression', 'using'), 24)\n",
      "(('premature', 'termination'), 24)\n",
      "(('inhibit', 'activation'), 24)\n",
      "(('results', 'observed'), 24)\n",
      "(('patients', 'cancer'), 24)\n",
      "(('variants', 'demonstrate'), 24)\n",
      "(('allow', 'us'), 24)\n",
      "(('molecular', 'level'), 24)\n",
      "(('five', 'mutants'), 24)\n",
      "(('ligation-dependent', 'probe'), 24)\n",
      "(('segregation', 'analysis,'), 24)\n",
      "(('analysis', 'patients'), 24)\n",
      "(('commonly', 'observed'), 24)\n",
      "(('however,', 'recent'), 24)\n",
      "(('contrast,', 'mutations'), 24)\n",
      "(('mutation', 'although'), 24)\n",
      "(('colorectal', 'tumors'), 24)\n",
      "(('tumors,', 'respectively,'), 24)\n",
      "(('factors,', 'including'), 24)\n",
      "(('thus', 'far,'), 24)\n",
      "(('1', '4'), 24)\n",
      "(('even', 'low'), 24)\n",
      "(('rt-pcr', 'products'), 24)\n",
      "(('three', 'genes'), 24)\n",
      "(('known', 'mutations'), 24)\n",
      "(('syndrome', '(mds),'), 24)\n",
      "(('whose', 'disease'), 24)\n",
      "(('lines', 'cell'), 24)\n",
      "(('rna', 'splicing'), 24)\n",
      "(('would', 'suggest'), 24)\n",
      "(('(figure', '7a)'), 24)\n",
      "(('(b', 'c)'), 24)\n",
      "(('could', 'explained'), 24)\n",
      "(('patients', 'showed'), 24)\n",
      "(('found', 'significantly'), 24)\n",
      "(('(figure', '2e)'), 24)\n",
      "(('blood', 'bone'), 24)\n",
      "(('occur', '20%'), 24)\n",
      "(('almost', 'half'), 24)\n",
      "(('genetic', 'lesions'), 24)\n",
      "(('multivariate', 'analysis'), 24)\n",
      "(('disease', 'phenotype'), 24)\n",
      "(('whole-exome', 'sequencing'), 24)\n",
      "(('erlotinib', 'sensitivity'), 24)\n",
      "(('supplementary', 'appendix)'), 24)\n",
      "(('lines', 'lung'), 24)\n",
      "(('targeting', 'mutant'), 24)\n",
      "(('reported', 'date,'), 24)\n",
      "(('identification', 'egfr'), 24)\n",
      "(('activity,', 'measured'), 24)\n",
      "(('adenocarcinomas', 'never'), 24)\n",
      "(('contain', 'egfr'), 24)\n",
      "(('pnas', 'web'), 24)\n",
      "(('insight', 'mechanisms'), 24)\n",
      "(('render', 'tumors'), 24)\n",
      "(('role', 'tumorigenesis,'), 24)\n",
      "(('gene', 'critical'), 24)\n",
      "(('predicted', 'change'), 24)\n",
      "(('common', 'mutation,'), 24)\n",
      "(('suggests', 'egfr'), 24)\n",
      "(('(24,', '25),'), 24)\n",
      "(('published', 'literature'), 24)\n",
      "(('adverse', 'effects'), 24)\n",
      "(('smoking', 'history'), 24)\n",
      "(('mutations', 'sensitivity'), 24)\n",
      "(('regions', 'within'), 24)\n",
      "(('sequenced', 'exons'), 24)\n",
      "(('models', 'expressing'), 24)\n",
      "(('exons', '19'), 24)\n",
      "(('two', 'egfr'), 24)\n",
      "(('mutation', 'among'), 24)\n",
      "(('fundamentally', 'different'), 24)\n",
      "(('activation', 'mitogen-activated'), 24)\n",
      "(('examined', 'exons'), 24)\n",
      "(('qualitatively', 'quantitatively'), 24)\n",
      "(('7', '15'), 24)\n",
      "(('whereas', '4'), 24)\n",
      "(('published', 'supporting'), 24)\n",
      "(('(fig', 'egfr'), 24)\n",
      "(('protein', 'activity'), 24)\n",
      "(('alterations', 'result'), 24)\n",
      "(('mutations', '16'), 24)\n",
      "(('proteins', 'altered'), 24)\n",
      "(('treatment', 'also'), 24)\n",
      "(('clinical', 'improvement'), 24)\n",
      "(('drug', 'concentration'), 24)\n",
      "(('predict', 'sensitivity'), 24)\n",
      "(('associated', 'response'), 24)\n",
      "(('nsclc', 'patients,'), 24)\n",
      "(('eight', 'nine'), 24)\n",
      "(('information', 'pnas'), 24)\n",
      "(('also', 'important'), 24)\n",
      "(('caused', 'similar'), 24)\n",
      "(('region', 'egfr'), 24)\n",
      "(('comprise', 'distinct'), 24)\n",
      "(('provide', 'important'), 24)\n",
      "(('use', 'site-directed'), 24)\n",
      "(('(28,', '29),'), 24)\n",
      "(('similar', 'control'), 24)\n",
      "(('mutations', 'started'), 24)\n",
      "(('negative', 'egfr'), 24)\n",
      "(('chromosome', '7,'), 24)\n",
      "(('associated', 'resistance'), 24)\n",
      "(('amplified', 'primers'), 24)\n",
      "(('model', 'tumor'), 24)\n",
      "(('mutation', 'wild-type'), 24)\n",
      "(('factors', 'including'), 24)\n",
      "(('gatekeeper', 'residue'), 24)\n",
      "(('irreversible', 'egfr'), 24)\n",
      "(('mutations', 'mutation'), 24)\n",
      "(('tumors', 'known'), 24)\n",
      "(('also', 'studied'), 24)\n",
      "(('group', 'performance'), 24)\n",
      "(('predicting', 'tumor'), 24)\n",
      "(('activating', 'egfr'), 24)\n",
      "(('exhibited', 'higher'), 24)\n",
      "(('expressed', 'purified'), 24)\n",
      "(('clinical', 'laboratory'), 24)\n",
      "(('mutations', 'exist'), 24)\n",
      "(('patient', 'populations'), 24)\n",
      "(('response', '(p'), 24)\n",
      "(('group', 'egfr'), 24)\n",
      "(('mutations', 'mutually'), 24)\n",
      "(('spectrum', 'egfr'), 24)\n",
      "(('age,', 'sex,'), 24)\n",
      "(('patients', 'current'), 24)\n",
      "(('clinical', 'trials,'), 24)\n",
      "(('patients', 'mutated'), 24)\n",
      "(('higher', 'egfr'), 24)\n",
      "(('kaplan-meier', 'plot'), 24)\n",
      "(('correlation', 'clinical'), 24)\n",
      "(('duplication', 'exon'), 24)\n",
      "(('mutations', 'indicated'), 24)\n",
      "(('egfr', 'reported'), 24)\n",
      "(('compounds', 'inhibit'), 24)\n",
      "(('showed', 'tumor'), 24)\n",
      "(('egfr', 'deletion'), 24)\n",
      "(('remaining', 'mutations'), 24)\n",
      "(('multivariate', 'analysis,'), 24)\n",
      "(('19', 'l858r'), 24)\n",
      "(('sequences', 'pcr'), 24)\n",
      "(('longer', 'progression-free'), 24)\n",
      "(('using', 'polymerase'), 24)\n",
      "(('10', 'nmol/l'), 24)\n",
      "(('(figure', '1,'), 24)\n",
      "(('progressive', 'disease;'), 24)\n",
      "(('mutation', 'p53'), 24)\n",
      "(('figure', '3)'), 24)\n",
      "(('(figure', '3a'), 24)\n",
      "(('cells', 'form'), 24)\n",
      "(('loop', 'mutations'), 24)\n",
      "(('elevated', 'expression'), 24)\n",
      "(('increased', 'ability'), 24)\n",
      "(('associated', 'amplification'), 24)\n",
      "(('pcr', '(rt-pcr)'), 24)\n",
      "(('published', 'reports,'), 24)\n",
      "(('extracellular', 'region'), 24)\n",
      "(('site', 'mutations'), 24)\n",
      "(('median', 'overall'), 24)\n",
      "(('showed', 'activity'), 24)\n",
      "(('indicate', 'poor'), 24)\n",
      "(('reach', 'statistical'), 24)\n",
      "(('also', 'introduced'), 24)\n",
      "(('1', 'hour'), 24)\n",
      "(('treated', 'increasing'), 24)\n",
      "(('confer', 'constitutive'), 24)\n",
      "(('structural', 'effects'), 24)\n",
      "(('small', 'in-frame'), 24)\n",
      "(('may', 'exist'), 24)\n",
      "(('(fig', 'structural'), 24)\n",
      "(('mutations', 'lie'), 24)\n",
      "(('used', 'clinical'), 24)\n",
      "(('reported', 'direct'), 24)\n",
      "(('methods', 'patients'), 24)\n",
      "(('directly', 'compare'), 24)\n",
      "(('five', 'patients'), 24)\n",
      "(('nucleic', 'acid'), 24)\n",
      "(('available', 'pathology'), 24)\n",
      "(('cases', 'showed'), 24)\n",
      "(('patients', 'found'), 24)\n",
      "(('lead', 'constitutive'), 24)\n",
      "(('enhanced', 'signaling'), 24)\n",
      "(('different', 'levels'), 24)\n",
      "(('analyzed', 'expression'), 24)\n",
      "(('positive', 'patients'), 24)\n",
      "(('(fig', 'normal'), 24)\n",
      "(('sequencing', 'confirmed'), 24)\n",
      "(('3', 'exon'), 24)\n",
      "(('al', 'results'), 24)\n",
      "(('myc', 'tag'), 24)\n",
      "(('10-fold', 'higher'), 24)\n",
      "(('conventional', 'chemotherapy'), 24)\n",
      "(('11', '1'), 24)\n",
      "(('cosmic', 'database'), 24)\n",
      "(('multiple', 'tumor'), 24)\n",
      "(('activity', 'absence'), 24)\n",
      "(('transfected', 'hek293'), 24)\n",
      "(('f', 'n'), 24)\n",
      "(('10', 'm'), 24)\n",
      "(('introduced', 'sequence'), 24)\n",
      "(('resistant', 'inhibition'), 24)\n",
      "(('present', 'within'), 24)\n",
      "(('adaptor', 'protein'), 24)\n",
      "(('salt', 'bridges'), 24)\n",
      "(('study', 'group'), 24)\n",
      "(('effect', 'mutant'), 24)\n",
      "(('also', 'transfected'), 24)\n",
      "(('reported', 'cases'), 24)\n",
      "(('number', 'changes'), 24)\n",
      "(('plasmid', 'using'), 24)\n",
      "(('samples', 'high'), 24)\n",
      "(('share', 'similar'), 24)\n",
      "(('stable', 'transfection'), 24)\n",
      "(('transactivation', 'activity'), 24)\n",
      "(('suggesting', 'may'), 24)\n",
      "(('two', 'samples'), 24)\n",
      "(('data', 'implicate'), 24)\n",
      "(('published', 'study'), 24)\n",
      "(('activated', 'wild-type'), 24)\n",
      "(('may', 'increase'), 24)\n",
      "(('human', 'prostate'), 24)\n",
      "(('patients', '(n'), 24)\n",
      "(('cells,', 'express'), 24)\n",
      "(('shown', 'affect'), 24)\n",
      "(('protein', 'present'), 24)\n",
      "(('gene', 'rearrangement'), 24)\n",
      "(('(fig', '3)'), 24)\n",
      "(('activity', 'similar'), 24)\n",
      "(('transgenic', 'mouse'), 24)\n",
      "(('marrow', 'samples'), 24)\n",
      "(('susceptibility', 'genes'), 24)\n",
      "(('reduce', 'stability'), 24)\n",
      "(('binding', 'wild-type'), 24)\n",
      "(('remaining', 'eight'), 24)\n",
      "(('a1,', 'online'), 24)\n",
      "(('take', 'place'), 24)\n",
      "(('could', 'inhibit'), 24)\n",
      "(('figure', '4b)'), 24)\n",
      "(('also', 'indicate'), 24)\n",
      "(('focal', 'adhesion'), 24)\n",
      "(('11', 'mutant'), 24)\n",
      "(('human', 'colon'), 24)\n",
      "(('microarray', 'data'), 24)\n",
      "(('mutation', 'smad4'), 24)\n",
      "(('mutation', 'effect'), 24)\n",
      "(('signaling', 'cells'), 24)\n",
      "(('results', 'showing'), 24)\n",
      "(('smaller', 'size'), 24)\n",
      "(('estrogen', 'receptor'), 24)\n",
      "(('however,', 'frequency'), 24)\n",
      "(('diffuse', 'large'), 24)\n",
      "(('encode', 'full-length'), 24)\n",
      "(('therapeutic', 'efficacy'), 24)\n",
      "(('expressing', 'empty'), 24)\n",
      "(('cells', 'increase'), 24)\n",
      "(('(figure', 's2)'), 24)\n",
      "(('transform', 'ba/f3'), 24)\n",
      "(('leukemia-like', 'disease'), 24)\n",
      "(('functional', 'validation'), 24)\n",
      "(('normal', 'dna'), 24)\n",
      "(('compared', 'activity'), 24)\n",
      "(('strong', 'family'), 24)\n",
      "(('mechanistic', 'understanding'), 24)\n",
      "(('three', 'main'), 24)\n",
      "(('functional', 'consequence'), 24)\n",
      "(('', '(pdgfra)'), 24)\n",
      "(('represent', 'passenger'), 24)\n",
      "(('patients', '(fig'), 24)\n",
      "(('higher', 'potency'), 24)\n",
      "(('followed', 'sequential'), 24)\n",
      "(('mutated', 'kit'), 24)\n",
      "(('activity', 'gist'), 24)\n",
      "(('extracts', 'analyzed'), 24)\n",
      "(('using', 'anti-kit'), 24)\n",
      "(('imatinib', 'mesylate,'), 24)\n",
      "(('imatinib', 'inhibits'), 24)\n",
      "(('kit', 'exons'), 24)\n",
      "(('might', 'effective'), 24)\n",
      "(('inhibited', 'imatinib'), 24)\n",
      "(('membranes', 'probed'), 24)\n",
      "(('cells', 'effects'), 24)\n",
      "(('kit', 'amplification'), 24)\n",
      "(('akt', 'activation'), 24)\n",
      "(('previously', 'unknown'), 24)\n",
      "(('data', 'revealed'), 24)\n",
      "(('product', 'corresponding'), 24)\n",
      "(('cells', 'display'), 24)\n",
      "(('pathway', 'inhibitors'), 24)\n",
      "(('proliferation', 'tumor'), 24)\n",
      "(('plasmid', 'expressing'), 24)\n",
      "(('structure,', 'predicted'), 24)\n",
      "(('wild-type', 'braf'), 24)\n",
      "(('conserved', 'region'), 24)\n",
      "(('dna', '(cdna)'), 24)\n",
      "(('england', 'biolabs,'), 24)\n",
      "(('domain', '(ctd)'), 24)\n",
      "(('squamous', 'carcinomas,'), 24)\n",
      "(('signaling', 'nsclc'), 24)\n",
      "(('higher', 'compared'), 24)\n",
      "(('signaling', 'tumor'), 24)\n",
      "(('cdna', 'transfected'), 24)\n",
      "(('factor-independent', 'growth'), 24)\n",
      "(('survival', 'despite'), 24)\n",
      "(('examined', 'vitro'), 24)\n",
      "(('provides', 'resistance'), 24)\n",
      "(('14', 'additional'), 24)\n",
      "(('cells', 'endogenous'), 24)\n",
      "(('could', 'involved'), 24)\n",
      "(('structure', 'elements'), 24)\n",
      "(('acids', '(aa)'), 24)\n",
      "(('mmr', 'activity'), 24)\n",
      "(('sh2', 'domain'), 24)\n",
      "(('ligand-independent', 'dimerization'), 24)\n",
      "(('ligand-independent', 'phosphorylation'), 24)\n",
      "(('inhibition', 'receptor'), 24)\n",
      "(('single-agent', 'activity'), 24)\n",
      "(('activity', 'ponatinib'), 24)\n",
      "(('refractory', 'patients'), 24)\n",
      "(('mutations', 'characteristic'), 24)\n",
      "(('level', 'mutants'), 24)\n",
      "(('flt3,', 'whereas'), 24)\n",
      "(('acids', '1'), 24)\n",
      "(('metastatic', 'melanoma,'), 24)\n",
      "(('50%', 'tumors'), 24)\n",
      "(('36,', '37,'), 24)\n",
      "(('tumors', '(17'), 24)\n",
      "(('a121v,', 'g129a'), 24)\n",
      "(('pten', 'mutants'), 24)\n",
      "(('9', 'none'), 24)\n",
      "(('c', 'terminal'), 24)\n",
      "(('four', 'vus'), 24)\n",
      "(('testing', 'brca1'), 24)\n",
      "(('novel', 'braf'), 24)\n",
      "(('brca2', 'missense'), 24)\n",
      "(('five', 'brca1'), 24)\n",
      "(('imatinib,', 'sunitinib'), 24)\n",
      "(('v560d', 'v654a'), 24)\n",
      "(('none', '(0/2)'), 24)\n",
      "(('study,', 'report'), 23)\n",
      "(('finger', 'domain'), 23)\n",
      "(('completely', 'conserved'), 23)\n",
      "(('analysis', 'genomic'), 23)\n",
      "(('deletion', 'mutants,'), 23)\n",
      "(('revealed', 'difference'), 23)\n",
      "(('enhanced', 'phosphorylation'), 23)\n",
      "(('effect', 'expression'), 23)\n",
      "((\"manufacturer's\", 'protocol'), 23)\n",
      "(('although', 'expression'), 23)\n",
      "(('investigate', 'effects'), 23)\n",
      "(('egf', 'stimulation'), 23)\n",
      "(('indicated', 'phosphorylation'), 23)\n",
      "(('32d', 'cells,'), 23)\n",
      "(('single', 'case'), 23)\n",
      "(('cells', 'dependent'), 23)\n",
      "(('similar', 'level'), 23)\n",
      "(('cho', 'cells'), 23)\n",
      "(('normal', 'karyotype'), 23)\n",
      "(('protein', '(gfp)'), 23)\n",
      "(('found', 'three'), 23)\n",
      "(('reading', 'frames'), 23)\n",
      "(('mutations', 'promote'), 23)\n",
      "(('although', 'mutations'), 23)\n",
      "(('similar', 'previously'), 23)\n",
      "(('lower', 'frequency'), 23)\n",
      "(('internal', 'tandem'), 23)\n",
      "(('overall', 'mutation'), 23)\n",
      "(('significant', 'change'), 23)\n",
      "(('0', '5'), 23)\n",
      "(('majority', 'cases'), 23)\n",
      "(('novel', 'mechanism'), 23)\n",
      "(('older', 'patients'), 23)\n",
      "(('mutations', 'conferred'), 23)\n",
      "(('mutations', 'although'), 23)\n",
      "(('obtained', 'cell'), 23)\n",
      "(('reported', 'occur'), 23)\n",
      "(('mutations', 'provide'), 23)\n",
      "(('activity', 'human'), 23)\n",
      "(('evidence', 'support'), 23)\n",
      "(('published', 'studies'), 23)\n",
      "(('human', 'genome'), 23)\n",
      "(('whole', 'genome'), 23)\n",
      "(('cells', 'demonstrated'), 23)\n",
      "(('head', 'neck,'), 23)\n",
      "(('including', 'two'), 23)\n",
      "(('amplification', 'chromosome'), 23)\n",
      "(('two', 'residues'), 23)\n",
      "(('sirnas', 'targeting'), 23)\n",
      "(('control', '(figure'), 23)\n",
      "(('markedly', 'low'), 23)\n",
      "(('less', '10%'), 23)\n",
      "(('least', 'part,'), 23)\n",
      "(('detected', 'patient'), 23)\n",
      "(('tumors', 'developed'), 23)\n",
      "(('deletion', 'chromosome'), 23)\n",
      "(('growth', 'differentiation'), 23)\n",
      "(('gastric', 'cancers'), 23)\n",
      "(('effects', 'cell'), 23)\n",
      "(('observed', 'significant'), 23)\n",
      "(('(17)', 'interestingly,'), 23)\n",
      "(('cancer', 'gene'), 23)\n",
      "(('al', 'interestingly,'), 23)\n",
      "(('conserved', 'across'), 23)\n",
      "(('mapk', 'pathways'), 23)\n",
      "(('aggressive', 'tumor'), 23)\n",
      "(('tumors', 'significantly'), 23)\n",
      "(('papillary', 'renal'), 23)\n",
      "(('observations', 'made'), 23)\n",
      "(('cancer,', 'including'), 23)\n",
      "(('activation', '(figure'), 23)\n",
      "(('(fig', 'deletions'), 23)\n",
      "(('(figure', '1a,'), 23)\n",
      "(('mutation', 'alone'), 23)\n",
      "(('heterozygosity', '(loh)'), 23)\n",
      "(('necessary', 'binding'), 23)\n",
      "(('variant', 'also'), 23)\n",
      "(('mutations', 'appeared'), 23)\n",
      "(('cambridge,', 'ma)'), 23)\n",
      "(('express', 'mutant'), 23)\n",
      "(('inhibition', 'tyrosine'), 23)\n",
      "(('(fig', '2a)'), 23)\n",
      "(('spectrum', 'mutations'), 23)\n",
      "(('substrate', 'binding'), 23)\n",
      "(('experimental', 'evidence'), 23)\n",
      "(('developing', 'cancer'), 23)\n",
      "(('results', 'different'), 23)\n",
      "(('mutator', 'phenotype'), 23)\n",
      "(('therapeutic', 'intervention'), 23)\n",
      "(('higher', 'expression'), 23)\n",
      "(('underlying', 'mechanism'), 23)\n",
      "(('decrease', 'expression'), 23)\n",
      "(('rapid', 'development'), 23)\n",
      "(('show', 'similar'), 23)\n",
      "(('first', 'step'), 23)\n",
      "(('higher', 'affinity'), 23)\n",
      "(('(fig', 'two'), 23)\n",
      "(('sequencing', 'identified'), 23)\n",
      "(('identified', 'heterozygous'), 23)\n",
      "(('protein', 'wild-type'), 23)\n",
      "(('kinetic', 'parameters'), 23)\n",
      "(('due', 'low'), 23)\n",
      "(('proteins', 'could'), 23)\n",
      "(('patients', '1'), 23)\n",
      "(('methylation', 'status'), 23)\n",
      "(('also', 'frequently'), 23)\n",
      "(('gene', 'may'), 23)\n",
      "(('shown', 'associated'), 23)\n",
      "(('mutation', 'caused'), 23)\n",
      "(('nucleotide', 'changes'), 23)\n",
      "(('soft', 'agar,'), 23)\n",
      "(('expression', 'endogenous'), 23)\n",
      "(('mutant', 'proteins,'), 23)\n",
      "(('4b)', 'results'), 23)\n",
      "(('immunofluorescence', 'staining'), 23)\n",
      "(('mutant', 'mice'), 23)\n",
      "(('conditions', '(fig'), 23)\n",
      "(('expressed', 'early'), 23)\n",
      "(('shrna', 'knockdown'), 23)\n",
      "(('lethally', 'irradiated'), 23)\n",
      "(('normal', 'levels'), 23)\n",
      "(('rna-seq', 'data'), 23)\n",
      "(('genes', 'significantly'), 23)\n",
      "(('investigated', 'two'), 23)\n",
      "(('steric', 'hindrance'), 23)\n",
      "(('patients', 'three'), 23)\n",
      "(('(b)', 'western'), 23)\n",
      "(('therapeutic', 'approaches'), 23)\n",
      "(('frequent', 'mutation'), 23)\n",
      "(('significantly', 'mutated'), 23)\n",
      "(('mutations', 'additional'), 23)\n",
      "(('whereas', 'remaining'), 23)\n",
      "(('al', 'recently'), 23)\n",
      "(('untreated', 'patients'), 23)\n",
      "(('negative', 'predictive'), 23)\n",
      "(('better', 'overall'), 23)\n",
      "(('regression', 'model'), 23)\n",
      "(('kinase', 'dead'), 23)\n",
      "(('induced', 'phosphorylation'), 23)\n",
      "(('g719s', 'l858r'), 23)\n",
      "(('induced', 'cells'), 23)\n",
      "(('vectors', '293t'), 23)\n",
      "(('found', '13'), 23)\n",
      "(('mutations', 'inhibited'), 23)\n",
      "(('targets', 'egfr'), 23)\n",
      "(('within', 'egfr'), 23)\n",
      "(('effective', 'treating'), 23)\n",
      "(('egfr', 'two'), 23)\n",
      "(('differential', 'sensitivities'), 23)\n",
      "(('access', 'program'), 23)\n",
      "(('another', 'egfr'), 23)\n",
      "(('expanded', 'access'), 23)\n",
      "(('may', 'essential'), 23)\n",
      "(('acquired', 'egfr'), 23)\n",
      "(('diagnostic', 'testing'), 23)\n",
      "(('result', 'activated'), 23)\n",
      "(('paez', 'et'), 23)\n",
      "(('growth,', 'whereas'), 23)\n",
      "(('abundantly', 'expressed'), 23)\n",
      "(('mutational', 'profiling'), 23)\n",
      "(('would', 'indicate'), 23)\n",
      "(('2', '25'), 23)\n",
      "(('neither', 'wild-type'), 23)\n",
      "(('quantitatively', 'different'), 23)\n",
      "(('response', 'may'), 23)\n",
      "(('tumors,', 'identified'), 23)\n",
      "(('histologic', 'types'), 23)\n",
      "(('(nsclc)', 'patients'), 23)\n",
      "(('remaining', 'cell'), 23)\n",
      "(('patients', 'could'), 23)\n",
      "(('vertebrate', 'species'), 23)\n",
      "(('4figure', '4'), 23)\n",
      "(('related', 'kinase'), 23)\n",
      "(('greater', 'ability'), 23)\n",
      "(('failed', 'show'), 23)\n",
      "(('mutation', 'response'), 23)\n",
      "(('detected', 'immunohistochemistry'), 23)\n",
      "(('14', 'tumors'), 23)\n",
      "(('(exons', '18'), 23)\n",
      "(('inhibitors', 'yet'), 23)\n",
      "(('selected', 'tumor'), 23)\n",
      "(('carcinoma', 'tissue'), 23)\n",
      "(('specific', 'regions'), 23)\n",
      "(('exon', 'sequencing'), 23)\n",
      "(('suggest', 'egfr'), 23)\n",
      "(('inhibited', 'egfr'), 23)\n",
      "(('constitutive,', 'ligand-independent'), 23)\n",
      "(('include:', '(i)'), 23)\n",
      "(('egfr', 'alterations'), 23)\n",
      "(('causes', 'tumor'), 23)\n",
      "(('whether', 'resistance'), 23)\n",
      "(('thus,', 'egfr'), 23)\n",
      "(('pathway', 'tumor'), 23)\n",
      "(('require', 'investigation'), 23)\n",
      "(('egfr', 'using'), 23)\n",
      "(('elucidate', 'molecular'), 23)\n",
      "(('egfr', 'kras'), 23)\n",
      "(('mutations', 'exclusively'), 23)\n",
      "(('(13,', '14,'), 23)\n",
      "(('survival', 'gefitinib-treated'), 23)\n",
      "(('mutations,', 'well'), 23)\n",
      "(('gene', 'genomic'), 23)\n",
      "(('gene', 'investigated'), 23)\n",
      "(('mutations', 'furthermore,'), 23)\n",
      "(('2', 'independent'), 23)\n",
      "(('institutional', 'review'), 23)\n",
      "(('reversible', 'egfr'), 23)\n",
      "(('copy', 'number,'), 23)\n",
      "(('cases', 'egfr'), 23)\n",
      "(('mutations', 'position'), 23)\n",
      "(('region', 'oncogenic'), 23)\n",
      "(('thus,', 'analysis'), 23)\n",
      "(('egfr', 'erbb2'), 23)\n",
      "(('three', 'egfr'), 23)\n",
      "(('11', 'deletions'), 23)\n",
      "(('thus', 'explaining'), 23)\n",
      "(('bearing', 'egfr'), 23)\n",
      "(('2,', '4,'), 23)\n",
      "(('patients', 'used'), 23)\n",
      "(('mutants', 'oncogenic'), 23)\n",
      "(('(in', 'case'), 23)\n",
      "(('response', 'erlotinib'), 23)\n",
      "(('resistant', 'gefitinib'), 23)\n",
      "(('gefitinib-treated', 'patients'), 23)\n",
      "(('often', 'associated'), 23)\n",
      "(('tkis', 'egfr'), 23)\n",
      "(('studied', 'using'), 23)\n",
      "(('antitumor', 'effects'), 23)\n",
      "(('mutations', 'relative'), 23)\n",
      "(('number', 'analysis'), 23)\n",
      "(('mutations,', 'reported'), 23)\n",
      "(('recently', 'showed'), 23)\n",
      "(('mutations', 'predictive'), 23)\n",
      "(('apoptosis', 'cells'), 23)\n",
      "(('mainly', 'consists'), 23)\n",
      "(('4', 'exons'), 23)\n",
      "(('inhibitor', 'sensitivity'), 23)\n",
      "(('value', 'egfr'), 23)\n",
      "(('stage', 'iv'), 23)\n",
      "(('results', 'construction'), 23)\n",
      "(('stat3', 'phosphorylation'), 23)\n",
      "(('t790m', 'mutation'), 23)\n",
      "(('patients', 'heterozygous'), 23)\n",
      "(('pd,', 'progressive'), 23)\n",
      "(('although', 'data'), 23)\n",
      "(('transfected', 'cell'), 23)\n",
      "(('phosphorylation', 'wild-type'), 23)\n",
      "(('coding', 'sequences'), 23)\n",
      "(('expression', 'tumor'), 23)\n",
      "(('receptor', '2'), 23)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('signaling', 'cascade'), 23)\n",
      "(('recently,', 'shown'), 23)\n",
      "(('brain', 'tumors'), 23)\n",
      "(('mutation', 'spectrum'), 23)\n",
      "(('kinase', 'conformation'), 23)\n",
      "(('mutants', 'containing'), 23)\n",
      "(('cells,', 'thus'), 23)\n",
      "(('purchased', 'santa'), 23)\n",
      "(('factorindependent', 'proliferation'), 23)\n",
      "(('oncogenic', 'properties'), 23)\n",
      "(('gene', '(fig'), 23)\n",
      "(('site', 'within'), 23)\n",
      "(('inhibitor', '(fig'), 23)\n",
      "(('10%', '15%'), 23)\n",
      "(('among', 'these,'), 23)\n",
      "(('study,', 'analyzed'), 23)\n",
      "(('effective', 'patients'), 23)\n",
      "(('could', 'explain'), 23)\n",
      "(('increased', 'tumor'), 23)\n",
      "(('growth', 'compared'), 23)\n",
      "(('k-ras', 'mutations'), 23)\n",
      "(('molecular', 'characteristics'), 23)\n",
      "(('constitutive', 'autophosphorylation'), 23)\n",
      "(('lower', 'ic50'), 23)\n",
      "(('nanomolar', 'range'), 23)\n",
      "(('exposed', 'surface'), 23)\n",
      "(('immortalized', 'human'), 23)\n",
      "(('primary', 'antibody'), 23)\n",
      "(('presence', 'two'), 23)\n",
      "(('recurrent', 'mutation'), 23)\n",
      "(('overall,', 'data'), 23)\n",
      "(('nucleotide', 'substitution'), 23)\n",
      "(('binding', 'assay'), 23)\n",
      "(('secondary', 'antibody'), 23)\n",
      "(('cells', 'phosphorylation'), 23)\n",
      "(('mutations', 'render'), 23)\n",
      "(('predicted', 'structural'), 23)\n",
      "(('mutations', 'go'), 23)\n",
      "(('expressed', 'either'), 23)\n",
      "(('upon', 'stimulation'), 23)\n",
      "(('used', 'previous'), 23)\n",
      "(('two', 'salt'), 23)\n",
      "(('determine', 'effect'), 23)\n",
      "(('taking', 'account'), 23)\n",
      "(('data', 'agreement'), 23)\n",
      "(('activation', 'may'), 23)\n",
      "(('analysis', 'gene'), 23)\n",
      "(('resulting', 'increased'), 23)\n",
      "(('among', 'different'), 23)\n",
      "(('produced', 'similar'), 23)\n",
      "(('control', 'plasmid'), 23)\n",
      "(('encoding', 'amino'), 23)\n",
      "(('different', 'domains'), 23)\n",
      "(('study,', 'demonstrate'), 23)\n",
      "(('resolution', 'image'), 23)\n",
      "(('previous', 'observations'), 23)\n",
      "(('snp', 'array'), 23)\n",
      "(('showed', 'three'), 23)\n",
      "(('plays', 'central'), 23)\n",
      "(('independent', 'experiments'), 23)\n",
      "(('structure', 'function'), 23)\n",
      "(('failed', 'stimulate'), 23)\n",
      "(('similar', 'increase'), 23)\n",
      "(('cells', 'therefore,'), 23)\n",
      "(('resistance', 'crizotinib'), 23)\n",
      "(('il-3', 'independence'), 23)\n",
      "(('intervening', 'sequence'), 23)\n",
      "(('tumors', 'used'), 23)\n",
      "(('kinase', '(rtk),'), 23)\n",
      "(('endometrial', 'cancer,'), 23)\n",
      "(('next', 'investigated'), 23)\n",
      "(('genomic', 'region'), 23)\n",
      "(('activity', 'assay'), 23)\n",
      "(('l1', 'loop'), 23)\n",
      "(('tumor', 'derived'), 23)\n",
      "(('familial', 'cancer'), 23)\n",
      "(('translocate', 'nucleus'), 23)\n",
      "(('mutation', 'study'), 23)\n",
      "(('however,', 'mutations'), 23)\n",
      "(('hydroxyl', 'group'), 23)\n",
      "(('four', 'mutant'), 23)\n",
      "(('comparative', 'analysis'), 23)\n",
      "(('human', 'diseases'), 23)\n",
      "(('individuals', 'carry'), 23)\n",
      "(('determine', 'effects'), 23)\n",
      "(('significant', 'resistance'), 23)\n",
      "(('proliferation', 'invasion'), 23)\n",
      "(('loss', 'interaction'), 23)\n",
      "(('wild-type', '(wt),'), 23)\n",
      "(('nervous', 'system,'), 23)\n",
      "(('signaling', 'promotes'), 23)\n",
      "(('lysates', 'transfected'), 23)\n",
      "(('high', 'number'), 23)\n",
      "(('relative', 'normal'), 23)\n",
      "(('review,', 'see'), 23)\n",
      "(('(for', 'review,'), 23)\n",
      "(('9,', '10,'), 23)\n",
      "(('hydrophobic', 'interactions'), 23)\n",
      "(('tested', 'study'), 23)\n",
      "(('newly', 'synthesized'), 23)\n",
      "(('using', 'phusion'), 23)\n",
      "(('raise', 'possibility'), 23)\n",
      "(('entire', 'set'), 23)\n",
      "(('residue', 'changes'), 23)\n",
      "(('kit', '(applied'), 23)\n",
      "(('however,', 'none'), 23)\n",
      "(('reverse', 'transcribed'), 23)\n",
      "(('found', 'lower'), 23)\n",
      "(('l', 'n'), 23)\n",
      "(('activation', 'thus,'), 23)\n",
      "(('listed', 'supplementary'), 23)\n",
      "(('imatinib', 'crenolanib'), 23)\n",
      "(('pdgfra', 'imatinib'), 23)\n",
      "(('mutations', 'acute'), 23)\n",
      "(('activating', 'point'), 23)\n",
      "(('jm', 'domain,'), 23)\n",
      "(('pdgfra', 'mutations,'), 23)\n",
      "(('potency', 'imatinib'), 23)\n",
      "(('gist', 'primary'), 23)\n",
      "(('potent', 'imatinib'), 23)\n",
      "(('development', 'even'), 23)\n",
      "(('9', '4'), 23)\n",
      "(('sequential', 'immunoblotting'), 23)\n",
      "(('wild-type', 'gists'), 23)\n",
      "(('cancer', 'institute,'), 23)\n",
      "(('d842v', 'mutation'), 23)\n",
      "(('study', 'crenolanib'), 23)\n",
      "(('imatinib', 'binds'), 23)\n",
      "(('located', 'activation'), 23)\n",
      "(('gist', 'imatinib'), 23)\n",
      "(('mutations', 'amplifications'), 23)\n",
      "(('amplification', 'status'), 23)\n",
      "(('previously', 'classified'), 23)\n",
      "(('domain', 'flt3'), 23)\n",
      "(('similar', 'described'), 23)\n",
      "(('variants,', 'although'), 23)\n",
      "(('neutral', 'variant,'), 23)\n",
      "(('tested', 'vitro'), 23)\n",
      "(('braf', 'inhibitor'), 23)\n",
      "(('pathways', 'found'), 23)\n",
      "(('15', '30'), 23)\n",
      "(('met', 'kinase'), 23)\n",
      "(('immunoprecipitated', 'anti-ha'), 23)\n",
      "(('activation', 'could'), 23)\n",
      "(('author', 'information'), 23)\n",
      "(('introduction', 'results'), 23)\n",
      "(('results', 'discussion'), 23)\n",
      "(('information', 'supplementary'), 23)\n",
      "(('discussion', 'methods'), 23)\n",
      "(('abstract', 'introduction'), 23)\n",
      "(('acknowledgments', 'author'), 23)\n",
      "(('study', '(fig'), 23)\n",
      "(('transcription', 'target'), 23)\n",
      "(('interaction', 'sites'), 23)\n",
      "(('assays', '(fig'), 23)\n",
      "(('scid', 'mice'), 23)\n",
      "(('indicating', 'activation'), 23)\n",
      "(('kinase', 'activation,'), 23)\n",
      "(('variants', 'reported'), 23)\n",
      "(('variant', 'predicted'), 23)\n",
      "(('mutations', 'protein'), 23)\n",
      "(('activity', 'downstream'), 23)\n",
      "(('cluster', 'region'), 23)\n",
      "(('different', 'lengths'), 23)\n",
      "(('two', 'deleterious'), 23)\n",
      "(('tandem', 'duplications'), 23)\n",
      "(('wild-type', 'fgfr2'), 23)\n",
      "(('molecular', 'brake'), 23)\n",
      "(('and,', 'particular,'), 23)\n",
      "(('relative', 'proportion'), 23)\n",
      "(('mutations', 'interestingly,'), 23)\n",
      "(('alk', 'phosphorylation'), 23)\n",
      "(('alk', 'protein'), 23)\n",
      "(('mutant', 'alk'), 23)\n",
      "(('tumor-associated', 'mutations'), 23)\n",
      "(('flt3', 'inhibition'), 23)\n",
      "(('aml', 'cells'), 23)\n",
      "(('flt3', 'inhibitors'), 23)\n",
      "(('ponatinib', 'patients'), 23)\n",
      "(('flt3', 'activation'), 23)\n",
      "(('favors', 'dimer'), 23)\n",
      "(('prophylactic', 'surgery'), 23)\n",
      "(('pathogenicity', 'vus'), 23)\n",
      "(('functional', 'tests'), 23)\n",
      "(('t131l)', 'fully'), 23)\n",
      "(('mutations', '(h123y,'), 23)\n",
      "(('k125m,', 'a126g'), 23)\n",
      "(('(h123y,', 'c124n,'), 23)\n",
      "(('(a121v,', 'i122v,'), 23)\n",
      "(('affected', 'partially'), 23)\n",
      "(('r130a,', 'r130k,'), 23)\n",
      "(('g129d,', 'r130a,'), 23)\n",
      "(('a126g', 'k128r)'), 23)\n",
      "(('groups', 'affected'), 23)\n",
      "(('(i122l,', 'k125l,'), 23)\n",
      "(('partially', '(i122l,'), 23)\n",
      "(('abrogated', 'pip3'), 23)\n",
      "(('t131s)', '(fig'), 23)\n",
      "(('g127n,', 'k128t,'), 23)\n",
      "(('yeast,', 'whereas'), 23)\n",
      "(('activity', '(a121v,'), 23)\n",
      "(('k128t,', 'g129d,'), 23)\n",
      "(('r130k,', 't131i'), 23)\n",
      "(('a126d,', 'g127n,'), 23)\n",
      "(('pten', 'yeast,'), 23)\n",
      "(('c124n,', 'a126d,'), 23)\n",
      "(('k128r)', 'affect'), 23)\n",
      "(('pten', 'pip3'), 23)\n",
      "(('i122v,', 'k125r,'), 23)\n",
      "(('k125r,', 'g129a'), 23)\n",
      "(('k125l,', 'k125m,'), 23)\n",
      "(('g129a', 't131s)'), 23)\n",
      "(('t131i', 't131l)'), 23)\n",
      "(('r130g,', 'y155c,'), 23)\n",
      "(('phosphatase', 'activities'), 23)\n",
      "(('activation', 'braf'), 23)\n",
      "(('(20)', 'using'), 23)\n",
      "(('introduced', 'splicing'), 23)\n",
      "(('brca1', 'brca2,'), 23)\n",
      "(('showing', 'brca1'), 23)\n",
      "(('one', 'brca1'), 23)\n",
      "(('tumors', 'brca1'), 23)\n",
      "(('brca1', 'nine'), 23)\n",
      "(('one', 'brca2'), 23)\n",
      "(('diagnosis', 'breast'), 23)\n",
      "(('wild-type', 'brca2'), 23)\n",
      "(('criteria', 'included'), 23)\n",
      "(('gists', 'secondary'), 23)\n",
      "(('y823d', '(1/1)'), 23)\n",
      "(('awd', '5'), 23)\n",
      "(('59/m', '400'), 23)\n",
      "(('stomach', 'sd'), 23)\n",
      "(('54/m', '400'), 23)\n",
      "(('dod', '30'), 23)\n",
      "(('43/m', '400'), 23)\n",
      "(('20', '59/f'), 23)\n",
      "(('(1/1)', 'awd'), 23)\n",
      "(('sd', '403'), 23)\n",
      "(('10', '49/m'), 23)\n",
      "(('del', 'kpmyevqwk550-558'), 23)\n",
      "(('pmye551-554', 'none'), 23)\n",
      "(('del', 'wkvve557-561'), 23)\n",
      "(('411', '1,028'), 23)\n",
      "(('pr', '275'), 23)\n",
      "(('vv559-560', 'none'), 23)\n",
      "(('pr', '613'), 23)\n",
      "(('141', '499'), 23)\n",
      "(('insertion', 'k558np'), 23)\n",
      "(('pr', '952'), 23)\n",
      "(('del', 'wk557-558'), 23)\n",
      "(('delyevqwk553-558', 'kit'), 23)\n",
      "(('275', '252'), 23)\n",
      "(('k558np', 'none'), 23)\n",
      "(('v559d', 'none'), 23)\n",
      "(('44/f', '400'), 23)\n",
      "(('awd', '10'), 23)\n",
      "(('del', 'myevqw552-557'), 23)\n",
      "(('23', '70/f'), 23)\n",
      "(('63/m', '400'), 23)\n",
      "(('252', 'kit'), 23)\n",
      "(('(mutations/samples)', 'status'), 23)\n",
      "(('pr', '1,609'), 23)\n",
      "(('400', 'stomach'), 23)\n",
      "(('883', 'kit'), 23)\n",
      "(('400', 'iab'), 23)\n",
      "(('1,580', 'kit'), 23)\n",
      "(('pr', '58'), 23)\n",
      "(('pr', '411'), 23)\n",
      "(('awd', '28'), 23)\n",
      "(('13', 'insertion'), 23)\n",
      "(('21', '54/m'), 23)\n",
      "(('53/f', '400'), 23)\n",
      "(('17', 'c809g'), 23)\n",
      "(('d816h', '(2/4)'), 23)\n",
      "(('(0/2)', 'dod'), 23)\n",
      "(('2', '53/f'), 23)\n",
      "(('(2/4)', 'dod'), 23)\n",
      "(('16', '52/m'), 23)\n",
      "(('pr', '141'), 23)\n",
      "(('(days)', 'ttf'), 23)\n",
      "(('w557g', 'kit'), 23)\n",
      "(('25/m', '600'), 23)\n",
      "(('952', '366'), 23)\n",
      "(('pr', '778'), 23)\n",
      "(('mutation', '(mutations/samples)'), 23)\n",
      "(('v561d', 'pdgfra'), 23)\n",
      "(('treatment', 'failure;'), 23)\n",
      "(('498', 'kit'), 23)\n",
      "(('4', '25/m'), 23)\n",
      "(('del', 'wkvv557-560f'), 23)\n",
      "(('67/m', '400'), 23)\n",
      "(('167', '462'), 23)\n",
      "(('v560d', 'np'), 23)\n",
      "(('591', '504'), 23)\n",
      "(('77/m', '400'), 23)\n",
      "(('del', 'kv558-559'), 23)\n",
      "(('y823d', '(1/4)'), 23)\n",
      "(('sd', '155'), 23)\n",
      "(('si', 'pr'), 23)\n",
      "(('104', '498'), 23)\n",
      "(('27', '246'), 23)\n",
      "(('1,297', 'kit'), 23)\n",
      "(('(days)', 'primary'), 23)\n",
      "(('(4/4)', 'awd'), 23)\n",
      "(('(mg)', 'primary'), 23)\n",
      "(('ttf,', 'time'), 23)\n",
      "(('6', '54/m'), 23)\n",
      "(('pr', '1,267'), 23)\n",
      "(('747', 'kit'), 23)\n",
      "(('717', 'kit'), 23)\n",
      "(('14', '63/m'), 23)\n",
      "(('status', '1'), 23)\n",
      "(('64/f', '400'), 23)\n",
      "(('pr', '1,037'), 23)\n",
      "(('insertion', 'kit'), 23)\n",
      "(('45/f', '600'), 23)\n",
      "(('71/m', '600'), 23)\n",
      "(('deletion;', 'awd,'), 23)\n",
      "(('65/m', '400'), 23)\n",
      "(('497', 'pdgfra'), 23)\n",
      "(('dod', '23'), 23)\n",
      "(('27', '53/f'), 23)\n",
      "(('34', '77/m'), 23)\n",
      "(('awd', '16'), 23)\n",
      "(('63/m', '600'), 23)\n",
      "(('dod', '34'), 23)\n",
      "(('600', 'stomach'), 23)\n",
      "(('awd', '2'), 23)\n",
      "(('response;', 'del,'), 23)\n",
      "(('awd', '18'), 23)\n",
      "(('d842v', '(1/1)'), 23)\n",
      "(('11', 'delkpmyevqwk550-558'), 23)\n",
      "(('l576p', 'kit'), 23)\n",
      "(('iab', 'sd'), 23)\n",
      "(('1,564', 'kit'), 23)\n",
      "(('del', 'kpmyevqw550-557fl'), 23)\n",
      "(('dod', '12'), 23)\n",
      "(('507', '747'), 23)\n",
      "(('kpmyevqwk550-558', 'none'), 23)\n",
      "(('62/f', '600'), 23)\n",
      "(('awd', 'abbreviations:'), 23)\n",
      "(('dod', '8'), 23)\n",
      "(('pr', '2,120'), 23)\n",
      "(('v654a', '(1/1)'), 23)\n",
      "(('v560g', 'kit'), 23)\n",
      "(('d820g', '(1/2)'), 23)\n",
      "(('613', '417'), 23)\n",
      "(('pr', '138'), 23)\n",
      "(('failure;', 'f,'), 23)\n",
      "(('female;', 'si,'), 23)\n",
      "(('pr', '348'), 23)\n",
      "(('del', '554-558evqwk'), 23)\n",
      "(('(2/2)', 'dod'), 23)\n",
      "(('v654a', '(2/2)'), 23)\n",
      "(('237', 'kit'), 23)\n",
      "(('intra-abdominal;', 'np,'), 23)\n",
      "(('v654a', '(3/4)'), 23)\n",
      "(('pmye551-554', 'kit'), 23)\n",
      "(('400', 'si'), 23)\n",
      "(('pr', '2,329'), 23)\n",
      "(('17', '54/f'), 23)\n",
      "(('643', '1,564'), 23)\n",
      "(('intestine;', 'pr,'), 23)\n",
      "(('pr', '448'), 23)\n",
      "(('kpmyevqwk550-558', 'kit'), 23)\n",
      "(('v654a', '(3/3)'), 23)\n",
      "(('155', '215'), 23)\n",
      "(('54/f', '400'), 23)\n",
      "(('k642e', 'kit'), 23)\n",
      "(('512', 'kit'), 23)\n",
      "(('60', '384'), 23)\n",
      "(('15', '67/m'), 23)\n",
      "(('pr', '507'), 23)\n",
      "(('17', 'n822h'), 23)\n",
      "(('del', 'pmye551-554'), 23)\n",
      "(('(1/2)', 'dod'), 23)\n",
      "(('np', 'awd'), 23)\n",
      "(('9', '58/f'), 23)\n",
      "(('11', 'v560g'), 23)\n",
      "(('60/m', '400'), 23)\n",
      "(('58', '1,297'), 23)\n",
      "(('(0/1)', 'dod'), 23)\n",
      "(('846', 'kit'), 23)\n",
      "(('age/sex', 'initial'), 23)\n",
      "(('del', 'kv558-559n'), 23)\n",
      "(('dod', '17'), 23)\n",
      "(('pr', '122'), 23)\n",
      "(('dod', '20'), 23)\n",
      "(('2,921', '883'), 23)\n",
      "(('50/f', '400'), 23)\n",
      "(('19', '59/m'), 23)\n",
      "(('33', '60/m'), 23)\n",
      "(('600', 'iab'), 23)\n",
      "(('7', '43/m'), 23)\n",
      "(('39/m', '400'), 23)\n",
      "(('2,329', '379'), 23)\n",
      "(('3,615', '349'), 23)\n",
      "(('wk557-558', 'kit'), 23)\n",
      "(('del', 'evqwk554-558'), 23)\n",
      "(('dod', '32'), 23)\n",
      "(('disease;', 'dod,'), 23)\n",
      "(('11', '50/f'), 23)\n",
      "(('881', 'kit'), 23)\n",
      "(('2,120', '1,580'), 23)\n",
      "(('wkvv557-560f', 'kit'), 23)\n",
      "(('29', '60/m'), 23)\n",
      "(('dod', '33'), 23)\n",
      "(('wkvve557-561', 'kit'), 23)\n",
      "(('none', '(0/1)'), 23)\n",
      "(('del', 'vv559-560'), 23)\n",
      "(('n822k', '(4/4)'), 23)\n",
      "(('(1/4)', 'dod'), 23)\n",
      "(('766', 'kit'), 23)\n",
      "(('y823d', '(1/3)'), 23)\n",
      "(('384', 'kit'), 23)\n",
      "(('awd', '11'), 23)\n",
      "(('379', 'kit'), 23)\n",
      "(('pr', '380'), 23)\n",
      "(('dod', '21'), 23)\n",
      "(('701', 'kit'), 23)\n",
      "(('abbreviations:', 'ttf,'), 23)\n",
      "(('587', 'kit'), 23)\n",
      "(('5', '39/m'), 23)\n",
      "(('215', 'none'), 23)\n",
      "(('28', '63/m'), 23)\n",
      "(('164', '701'), 23)\n",
      "(('del', 'evqwkvveeingnnyvyi554-571'), 23)\n",
      "(('pr', '2,921'), 23)\n",
      "(('dod', '3'), 23)\n",
      "(('n822h', 'kit'), 23)\n",
      "(('del', 'wkvv557-560c'), 23)\n",
      "(('(1/1)', 'dod'), 23)\n",
      "(('554-558evqwk', 'none'), 23)\n",
      "(('disease;', 'm,'), 23)\n",
      "(('v560d', 'kit'), 23)\n",
      "(('38/m', '400'), 23)\n",
      "(('49/m', '600'), 23)\n",
      "(('dod', '31'), 23)\n",
      "(('patient', 'age/sex'), 23)\n",
      "(('8', '49/m'), 23)\n",
      "(('n822k', '(1/1)'), 23)\n",
      "(('delkpmyevqwk550-558', 'kit'), 23)\n",
      "(('13', '46/m'), 23)\n",
      "(('dod', '9'), 23)\n",
      "(('500', 'kit'), 23)\n",
      "(('33', '416'), 23)\n",
      "(('1,037', '512'), 23)\n",
      "(('504', 'kit'), 23)\n",
      "(('kv558-559n', 'kit'), 23)\n",
      "(('imatinib', '(days)'), 23)\n",
      "(('dod', '25'), 23)\n",
      "(('pr', '125'), 23)\n",
      "(('iab,', 'intra-abdominal;'), 23)\n",
      "(('iab', 'pr'), 23)\n",
      "(('122', '237'), 23)\n",
      "(('dod', '22'), 23)\n",
      "(('awd', '24'), 23)\n",
      "(('d820g', '(1/1)'), 23)\n",
      "(('462', 'kit'), 23)\n",
      "(('myevqw552-557', 'kit'), 23)\n",
      "(('insertion', 'v654a'), 23)\n",
      "(('alive', 'disease;'), 23)\n",
      "(('d816h', '(1/4)'), 23)\n",
      "(('pr', '60'), 23)\n",
      "(('448', '881'), 23)\n",
      "(('male;', 'sd,'), 23)\n",
      "(('31', '71/m'), 23)\n",
      "(('417', 'kit'), 23)\n",
      "(('ttf', '(days)'), 23)\n",
      "(('14', 't6701'), 23)\n",
      "(('1,028', 'kit'), 23)\n",
      "(('403', '508'), 23)\n",
      "(('(2/3)', 'dod'), 23)\n",
      "(('46/m', '600'), 23)\n",
      "(('32', '38/m'), 23)\n",
      "(('1,609', '584'), 23)\n",
      "(('evqwk554-558', 'kit'), 23)\n",
      "(('kv558-559', 'none'), 23)\n",
      "(('awd,', 'alive'), 23)\n",
      "(('c809g', '(2/2)'), 23)\n",
      "(('930', 'kit'), 23)\n",
      "(('pr', '104'), 23)\n",
      "(('1,267', '766'), 23)\n",
      "(('416', 'kit'), 23)\n",
      "(('508', 'none'), 23)\n",
      "(('pr', '3,615'), 23)\n",
      "(('246', 'kit'), 23)\n",
      "(('stomach', 'pr'), 23)\n",
      "(('awd', '26'), 23)\n",
      "(('evqwkvveeingnnyvyi554-571', 'none'), 23)\n",
      "(('k642e', 'exon'), 23)\n",
      "(('si', 'sd'), 23)\n",
      "(('imatinib', 'dose'), 23)\n",
      "(('348', '717'), 23)\n",
      "(('pr', '1,252'), 23)\n",
      "(('small', 'intestine;'), 23)\n",
      "(('(1/3)', 'dod'), 23)\n",
      "(('50/m', '400'), 23)\n",
      "(('site', 'best'), 23)\n",
      "(('584', 'kit'), 23)\n",
      "(('24', '45/f'), 23)\n",
      "(('pr', '27'), 23)\n",
      "(('t6701', '(2/3)'), 23)\n",
      "(('pr', '164'), 23)\n",
      "(('11', 'delyevqwk553-558'), 23)\n",
      "(('26', '37/m'), 23)\n",
      "(('349', 'kit'), 23)\n",
      "(('125', '587'), 23)\n",
      "(('kpmyevqw550-557fl', 'kit'), 23)\n",
      "(('dose', '(mg)'), 23)\n",
      "(('(3/4)', 'kit'), 23)\n",
      "(('37/m', '600'), 23)\n",
      "(('diagnosis', 'imatinib'), 23)\n",
      "(('1', '64/f'), 23)\n",
      "(('awd', '14'), 23)\n",
      "(('dod', '13'), 23)\n",
      "(('pr', '664'), 23)\n",
      "(('380', '500'), 23)\n",
      "(('600', 'si'), 23)\n",
      "(('sd', '33'), 23)\n",
      "(('664', '930'), 23)\n",
      "(('138', '505'), 23)\n",
      "(('dod', '15'), 23)\n",
      "(('22', '65/m'), 23)\n",
      "(('52/m', '400'), 23)\n",
      "(('pr', '167'), 23)\n",
      "(('si,', 'small'), 23)\n",
      "(('25', '44/f'), 23)\n",
      "(('awd', '4'), 23)\n",
      "(('505', 'kit'), 23)\n",
      "(('disease;', 'iab,'), 23)\n",
      "(('2,285', '846'), 23)\n",
      "(('18', '67/m'), 23)\n",
      "(('dod', '7'), 23)\n",
      "(('pr', '643'), 23)\n",
      "(('53/f', '600'), 23)\n",
      "(('(0/1)', 'awd'), 23)\n",
      "(('54/m', '600'), 23)\n",
      "(('778', '766'), 23)\n",
      "(('wkvv557-560c', 'kit'), 23)\n",
      "(('70/f', '400'), 23)\n",
      "(('3', '50/m'), 23)\n",
      "(('30', '63/m'), 23)\n",
      "(('58/f', '600'), 23)\n",
      "(('12', '62/f'), 23)\n",
      "(('dod', '29'), 23)\n",
      "(('d820a', '(2/2)'), 23)\n",
      "(('(3/3)', 'awd'), 23)\n",
      "(('499', 'kit'), 23)\n",
      "(('pr', '591'), 23)\n",
      "(('59/f', '600'), 23)\n",
      "(('366', 'kit'), 23)\n",
      "(('pr', '2,285'), 23)\n",
      "(('dod', '27'), 23)\n",
      "(('dod', '19'), 23)\n",
      "(('dod', '6'), 23)\n",
      "(('1,252', '497'), 23)\n",
      "(('sequence', 'homology'), 22)\n",
      "(('number', 'variation'), 22)\n",
      "(('derived', 'patient'), 22)\n",
      "(('activity,', 'whereas'), 22)\n",
      "(('wild-type', 'flt3'), 22)\n",
      "(('mutants', 'induce'), 22)\n",
      "(('leukemia', '(all)'), 22)\n",
      "(('analysis', 'large'), 22)\n",
      "(('mutants', 'show'), 22)\n",
      "(('survival', 'absence'), 22)\n",
      "(('(fig', 'finally,'), 22)\n",
      "(('upon', 'treatment'), 22)\n",
      "(('mutant', 'cells,'), 22)\n",
      "(('order', 'investigate'), 22)\n",
      "(('noonan', 'syndrome'), 22)\n",
      "(('pcr', 'amplicons'), 22)\n",
      "(('presence', 'wild-type'), 22)\n",
      "(('addition,', 'found'), 22)\n",
      "(('factor', 'independence'), 22)\n",
      "(('kit', 'receptors'), 22)\n",
      "(('class', 'iii'), 22)\n",
      "(('presence', 'kit'), 22)\n",
      "(('akt,', 'erk'), 22)\n",
      "(('endogenous', 'wild-type'), 22)\n",
      "(('(fig', 'next'), 22)\n",
      "(('first', 'studied'), 22)\n",
      "(('mutations,', 'first'), 22)\n",
      "(('expressing', 'human'), 22)\n",
      "(('explored', 'whether'), 22)\n",
      "(('cell', 'fate'), 22)\n",
      "(('braf', 'nras'), 22)\n",
      "(('showed', 'variant'), 22)\n",
      "(('number', 'cell'), 22)\n",
      "(('(table', 's3),'), 22)\n",
      "(('mutations', 'large'), 22)\n",
      "(('reported', 'association'), 22)\n",
      "(('mutations', 'include'), 22)\n",
      "(('factors', 'may'), 22)\n",
      "(('similar', 'pattern'), 22)\n",
      "(('used', 'two'), 22)\n",
      "(('site', '(fig'), 22)\n",
      "(('nuclear', 'extracts'), 22)\n",
      "(('identify', 'genes'), 22)\n",
      "(('sequencing', 'results'), 22)\n",
      "(('domain,', 'compared'), 22)\n",
      "(('mechanisms', 'may'), 22)\n",
      "(('table', '1),'), 22)\n",
      "(('single', 'copy'), 22)\n",
      "(('acute', 'myelogenous'), 22)\n",
      "(('alone', '(fig'), 22)\n",
      "(('cells', 'inhibited'), 22)\n",
      "(('high', 'rate'), 22)\n",
      "(('promoter', 'methylation'), 22)\n",
      "(('mutant', 'exhibited'), 22)\n",
      "(('studies', 'confirm'), 22)\n",
      "(('assessed', 'using'), 22)\n",
      "(('growth', 'vivo'), 22)\n",
      "(('cells,', 'indicating'), 22)\n",
      "(('fetal', 'bovine'), 22)\n",
      "(('n-terminal', 'domain'), 22)\n",
      "(('deletion', 'found'), 22)\n",
      "(('cannot', 'excluded'), 22)\n",
      "(('16', '17'), 22)\n",
      "(('line', 'heterozygous'), 22)\n",
      "(('cancer', 'families'), 22)\n",
      "(('fibroblasts', '(mefs)'), 22)\n",
      "(('may', 'exert'), 22)\n",
      "(('single', 'mutants'), 22)\n",
      "(('(fig', '1a)'), 22)\n",
      "(('luciferase', 'assay'), 22)\n",
      "(('would', 'predict'), 22)\n",
      "(('domain', 'protein'), 22)\n",
      "(('using', 'fugene'), 22)\n",
      "(('suggesting', 'mutant'), 22)\n",
      "(('shown)', 'thus,'), 22)\n",
      "(('25%', 'ovarian'), 22)\n",
      "(('predominant', 'mechanism'), 22)\n",
      "(('however,', 'cell'), 22)\n",
      "(('rna', 'expression'), 22)\n",
      "(('required', 'determine'), 22)\n",
      "(('involves', 'two'), 22)\n",
      "(('affinity', 'atp'), 22)\n",
      "(('deep', 'sequencing'), 22)\n",
      "(('individuals', 'known'), 22)\n",
      "(('(see', 'materials'), 22)\n",
      "(('figure', '7,'), 22)\n",
      "(('mutations', 'directly'), 22)\n",
      "(('lower', 'observed'), 22)\n",
      "(('mlh1', 'mutations'), 22)\n",
      "(('tumors', 'assessed'), 22)\n",
      "(('seven', 'cases'), 22)\n",
      "(('1,', 'figure'), 22)\n",
      "(('respectively', '(table'), 22)\n",
      "(('analyzed', 'mutations'), 22)\n",
      "(('et', 'al12'), 22)\n",
      "(('amplify', 'exons'), 22)\n",
      "(('mutations', 'preferentially'), 22)\n",
      "(('hot', 'spots'), 22)\n",
      "(('changes', 'occurring'), 22)\n",
      "(('mutations', 'predominantly'), 22)\n",
      "(('phenotypes', 'associated'), 22)\n",
      "(('fibroblast', 'cell'), 22)\n",
      "(('gene', 'human'), 22)\n",
      "(('previously,', 'shown'), 22)\n",
      "(('region', 'kinase'), 22)\n",
      "(('human', 'mouse'), 22)\n",
      "(('briefly,', 'cells'), 22)\n",
      "(('examined', 'expression'), 22)\n",
      "(('specific', 'primer'), 22)\n",
      "(('supported', 'observation'), 22)\n",
      "(('activity', 'required'), 22)\n",
      "(('dramatically', 'reduced'), 22)\n",
      "(('knockout', 'mice'), 22)\n",
      "(('also', 'contribute'), 22)\n",
      "(('data', 'shown),'), 22)\n",
      "(('favorable', 'clinical'), 22)\n",
      "(('investigated', 'effects'), 22)\n",
      "(('specific', 'cellular'), 22)\n",
      "(('zinc', 'finger'), 22)\n",
      "(('5)', '(fig'), 22)\n",
      "(('alternatively', 'spliced'), 22)\n",
      "(('positively', 'charged'), 22)\n",
      "(('0', '4'), 22)\n",
      "(('given', 'low'), 22)\n",
      "(('ma,', 'usa)'), 22)\n",
      "(('(figure', '1b'), 22)\n",
      "(('2', '4'), 22)\n",
      "(('already', 'known'), 22)\n",
      "(('expression', 'studies'), 22)\n",
      "(('human', 'malignancies,'), 22)\n",
      "(('transduction', 'laboratories),'), 22)\n",
      "(('using', 'enhanced'), 22)\n",
      "(('tumors', '2'), 22)\n",
      "(('erlotinib', 'egfr'), 22)\n",
      "(('detection', 'rate'), 22)\n",
      "(('associated', 'altered'), 22)\n",
      "(('disease', 'progression,'), 22)\n",
      "(('vitro', 'experiments'), 22)\n",
      "(('might', 'reveal'), 22)\n",
      "(('phosphotyrosine,', 'whereas'), 22)\n",
      "(('sensitive', 'drug,'), 22)\n",
      "(('per', 'sample'), 22)\n",
      "(('analyzed', 'exons'), 22)\n",
      "(('positive', 'egfr'), 22)\n",
      "(('ten', 'tumors'), 22)\n",
      "(('thus,', 'similar'), 22)\n",
      "(('form', 'egfr'), 22)\n",
      "(('activity,', 'ability'), 22)\n",
      "(('whether', 'receptors'), 22)\n",
      "(('whereas', 'l858r'), 22)\n",
      "(('egfr', 'extracellular'), 22)\n",
      "(('sense', 'antisense'), 22)\n",
      "(('kinase', '1/2'), 22)\n",
      "(('autophosphorylation', 'egfr'), 22)\n",
      "(('contrast', 'l858r'), 22)\n",
      "(('mutations', 'correlate'), 22)\n",
      "(('harbored', 'egfr'), 22)\n",
      "(('marked', 'clinical'), 22)\n",
      "(('amplification', 'deletions'), 22)\n",
      "(('shown)', 'since'), 22)\n",
      "(('received', 'tki'), 22)\n",
      "(('applied', 'science)'), 22)\n",
      "(('whether', 'wild-type'), 22)\n",
      "(('importance', 'egfr'), 22)\n",
      "(('receptors', 'harboring'), 22)\n",
      "(('rationale', 'use'), 22)\n",
      "(('autophosphorylation', '(figure'), 22)\n",
      "(('specimens', 'obtained'), 22)\n",
      "(('similar', 'sensitivities'), 22)\n",
      "(('egfr', 'associated'), 22)\n",
      "(('two', 'drugs'), 22)\n",
      "(('aa', 'substitution'), 22)\n",
      "(('deletion', '(del'), 22)\n",
      "(('properties', 'egfr'), 22)\n",
      "(('mutation,', 'predicted'), 22)\n",
      "(('none', 'eight'), 22)\n",
      "(('dimerization', 'egfr'), 22)\n",
      "(('enhanced', 'chemiluminescence'), 22)\n",
      "(('mutations', '96'), 22)\n",
      "(('15', '(47%)'), 22)\n",
      "(('adjuvant', 'setting'), 22)\n",
      "(('crystallographic', 'studies'), 22)\n",
      "(('tumors', 'never'), 22)\n",
      "(('appendix)', 'functional'), 22)\n",
      "(('factor', '(egf),'), 22)\n",
      "(('still', 'small'), 22)\n",
      "(('therefore', 'provide'), 22)\n",
      "(('respond', 'gefitinib'), 22)\n",
      "(('inhibitor', 'gefitinib'), 22)\n",
      "(('activity', '293t'), 22)\n",
      "(('human', 'nsclc'), 22)\n",
      "(('egfr', 'del'), 22)\n",
      "(('expressed', 'lung'), 22)\n",
      "(('obtained', 'use'), 22)\n",
      "(('mediate', 'binding'), 22)\n",
      "(('levels', 'phosphotyrosine,'), 22)\n",
      "(('kinase', 'cascade,'), 22)\n",
      "(('summary', 'mutations'), 22)\n",
      "(('gain', 'insight,'), 22)\n",
      "(('cancers', 'somatic'), 22)\n",
      "(('egf', '(data'), 22)\n",
      "(('(fig', 'summary'), 22)\n",
      "(('l861q', 'mutation'), 22)\n",
      "(('21', 'point'), 22)\n",
      "(('heterozygous', 'nature'), 22)\n",
      "(('gefitinib', 'may'), 22)\n",
      "(('polymerase-chain-reaction', '(pcr)'), 22)\n",
      "(('3700', 'dna'), 22)\n",
      "(('gefitinib', 'response'), 22)\n",
      "(('prior', 'chemotherapy'), 22)\n",
      "(('mutations', 'another'), 22)\n",
      "(('egfr', 'tkis,'), 22)\n",
      "(('patients', 'kras'), 22)\n",
      "(('expressed', 'epithelial'), 22)\n",
      "(('finding', 'similar'), 22)\n",
      "(('contrast,', 'none'), 22)\n",
      "(('akt,', 'erk,'), 22)\n",
      "(('fine', 'needle'), 22)\n",
      "(('erlotinib', 'patients'), 22)\n",
      "(('selective', 'egfr'), 22)\n",
      "(('specimens', 'egfr'), 22)\n",
      "(('recruitment', 'egfr'), 22)\n",
      "(('exist', 'exons'), 22)\n",
      "(('expressing', 'egfr'), 22)\n",
      "(('dose-dependent', 'cell'), 22)\n",
      "(('resected', 'tumors'), 22)\n",
      "(('20', 'insertions'), 22)\n",
      "(('mtor', 'inhibitor'), 22)\n",
      "(('amplifications', 'purified'), 22)\n",
      "(('whose', 'resected'), 22)\n",
      "(('used', 'present'), 22)\n",
      "(('automatic', 'abi'), 22)\n",
      "(('published', 'cases'), 22)\n",
      "(('fish', 'assay'), 22)\n",
      "(('average', 'ic50'), 22)\n",
      "(('however,', 'four'), 22)\n",
      "(('purified', 'sequenced'), 22)\n",
      "(('prospective', 'trials'), 22)\n",
      "(('retrospective', 'nsclc'), 22)\n",
      "(('(fig', 'patients'), 22)\n",
      "(('mutations', 'residues'), 22)\n",
      "(('reported', 'literature,'), 22)\n",
      "(('mutation,', 'although'), 22)\n",
      "(('consecutive', 'recruitment'), 22)\n",
      "(('may', 'impact'), 22)\n",
      "(('effusions', 'july'), 22)\n",
      "(('different', 'patterns'), 22)\n",
      "(('gefitinib', 'erlotinib,'), 22)\n",
      "(('exquisite', 'sensitivity'), 22)\n",
      "(('bank', '(pdb)'), 22)\n",
      "(('thoracentesis', 'pleural'), 22)\n",
      "(('fully', 'inhibited'), 22)\n",
      "(('investigated', 'association'), 22)\n",
      "(('dna', 'analyzer'), 22)\n",
      "(('sanger', 'sequencing,'), 22)\n",
      "(('sequenced', 'automatic'), 22)\n",
      "(('tumors', 'retrospectively'), 22)\n",
      "(('clinical', 'evidence'), 22)\n",
      "(('encompass', 'tyrosine'), 22)\n",
      "(('kinase', 'mutation'), 22)\n",
      "(('orally', 'active'), 22)\n",
      "(('tumors', 'lacked'), 22)\n",
      "(('rate', 'wild-type'), 22)\n",
      "(('second-line', 'treatment'), 22)\n",
      "(('control', 'rate'), 22)\n",
      "(('preclinical', 'studies,'), 22)\n",
      "(('cases', 'nsclc'), 22)\n",
      "(('predictive', 'prognostic'), 22)\n",
      "(('however,', 'finding'), 22)\n",
      "(('gene', 'using'), 22)\n",
      "(('18', 'mutations'), 22)\n",
      "(('carrying', 'wild-type'), 22)\n",
      "(('using', 'wt'), 22)\n",
      "(('wt', 'mutated'), 22)\n",
      "(('kinase', 'mutations,'), 22)\n",
      "(('increase', 'phosphorylation'), 22)\n",
      "(('response;', 'sd,'), 22)\n",
      "(('2', 'different'), 22)\n",
      "(('cells', 'taken'), 22)\n",
      "(('molecule', 'kinase'), 22)\n",
      "(('(atp)', 'binding'), 22)\n",
      "(('conventional', 'chemotherapy,'), 22)\n",
      "(('received', 'chemotherapy'), 22)\n",
      "(('resistance', 'chemotherapy'), 22)\n",
      "(('xenograft', 'growth'), 22)\n",
      "(('variant', 'previously'), 22)\n",
      "(('ins,', 'insertion;'), 22)\n",
      "(('mutation', 'indicated'), 22)\n",
      "(('here,', 'describe'), 22)\n",
      "(('(table', '3;'), 22)\n",
      "(('mutation', '(exon'), 22)\n",
      "(('findings', 'confirmed'), 22)\n",
      "(('may', 'suggest'), 22)\n",
      "(('domain;', 'tm,'), 22)\n",
      "(('protein', 'involved'), 22)\n",
      "(('patients', 'develop'), 22)\n",
      "(('dna', 'prepared'), 22)\n",
      "(('large', 'international'), 22)\n",
      "(('mutations', 'important'), 22)\n",
      "(('mutations', 'exhibit'), 22)\n",
      "(('(n', '1)'), 22)\n",
      "(('fluorescent', 'situ'), 22)\n",
      "(('that,', 'although'), 22)\n",
      "(('constructs', 'created'), 22)\n",
      "(('four', 'mutants'), 22)\n",
      "(('mutations,', 'compared'), 22)\n",
      "(('higher', 'patients'), 22)\n",
      "(('stimulation', '(figure'), 22)\n",
      "(('two', 'highly'), 22)\n",
      "(('formation', 'assays'), 22)\n",
      "(('previously', 'unreported'), 22)\n",
      "(('mutants', '(supplementary'), 22)\n",
      "(('c', 'helix'), 22)\n",
      "(('mutation', 'pdgfra'), 22)\n",
      "(('signaling', 'wild-type'), 22)\n",
      "(('frequently', 'identified'), 22)\n",
      "(('little', 'known'), 22)\n",
      "(('autophosphorylation', 'status'), 22)\n",
      "(('confirmed', 'western'), 22)\n",
      "(('observations', 'may'), 22)\n",
      "(('(figure', 's3)'), 22)\n",
      "(('3t3', 'fibroblasts'), 22)\n",
      "(('kinase', '(fig'), 22)\n",
      "(('less', 'responsive'), 22)\n",
      "(('one', 'common'), 22)\n",
      "(('three', 'additional'), 22)\n",
      "(('growth', 'tumors'), 22)\n",
      "(('interestingly,', 'two'), 22)\n",
      "(('phosphorylated', 'activated'), 22)\n",
      "(('primers', 'described'), 22)\n",
      "(('overexpressing', 'wild-type'), 22)\n",
      "(('hek293', 'cells,'), 22)\n",
      "(('mutants', 'examined'), 22)\n",
      "(('found', 'study'), 22)\n",
      "(('survival', 'signals'), 22)\n",
      "(('also', 'promote'), 22)\n",
      "(('support', 'idea'), 22)\n",
      "(('1', '6'), 22)\n",
      "(('corresponding', 'mutation'), 22)\n",
      "(('expressed', 'normal'), 22)\n",
      "(('number', 'gain'), 22)\n",
      "(('impaired', 'ability'), 22)\n",
      "(('cdna', 'using'), 22)\n",
      "(('mutation', 'braf'), 22)\n",
      "(('figure', '2,'), 22)\n",
      "(('c-terminal', 'residues'), 22)\n",
      "(('found', 'several'), 22)\n",
      "(('thus', 'leading'), 22)\n",
      "(('lower', 'extent'), 22)\n",
      "(('21,', '22,'), 22)\n",
      "(('variant', 'suggests'), 22)\n",
      "(('one', '16'), 22)\n",
      "(('given', 'table'), 22)\n",
      "(('similar', 'findings'), 22)\n",
      "(('prediction', 'models'), 22)\n",
      "(('expression', 'patterns'), 22)\n",
      "(('fusion', 'protein,'), 22)\n",
      "(('new', 'window'), 22)\n",
      "(('contrast,', 'mutation'), 22)\n",
      "(('region', 'contains'), 22)\n",
      "(('respectively', 'mutations'), 22)\n",
      "(('alk', 'mutations,'), 22)\n",
      "(('patient', 'sample'), 22)\n",
      "(('using', 'reverse'), 22)\n",
      "(('titration', 'calorimetry'), 22)\n",
      "(('finding', 'consistent'), 22)\n",
      "(('isothermal', 'titration'), 22)\n",
      "(('several', 'small'), 22)\n",
      "(('cdna', 'clones'), 22)\n",
      "(('gatekeeper', 'mutations'), 22)\n",
      "(('human', 'mammary'), 22)\n",
      "(('novo', 'aml'), 22)\n",
      "(('mutation', 'resulted'), 22)\n",
      "(('amplification', '(mlpa)'), 22)\n",
      "(('exon', '16'), 22)\n",
      "(('cellular', 'localization'), 22)\n",
      "(('expressed', 'similar'), 22)\n",
      "(('aml', 'mds'), 22)\n",
      "(('(figure', '2b),'), 22)\n",
      "(('p53', 'function'), 22)\n",
      "(('mutants,', 'well'), 22)\n",
      "(('underlying', 'biology'), 22)\n",
      "(('whereas', 'mutants'), 22)\n",
      "(('binding', 'affinity,'), 22)\n",
      "(('binding', 'mutant'), 22)\n",
      "(('binding', 'capacity'), 22)\n",
      "(('gene', 'cancer'), 22)\n",
      "(('protein', 'extracts'), 22)\n",
      "(('cells', 'towards'), 22)\n",
      "(('substantially', 'higher'), 22)\n",
      "(('adhesion', 'kinase'), 22)\n",
      "(('cells,', 'consistent'), 22)\n",
      "(('class', '2'), 22)\n",
      "(('wild-type', 'levels'), 22)\n",
      "(('ability', 'interact'), 22)\n",
      "(('morphological', 'changes'), 22)\n",
      "(('critical', 'interaction'), 22)\n",
      "(('vivo', 'data'), 22)\n",
      "(('activity', 'enzyme'), 22)\n",
      "(('along', 'entire'), 22)\n",
      "(('associated', 'severe'), 22)\n",
      "(('found', 'that,'), 22)\n",
      "(('associated', 'adverse'), 22)\n",
      "(('cytogenetically', 'normal'), 22)\n",
      "(('show', 'cell'), 22)\n",
      "(('vitro', 'mutagenesis'), 22)\n",
      "(('functional', 'assays,'), 22)\n",
      "(('al', 'mutations'), 22)\n",
      "(('less', 'certain'), 22)\n",
      "(('erbb2', 'mutants'), 22)\n",
      "(('pi3', 'kinase'), 22)\n",
      "(('would', 'provide'), 22)\n",
      "(('biosciences,', 'san'), 22)\n",
      "(('strong', 'inhibition'), 22)\n",
      "(('significantly', 'improved'), 22)\n",
      "(('dna', 'lesions'), 22)\n",
      "(('harboring', 'wt'), 22)\n",
      "(('interpretation', 'missense'), 22)\n",
      "(('restriction', 'sites'), 22)\n",
      "(('biochemical', 'assays'), 22)\n",
      "(('major', 'band'), 22)\n",
      "(('fig', '(a)'), 22)\n",
      "(('transferred', 'nitrocellulose'), 22)\n",
      "(('imatinib', 'resistant'), 22)\n",
      "(('cdna', 'expressed'), 22)\n",
      "(('transgenic', 'mice,'), 22)\n",
      "(('sensitive', 'small'), 22)\n",
      "(('mutant', 'pdgfra'), 22)\n",
      "(('alpha', '(pdgfra)'), 22)\n",
      "(('phosphorylated', 'erk'), 22)\n",
      "(('therefore,', 'investigated'), 22)\n",
      "(('abbreviations:', 'pdgfra,'), 22)\n",
      "(('1', 'yes'), 22)\n",
      "(('therapy', 'study'), 22)\n",
      "(('melanoma', 'determined'), 22)\n",
      "(('', '104'), 22)\n",
      "(('relative', 'potency'), 22)\n",
      "(('kinases', '(table'), 22)\n",
      "(('studies', 'crenolanib'), 22)\n",
      "(('patients', 'gist'), 22)\n",
      "(('second', 'tyrosine'), 22)\n",
      "(('addition,', 'shown'), 22)\n",
      "(('activating', 'kit'), 22)\n",
      "(('using', 'various'), 22)\n",
      "(('reported', 'data'), 22)\n",
      "(('pik3ca', 'mutations,'), 22)\n",
      "(('cyclin', 'd1,'), 22)\n",
      "(('expressed', 'cos'), 22)\n",
      "(('changes', 'reported'), 22)\n",
      "(('specific', 'human'), 22)\n",
      "(('strategy', 'used'), 22)\n",
      "(('previously', '(5)'), 22)\n",
      "(('cause', 'increased'), 22)\n",
      "(('cells', 'injected'), 22)\n",
      "(('phenotype', '(figure'), 22)\n",
      "(('detected', 'mutant'), 22)\n",
      "(('although', 'role'), 22)\n",
      "(('oncogenic', 'braf'), 22)\n",
      "(('additional', 'functions'), 22)\n",
      "(('restriction', 'enzymes'), 22)\n",
      "(('phosphorylation', 'decreased'), 22)\n",
      "(('proliferation', 'assays,'), 22)\n",
      "(('met', 'mutations'), 22)\n",
      "(('also', 'agreement'), 22)\n",
      "(('detected', 'cells'), 22)\n",
      "(('(exons', '13'), 22)\n",
      "(('took', 'advantage'), 22)\n",
      "(('codes', 'references'), 22)\n",
      "(('references', 'acknowledgments'), 22)\n",
      "(('accession', 'codes'), 22)\n",
      "(('methods', 'accession'), 22)\n",
      "(('mutants', 'p110'), 22)\n",
      "(('reported', 'mutated'), 22)\n",
      "(('mutants', 'strongly'), 22)\n",
      "(('activation', 'assay'), 22)\n",
      "(('induced', 'transformation'), 22)\n",
      "(('allele;', 'however,'), 22)\n",
      "(('located,', 'respectively,'), 22)\n",
      "(('mutations', 'close'), 22)\n",
      "(('partially', 'suppressed'), 22)\n",
      "(('three', 'classes:'), 22)\n",
      "(('regulatory', 'domain,'), 22)\n",
      "(('variant', 'lacking'), 22)\n",
      "(('strong', 'mutator'), 22)\n",
      "(('interacts', 'proteins'), 22)\n",
      "(('region', 'mlh1'), 22)\n",
      "(('predict', 'pathogenicity'), 22)\n",
      "(('resistance', 'cell'), 22)\n",
      "(('fgfr2', 'kinase'), 22)\n",
      "(('clinically', 'applicable'), 22)\n",
      "(('activity', 'primary'), 22)\n",
      "(('seen', 'three'), 22)\n",
      "(('cells', 'lysates'), 22)\n",
      "(('associated', 'normal'), 22)\n",
      "(('blast', 'cells'), 22)\n",
      "(('additional', 'flt3'), 22)\n",
      "(('activating', 'flt3'), 22)\n",
      "(('3', '(flt3)'), 22)\n",
      "(('allelic', 'ratio'), 22)\n",
      "(('clinical', 'problem'), 22)\n",
      "(('destabilizes', 'fold'), 22)\n",
      "(('flt3', 'antibody'), 22)\n",
      "(('dna', 'replication,'), 22)\n",
      "(('development', 'disease'), 22)\n",
      "(('wild-type', 'murine'), 22)\n",
      "(('cleaved', 'within'), 22)\n",
      "(('constructs', '(a)'), 22)\n",
      "(('mutations', '5'), 22)\n",
      "(('tumors', '(10'), 22)\n",
      "(('variant,', 'although'), 22)\n",
      "(('cancer', 'carry'), 22)\n",
      "(('mutant', 'braf'), 22)\n",
      "(('cleavage', 'site'), 22)\n",
      "(('min', 'incubation'), 22)\n",
      "(('context', 'full-length'), 22)\n",
      "(('entire', 'pten'), 22)\n",
      "(('method', 'predict'), 22)\n",
      "(('variants', 'r1443g,'), 22)\n",
      "(('l1664p,', 't1700a,'), 22)\n",
      "(('p1806a)', 'activity'), 22)\n",
      "(('clinical,', 'genetic,'), 22)\n",
      "(('terminus', '(amino'), 22)\n",
      "(('genetic,', 'structural'), 22)\n",
      "(('g1788d,', 'e1794d,'), 22)\n",
      "(('v1804d,', 'p1806a)'), 22)\n",
      "(('interpret', 'findings'), 22)\n",
      "(('mutations', 'r1443g,'), 22)\n",
      "(('light', 'available'), 22)\n",
      "(('(r1443g,', 'v1534m,'), 22)\n",
      "(('plasmid', 'p385-brca1'), 22)\n",
      "(('t1700a,', 'g1706e,'), 22)\n",
      "(('r1443g,', 'v1534m,'), 22)\n",
      "(('effect', '22'), 22)\n",
      "(('transcription-based', 'assay'), 22)\n",
      "(('v1713a,', 'v1736a,'), 22)\n",
      "(('r1753t,', 'i1766s,'), 22)\n",
      "(('s1655f,', 'l1664p,'), 22)\n",
      "(('l1564p,', 'p1614s,'), 22)\n",
      "(('e1644g,', 's1655f,'), 22)\n",
      "(('g1706e,', 'v1713a,'), 22)\n",
      "(('pcr', '(20)'), 22)\n",
      "(('v1534m,', 'd1546n,'), 22)\n",
      "(('e1794d,', 'v1804d,'), 22)\n",
      "(('l1764p,', 'q1785h,'), 22)\n",
      "(('i1766s,', 'l1764p,'), 22)\n",
      "(('v1736a,', 'g1738r,'), 22)\n",
      "(('v1804d,', 'p1806a'), 22)\n",
      "(('p1806a', 'introduced'), 22)\n",
      "(('p385-brca1', 'template'), 22)\n",
      "(('g1738e,', 'r1753t,'), 22)\n",
      "(('q1785h,', 'g1788d,'), 22)\n",
      "(('22', 'variants'), 22)\n",
      "(('p1614s,', 'e1644g,'), 22)\n",
      "(('d1546n,', 'l1564p,'), 22)\n",
      "(('findings', 'light'), 22)\n",
      "(('brca1', 'cooh'), 22)\n",
      "(('apply', 'transcription-based'), 22)\n",
      "(('g1738r,', 'g1738e,'), 22)\n",
      "(('1,3961,863)', 'interpret'), 22)\n",
      "(('variants', '(r1443g,'), 22)\n",
      "(('r1753t,', 'l1764p,'), 22)\n",
      "(('available', 'clinical,'), 22)\n",
      "(('acids', '1,3961,863)'), 22)\n",
      "(('characteristic', 'brca1'), 22)\n",
      "(('brca1', 'function'), 22)\n",
      "(('outside', 'conserved'), 22)\n",
      "(('variant', 'uncertain'), 22)\n",
      "(('braf', 'craf'), 22)\n",
      "(('samples', '1'), 22)\n",
      "(('(exons', '11'), 22)\n",
      "(('provide', 'extensive'), 22)\n",
      "(('however,', 'brca2'), 22)\n",
      "(('brca2', 'allele'), 22)\n",
      "(('human', 'kit'), 22)\n",
      "(('kit', 'receptor'), 22)\n",
      "(('14,', '17'), 22)\n",
      "(('imatinib-resistant', 'tumors'), 22)\n",
      "(('secondary', 'kinase'), 22)\n",
      "(('risk', 'acquisition'), 22)\n",
      "(('schematic', 'structure'), 21)\n",
      "(('mutation', 'c'), 21)\n",
      "(('identified', 'lung'), 21)\n",
      "(('cells', 'coexpressing'), 21)\n",
      "(('flt3', 'ligand'), 21)\n",
      "(('genomic', 'deletion'), 21)\n",
      "(('found', 'aml'), 21)\n",
      "(('mutations', 'targeting'), 21)\n",
      "(('activation', 'kinase'), 21)\n",
      "(('lower', 'expression'), 21)\n",
      "(('samples', '(fig'), 21)\n",
      "(('transfected', 'control'), 21)\n",
      "(('hematopoietic', 'cell'), 21)\n",
      "(('decreased', 'levels'), 21)\n",
      "(('however,', 'cells'), 21)\n",
      "(('growth', '32d'), 21)\n",
      "(('oncogenic', 'mutation'), 21)\n",
      "(('bone', 'marrow,'), 21)\n",
      "(('could', 'help'), 21)\n",
      "(('family', 'genes'), 21)\n",
      "(('effective', 'treatment'), 21)\n",
      "(('frequent', 'genetic'), 21)\n",
      "(('next', 'table'), 21)\n",
      "(('cells', 'peripheral'), 21)\n",
      "(('within', '3'), 21)\n",
      "(('gene', 'mouse'), 21)\n",
      "(('akt', 'pathways'), 21)\n",
      "(('strongly', 'suggesting'), 21)\n",
      "(('mutant', 'protein,'), 21)\n",
      "(('patients', 'harbouring'), 21)\n",
      "(('recurrent', 'somatic'), 21)\n",
      "(('thyroid', 'cancers'), 21)\n",
      "(('glioblastoma', 'multiforme'), 21)\n",
      "(('one', 'potential'), 21)\n",
      "(('table', 's5)'), 21)\n",
      "(('mutations', 'relatively'), 21)\n",
      "(('found', 'associated'), 21)\n",
      "(('tumours', 'harbouring'), 21)\n",
      "(('tumour', 'samples'), 21)\n",
      "(('clear', 'whether'), 21)\n",
      "(('signalling', 'pathway'), 21)\n",
      "(('suggest', 'although'), 21)\n",
      "(('electrophoretic', 'mobility'), 21)\n",
      "(('transfection', 'reagent'), 21)\n",
      "(('fold', 'change'), 21)\n",
      "(('whereas', 'wt'), 21)\n",
      "(('additional', 'mechanisms'), 21)\n",
      "(('tumour', 'development'), 21)\n",
      "(('oral', 'squamous'), 21)\n",
      "(('harboring', 'mutation'), 21)\n",
      "(('may', 'reduce'), 21)\n",
      "(('consistent', 'increased'), 21)\n",
      "(('phosphorylation', 'endogenous'), 21)\n",
      "(('phosphorylation', 'wt'), 21)\n",
      "(('(23)', 'thus,'), 21)\n",
      "(('led', 'significant'), 21)\n",
      "(('whereas', 'mutant'), 21)\n",
      "(('serum', 'starved'), 21)\n",
      "(('loss', 'pten'), 21)\n",
      "(('reduced', 'compared'), 21)\n",
      "(('mutated', 'allele'), 21)\n",
      "(('strong', 'correlation'), 21)\n",
      "(('hereditary', 'cancer'), 21)\n",
      "(('four', 'novel'), 21)\n",
      "(('conserved', 'residue'), 21)\n",
      "(('proteinprotein', 'interactions'), 21)\n",
      "(('observed', 'unselected'), 21)\n",
      "(('significantly', 'shorter'), 21)\n",
      "(('next', 'assessed'), 21)\n",
      "(('cells,', 'particularly'), 21)\n",
      "(('five', 'six'), 21)\n",
      "(('frame', 'shift'), 21)\n",
      "(('protein', 'levels,'), 21)\n",
      "(('hct116', 'cells'), 21)\n",
      "(('interaction', '(fig'), 21)\n",
      "(('compelling', 'evidence'), 21)\n",
      "(('highly', 'significant'), 21)\n",
      "(('confer', 'increased'), 21)\n",
      "(('page', 'discussion'), 21)\n",
      "(('blood', 'samples'), 21)\n",
      "(('cancer', 'number'), 21)\n",
      "(('variants', 'increased'), 21)\n",
      "(('cells', 'acquired'), 21)\n",
      "(('expressed', 'receptor'), 21)\n",
      "(('mutation', 'nucleotide'), 21)\n",
      "(('constructs', 'demonstrated'), 21)\n",
      "(('0', 'comparison'), 21)\n",
      "(('clear', 'evidence'), 21)\n",
      "(('examine', 'effect'), 21)\n",
      "(('primer', 'sets'), 21)\n",
      "(('++', '++'), 21)\n",
      "(('10', '20'), 21)\n",
      "(('last', 'exon'), 21)\n",
      "(('long', 'latency'), 21)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(('although', 'many'), 21)\n",
      "(('(c)', 'deletion'), 21)\n",
      "(('despite', 'fact'), 21)\n",
      "(('single', 'base'), 21)\n",
      "(('2', '7'), 21)\n",
      "(('msh2', 'mutations'), 21)\n",
      "(('deletions', 'identified'), 21)\n",
      "(('deletion', 'second'), 21)\n",
      "(('study', 'demonstrated'), 21)\n",
      "(('showed', 'presence'), 21)\n",
      "(('mutation', 'causes'), 21)\n",
      "(('tissue', 'sections'), 21)\n",
      "(('(salt', 'lake'), 21)\n",
      "(('laboratories', '(salt'), 21)\n",
      "(('lake', 'city,'), 21)\n",
      "(('small', 'fraction'), 21)\n",
      "(('early', 'onset'), 21)\n",
      "(('germline', 'somatic'), 21)\n",
      "(('conformation', 'polymorphism'), 21)\n",
      "(('genes', '(fig'), 21)\n",
      "(('event-free', 'survival'), 21)\n",
      "(('showed', 'two'), 21)\n",
      "(('14', 'days'), 21)\n",
      "(('normal', 'function'), 21)\n",
      "(('lower', 'expected'), 21)\n",
      "(('apoptotic', 'signals'), 21)\n",
      "(('two', 'five'), 21)\n",
      "(('mouse', 'human'), 21)\n",
      "(('cells', 'tumor'), 21)\n",
      "(('kinases', 'play'), 21)\n",
      "(('associated', 'tumor'), 21)\n",
      "(('compared', 'expression'), 21)\n",
      "(('mice', 'generated'), 21)\n",
      "(('mefs', 'expressing'), 21)\n",
      "(('exons', '12'), 21)\n",
      "(('show', 'first'), 21)\n",
      "(('index', 'case'), 21)\n",
      "(('9', 'exon'), 21)\n",
      "(('shown),', 'suggesting'), 21)\n",
      "(('mutant', 'cases'), 21)\n",
      "(('associated', 'presence'), 21)\n",
      "(('aberrant', 'expression'), 21)\n",
      "(('multiple', 'cancer'), 21)\n",
      "(('deleterious', 'effects'), 21)\n",
      "(('given', 'apparent'), 21)\n",
      "(('commonly', 'associated'), 21)\n",
      "(('80', '100'), 21)\n",
      "(('flow', 'cytometric'), 21)\n",
      "(('(n', '20)'), 21)\n",
      "(('cells', '24'), 21)\n",
      "(('cells', 'plated'), 21)\n",
      "(('bm', 'cells'), 21)\n",
      "(('also', 'lead'), 21)\n",
      "(('cancer', 'genes'), 21)\n",
      "(('differences', 'frequency'), 21)\n",
      "(('hematopoietic', 'progenitors,'), 21)\n",
      "(('positive', 'predictive'), 21)\n",
      "(('assays', 'parameters'), 21)\n",
      "(('mutant,', 'even'), 21)\n",
      "(('initial', 'treatment'), 21)\n",
      "(('patient', 'population,'), 21)\n",
      "(('predictive', 'marker'), 21)\n",
      "(('multiple', 'egfr'), 21)\n",
      "(('signaling', 'identified'), 21)\n",
      "(('erlotinib', 'also'), 21)\n",
      "(('makes', 'possible'), 21)\n",
      "(('erlotinib', 'wild-type'), 21)\n",
      "(('many', 'epithelial'), 21)\n",
      "(('treated', 'drug'), 21)\n",
      "(('cancer', 'growth'), 21)\n",
      "(('proteins', 'general'), 21)\n",
      "(('transforming', 'alleles'), 21)\n",
      "(('egfrs', '[wild'), 21)\n",
      "(('phosphorylate', 'tyrosine'), 21)\n",
      "(('lysates', 'transiently'), 21)\n",
      "(('type,', 'exon'), 21)\n",
      "(('observed', 'japanese'), 21)\n",
      "(('history', 'smoking,'), 21)\n",
      "(('mutations', 'g719s'), 21)\n",
      "(('egfr', 'gefitinib'), 21)\n",
      "(('since', 'amino'), 21)\n",
      "(('mutation', 'gefitinib'), 21)\n",
      "(('proteins', 'lung'), 21)\n",
      "(('predicted', 'lie'), 21)\n",
      "(('various', 'aspects'), 21)\n",
      "(('activity,', 'still'), 21)\n",
      "(('gene', 'suggests'), 21)\n",
      "(('(table', '3'), 21)\n",
      "(('cellular', 'tyrosine'), 21)\n",
      "(('substitution', '(l858r)]'), 21)\n",
      "(('2figure', '2'), 21)\n",
      "(('leading', 'deletion'), 21)\n",
      "(('achieved', 'use'), 21)\n",
      "(('egfr', 'also'), 21)\n",
      "(('adaptor', 'molecules'), 21)\n",
      "(('association', 'response'), 21)\n",
      "(('expression,', 'shown'), 21)\n",
      "(('gefitinib-responsive', 'patients'), 21)\n",
      "(('figure', '2figure'), 21)\n",
      "(('loop', 'egfr'), 21)\n",
      "(('egfr', 'absence'), 21)\n",
      "(('(del', 'l747-s752),'), 21)\n",
      "(('striking', 'differences'), 21)\n",
      "(('advanced', 'nonsmall-cell'), 21)\n",
      "(('none', '11'), 21)\n",
      "(('tumors', 'functional'), 21)\n",
      "(('parameters', 'activity,'), 21)\n",
      "(('altered', 'egfr'), 21)\n",
      "(('cancer', 'defined'), 21)\n",
      "(('adenocarcinomas', 'patients'), 21)\n",
      "(('allele', 'mutations'), 21)\n",
      "(('b-raf', 'kinase'), 21)\n",
      "(('[wild', 'type,'), 21)\n",
      "(('treatment', 'option'), 21)\n",
      "(('suppress', 'wild-type'), 21)\n",
      "(('web', 'site)'), 21)\n",
      "(('cytotoxic', 'therapy'), 21)\n",
      "(('total', '16'), 21)\n",
      "(('various', 'egfrs'), 21)\n",
      "(('cancers', 'patients'), 21)\n",
      "(('l858r', 'amino'), 21)\n",
      "(('study', 'various'), 21)\n",
      "(('gefitinib-sensitive', 'lung'), 21)\n",
      "(('alone', '(v)'), 21)\n",
      "(('(18)', 'found'), 21)\n",
      "(('finally,', 'assessed'), 21)\n",
      "(('gefitinib', '(iressa)'), 21)\n",
      "(('levels', 'tyrosine-phosphorylated'), 21)\n",
      "(('important', 'insight'), 21)\n",
      "(('95', 'primary'), 21)\n",
      "(('detected,', 'suggesting'), 21)\n",
      "(('l747-s752),', 'exon'), 21)\n",
      "(('highest', 'fraction'), 21)\n",
      "(('egfr', 'phosphorylate'), 21)\n",
      "(('residues', 'cell'), 21)\n",
      "(('could', 'similarly'), 21)\n",
      "(('mutagenesis', 'inserted'), 21)\n",
      "(('determining', 'gefitinib'), 21)\n",
      "(('tkis,', 'whereas'), 21)\n",
      "(('see', 'supporting'), 21)\n",
      "(('transfection', 'either'), 21)\n",
      "(('setting', 'surgical'), 21)\n",
      "(('constructs', 'l858r'), 21)\n",
      "(('assays', 'required'), 21)\n",
      "(('questions', 'remain'), 21)\n",
      "(('mutation,', 'well'), 21)\n",
      "(('used', 'separate'), 21)\n",
      "(('pretreated', 'gefitinib'), 21)\n",
      "(('gefitinib', '(fig'), 21)\n",
      "(('egfr', 'receptor'), 21)\n",
      "(('extraordinary', 'drug'), 21)\n",
      "(('egf-induced', 'activation'), 21)\n",
      "(('separate', 'transfection'), 21)\n",
      "(('histology', '(adenocarcinoma'), 21)\n",
      "(('21', 'aa'), 21)\n",
      "(('insight,', 'begun'), 21)\n",
      "(('line', 'identification'), 21)\n",
      "(('(10)', 'however,'), 21)\n",
      "(('conferred', 'upon'), 21)\n",
      "(('tyrosine', '1092'), 21)\n",
      "(('similar', 'types'), 21)\n",
      "(('0', 'identified'), 21)\n",
      "(('(l858r)]', 'cultured'), 21)\n",
      "(('transfection', 'efficiencies,'), 21)\n",
      "(('sensitivity', 'inhibitor'), 21)\n",
      "(('identify', 'subgroup'), 21)\n",
      "(('sequences', 'mouse'), 21)\n",
      "(('egfr', 'lung'), 21)\n",
      "(('1092', 'egfr'), 21)\n",
      "(('anti-egfr', 'monoclonal'), 21)\n",
      "(('randomized', 'trials'), 21)\n",
      "(('patient', 'may'), 21)\n",
      "(('thus,', 'patients'), 21)\n",
      "(('susceptibility', 'tkis'), 21)\n",
      "(('binding', 'growth'), 21)\n",
      "(('cell', 'proteins'), 21)\n",
      "(('cancer-derived', 'cell'), 21)\n",
      "(('c-kit', 'mutant'), 21)\n",
      "(('begun', 'study'), 21)\n",
      "(('contaminating', 'normal'), 21)\n",
      "(('cytomegalovirus', 'promoter-driven'), 21)\n",
      "(('significant', 'p'), 21)\n",
      "(('cdna', 'four'), 21)\n",
      "(('results:', 'egfr'), 21)\n",
      "(('mutation', 'exons'), 21)\n",
      "(('mutations', 'respond'), 21)\n",
      "(('tkis', 'treatment'), 21)\n",
      "(('low', 'tumor'), 21)\n",
      "(('mutations', 'amplification'), 21)\n",
      "(('second-line', 'therapy'), 21)\n",
      "(('comparable', 'efficacy'), 21)\n",
      "(('combined', 'deletion'), 21)\n",
      "(('20', '21'), 21)\n",
      "(('major', 'egfr'), 21)\n",
      "(('mutations,', 'patients'), 21)\n",
      "(('due', 'acquired'), 21)\n",
      "(('one', 'present'), 21)\n",
      "(('testing', 'identifies'), 21)\n",
      "(('treatment', 'strategies'), 21)\n",
      "(('ki', 'values'), 21)\n",
      "(('(a)', 'wild-type'), 21)\n",
      "(('c', 'helix'), 21)\n",
      "(('first', 'tested'), 21)\n",
      "(('reported', 'recent'), 21)\n",
      "(('understanding', 'molecular'), 21)\n",
      "(('subgroup', 'analysis'), 21)\n",
      "(('20', 'mutations'), 21)\n",
      "(('mutations', 'age'), 21)\n",
      "(('carcinomas', '1'), 21)\n",
      "(('approximately', '3%'), 21)\n",
      "(('(14)', 'although'), 21)\n",
      "(('protein,', 'whereas'), 21)\n",
      "(('outcome', 'patients'), 21)\n",
      "(('others', 'reported'), 21)\n",
      "(('suggest', 'sequencing'), 21)\n",
      "(('kaplan-meier', 'plots'), 21)\n",
      "(('mutations', 'second,'), 21)\n",
      "(('using', 'fluorescence'), 21)\n",
      "(('pathway', 'developed'), 21)\n",
      "(('cancer', '(nsclc),'), 21)\n",
      "(('clinically', 'achievable'), 21)\n",
      "(('result', 'malignant'), 21)\n",
      "(('(egfr)', 'mutations'), 21)\n",
      "(('tumors', 'positive'), 21)\n",
      "(('treatment', 'efficacy'), 21)\n",
      "(('using', 'egfr'), 21)\n",
      "(('identified', '33'), 21)\n",
      "(('different', 'exons'), 21)\n",
      "(('survival', 'start'), 21)\n",
      "(('20', 'insertion'), 21)\n",
      "(('comparable', 'results'), 21)\n",
      "(('literature,', '2'), 21)\n",
      "(('tk', 'inhibitors'), 21)\n",
      "(('exons', '17'), 21)\n",
      "(('irreversible', 'inhibitors'), 21)\n",
      "(('development', 'resistance'), 21)\n",
      "(('code', 'tk'), 21)\n",
      "(('increased', 'egfr'), 21)\n",
      "(('tkis', 'patients'), 21)\n",
      "(('rate', 'nsclc'), 21)\n",
      "(('mutual', 'exclusivity'), 21)\n",
      "(('sequence,', 'exons'), 21)\n",
      "(('patients', 'objective'), 21)\n",
      "(('inhibitors', 'result'), 21)\n",
      "(('levels', 'mutant'), 21)\n",
      "(('another', 'interesting'), 21)\n",
      "(('underwent', 'fine'), 21)\n",
      "(('therapeutic', 'options'), 21)\n",
      "(('treatment', 'regimens'), 21)\n",
      "(('response', 'noted'), 21)\n",
      "(('egfr,', 'kras,'), 21)\n",
      "(('2', 'egfr'), 21)\n",
      "(('absence', 'exogenous'), 21)\n",
      "(('study', 'investigate'), 21)\n",
      "(('overall', 'frequency'), 21)\n",
      "(('samples', 'available'), 21)\n",
      "(('2', 'weeks'), 21)\n",
      "(('primary', 'breast'), 21)\n",
      "(('negative', 'prognostic'), 21)\n",
      "(('molecular', 'mass'), 21)\n",
      "(('low', 'serum'), 21)\n",
      "(('vector', 'wild-type'), 21)\n",
      "(('cells', 'ba/f3'), 21)\n",
      "(('patients,', 'including'), 21)\n",
      "(('high', 'molecular'), 21)\n",
      "(('reported', 'literature'), 21)\n",
      "(('mutations', 'following'), 21)\n",
      "(('resistance', 'mechanisms'), 21)\n",
      "(('(fig', 'together,'), 21)\n",
      "(('wide', 'array'), 21)\n",
      "(('tumor', 'biopsy'), 21)\n",
      "(('oncogenic', 'signaling'), 21)\n",
      "(('fig', 'although'), 21)\n",
      "(('2', 'years'), 21)\n",
      "(('smoking', 'status'), 21)\n",
      "(('lymphoma', 'kinase'), 21)\n",
      "(('frequent', 'patients'), 21)\n",
      "(('anaplastic', 'lymphoma'), 21)\n",
      "(('also', 'used'), 21)\n",
      "(('30', 'min,'), 21)\n",
      "(('functional', 'roles'), 21)\n",
      "(('mutations,', 'cells'), 21)\n",
      "(('(figure', '4a'), 21)\n",
      "(('9', '2'), 21)\n",
      "(('resistance', 'profiles'), 21)\n",
      "(('mutation', '(table'), 21)\n",
      "(('control,', 'wt'), 21)\n",
      "(('rate', 'significantly'), 21)\n",
      "(('different', 'activating'), 21)\n",
      "(('molecular', 'analyses'), 21)\n",
      "(('treatment', 'cells'), 21)\n",
      "(('mutations', 'structural'), 21)\n",
      "(('retroviruses', 'encoding'), 21)\n",
      "(('domain', 'receptor'), 21)\n",
      "(('mutated', 'tumors'), 21)\n",
      "(('mutation', 'primary'), 21)\n",
      "(('decreased', 'cell'), 21)\n",
      "(('identified', 'new'), 21)\n",
      "(('figures', 'index'), 21)\n",
      "(('showed', 'low'), 21)\n",
      "(('although', 'exact'), 21)\n",
      "(('inhibition', 'either'), 21)\n",
      "(('cell', 'invasion'), 21)\n",
      "(('genetic', 'changes'), 21)\n",
      "(('constructs', 'containing'), 21)\n",
      "(('active', 'forms'), 21)\n",
      "(('receptor', '(pdgfr)'), 21)\n",
      "(('6', '10'), 21)\n",
      "(('contrast,', 'three'), 21)\n",
      "(('cancer', 'reported'), 21)\n",
      "(('resistance', 'mechanism'), 21)\n",
      "(('mutation,', 'suggesting'), 21)\n",
      "(('levels', 'figure'), 21)\n",
      "(('expression', 'proteins'), 21)\n",
      "(('mutation', 'data'), 21)\n",
      "(('patients', 'using'), 21)\n",
      "(('may', 'allow'), 21)\n",
      "(('to:', 'methods'), 21)\n",
      "(('association', 'cancer'), 21)\n",
      "(('clinicopathological', 'features'), 21)\n",
      "(('differences', 'cell'), 21)\n",
      "(('specific', 'gene'), 21)\n",
      "(('c-terminal', 'portion'), 21)\n",
      "(('lipofectamine', 'plus'), 21)\n",
      "(('plays', 'key'), 21)\n",
      "(('potent', 'activity'), 21)\n",
      "(('alk', 'mutation'), 21)\n",
      "(('shown', 'inhibit'), 21)\n",
      "(('studies', 'assessed'), 21)\n",
      "(('study,', 'characterized'), 21)\n",
      "(('inhibits', 'proliferation'), 21)\n",
      "(('small-molecule', 'tyrosine'), 21)\n",
      "(('target', 'inhibition'), 21)\n",
      "(('n-terminal', 'c-terminal'), 21)\n",
      "(('aml', 'samples'), 21)\n",
      "(('flt3-itd', 'mutations'), 21)\n",
      "(('exon', '13,'), 21)\n",
      "(('currently', 'tested'), 21)\n",
      "(('remaining', 'variants'), 21)\n",
      "(('mouse', 'xenograft'), 21)\n",
      "(('western', 'analysis'), 21)\n",
      "(('wt', 'p53'), 21)\n",
      "(('p53', 'proteins'), 21)\n",
      "(('11%', 'patients'), 21)\n",
      "(('(n', '3),'), 21)\n",
      "(('results', 'demonstrated'), 21)\n",
      "(('dramatic', 'reduction'), 21)\n",
      "(('4', 'h'), 21)\n",
      "(('gene', 'transfer'), 21)\n",
      "(('control', 'samples'), 21)\n",
      "(('negligible', 'effect'), 21)\n",
      "(('helix', '1'), 21)\n",
      "(('helix', '2,'), 21)\n",
      "(('performed', 'yeast'), 21)\n",
      "(('2', ''), 21)\n",
      "(('(table', 'iii)'), 21)\n",
      "(('loop', 'mutants'), 21)\n",
      "(('similar', 'wild'), 21)\n",
      "(('deletion', 'amino'), 21)\n",
      "(('two', 'scenarios'), 21)\n",
      "(('mutation', 'induces'), 21)\n",
      "(('heteromeric', 'complex'), 21)\n",
      "(('structures', 'reveal'), 21)\n",
      "(('mice', 'also'), 21)\n",
      "(('mammary', 'tumor'), 21)\n",
      "(('protein', 'could'), 21)\n",
      "(('form', 'heterodimers'), 21)\n",
      "(('exhibit', 'increased'), 21)\n",
      "(('amplification', 'present'), 21)\n",
      "(('impact', 'protein'), 21)\n",
      "(('biological', 'activity'), 21)\n",
      "(('transcriptional', 'repression'), 21)\n",
      "(('patients', 'acute'), 21)\n",
      "(('protein', 'degradation'), 21)\n",
      "(('negatively', 'affect'), 21)\n",
      "(('patients', 'affected'), 21)\n",
      "(('major', 'cause'), 21)\n",
      "(('variants', 'reduced'), 21)\n",
      "(('familial', 'cases'), 21)\n",
      "(('carboxy', 'terminal'), 21)\n",
      "(('epigenetic', 'changes'), 21)\n",
      "(('erbb2', 'kinase'), 21)\n",
      "(('agar', 'colony'), 21)\n",
      "(('erbb2', 'mutations'), 21)\n",
      "(('consistent', 'immunoblot'), 21)\n",
      "(('combination', 'agents,'), 21)\n",
      "(('pten', 'loss'), 21)\n",
      "(('consistent', 'observed'), 21)\n",
      "(('whole-cell', 'lysates'), 21)\n",
      "(('surrounding', 'matrix'), 21)\n",
      "(('hierarchical', 'clustering'), 21)\n",
      "(('cellular', 'ic50'), 21)\n",
      "(('1b)', 'results'), 21)\n",
      "(('a,', 'n'), 21)\n",
      "(('increase', 'proliferation'), 21)\n",
      "(('predicted', 'disrupt'), 21)\n",
      "(('forms', 'part'), 21)\n",
      "(('may', 'explained'), 21)\n",
      "(('tcga', 'cohort'), 21)\n",
      "(('0', 'moreover,'), 21)\n",
      "(('(fig', 'ic50'), 21)\n",
      "(('using', 'abi'), 21)\n",
      "(('mutations', 'individuals'), 21)\n",
      "(('analyzed', 'complete'), 21)\n",
      "(('cycle', 'sequencing'), 21)\n",
      "(('manner', '(figure'), 21)\n",
      "(('confirming', 'functional'), 21)\n",
      "(('allele', 'breast'), 21)\n",
      "(('well', 'tyrosine'), 21)\n",
      "(('phosphorylation', 'induced'), 21)\n",
      "(('analyzed', 'effects'), 21)\n",
      "(('patient', '14'), 21)\n",
      "(('found', '40'), 21)\n",
      "(('androgen', 'receptor'), 21)\n",
      "(('break', 'repair'), 21)\n",
      "(('line', 'ba/f3'), 21)\n",
      "(('dna', 'using'), 21)\n",
      "(('mutations,', 'analyzed'), 21)\n",
      "(('pdgfra', 'isoforms'), 21)\n",
      "(('2', 'yes'), 21)\n",
      "(('mutation', 'juxtamembrane'), 21)\n",
      "(('activation', 'status'), 21)\n",
      "(('crenolanib', 'patients'), 21)\n",
      "(('rabbit', 'anti-human'), 21)\n",
      "(('found', 'braf'), 21)\n",
      "(('pathogenesis', 'gastrointestinal'), 21)\n",
      "(('kit', 'platelet-derived'), 21)\n",
      "(('gists', 'express'), 21)\n",
      "(('transfected', 'mutated'), 21)\n",
      "(('gist', 'cell'), 21)\n",
      "(('compounds', 'may'), 21)\n",
      "(('structural', 'analyses,'), 21)\n",
      "(('activation', 'mutations'), 21)\n",
      "(('second', 'common'), 21)\n",
      "(('clinical', 'study,'), 21)\n",
      "(('used', 'experiments'), 21)\n",
      "(('moreover,', 'overexpression'), 21)\n",
      "(('protein', 'products'), 21)\n",
      "(('juxtamembrane', '(jm)'), 21)\n",
      "(('lead', 'loss'), 21)\n",
      "(('methods', 'used'), 21)\n",
      "(('associated', 'unfavorable'), 21)\n",
      "(('consistent', 'hypothesis'), 21)\n",
      "(('(31,', '32)'), 21)\n",
      "(('pi3k-akt', 'pathway'), 21)\n",
      "(('band', 'detected'), 21)\n",
      "(('mutations', 'leading'), 21)\n",
      "(('compared', 'tumors'), 21)\n",
      "(('4', '9'), 21)\n",
      "(('cancers', 'well,'), 21)\n",
      "(('49,', '50'), 21)\n",
      "(('cell', 'system'), 21)\n",
      "(('experiment', 'showed'), 21)\n",
      "(('brafv600e', 'mutation'), 21)\n",
      "(('distributed', 'along'), 21)\n",
      "(('protein', 'compared'), 21)\n",
      "(('mutations', 'met'), 21)\n",
      "(('activate', 'mapk'), 21)\n",
      "(('however,', 'encouraging'), 21)\n",
      "(('individuals', 'mutation'), 21)\n",
      "(('cultured', 'cell'), 21)\n",
      "(('es', 'cells'), 21)\n",
      "(('cell', 'lineage'), 21)\n",
      "(('4', 'different'), 21)\n",
      "(('figures', '3'), 21)\n",
      "(('deletion', 'removes'), 21)\n",
      "(('carrying', 'mutations'), 21)\n",
      "(('analysis', 'impact'), 21)\n",
      "(('mlh1', 'expression'), 21)\n",
      "(('proteins', 'common'), 21)\n",
      "(('accessible', 'surface'), 21)\n",
      "(('regulatory', 'domains'), 21)\n",
      "(('basel,', 'switzerland)'), 21)\n",
      "(('frequencies', 'loss'), 21)\n",
      "(('1', 'schematic'), 21)\n",
      "(('highly', 'activating'), 21)\n",
      "(('mutations', 'intracellular'), 21)\n",
      "(('leads', 'ligand-independent'), 21)\n",
      "(('conserved', 'domains:'), 21)\n",
      "(('also', 'stimulated'), 21)\n",
      "(('loop', '(a-loop)'), 21)\n",
      "(('findings', 'confirm'), 21)\n",
      "(('al', 'recent'), 21)\n",
      "(('mutation', 'alk'), 21)\n",
      "(('frequently', 'occurring'), 21)\n",
      "(('flt3-tkd', 'point'), 21)\n",
      "(('strategy', 'treatment'), 21)\n",
      "(('substantial', 'proportion'), 21)\n",
      "(('flt3', 'inhibitor'), 21)\n",
      "(('flt3-itd', 'cell'), 21)\n",
      "(('sensitivity', 'flt3'), 21)\n",
      "(('different', 'flt3'), 21)\n",
      "(('flt3', 'found'), 21)\n",
      "(('flt3-wt', 'cells'), 21)\n",
      "(('introduced', 'wild'), 21)\n",
      "(('compared', 'flt3-wt'), 21)\n",
      "(('mutations', 'kit,'), 21)\n",
      "(('mutation-negative', 'individuals'), 21)\n",
      "(('three', 'conserved'), 21)\n",
      "(('activity', '(supplementary'), 21)\n",
      "(('interestingly,', 'presence'), 21)\n",
      "(('mutant', 'melanoma'), 21)\n",
      "(('carry', 'deleterious'), 21)\n",
      "(('p', '5'), 21)\n",
      "(('reported', 'outside'), 21)\n",
      "(('a375', 'cells'), 21)\n",
      "(('serum', 'growth'), 21)\n",
      "(('acquired', 'enhanced'), 21)\n",
      "(('studies', 'one'), 21)\n",
      "(('following', 'three'), 21)\n",
      "(('features', 'predictive'), 21)\n",
      "(('function', 'panel'), 21)\n",
      "(('obtained', 'analyzing'), 21)\n",
      "(('(s10n,', 'l42r,'), 21)\n",
      "(('brca2-positive', 'breast'), 21)\n",
      "(('observed', 'loh'), 21)\n",
      "(('5', 'model'), 21)\n",
      "(('loh', 'wild-type'), 21)\n",
      "(('moderate-risk', 'variant'), 21)\n",
      "(('(b)', 's1655f,'), 21)\n",
      "(('variants;', '(c)'), 21)\n",
      "(('l1546n,', 'l1564p,'), 21)\n",
      "(('l1764p,', 'i1766s'), 21)\n",
      "(('likely', 'neutral;'), 21)\n",
      "(('l1664p,', 'v1804d,'), 21)\n",
      "(('indicate', '(a)'), 21)\n",
      "(('represent', 'deleterious/high-risk'), 21)\n",
      "(('variants;', '(d)'), 21)\n",
      "(('neutral;', '(b)'), 21)\n",
      "(('brca1', 'variants,'), 21)\n",
      "(('deleterious/high-risk', 'variants;'), 21)\n",
      "(('g1788d', 'represent'), 21)\n",
      "(('instead', 'deleterious'), 21)\n",
      "(('s1655f,', 't1700a,'), 21)\n",
      "(('(a)', 'v1534m,'), 21)\n",
      "(('e1794d', 'represent'), 21)\n",
      "(('p1806a', 'represent'), 21)\n",
      "(('represent', 'moderate-risk'), 21)\n",
      "(('using', 'threshold,'), 21)\n",
      "(('high-risk', 'variants,'), 21)\n",
      "(('l1564p,', 'l1664p,'), 21)\n",
      "(('(c)', 'variants'), 21)\n",
      "(('moderate-risk', 'variants;'), 21)\n",
      "(('i1766s', 'represent'), 21)\n",
      "(('threshold,', 'data'), 21)\n",
      "(('variant', 'instead'), 21)\n",
      "(('q1785h,', 'e1794d'), 21)\n",
      "(('e1644g,', 'q1785h,'), 21)\n",
      "(('(d)', 'variant'), 21)\n",
      "(('r1443g,', 'l1546n,'), 21)\n",
      "(('variant', 'g1788d'), 21)\n",
      "(('high-risk', 'variants'), 21)\n",
      "(('v1534m,', 'p1614s,'), 21)\n",
      "(('deleterious', 'one'), 21)\n",
      "(('represent', 'high-risk'), 21)\n",
      "(('brca2', 'breast'), 21)\n",
      "(('analysis', 'brca1'), 21)\n",
      "(('brca1', 'breast'), 21)\n",
      "(('k', 'n'), 21)\n",
      "(('g', 'loop'), 21)\n",
      "(('upstream', 'brct'), 21)\n",
      "(('thus', 'predicted'), 21)\n",
      "(('', 'sheet'), 21)\n",
      "(('suggesting', 'probably'), 21)\n",
      "(('likelihood', 'ratios'), 21)\n",
      "(('increased', 'lifetime'), 21)\n",
      "(('making', 'clinical'), 21)\n",
      "(('high-risk', 'individuals'), 21)\n",
      "(('mutation-positive', 'mutation-negative'), 21)\n",
      "(('models', 'loh'), 21)\n",
      "(('brca2', 'confer'), 21)\n",
      "(('wild-type', 'allele;'), 21)\n",
      "(('mutation', 'trans'), 21)\n",
      "(('expected', 'lethal'), 21)\n",
      "(('brca2', 'deleterious'), 21)\n",
      "(('likelihood', 'detecting'), 21)\n",
      "(('earlier', 'stage'), 21)\n",
      "(('identify', 'tumor'), 21)\n",
      "(('11', 'deletion'), 21)\n",
      "(('imatinib-resistant', 'kit'), 21)\n",
      "(('activated', 'kit'), 21)\n",
      "(('np,', 'performed'), 21)\n",
      "(('kit', 'secondary'), 21)\n",
      "(('heterozygous', 'kit'), 21)\n",
      "(('kit', 'allele'), 21)\n",
      "(('cells', 'three'), 20)\n",
      "(('cases', '(fig'), 20)\n",
      "(('ring', 'finger'), 20)\n",
      "(('b-cell', 'lymphoma'), 20)\n",
      "(('induced', 'rapid'), 20)\n",
      "(('mutant', 'able'), 20)\n",
      "(('cells', 'starved'), 20)\n",
      "(('induction', 'apoptosis'), 20)\n",
      "(('treated', '2'), 20)\n",
      "(('time', 'course'), 20)\n",
      "(('(figure', '4b),'), 20)\n",
      "(('proliferation,', 'survival,'), 20)\n",
      "(('next', 'analyzed'), 20)\n",
      "(('strongly', 'expressed'), 20)\n",
      "(('samples', 'detected'), 20)\n",
      "(('family', 'proteins'), 20)\n",
      "(('risk', 'developing'), 20)\n",
      "(('proteins', 'showed'), 20)\n",
      "(('heterozygous', 'germline'), 20)\n",
      "(('region', 'chromosome'), 20)\n",
      "(('leukemia', 'samples'), 20)\n",
      "(('however,', 'found'), 20)\n",
      "(('asterisks', 'denote'), 20)\n",
      "(('increased', 'colony'), 20)\n",
      "(('contained', 'either'), 20)\n",
      "(('region', 'containing'), 20)\n",
      "(('human', 'melanoma'), 20)\n",
      "(('mutations', 'remains'), 20)\n",
      "(('1', 'supplementary'), 20)\n",
      "(('basal', 'squamous'), 20)\n",
      "(('tumours', 'harbour'), 20)\n",
      "(('mutations', 'positions'), 20)\n",
      "(('mutations,', 'one'), 20)\n",
      "(('activated', 'braf'), 20)\n",
      "(('amplification', 'defined'), 20)\n",
      "(('frequently', 'detected'), 20)\n",
      "(('chromosomal', 'region'), 20)\n",
      "(('address', 'issues'), 20)\n",
      "(('support', 'model'), 20)\n",
      "(('(figure', '3a),'), 20)\n",
      "(('compared', 'two'), 20)\n",
      "(('report', 'patient'), 20)\n",
      "(('studies', 'demonstrate'), 20)\n",
      "(('performed', 'gene'), 20)\n",
      "(('gene', 'reported'), 20)\n",
      "(('colonies', 'compared'), 20)\n",
      "(('well', 'cells'), 20)\n",
      "(('tumor-derived', 'mutations'), 20)\n",
      "(('fishers', 'exact'), 20)\n",
      "(('normal', 'tissues,'), 20)\n",
      "(('activity', 'significantly'), 20)\n",
      "(('gap', 'activity'), 20)\n",
      "(('increased', 'activation'), 20)\n",
      "(('mutant', 'wt'), 20)\n",
      "(('collection', 'refinement'), 20)\n",
      "(('diverse', 'cellular'), 20)\n",
      "(('mutant', 'alone'), 20)\n",
      "(('cellular', 'growth'), 20)\n",
      "(('transfected', '293t'), 20)\n",
      "(('mutation', 'increased'), 20)\n",
      "(('(a)', 'immunoblot'), 20)\n",
      "(('cells,', 'examined'), 20)\n",
      "(('value', 'patients'), 20)\n",
      "(('activation', 'observed'), 20)\n",
      "(('3', 'weeks'), 20)\n",
      "(('leads', 'constitutive'), 20)\n",
      "(('found', 'evidence'), 20)\n",
      "(('changes', 'observed'), 20)\n",
      "(('blood', 'lymphocytes'), 20)\n",
      "(('ii', 'table'), 20)\n",
      "(('mutation', 'affecting'), 20)\n",
      "(('tumors', 'examined'), 20)\n",
      "(('signaling', 'context'), 20)\n",
      "(('patients', 'metastatic'), 20)\n",
      "(('dna', 'sequences'), 20)\n",
      "(('resulted', 'complete'), 20)\n",
      "(('rapid', 'amplification'), 20)\n",
      "(('amplification', 'fish'), 20)\n",
      "(('found', '15'), 20)\n",
      "(('initiation', 'codon'), 20)\n",
      "(('centrosome', 'amplification'), 20)\n",
      "(('regulatory', 'subunit'), 20)\n",
      "(('lung', 'carcinoma,'), 20)\n",
      "(('mechanisms,', 'including'), 20)\n",
      "(('performed', 'vitro'), 20)\n",
      "(('105', 'cells'), 20)\n",
      "(('messenger', 'rna'), 20)\n",
      "(('u2os', 'cells,'), 20)\n",
      "(('cyclin-dependent', 'kinase'), 20)\n",
      "(('e', 'f'), 20)\n",
      "(('discussion', 'present'), 20)\n",
      "(('signals', 'growth'), 20)\n",
      "(('transfected', 'human'), 20)\n",
      "(('thus,', 'mutant'), 20)\n",
      "(('activity', 'however,'), 20)\n",
      "(('generated', 'stable'), 20)\n",
      "(('equal', 'amounts'), 20)\n",
      "(('relative', 'intensities'), 20)\n",
      "(('somatic', 'point'), 20)\n",
      "(('sequencing', 'studies'), 20)\n",
      "(('(figure', '3e)'), 20)\n",
      "(('may', 'involve'), 20)\n",
      "(('ha', 'tag'), 20)\n",
      "(('development', 'new'), 20)\n",
      "(('many', 'tumor'), 20)\n",
      "(('transcriptome', 'sequencing'), 20)\n",
      "(('lines', 'revealed'), 20)\n",
      "(('experiments', 'suggest'), 20)\n",
      "(('5', '1'), 20)\n",
      "(('tumors', 'show'), 20)\n",
      "(('presence', 'deletion'), 20)\n",
      "(('index', 'cases,'), 20)\n",
      "(('large', 'deletions,'), 20)\n",
      "(('analysis', 'carried'), 20)\n",
      "(('high-fidelity', 'dna'), 20)\n",
      "(('skipping', 'exon'), 20)\n",
      "(('mlh1', 'pms2'), 20)\n",
      "(('dna', 'available'), 20)\n",
      "(('induced', 'expression'), 20)\n",
      "(('base', 'pair'), 20)\n",
      "(('small', 'proportion'), 20)\n",
      "(('approximate', 'molecular'), 20)\n",
      "(('determine', 'relative'), 20)\n",
      "(('shown', 'previously,'), 20)\n",
      "(('showed', 'intermediate'), 20)\n",
      "(('mutations', 'endometrial'), 20)\n",
      "(('mlh1', 'msh2,'), 20)\n",
      "(('mutations', 'evaluated'), 20)\n",
      "(('expression', 'due'), 20)\n",
      "(('hours', 'later'), 20)\n",
      "(('fig', 'together,'), 20)\n",
      "(('key', 'words:'), 20)\n",
      "(('cell', 'number'), 20)\n",
      "(('assay', 'also'), 20)\n",
      "(('induces', 'cell'), 20)\n",
      "(('table', '5)'), 20)\n",
      "(('many', 'mutations'), 20)\n",
      "(('reported', 'two'), 20)\n",
      "(('1a)', 'figure'), 20)\n",
      "(('cases', 'first'), 20)\n",
      "(('without', 'mutation'), 20)\n",
      "(('figure', 's2)'), 20)\n",
      "(('disorder', 'hematopoietic'), 20)\n",
      "(('heterozygous', 'point'), 20)\n",
      "(('effects', 'mutant'), 20)\n",
      "(('significant', 'enrichment'), 20)\n",
      "(('study', 'determine'), 20)\n",
      "(('similar', 'fashion'), 20)\n",
      "(('expressed', 'significantly'), 20)\n",
      "(('could', 'play'), 20)\n",
      "(('exon', '10,'), 20)\n",
      "(('patients,', 'clustered'), 20)\n",
      "(('blood', 'cells'), 20)\n",
      "(('0', '3'), 20)\n",
      "(('cells,', 'cells'), 20)\n",
      "(('recipient', 'mice'), 20)\n",
      "(('displayed', 'higher'), 20)\n",
      "(('controls', '(figure'), 20)\n",
      "(('primary', 'cells'), 20)\n",
      "(('whereas', 'mutations'), 20)\n",
      "(('study', 'aimed'), 20)\n",
      "(('al', 'found'), 20)\n",
      "(('trend', 'toward'), 20)\n",
      "(('randomized', 'phase'), 20)\n",
      "(('transfection', 'levels'), 20)\n",
      "(('together', 'published'), 20)\n",
      "(('chi-square', 'test'), 20)\n",
      "(('egfr,', 'del'), 20)\n",
      "(('reflect', 'less'), 20)\n",
      "(('drug', 'however,'), 20)\n",
      "(('tissue', 'adjacent'), 20)\n",
      "(('h3255', '(table'), 20)\n",
      "(('10-fold', 'lower'), 20)\n",
      "(('patients', 'contain'), 20)\n",
      "(('observed', 'l858r'), 20)\n",
      "(('positions', 'missense'), 20)\n",
      "(('egfr', '(wt)'), 20)\n",
      "(('course', 'treatment'), 20)\n",
      "(('antibody', 'l858r'), 20)\n",
      "(('demonstrated', 'partial'), 20)\n",
      "(('cleft', 'within'), 20)\n",
      "(('domain,', 'contrast'), 20)\n",
      "(('finally,', 'striking'), 20)\n",
      "(('egfr', 'particularly'), 20)\n",
      "(('inhibitor', 'initial'), 20)\n",
      "(('receptors', 'demonstrated'), 20)\n",
      "(('mutation,', 'loop'), 20)\n",
      "(('deletion', 'codons'), 20)\n",
      "(('murine', 'thymoma'), 20)\n",
      "(('(figs', 'compared'), 20)\n",
      "(('received', 'gefitinib'), 20)\n",
      "(('gefitinib', 'reported'), 20)\n",
      "(('receptor', 'l747p753inss'), 20)\n",
      "(('rapid', 'reliable'), 20)\n",
      "(('domain', 'makes'), 20)\n",
      "(('first', 'looked'), 20)\n",
      "(('mutant', '(l858r)'), 20)\n",
      "(('progressed', 'treatment'), 20)\n",
      "(('highlighted', 'yellow'), 20)\n",
      "(('mutations', 'reside'), 20)\n",
      "(('protein,', 'exon'), 20)\n",
      "(('receptor', 'vitro'), 20)\n",
      "(('l747-s752', 'l858r'), 20)\n",
      "(('egfr', 'molecules'), 20)\n",
      "(('caused', 'varying'), 20)\n",
      "(('lines', 'representing'), 20)\n",
      "(('t-egfr', 'another'), 20)\n",
      "(('avian', 'erythroblastosis'), 20)\n",
      "(('domain', 'findings'), 20)\n",
      "(('(5%)', 'mutations'), 20)\n",
      "(('former', 'current'), 20)\n",
      "(('suggest', 'l858r'), 20)\n",
      "(('mutation', 'cultured'), 20)\n",
      "(('selected', 'regions'), 20)\n",
      "(('related', 'receptors,'), 20)\n",
      "(('autophosphorylation', 'tyrosine'), 20)\n",
      "(('concentrations', 'tki,'), 20)\n",
      "(('gefitinib-induced', 'inhibition'), 20)\n",
      "(('finding', 'egfr'), 20)\n",
      "(('testing', 'guide'), 20)\n",
      "(('(l858r)', 'comparable'), 20)\n",
      "(('sensitivity', 'del'), 20)\n",
      "(('expanded', 'view'), 20)\n",
      "(('alter', 'atp'), 20)\n",
      "(('two', 'published'), 20)\n",
      "(('(amersham', 'pharmacia)'), 20)\n",
      "(('(egfr)', 'protein'), 20)\n",
      "(('grb2,', 'leads'), 20)\n",
      "(('dramatic', 'responses'), 20)\n",
      "(('l858r', 'missense'), 20)\n",
      "(('furthermore,', 'contrast'), 20)\n",
      "(('vitro', 'mutational'), 20)\n",
      "(('v-akt', 'murine'), 20)\n",
      "(('enhanced', 'activity,'), 20)\n",
      "(('population', 'tumors'), 20)\n",
      "(('inhibitor', 'found'), 20)\n",
      "(('regions', 'egfr'), 20)\n",
      "(('(puse,', 'upstate)'), 20)\n",
      "(('tumors', 'adenocarcinomas'), 20)\n",
      "(('receptors', 'data'), 20)\n",
      "(('sensitivities', 'wild-type'), 20)\n",
      "(('normal', 'egfr'), 20)\n",
      "(('ability', 'autophosphorylate'), 20)\n",
      "(('(bd', 'transduction'), 20)\n",
      "(('correlated', 'level'), 20)\n",
      "(('rearrangements', 'egfr'), 20)\n",
      "(('0', 'results'), 20)\n",
      "(('present', 'dna'), 20)\n",
      "(('patients', 'u,'), 20)\n",
      "(('varying', 'transfection'), 20)\n",
      "(('2)', 'taken'), 20)\n",
      "(('levels', 't-egfr'), 20)\n",
      "(('sensitivity', 'multiple'), 20)\n",
      "(('domain;', 'previously'), 20)\n",
      "(('cancer', '(23),'), 20)\n",
      "(('81', '(5%)'), 20)\n",
      "(('correlate', 'presence'), 20)\n",
      "(('explain', 'selective'), 20)\n",
      "(('analyses', 'mutant'), 20)\n",
      "(('proteins', 'study'), 20)\n",
      "(('egfr', 'produced'), 20)\n",
      "(('three', 'remaining'), 20)\n",
      "(('tumor', 'displayed'), 20)\n",
      "(('deletion', 'receptor'), 20)\n",
      "(('shown', 'use'), 20)\n",
      "(('anti-mouse', 'igg'), 20)\n",
      "(('sequencing', 'kinase'), 20)\n",
      "(('l747p753inss', 'deletion'), 20)\n",
      "(('specific', 'proteins'), 20)\n",
      "(('efficiencies,', 'equal'), 20)\n",
      "(('deletion', 'three-dimensional'), 20)\n",
      "(('quantified', 'measuring'), 20)\n",
      "(('inhibited', '10-fold'), 20)\n",
      "(('indicate', 'essential'), 20)\n",
      "(('egfr', 'common'), 20)\n",
      "(('receptor', 'l858r'), 20)\n",
      "(('overlapping', 'sets'), 20)\n",
      "(('characterize', 'biochemical'), 20)\n",
      "(('tumors', '(15)'), 20)\n",
      "(('also', 'sought'), 20)\n",
      "(('(11)', 'three'), 20)\n",
      "(('schematic', 'view'), 20)\n",
      "(('higher', 'del'), 20)\n",
      "(('adenocarcinoma', '(fig'), 20)\n",
      "(('occur', 'two'), 20)\n",
      "(('history,', 'egfr'), 20)\n",
      "(('ratio', 'p-egfr/t-egfr'), 20)\n",
      "(('egfr', 'key'), 20)\n",
      "(('within', 'codons'), 20)\n",
      "(('egfr', 'demonstrated'), 20)\n",
      "(('specific', 'activating'), 20)\n",
      "(('peroxidase', '(hrp)-conjugated'), 20)\n",
      "(('(iii)', 'mutations'), 20)\n",
      "(('gefitinib', 'revealed'), 20)\n",
      "(('protein', 'kinase/extracellular'), 20)\n",
      "(('compared', '1'), 20)\n",
      "(('t-egfr', 'likely'), 20)\n",
      "(('(table', '2table'), 20)\n",
      "(('gefitinib,', 'assessed'), 20)\n",
      "(('28', 'exons'), 20)\n",
      "(('del', 'l747-s752,'), 20)\n",
      "(('sensitivity', 'specific'), 20)\n",
      "(('125', 'patients'), 20)\n",
      "(('(v)', 'vector'), 20)\n",
      "(('even', 'addition'), 20)\n",
      "(('activation', 'gefitinib-induced'), 20)\n",
      "(('tkis,', 'measuring'), 20)\n",
      "(('tumors', 'japan'), 20)\n",
      "(('spanning', 'codons'), 20)\n",
      "(('observed', 'eight'), 20)\n",
      "(('results', 'y1092-specific'), 20)\n",
      "(('gefitinib-responsive', 'tumors'), 20)\n",
      "(('unselected', 'tumors,'), 20)\n",
      "(('size', 'mutant'), 20)\n",
      "(('activation', 'receptor,'), 20)\n",
      "(('egfr', 'receptors'), 20)\n",
      "(('egfr', 'inhibited'), 20)\n",
      "(('del-1', 'deletion'), 20)\n",
      "(('(19)', 'shown'), 20)\n",
      "(('p-egfr/t-egfr', 'lysates'), 20)\n",
      "(('inhibitor,', 'erlotinib,'), 20)\n",
      "(('assess', 'various'), 20)\n",
      "(('total', '95'), 20)\n",
      "(('egfr', 'present'), 20)\n",
      "(('encoded', 'mutations,'), 20)\n",
      "(('1092', 'exon'), 20)\n",
      "(('increase', 'egf-induced'), 20)\n",
      "(('tki,', 'del'), 20)\n",
      "(('multivariate', 'analyses'), 20)\n",
      "(('l747-s752', 'mutant'), 20)\n",
      "(('detection', 'mutant'), 20)\n",
      "(('virtually', 'indistinguishable'), 20)\n",
      "(('sensitivity', 'lung'), 20)\n",
      "(('sensitivity', 'h3255'), 20)\n",
      "(('gefitinib', 'achieved'), 20)\n",
      "(('similar', 'mechanisms'), 20)\n",
      "(('mutations', 'h3255'), 20)\n",
      "(('binding', 'adaptor'), 20)\n",
      "(('phosphotyrosine', 'position'), 20)\n",
      "(('horseradish', 'peroxidase'), 20)\n",
      "(('del', 'l747-s752'), 20)\n",
      "(('differences', 't-egfr'), 20)\n",
      "(('tk', 'domain;'), 20)\n",
      "(('gefitinib', 'somatic'), 20)\n",
      "(('promoter-driven', 'expression'), 20)\n",
      "(('genotype-directed', 'clinical'), 20)\n",
      "(('properties', 'encoded'), 20)\n",
      "(('induced', 'diminished'), 20)\n",
      "(('cancers,', 'egfr'), 20)\n",
      "(('lie', 'adjacent'), 20)\n",
      "(('selected', 'growth'), 20)\n",
      "(('activated', 'transforming'), 20)\n",
      "(('(figure', '3afigure'), 20)\n",
      "(('show', 'striking'), 20)\n",
      "(('l747-s752,', 'l858r'), 20)\n",
      "(('sequenced', 'analyzed'), 20)\n",
      "(('sequence', 'use'), 20)\n",
      "(('one', 'gefitinib-sensitive'), 20)\n",
      "(('2table', '2'), 20)\n",
      "(('tumors', '10'), 20)\n",
      "(('inserted', 'cytomegalovirus'), 20)\n",
      "(('encoding', 'tk'), 20)\n",
      "(('(ii)', 'mutations'), 20)\n",
      "(('showed', 'l858r'), 20)\n",
      "(('construct', '(puse,'), 20)\n",
      "(('using', 'anti-egfr'), 20)\n",
      "(('contrast', 'deletion'), 20)\n",
      "(('mutants', 'tkis,'), 20)\n",
      "(('tumors', 'four'), 20)\n",
      "(('egfr', 'assessed'), 20)\n",
      "(('(i)', 'tumor'), 20)\n",
      "(('effort', 'explain'), 20)\n",
      "(('finding', 'frequent'), 20)\n",
      "(('s3),', 'whereas'), 20)\n",
      "(('starting', 'nucleotide'), 20)\n",
      "(('patients', 'drug'), 20)\n",
      "(('wild-type,', 'del'), 20)\n",
      "(('obtained', '5'), 20)\n",
      "(('3afigure', '3'), 20)\n",
      "(('accounts', 'extraordinary'), 20)\n",
      "(('mutant', 'egfrs,'), 20)\n",
      "(('striking', 'correlation'), 20)\n",
      "(('four', '81'), 20)\n",
      "(('mutations', 'subset'), 20)\n",
      "(('tyrosine-phosphorylated', 'proteins'), 20)\n",
      "(('used', 'marker'), 20)\n",
      "(('therapy', 'without'), 20)\n",
      "(('21', 'also'), 20)\n",
      "(('(wt)', 'egfr,'), 20)\n",
      "(('carried', 'one'), 20)\n",
      "(('cancer', 'received'), 20)\n",
      "(('gefitinib', 'responders'), 20)\n",
      "(('thymoma', 'viral'), 20)\n",
      "(('21', '(table'), 20)\n",
      "(('required', 'suppress'), 20)\n",
      "(('l858r', 'dell747p753inss'), 20)\n",
      "(('h3255', 'cell'), 20)\n",
      "(('egfr', 'tumors'), 20)\n",
      "(('(fig', 'positions'), 20)\n",
      "(('account', 'apparent'), 20)\n",
      "(('mutations,', 'sequenced'), 20)\n",
      "(('inhibited', '10-fold'), 20)\n",
      "(('heterodimerization', 'partner'), 20)\n",
      "(('(tki)', 'gefitinib'), 20)\n",
      "(('second', 'wild-type'), 20)\n",
      "(('mutant,', 'results'), 20)\n",
      "(('contain', 'similar'), 20)\n",
      "(('glioblastomas', 'egfr'), 20)\n",
      "(('egfr', 'drug'), 20)\n",
      "(('signal-related', 'kinase'), 20)\n",
      "(('additional', 'lung'), 20)\n",
      "(('drug', 'phosphorylation'), 20)\n",
      "(('18', '24)'), 20)\n",
      "(('demonstrate', 'egfr'), 20)\n",
      "(('dell747p753inss', 'mutations'), 20)\n",
      "(('signaling', 'triggered'), 20)\n",
      "(('adjacent', 'activation'), 20)\n",
      "(('measuring', 'ratio'), 20)\n",
      "(('y1092-specific', 'antibody'), 20)\n",
      "(('demonstrated', 'autophosphorylation'), 20)\n",
      "(('egfr', 'and/or'), 20)\n",
      "(('treatment', 'investigate'), 20)\n",
      "(('fraction', 'egfr'), 20)\n",
      "(('serum-starved', 'pretreated'), 20)\n",
      "(('study,', 'confirm'), 20)\n",
      "(('histology', 'egfr'), 20)\n",
      "(('clustering', 'egfr'), 20)\n",
      "(('l858r', 'del-1'), 20)\n",
      "(('enzyme', 'likely'), 20)\n",
      "(('versus', 'wild-type'), 20)\n",
      "(('difference', 'patients'), 20)\n",
      "(('control', 'tumor'), 20)\n",
      "(('(table', '5)'), 20)\n",
      "(('known', 'egfr'), 20)\n",
      "(('neck', 'cancer'), 20)\n",
      "(('detected', 'immunoblotting'), 20)\n",
      "(('using', 'kaplan-meier'), 20)\n",
      "(('region', 'within'), 20)\n",
      "(('case', 'resulted'), 20)\n",
      "(('numerous', 'human'), 20)\n",
      "(('sequence', 'egfr'), 20)\n",
      "(('mutations', '(85%)'), 20)\n",
      "(('response', 'duration'), 20)\n",
      "(('nsclc', '(14)'), 20)\n",
      "(('would', 'clinical'), 20)\n",
      "(('response', 'mutation'), 20)\n",
      "(('rates', 'patients'), 20)\n",
      "(('although', 'egfr'), 20)\n",
      "(('also', 'higher'), 20)\n",
      "(('10', '000'), 20)\n",
      "(('reports', 'describing'), 20)\n",
      "(('two', 'hundred'), 20)\n",
      "(('contrast', 'primary'), 20)\n",
      "(('statistical', 'difference'), 20)\n",
      "(('number', 'metastatic'), 20)\n",
      "(('insertions', 'exon'), 20)\n",
      "(('using', 'wizard'), 20)\n",
      "(('little', 'information'), 20)\n",
      "(('cases', 'tested'), 20)\n",
      "(('therapy', 'erlotinib'), 20)\n",
      "(('report', 'egfr'), 20)\n",
      "(('however,', 'association'), 20)\n",
      "(('gefitinib', 'showed'), 20)\n",
      "(('achieving', 'partial'), 20)\n",
      "(('stage', 'iiib'), 20)\n",
      "(('previously', '(13,'), 20)\n",
      "(('complex', 'inhibitor'), 20)\n",
      "(('egfr', 'inhibitors,'), 20)\n",
      "(('lung', 'cell'), 20)\n",
      "(('gefitinib', 'presence'), 20)\n",
      "(('unlike', 'egfr'), 20)\n",
      "(('status', 'egfr,'), 20)\n",
      "(('reported', 'regarding'), 20)\n",
      "(('show', 'higher'), 20)\n",
      "(('conclusion,', 'reported'), 20)\n",
      "(('medical', 'center'), 20)\n",
      "(('egfr', 'due'), 20)\n",
      "(('21', 'patients'), 20)\n",
      "(('analysis', 'overall'), 20)\n",
      "(('shown', 'strongly'), 20)\n",
      "(('chemotherapy', 'regimens'), 20)\n",
      "(('predict', 'tumor'), 20)\n",
      "(('mutation', 'included'), 20)\n",
      "(('among', 'many'), 20)\n",
      "(('wizard', 'sv'), 20)\n",
      "(('patients', 'undergone'), 20)\n",
      "(('us', 'investigate'), 20)\n",
      "(('bearing', 'mutant'), 20)\n",
      "(('heat', 'shock'), 20)\n",
      "(('analyzed', '16'), 20)\n",
      "(('patients', 'conclusion,'), 20)\n",
      "(('reports', 'identified'), 20)\n",
      "(('comparison', 'patients'), 20)\n",
      "(('assays', 'showed'), 20)\n",
      "(('three', 'groups:'), 20)\n",
      "(('dramatically', 'increased'), 20)\n",
      "(('subjected', 'direct'), 20)\n",
      "(('resulting', 'reduced'), 20)\n",
      "(('material', 'available'), 20)\n",
      "(('tissues', 'using'), 20)\n",
      "(('additional', 'patients'), 20)\n",
      "(('l858r', 'mutation'), 20)\n",
      "(('parental', 'cell'), 20)\n"
     ]
    }
   ],
   "source": [
    "flat_list = [item for sublist in merged[\"bigramsfiltered\"].tolist() for item in set(sublist)]\n",
    "\n",
    "bigs = {}\n",
    "for i in flat_list:\n",
    "    bigs[i] = bigs.get(i,0)+1\n",
    "sortedbgs=sorted(bigs.items(), key=lambda c: c[1], reverse=True)\n",
    "for i in range(10000):\n",
    "    print(sortedbgs[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "776635"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sortedbgs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "% matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "freqs = []\n",
    "for (k,v) in sortedbgs:\n",
    "    freqs.append(np.log(v))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "counts = merged.groupby(\"Text\").count()\n",
    "sortedcounts = sorted(counts[\"ID\"].tolist(), reverse = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "abridgedsortedbgs = {}\n",
    "for (k,v) in sortedbgs:\n",
    "    if v< len(merged)*.5 and v>59:\n",
    "        abridgedsortedbgs[k]=v\n",
    "merged[\"bigramfeaturesfiltered\"]=merged[\"bigramsfiltered\"].map(lambda b: [int(word in b) for word in abridgedsortedbgs.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3862"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(merged[\"wordfeatures\"][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Class</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>split</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>...</th>\n",
       "      <th>wordsgenes</th>\n",
       "      <th>mutType</th>\n",
       "      <th>mutwordsgenes</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>sentenceWordBag</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>mutwordgenesSentences</th>\n",
       "      <th>bigrams</th>\n",
       "      <th>bigramfeatures</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>True</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[(Cyclin, M,), (M,, an), (an, orphan), (orphan...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>False</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>True</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>N454D</td>\n",
       "      <td>False</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>15 Apart from N454D, all residues affected by ...</td>\n",
       "      <td>[15, Apart, from, N454D, ,, all, residues, aff...</td>\n",
       "      <td>{'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[(15, Apart), (Apart, from), (from, N454D,), (...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>L399V</td>\n",
       "      <td>True</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>Finally, the third group constituted mutatio...</td>\n",
       "      <td>[Finally, ,, the, third, group, constituted, m...</td>\n",
       "      <td>{'Finally': 1, 'third': 1, 'group': 1, 'consti...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 0, ...</td>\n",
       "      <td>[(Finally,, the), (the, third), (third, group)...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID  Class    Gene                                               Text  \\\n",
       "0   0      1  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   1      2     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "2   2      2     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "3   3      3     CBL  Recent evidence has demonstrated that acquired...   \n",
       "4   4      4     CBL  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "              Variation  split  \\\n",
       "0  Truncating Mutations   True   \n",
       "1                 W802*  False   \n",
       "2                 Q249E   True   \n",
       "3                 N454D  False   \n",
       "4                 L399V   True   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "1                        ...                           \n",
       "2                        ...                           \n",
       "3                        ...                           \n",
       "4                        ...                           \n",
       "\n",
       "                                          wordsgenes                mutType  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 1, 0, 0]   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 1]   \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "4  [1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "\n",
       "                                       mutwordsgenes  \\\n",
       "0  [0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 0, ...   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "1   Using select c-CBL somatic mutations such as ...   \n",
       "2   Using select c-CBL somatic mutations such as ...   \n",
       "3  15 Apart from N454D, all residues affected by ...   \n",
       "4    Finally, the third group constituted mutatio...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "2  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "3  [15, Apart, from, N454D, ,, all, residues, aff...   \n",
       "4  [Finally, ,, the, third, group, constituted, m...   \n",
       "\n",
       "                                     sentenceWordBag  \\\n",
       "0  {'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...   \n",
       "1  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "2  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "3  {'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...   \n",
       "4  {'Finally': 1, 'third': 1, 'group': 1, 'consti...   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                               mutwordgenesSentences  \\\n",
       "0  [0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 0, ...   \n",
       "\n",
       "                                             bigrams  \\\n",
       "0  [(Cyclin, M,), (M,, an), (an, orphan), (orphan...   \n",
       "1  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "2  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "3  [(15, Apart), (Apart, from), (from, N454D,), (...   \n",
       "4  [(Finally,, the), (the, third), (third, group)...   \n",
       "\n",
       "                                      bigramfeatures  \n",
       "0  [1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...  \n",
       "1  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "2  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "4  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"mutwordgenesSentencesbgs\"]=merged[\"mutwordgenesSentences\"]+merged[\"bigramfeatures\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.951012080346\n",
      "Accuracy on training set:\n",
      "0.412082914614\n",
      "Accuracy on test set:\n",
      "0.938232281066\n",
      "Accuracy on training set:\n",
      "0.362910870674\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutwordgenesSentences\", \"Class\")\n",
    "logistictester(merged, \"mutwordgenesSentencesbgs\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.14894123866\n",
      "Accuracy on training set:\n",
      "0.864262017437\n",
      "Accuracy on test set:\n",
      "1.03921249338\n",
      "Accuracy on training set:\n",
      "0.144731577792\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"mutwordgenesSentencesbgs\", \"Class\", C=1e-3)\n",
    "logistictester(merged, \"mutwordgenesSentencesbgs\", \"Class\", C=1e-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.970707611738\n",
      "Accuracy on training set:\n",
      "0.357339259303\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"all\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>sentenceWordBag</th>\n",
       "      <th>bigrams</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>wordfeatures</th>\n",
       "      <th>mutType</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>bigramfeatures</th>\n",
       "      <th>all</th>\n",
       "      <th>split</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...</td>\n",
       "      <td>[(Cyclin, M,), (M,, an), (an, orphan), (orphan...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene                                               Text  \\\n",
       "0   0  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   1     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "\n",
       "              Variation  Class  \\\n",
       "0  Truncating Mutations      1   \n",
       "1                 W802*      2   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "1   Using select c-CBL somatic mutations such as ...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "\n",
       "                                     sentenceWordBag  \\\n",
       "0  {'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...   \n",
       "1  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "\n",
       "                                             bigrams  \\\n",
       "0  [(Cyclin, M,), (M,, an), (an, orphan), (orphan...   \n",
       "1  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        wordfeatures                mutType  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 1, 0, 0]   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 1]   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      bigramfeatures  \\\n",
       "0  [1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...   \n",
       "1  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                                 all  split  \n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  False  \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   True  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def logistictester(df, X, Y, C=1e-2):\n",
    "    training = df[df[\"split\"]]\n",
    "    test = df[df[\"split\"]==False]\n",
    "    X_train = training[X].tolist()\n",
    "    y_train = training[Y].tolist()\n",
    "    X_test=test[X].tolist()\n",
    "\n",
    "    lr = LogisticRegression( solver = \"lbfgs\",\n",
    "                                multi_class=\"multinomial\",\n",
    "                                penalty='l2',\n",
    "                                C=C,\n",
    "                                fit_intercept=True\n",
    "                                )\n",
    "    lr.fit(X_train, y_train)\n",
    "    y_pred = lr.predict_proba(X_test) \n",
    "    y_truth = \n",
    "    return y_pred"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "y_pred = logistictester(merged, \"all\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "691"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(y_pred)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(691, 9)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y_pred.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 0.21726421,  0.01403152,  0.00431362,  0.71930032,  0.00923844,\n",
       "         0.01001715,  0.01665731,  0.00424726,  0.00493017],\n",
       "       [ 0.03161272,  0.00675678,  0.00845566,  0.63130577,  0.23393571,\n",
       "         0.0654423 ,  0.01476029,  0.00333962,  0.00439114],\n",
       "       [ 0.08360373,  0.02692076,  0.01020205,  0.26481819,  0.01397709,\n",
       "         0.01301619,  0.56949028,  0.01007952,  0.00789219]])"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y_pred[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "test = merged[merged[\"split\"]==False]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(691, 20)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "truth = merged[merged[\"split\"]==False][\"Class\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(691,)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "truth.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     1\n",
       "6     5\n",
       "9     4\n",
       "10    4\n",
       "12    4\n",
       "Name: Class, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "truth.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "truth.iloc[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[  6.48947752e+01   5.67630178e+00   1.64435547e+00   2.95721198e+01\n",
      "    1.02130122e+01   5.88841904e+00   6.49792343e+00   7.75062800e-01\n",
      "    8.38030283e-01]\n",
      " [  7.22104919e+00   4.12944596e+01   1.11293273e+00   4.79002050e+00\n",
      "    3.21560404e+00   2.79120324e+00   3.51047585e+01   8.54621780e-01\n",
      "    6.15350459e-01]\n",
      " [  1.15325254e+00   6.04257599e-01   5.46266131e+00   5.46801460e+00\n",
      "    2.34805520e+00   4.72518212e-01   4.37520224e+00   5.47329895e-02\n",
      "    6.13053107e-02]\n",
      " [  3.28020434e+01   4.78623506e+00   2.85059023e+00   8.06058349e+01\n",
      "    5.52606101e+00   5.59542892e+00   9.06056300e+00   7.93044680e-01\n",
      "    9.80198796e-01]\n",
      " [  1.71328869e+01   2.19531218e+00   1.19815774e+00   7.85858416e+00\n",
      "    1.91238381e+01   3.51248025e+00   8.53011220e+00   2.28207819e-01\n",
      "    2.20420636e-01]\n",
      " [  4.85578901e+00   3.69969252e+00   1.41207090e+00   2.82593475e+00\n",
      "    3.40071482e+00   2.94456141e+01   4.78999114e+00   2.90792467e-01\n",
      "    2.79400235e-01]\n",
      " [  8.86516075e+00   2.61644816e+01   3.39275318e+00   8.20348525e+00\n",
      "    5.16381349e+00   4.22907663e+00   1.24261971e+02   9.75335724e-01\n",
      "    7.43922661e-01]\n",
      " [  6.42751469e-01   7.26753830e-01   6.18092379e-03   5.44666663e-02\n",
      "    7.92677364e-03   1.60738234e-02   4.83748546e-01   2.28564406e-02\n",
      "    3.92415277e-02]\n",
      " [  9.17159915e-01   6.26392674e-01   1.24010397e-01   1.19710323e+00\n",
      "    1.04331049e-01   1.97125488e-01   1.31880074e+00   4.93584470e-01\n",
      "    5.02149204e+00]]\n",
      "[ 126.   97.   20.  143.   60.   51.  182.    2.   10.]\n"
     ]
    }
   ],
   "source": [
    "#row of scores will correspond to the true class\n",
    "#columns will correspond to the predicted class\n",
    "scores = np.zeros((9,9))\n",
    "counts = np.zeros(9)\n",
    "for row, values in enumerate(y_pred):\n",
    "    trueclass = truth.iloc[row]-1\n",
    "    scores[trueclass]+=values\n",
    "    counts[trueclass]+=1\n",
    "print(scores)\n",
    "print(counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(9):\n",
    "    scores[i]/=counts[i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAPgAAAD8CAYAAABaQGkdAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADHJJREFUeJzt3X+o3fV9x/Hna0lcNbo6NF3XRBb/EEdmqbrg2jkc1XXo\nWpRB/1Box7ofMlhbHYVix6Bs/w1GqYVScGpXqFM6q6wUa2vR0pWtWWN0UxNlNtUmmc64YU1DNUbf\n++Meza1z3O+55/u5595Png+45J5zv3mf9wn3le/3fM/3vD+pKiT16Wfm3YCkdgy41DEDLnXMgEsd\nM+BSxwy41DEDLnXMgEsdM+BSx9a3KHr66amtW9OiNAC772979d2RptUXvL1x/Scb13+xcf12vz3H\n/GQFHqOVV4CqWvKfqUnAt24NO3duaFEagPPTNoKtwwFwb+P6f9K4/t7G9dv99hzz8Ao8RiuHB27n\nIbrUMQMudcyASx0z4FLHDLjUMQMudcyASx0bFPAklyZ5LMnjSa5r3ZSkcSwZ8CTrgM8ClwHbgKuS\nbGvdmKTZDdmDXwA8XlV7q+oIcBtwRdu2JI1hSMA3A/sW3d4/ue+nJLk6yc4kOw8edFKrtBqMdpKt\nqm6oqu1VtX3TppX4qICkpQwJ+AHgjEW3t0zuk7TKDQn494CzkpyZ5ATgSuArbduSNIYlPy5aVUeT\nfBj4OrAOuLmqHmnemaSZDfo8eFXdBdzVuBdJI/NKNqljBlzqmAGXOmbApY4ZcKljBlzqWKrGv278\n1KR+c/Sqx/zj7Q2LA6e9v219gLMa13+ucf2XGtf//kcbPwBw6mfa1n+lYe3DwMsD5qK7B5c6ZsCl\njhlwqWMGXOqYAZc6ZsCljhlwqWMGXOrYkLHJNyd5JslaXk5ZOi4N2YP/HXBp4z4kNbBkwKvq28D/\nrEAvkkbma3CpY4Nmsg2R5GrgaoATxyoqaSZNFj44YayikmbiIbrUsSFvk90K/AtwdpL9Sf6wfVuS\nxjBk4YOrVqIRSePzEF3qmAGXOmbApY4ZcKljBlzqmAGXOtZkLvopSf3q6FWPOdCwNsB/nN74AVbA\nXzzbtv4X2pbnnxrXB/iVFXiMVl7AuejScc+ASx0z4FLHDLjUMQMudcyASx0z4FLHDLjUsSEDH85I\ncl+S3UkeSXLNSjQmaXZDhi4eBT5WVbuSnALcn+SeqtrduDdJMxoyF/2pqto1+f4QsAfY3LoxSbOb\n6jV4kq3AecCOFs1IGtfguehJTga+DFxbVc+/wc9fm4v+s6O1J2kWg/bgSTawEO5bquqON9pm8Vz0\nDWN2KGnZhpxFD3ATsKeqPtW+JUljGbIHvxD4IHBxkgcnX7/TuC9JIxgyF/07wJIfLJe0+nglm9Qx\nAy51zIBLHTPgUscMuNQxAy51zIBLHWuy8MG5SX1j9KrHXNSwNsChxvUBDtTvNq3/9tzZtH5rb12B\nx1jLn5g6jAsfSMc9Ay51zIBLHTPgUscMuNQxAy51zIBLHRsy0eVNSf41yb9N5qL/5Uo0Jml2Q4Yu\nvghcXFU/nsxm+06Sr1XVdxv3JmlGQya6FPDjyc0Nk6/xL3+TNLqhU1XXJXkQeAa4p6rW8lV+0nFj\nUMCr6uWqOhfYAlyQ5JzXb5Pk6iQ7k+z877G7lLQsU51Fr6rngPuAS9/gZ6/NRT9trO4kzWTIWfRN\nSU6dfH8i8B7g0daNSZrdkLPovwh8Ick6Fv5D+FJVfbVtW5LGMOQs+r+zsOCgpDXGK9mkjhlwqWMG\nXOqYAZc6ZsCljhlwqWMGXOrYkAtdpi96Arxlc4vKC9b9oF3tlfLXjeeWP/TupuW5+L629a9oWx6A\nf16Bx5g39+BSxwy41DEDLnXMgEsdM+BSxwy41DEDLnVscMAngxcfSOKwB2mNmGYPfg2wp1UjksY3\ndGzyFuC9wI1t25E0pqF78E8DHwdeadiLpJENmar6PuCZqrp/ie1em4t+8OXR+pM0gyF78AuBy5M8\nAdwGXJzki6/faPFc9E3rRu5S0rIsGfCq+kRVbamqrcCVwL1V9YHmnUmame+DSx2b6vPgVfUt4FtN\nOpE0OvfgUscMuNQxAy51zIBLHTPgUscMuNQxAy51LFU1etFNSbWca32wYW2AhxvXB3hb4/qnNK5/\nV/1x0/p/lL9tWh/gS80foZ3DwMtVWWo79+BSxwy41DEDLnXMgEsdM+BSxwy41DEDLnXMgEsdGzTw\nYTKP7RDwMnC0qra3bErSOKaZ6PLuqnq2WSeSRuchutSxoQEv4JtJ7k9y9RttsHgu+k/G60/SDIYe\nov9GVR1I8hbgniSPVtW3F29QVTcAN8DCh01G7lPSMgzag1fVgcmfzwB3Ahe0bErSOIYsXbQxySmv\nfg/8NivziUpJMxpyiP4LwJ1JXt3+76vq7qZdSRrFkgGvqr3AO1agF0kj820yqWMGXOqYAZc6ZsCl\njhlwqWMGXOpYk7noJyd1zuhVj/m9hrUB/qpxfWg/F/1HjesfaVx/X4Pfy9fbmCXHiq9aL+BcdOm4\nZ8CljhlwqWMGXOqYAZc6ZsCljhlwqWMGXOrYoIAnOTXJ7UkeTbInybtaNyZpdkOHLl4P3F1V709y\nAnBSw54kjWTJgCd5M3AR8PsAVXWE9lcqShrBkEP0M4GDwOeTPJDkxsnwxZ+yeC76S6O3KWk5hgR8\nPXA+8LmqOg84DFz3+o2q6oaq2l5V2zeM3KSk5RkS8P3A/qraMbl9OwuBl7TKLRnwqnoa2Jfk7Mld\nlwC7m3YlaRRDz6J/BLhlcgZ9L/Chdi1JGsuggFfVg4BrgktrjFeySR0z4FLHDLjUMQMudcyASx0z\n4FLHDLjUsSYLH2z/udTOXxu97DHbGtYGNn6mbX2Atzau/3Tj+usa1/+vxvUBtjSu/6aGtQ8CR1z4\nQDq+GXCpYwZc6pgBlzpmwKWOGXCpYwZc6tiSAU9ydpIHF309n+TalWhO0myWHPhQVY8B5wIkWQcc\nAO5s3JekEUx7iH4J8P2qerJFM5LGNW3ArwRubdGIpPENDvhk4OLlwD/8Pz9/beGDg658IK0K0+zB\nLwN2VdUbfg5g8cIHm1z5QFoVpgn4VXh4Lq0pQ5cP3gi8B7ijbTuSxjR0Lvph4LTGvUgamVeySR0z\n4FLHDLjUMQMudcyASx0z4FLHDLjUsUHvg0/riUPwB99sUXnBvoa1oe0861d9oHH96xvX39i4/gqM\npuesxvW/+2K72tvfNWw79+BSxwy41DEDLnXMgEsdM+BSxwy41DEDLnVs6MCHP0vySJKHk9yaZCXe\nKpY0oyELH2wGPgpsr6pzWFj7/crWjUma3dBD9PXAiUnWAycB/9muJUljWTLgVXUA+Bvgh8BTwI+q\n6hutG5M0uyGH6D8PXAGcCbwN2Jjk/1xKvXgu+gvj9ylpGYYcov8W8IOqOlhVL7EwWfXXX7/R4rno\nnoGTVochAf8h8M4kJyUJC+uT7WnblqQxDHkNvgO4HdgFPDT5Ozc07kvSCIbORf8k8MnGvUgamVey\nSR0z4FLHDLjUMQMudcyASx0z4FLHDLjUsVTV+EWTg8CTU/yV04FnR29k5dj//K315zBt/79UVZuW\n2qhJwKeVZGdVbZ93H8tl//O31p9Dq/49RJc6ZsCljq2WgK/1D6/Y//yt9efQpP9V8RpcUhurZQ8u\nqYG5BjzJpUkeS/J4kuvm2ctyJDkjyX1Jdk/GSl8z756WI8m6JA8k+eq8e5lWklOT3J7k0SR7kgxc\nWHd1aD2SfG4BT7IO+CxwGbANuCrJtnn1s0xHgY9V1TbgncCfrsHnAHANa3dKz/XA3VX1y8A7WEPP\nYyVGks9zD34B8HhV7a2qI8BtLAx3XDOq6qmq2jX5/hALv1yb59vVdJJsAd4L3DjvXqaV5M3ARcBN\nAFV1pKqem29XU2s6knyeAd8M7Ft0ez9rLByLJdkKnAfsmG8nU/s08HHglXk3sgxnAgeBz09eYtyY\nZOO8mxpqJUaSe5JtBElOBr4MXFtVz8+7n6GSvA94pqrun3cvy7QeOB/4XFWdBxwG1sy5nKEjyWcx\nz4AfAM5YdHvL5L41JckGFsJ9S1XdMe9+pnQhcHmSJ1h4iXRxki/Ot6Wp7Af2TwaDwsJw0PPn2M+0\nBo0kn8U8A/494KwkZyY5gYWTC1+ZYz9Tm4yRvgnYU1Wfmnc/06qqT1TVlqraysK//71VNeoepKWq\nehrYl+TsyV2XALvn2NK0mo8kHzRVtYWqOprkw8DXWTh7eHNVPTKvfpbpQuCDwENJHpzc9+dVddcc\nezrefAS4ZbKT2At8aM79DFZVO5K8OpL8KPAAI1/R5pVsUsc8ySZ1zIBLHTPgUscMuNQxAy51zIBL\nHTPgUscMuNSx/wUZKhe4RxtfKAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x12a51ab38>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.imshow(scores, cmap='hot', interpolation='nearest')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "0, 3 lf \n",
    "1, 6 gf\n",
    "2, 4 neutral\n",
    "7, 8 switch\n",
    "inc 5\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAPgAAAD8CAYAAABaQGkdAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADDRJREFUeJzt3V+MpuVZx/Hvz9lFoCBoW6vsku42qRDSWCEjtmKaFJRA\n28CBHkBCExsTYtJWME0a2jMP9MiQ9oBUN0A1KUJ025qGIBRTmqaJYpc/WtilCSJ0dwV3iQLbjdtl\nl8uDeYHtSnmfd97nnnfm5vtJJsy8c+811wz72+fPPM/1pKqQ1KefWXQDktox4FLHDLjUMQMudcyA\nSx0z4FLHDLjUMQMudcyASx3b1KLoO05LbTuzReUVjx1sVxvgWNvyALy7cf3GPyJeblx/LRxddANz\neAWoqkxb1yTg286EXb/bovKK8/+iXW1oHw6AP2tc/y8b1/+vxvXXYtfymTX4Gq0cHrjOXXSpYwZc\n6pgBlzpmwKWOGXCpYwZc6pgBlzo2KOBJrkjygyRPJrmpdVOSxjE14EmWgFuAK4ELgGuTXNC6MUnz\nG7IFvxh4sqqeqqqjwF3A1W3bkjSGIQHfAuw94eN9k9d+QpLrk+xKsuvg/47VnqR5jHaSrap2VNVy\nVS2/87Sxqkqax5CA7wfOPeHjrZPXJK1zQwL+PeC9SbYnOQW4BvhG27YkjWHq7aJVdSzJp4D7gCXg\n9qp6vHlnkuY26H7wqroHuKdxL5JG5pVsUscMuNQxAy51zIBLHTPgUscMuNSxVNXoRU9P6ldGr/q6\nbzasDdCy91e9vXH9I43rH29c/+8b1we4fA2+RiuHgeMD5qK7BZc6ZsCljhlwqWMGXOqYAZc6ZsCl\njhlwqWMGXOrYkLHJtyc5kOSxtWhI0niGbMH/CriicR+SGpga8Kr6DvDfa9CLpJF5DC51bNBMtiGS\nXA9cD7B5rKKS5tLkwQej/ashaS7uoksdG/JrsjuBfwLOS7IvyR+0b0vSGIY8+ODatWhE0vjcRZc6\nZsCljhlwqWMGXOqYAZc6ZsCljjWZi74pqTNGr/q6VxrWBjincf210PruoMON6/9S4/oAz63B12jl\nCM5Fl97yDLjUMQMudcyASx0z4FLHDLjUMQMudcyASx0bMvDh3CQPJNmd5PEkN6xFY5LmN2R82jHg\nM1X1cJIzgYeS3F9Vuxv3JmlOQ+aiP1tVD0/ePwTsAba0bkzS/GY6Bk+yDbgQeLBFM5LGNXjCcZIz\ngK8CN1bVS2/w+dfmok+9Al7Smhh0N1mSzcDdwH1VdfO09d5NtnjeTTadd5MBSQLcBuwZEm5J68eQ\nY/BLgI8DlyZ5dPL2kcZ9SRrBkLno38XDamlD8ko2qWMGXOqYAZc6ZsCljhlwqWMGXOqYAZc6Nvha\n9Fm8B7ilReGJP2xYG+CFxvUBnnugbf3zP9y2/i+0Ld+8PsDBxvVbbj1/vA56kLRgBlzqmAGXOmbA\npY4ZcKljBlzqmAGXOjZkosupSf4lyb9O5qL/yVo0Jml+Qy50+TFwaVX9aDKb7btJ/qGq/rlxb5Lm\nNGSiSwE/mny4efI2fVKjpIUbdAyeZCnJo8AB4P6qci66tAEMCnhVHa+qXwO2Ahcned/Ja5Jcn2RX\nkl0vjt2lpFWZ6Sx6Vb0APABc8Qaf21FVy1W1fNZY3Umay5Cz6O9Mcvbk/dOA3wGeaN2YpPkNOYv+\ny8BfJ1li5R+Ev62qu9u2JWkMQ86i/xsrDxyUtMF4JZvUMQMudcyASx0z4FLHDLjUMQMudcyASx1r\nMhf9f4CdLQpPHGpYG+Bo4/oAv954bvmVbcs3/f8LcHbj+gDHN3D9obdzugWXOmbApY4ZcKljBlzq\nmAGXOmbApY4ZcKljgwM+Gbz4SBKHPUgbxCxb8BuAPa0akTS+oWOTtwIfBW5t246kMQ3dgn8B+Czw\nSsNeJI1syFTVjwEHquqhKetem4t+ZLT2JM1jyBb8EuCqJE8DdwGXJvnKyYtOnIt+6shNSlqdqQGv\nqs9V1daq2gZcA3yrqq5r3pmkufl7cKljM90PXlXfBr7dpBNJo3MLLnXMgEsdM+BSxwy41DEDLnXM\ngEsdM+BSx1I1dMLycD+b1NbRq76u9dzy1nPXAVpfzrvUuP7Njet/vnF9gOfW4Gu0cgQ4XpVp69yC\nSx0z4FLHDLjUMQMudcyASx0z4FLHDLjUMQMudWzQwIfJPLZDrDzT/FhVLbdsStI4Zpno8uGqer5Z\nJ5JG5y661LGhAS/gH5M8lOT6N1pw4lx0n44grQ9Dd9F/q6r2J/lF4P4kT1TVd05cUFU7gB2wcrPJ\nyH1KWoVBW/Cq2j/57wHg68DFLZuSNI4hjy56W5IzX30fuBx4rHVjkuY3ZBf9XcDXk7y6/m+q6t6m\nXUkaxdSAV9VTwPvXoBdJI/PXZFLHDLjUMQMudcyASx0z4FLHDLjUsSZz0c9I6ldHr/q66xrWBvjT\nxvUBzmlc/4XG9V9uXP/pBn8vT/ZzmTpWfN06jHPRpbc8Ay51zIBLHTPgUscMuNQxAy51zIBLHTPg\nUscGBTzJ2Ul2JnkiyZ4kH2zdmKT5DR26+EXg3qr6vSSnAKc37EnSSKYGPMlZwIeA3weoqqPA0bZt\nSRrDkF307cBB4MtJHkly62T44k84cS566+uUJQ0zJOCbgIuAL1XVhaxc537TyYuqakdVLVfV8uaR\nm5S0OkMCvg/YV1UPTj7eyUrgJa1zUwNeVc8Be5OcN3npMmB3064kjWLoWfRPA3dMzqA/BXyiXUuS\nxjIo4FX1KOAzwaUNxivZpI4ZcKljBlzqmAGXOmbApY4ZcKljBlzqWJMHHywl9f/uRtlAji+6AWmK\nI/jgA+ktz4BLHTPgUscMuNQxAy51zIBLHTPgUsemBjzJeUkePeHtpSQ3rkVzkuYz04UuSZaA/cBv\nVNUzP22dF7pIbbW60OUy4N/fLNyS1o9ZA34NcGeLRiSNb3DAJwMXrwL+7qd8/rUHH4x/dbuk1Rh8\nDJ7kauCTVXX5tLUeg0tttTgGvxZ3z6UNZdAWfPIssh8C76mqF6etdwsutTV0Cz50Lvph4O3zNiVp\nbXklm9QxAy51zIBLHTPgUscMuNQxAy51zIBLHRv0e3CNb3Pj+i83rt9a658PwCuN67+rYe29A9e5\nBZc6ZsCljhlwqWMGXOqYAZc6ZsCljhlwqWODAp7kj5M8nuSxJHcmObV1Y5LmN+TBB1uAPwKWq+p9\nwBIr01UlrXNDd9E3Aacl2QScDvxnu5YkjWVqwKtqP/DnrMxkexZ4saq+2boxSfMbsov+88DVwHbg\nHOBtSa57g3XORZfWmSG76L8N/EdVHayql4GvAb958qKq2lFVy1W1PHXUo6Q1MSTgPwQ+kOT0JGHl\n+WR72rYlaQxDjsEfBHYCDwPfn/yZHY37kjSCmR4fPJQPPpjO+8HfnPeDv7m9wJEGjw+WtIEYcKlj\nBlzqmAGXOmbApY4ZcKljBlzqWJO56K/A84fgmRn+yDuA51v0skbsf/HW3fdwaLbls/b/7iGLmlzo\nMqsku6pqedF9rJb9L95G/x5a9e8uutQxAy51bL0EfKPfvGL/i7fRv4cm/a+LY3BJbayXLbikBhYa\n8CRXJPlBkieT3LTIXlYjyblJHkiyezJW+oZF97QaSZaSPJLk7kX3MqskZyfZmeSJJHuSfHDRPc2i\n9UjyhQU8yRJwC3AlcAFwbZILFtXPKh0DPlNVFwAfAD65Ab8HgBvYuFN6vgjcW1XnA+9nA30fazGS\nfJFb8IuBJ6vqqao6CtzFynDHDaOqnq2qhyfvH2LlL9eWxXY1myRbgY8Cty66l1klOQv4EHAbQFUd\nraoXFtvVzJqOJF9kwLewMpjiVfvYYOE4UZJtwIXAg4vtZGZfAD5L+wEnLWwHDgJfnhxi3JpkwwwT\nWouR5J5kG0GSM4CvAjdW1UuL7meoJB8DDlTVQ4vuZZU2ARcBX6qqC4HDwIY5lzN0JPk8Fhnw/cC5\nJ3y8dfLahpJkMyvhvqOqvrbofmZ0CXBVkqdZOUS6NMlXFtvSTPYB+yaDQWFlOOhFC+xnVoNGks9j\nkQH/HvDeJNuTnMLKyYVvLLCfmU3GSN8G7Kmqmxfdz6yq6nNVtbWqtrHy8/9WVY26BWmpqp4D9iY5\nb/LSZcDuBbY0q+YjyZvcTTZEVR1L8ingPlbOHt5eVY8vqp9VugT4OPD9JI9OXvt8Vd2zwJ7eaj4N\n3DHZSDwFfGLB/QxWVQ8meXUk+THgEUa+os0r2aSOeZJN6pgBlzpmwKWOGXCpYwZc6pgBlzpmwKWO\nGXCpY/8HSiQYGX1kWYYAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x135a1a6a0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for i in range(9):\n",
    "    scores[i]*=counts[i]\n",
    "plt.imshow(scores, cmap='hot', interpolation='nearest')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'X_test' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-52-dbc6beba3b50>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX_train\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_train\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m \u001b[0my_predxgb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpredict_proba\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX_test\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'X_test' is not defined"
     ]
    }
   ],
   "source": [
    "model = xgb.XGBClassifier(eval_metric=\"mlogloss\", max_depth=6)\n",
    "\n",
    "training = merged[merged[\"split\"]]\n",
    "test = merged[merged[\"split\"]==False]\n",
    "X_train = np.array(training[\"all\"].tolist())\n",
    "y_train = np.array(training[\"Class\"].tolist())\n",
    "\n",
    "model.fit(X_train, y_train)\n",
    "\n",
    "X_test=test[\"all\"].tolist()\n",
    "y_predxgb = model.predict_proba(X_test) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "X_test=test[\"all\"].tolist()\n",
    "y_predxgb = model.predict_proba(X_test) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAPgAAAD8CAYAAABaQGkdAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADCBJREFUeJzt3V2oZfV5x/HvrzNj1YmNtrGhdaQzghhESJWDTWoJVFsx\nNSiFXigktFLwJrFaAsG0N22velFCUgiBQU0LsUqrEYJYrSWmaaCdZnxpfRkD02mSmYkyppI42k7H\nGZ9enK2Og3Gvffb6733O3+8HDp6XNc95tsxv1sv+r2elqpDUp59adgOS2jHgUscMuNQxAy51zIBL\nHTPgUscMuNQxAy51zIBLHdvcouj7tqa2n9Wi8qonD7arDXCsbXkAzm5c/+XG9Y82rr+IPc+rC/gd\nrbwGVFWmbdck4NvPgt03tag8qX9ru9oAL7YtD8DvNq7/z43r729c/9TG9QF+sIDf0cqRgdt5iC51\nzIBLHTPgUscMuNQxAy51zIBLHTPgUscGBTzJVUm+k2RvksbvQksay9SAJ9kEfBH4KHAhcH2SC1s3\nJml+Q/bglwJ7q2pfVR0F7gaubduWpDEMCfg5vHVl4oHJ994iyY1JdifZ/cIrY7UnaR6jXWSrqp1V\ntVJVK2dvHauqpHkMCfhB4NwTvt42+Z6kdW5IwL8NnJ9kR5JTgOuAr7VtS9IYpt4uWlXHknwKeAjY\nBNxRVU8370zS3AbdD15VDwAPNO5F0shcySZ1zIBLHTPgUscMuNQxAy51zIBLHUtVjV70p5PaNnrV\nNz3SsDbARY3rA/xs4/pDx+quVeuZ4vc2rg9w9QJ+RytHgOMD5qK7B5c6ZsCljhlwqWMGXOqYAZc6\nZsCljhlwqWMGXOrYkLHJdyQ5lOSpRTQkaTxD9uB/BVzVuA9JDUwNeFV9E3hxAb1IGpnn4FLHBs1k\nGyLJjcCNoxaVNJcmDz7wsEBaH8yi1LEhb5PdBfwLcEGSA0l+v31bksYw5MEH1y+iEUnj8xBd6pgB\nlzpmwKWOGXCpYwZc6pgBlzrWZC76pqS2jl71Tccb1gY4v3F9gMON67/SuH7r/s9oXB/av4aWnIsu\nyYBLPTPgUscMuNQxAy51zIBLHTPgUscMuNSxIQMfzk3ySJJnkjyd5OZFNCZpfkPmIx4DPl1VjyU5\nA3g0ycNV9Uzj3iTNachc9Oeq6rHJ54eBPcA5rRuTNL+ZzsGTbAcuBna1aEbSuAaPME/yHuBe4Jaq\neultfv7GXPSpK+AlLcSgu8mSbAHuBx6qqs9N2967yabzbrJ35t1k72y0u8mSBLgd2DMk3JLWjyHn\n4JcBnwAuT/LE5OO3GvclaQRD5qJ/C0+rpQ3JlWxSxwy41DEDLnXMgEsdM+BSxwy41DEDLnVs8Fr0\nWZwL/HGLwhN/1rA2wP7G9QH++7629bf/dtv6W9qWX8hy4XfDHVPuwaWOGXCpYwZc6pgBlzpmwKWO\nGXCpYwZc6tiQiS6nJvm3JP8+mYv+p4toTNL8hix0+T/g8qp6eTKb7VtJ/r6q/rVxb5LmNGSiSwEv\nT77cMvmYPqlR0tINOgdPsinJE8Ah4OGqejes8pM2vEEBr6rjVfXLwDbg0iQXnbxNkhuT7E6yeyOP\no5V6MtNV9Kr6EfAIcNXb/GxnVa1U1coiZlpLmm7IVfSzk5w5+fw04DeBZ1s3Jml+Q66i/wLw10k2\nsfoPwt9W1f1t25I0hiFX0f+D1QcOStpgXMkmdcyASx0z4FLHDLjUMQMudcyASx0z4FLHmsxFPwz8\nU4vCE//TsDbAq43rA3yg8dzyG9qW5y8b13+xcf13C/fgUscMuNQxAy51zIBLHTPgUscMuNQxAy51\nbHDAJ4MXH0/isAdpg5hlD34zsKdVI5LGN3Rs8jbgauC2tu1IGtPQPfjngc8ArzXsRdLIhkxV/Rhw\nqKoenbLdG3PRj4zWnqR5DNmDXwZck+S7wN3A5Um+cvJGJ85FP3XkJiWtzdSAV9Vnq2pbVW0HrgO+\nXlUfb96ZpLn5PrjUsZnuB6+qbwDfaNKJpNG5B5c6ZsCljhlwqWMGXOqYAZc6ZsCljhlwqWOpqtGL\nnpLU+0ev+qbWd7wcblwfoPVy3i2N6/954/p/0rg+wPML+B2tHAGOV2Xadu7BpY4ZcKljBlzqmAGX\nOmbApY4ZcKljBlzqmAGXOjZo4MNkHtth4DhwrKpWWjYlaRyzTHT59ar6YbNOJI3OQ3SpY0MDXsA/\nJnk0yY1vt8GJc9F9OoK0Pgw9RP+1qjqY5OeBh5M8W1XfPHGDqtoJ7ITVm01G7lPSGgzag1fVwcl/\nDwH3AZe2bErSOIY8umhrkjNe/xy4EniqdWOS5jfkEP39wH1JXt/+b6rqwaZdSRrF1IBX1T7ggwvo\nRdLIfJtM6pgBlzpmwKWOGXCpYwZc6pgBlzrWZC76aUmdN3rVN13ZsDbAnY3rw+rigpZaz3Y/0rj+\n8w3+Xp7sZzJ1rPhcjjes7Vx0SQZc6pkBlzpmwKWOGXCpYwZc6pgBlzpmwKWODQp4kjOT3JPk2SR7\nkny4dWOS5jd06OIXgAer6neSnAKc3rAnSSOZGvAk7wU+AvweQFUdBY62bUvSGIYcou8AXgC+nOTx\nJLdNhi++xYlz0VuuwZU03JCAbwYuAb5UVRcDrwC3nrxRVe2sqpWqWtk0cpOS1mZIwA8AB6pq1+Tr\ne1gNvKR1bmrAq+p5YH+SCybfugJ4pmlXkkYx9Cr6TcCdkyvo+4Ab2rUkaSyDAl5VTwA+E1zaYFzJ\nJnXMgEsdM+BSxwy41DEDLnXMgEsdM+BSx5o8+GBTUqeOXvWE+g1rQ9uB9dIYfPCBJAMu9cyASx0z\n4FLHDLjUMQMudcyASx2bGvAkFyR54oSPl5LcsojmJM1npoUuSTYBB4Ffqarv/aTtXOgitdVqocsV\nwH++U7glrR+zBvw64K4WjUga3+CATwYuXgP83U/4+RsPPhh/dbuktRh8Dp7kWuCTVXXltG09B5fa\nanEOfj0enksbyqA9+ORZZN8HzquqH0/b3j241NbQPfjQueivAD83b1OSFsuVbFLHDLjUMQMudcyA\nSx0z4FLHDLjUMQMudWzQ++DrjQtR+rdlAb/j1cb1z2tYe+/A7dyDSx0z4FLHDLjUMQMudcyASx0z\n4FLHDLjUsUEBT/KHSZ5O8lSSu5K0nOcgaSRDHnxwDvAHwEpVXcTqQJXrWjcmaX5DD9E3A6cl2Qyc\nDvygXUuSxjI14FV1EPgLVmeyPQf8uKr+oXVjkuY35BD9LOBaYAfwi8DWJB9/m+2ciy6tM0MO0X8D\n+K+qeqGqXgW+CvzqyRtV1c6qWqmqlamjHiUtxJCAfx/4UJLTk4TV55PtaduWpDEMOQffBdwDPAY8\nOfkzOxv3JWkEMz0+eKjWDz5Q/7wf/J3tBf63weODJW0gBlzqmAGXOmbApY4ZcKljBlzqmAGXOtbk\nffAkLwDfm+GPvA/44eiNLI79L99Gfw2z9v9LVXX2tI2aBHxWSXZX1cqy+1gr+1++jf4aWvXvIbrU\nMQMudWy9BHyj37xi/8u30V9Dk/7XxTm4pDbWyx5cUgNLDXiSq5J8J8neJLcus5e1SHJukkeSPDMZ\nK33zsntaiySbkjye5P5l9zKrJGcmuSfJs0n2JPnwsnuaReuR5EsLeJJNwBeBjwIXAtcnuXBZ/azR\nMeDTVXUh8CHgkxvwNQDczMad0vMF4MGq+gDwQTbQ61jESPJl7sEvBfZW1b6qOgrczepwxw2jqp6r\nqscmnx9m9S/XOcvtajZJtgFXA7ctu5dZJXkv8BHgdoCqOlpVP1puVzNrOpJ8mQE/B9h/wtcH2GDh\nOFGS7cDFwK7ldjKzzwOfAV5bdiNrsAN4Afjy5BTjtiRbl93UUIsYSe5FthEkeQ9wL3BLVb207H6G\nSvIx4FBVPbrsXtZoM3AJ8KWquhh4Bdgw13KGjiSfxzIDfhA494Svt02+t6Ek2cJquO+sqq8uu58Z\nXQZck+S7rJ4iXZ7kK8ttaSYHgAOTwaCwOhz0kiX2M6tBI8nnscyAfxs4P8mOJKewenHha0vsZ2aT\nMdK3A3uq6nPL7mdWVfXZqtpWVdtZ/f//9aoadQ/SUlU9D+xPcsHkW1cAzyyxpVk1H0m+ecxis6iq\nY0k+BTzE6tXDO6rq6WX1s0aXAZ8AnkzyxOR7f1RVDyyxp3ebm4A7JzuJfcANS+5nsKraleT1keTH\ngMcZeUWbK9mkjnmRTeqYAZc6ZsCljhlwqWMGXOqYAZc6ZsCljhlwqWP/D7RpGRhrm56tAAAAAElF\nTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x12adc3ef0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#row of scores will correspond to the true class\n",
    "#columns will correspond to the predicted class\n",
    "scores = np.zeros((9,9))\n",
    "counts = np.zeros(9)\n",
    "for row, values in enumerate(y_predxgb):\n",
    "    trueclass = truth.iloc[row]-1\n",
    "    scores[trueclass]+=values\n",
    "    counts[trueclass]+=1\n",
    "#print(scores)\n",
    "#print(counts)\n",
    "\n",
    "plt.imshow(scores, cmap='hot', interpolation='nearest')\n",
    "plt.show()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAPgAAAD8CAYAAABaQGkdAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADGtJREFUeJzt3W+MZfVdx/H3x2EJy0K6KrRRlnRpbCDYSCEb0rqmidAa\nsA190geQtEbUrA9sC6axoSba+MQH2jStSUOyYUGTIqgUkqbhj5hSm0bddvmjhd0lIqWwK3S3Nsgf\nhWWXrw/mAlOyZs6de3737vz6fiWTnXvnzPd+72Q/8zv3zLnfk6pCUp9+atENSGrHgEsdM+BSxwy4\n1DEDLnXMgEsdM+BSxwy41DEDLnXspBZFzzjj5Nq6dWOL0gA8cv9zzWoDvNK0+rK3N67/o8b1/7dx\n/XmsPC/P4TFaeRWoqqy2XZOAb926kT17trcoDcAv5q5mtQEONq2+7E8b1/+bxvX3Nq5/SuP6AP8+\nh8do5aWB27mLLnXMgEsdM+BSxwy41DEDLnXMgEsdM+BSxwYFPMllSR5N8liS61o3JWkcqwY8yRLw\nJeBy4HzgqiTnt25M0uyGrOAXA49V1eNVdQS4Ffhw27YkjWFIwM8Cnlpx+8Dkvh+TZEeSPUn2HD58\nZKz+JM1gtINsVbWzqrZV1bYzzzx5rLKSZjAk4AeBs1fc3sJ83o8haUZDAv4d4J1JzklyMnAl8NW2\nbUkaw6pvF62qo0k+DtwDLAE3VtUjzTuTNLNB7wevqjuBOxv3ImlknskmdcyASx0z4FLHDLjUMQMu\ndcyASx1LVY1e9LSkfmn0qm/4p5saFgc2X922PsA7Gtd/fp3Xf+a3Gz8AsGlX2/pLDWu/CBwbMBfd\nFVzqmAGXOmbApY4ZcKljBlzqmAGXOmbApY4ZcKljQ8Ym35jkUJKH59GQpPEMWcH/EriscR+SGlg1\n4FX1TeBHc+hF0sh8DS51bNBMtiGS7AB2ADgVXToxNLnwwYaxikqaibvoUseG/JnsFuCfgXOTHEgy\nh3fqShrDkAsfXDWPRiSNz110qWMGXOqYAZc6ZsCljhlwqWMGXOpYk7noG5P6hdGrvuEHDWsDHNrR\n+AGA53a2rf8Xbcvzucb1/7hxfYA/msNjtPISzkWXfuIZcKljBlzqmAGXOmbApY4ZcKljBlzqmAGX\nOjZk4MPZSe5LsjfJI0mumUdjkmY3ZOjiUeBTVfVAktOB+5PcW1V7G/cmaUZD5qI/XVUPTD5/HtgH\nnNW6MUmzm+o1eJKtwIXA7hbNSBrX4LnoSU4DvgJcW1XPHefrr89Fd2yydGIYtIIn2cByuG+uqtuP\nt83KuehLY3Yoac2GHEUPsAvYV1Wfb9+SpLEMWcG3Ax8DLkny0OTj1xv3JWkEQ+aifwtY9Y3lkk48\nnskmdcyASx0z4FLHDLjUMQMudcyASx0z4FLHmlz44Lykdo1e9Q2/0bA2wPON6wMcqoub1j8v325a\n/5Wm1eGCxvUB7pnDY7TihQ8kGXCpZwZc6pgBlzpmwKWOGXCpYwZc6tiQiS6nJPl2kn+dzEX/k3k0\nJml2Q4YuvgxcUlUvTGazfSvJXVX1L417kzSjIRNdCnhhcnPD5GP8098kjW7oVNWlJA8Bh4B7q8q5\n6NI6MCjgVXWsqt4NbAEuTvKuN2+TZEeSPUn2PDt2l5LWZKqj6FX1LHAfcNlxvvb6XPTNY3UnaSZD\njqKfmWTz5PONwAeA/a0bkzS7IUfRfw74qyRLLP9C+Nuq+lrbtiSNYchR9H9j+YKDktYZz2STOmbA\npY4ZcKljBlzqmAGXOmbApY4ZcKljQ050mdppwPaGvzo2vNquNsznt95vNZ5bvv8PmpbnvD9vW//c\ntuWB9T0XfShXcKljBlzqmAGXOmbApY4ZcKljBlzqmAGXOjY44JPBiw8mcdiDtE5Ms4JfA+xr1Yik\n8Q0dm7wF+CBwQ9t2JI1p6Ar+BeDTQOOTRCWNachU1Q8Bh6rq/lW2e30u+uHR2pM0iyEr+HbgiiRP\nALcClyT58ps3WjkX/cyRm5S0NqsGvKo+U1VbqmorcCXw9ar6aPPOJM3Mv4NLHZvq/eBV9Q3gG006\nkTQ6V3CpYwZc6pgBlzpmwKWOGXCpYwZc6pgBlzqWqhq96M8kdenoVd/wXw1rAzzauD5A69N539a4\n/j11edP6v5O7mtYHuKX5I7TzEnCsKqtt5woudcyASx0z4FLHDLjUMQMudcyASx0z4FLHDLjUsUED\nHybz2J4HjgFHq2pby6YkjWOaiS6/WlU/bNaJpNG5iy51bGjAC/iHJPcn2XG8DVbORX95vP4kzWDo\nLvqvVNXBJG8F7k2yv6q+uXKDqtoJ7ITlN5uM3KekNRi0glfVwcm/h4A7gItbNiVpHEMuXbQpyemv\nfQ78GvBw68YkzW7ILvrbgDuSvLb9X1fV3U27kjSKVQNeVY8DF8yhF0kj889kUscMuNQxAy51zIBL\nHTPgUscMuNSxJnPRNyb1jtGrvuH9DWvDfOZlt56L/mLj+v/TuP6hBv8v32xTVh0rfsJyLrokAy71\nzIBLHTPgUscMuNQxAy51zIBLHTPgUscGBTzJ5iS3JdmfZF+S97ZuTNLshg5d/CJwd1V9JMnJwKkN\ne5I0klUDnuQtwPuA3wSoqiPAkbZtSRrDkF30c4DDwE1JHkxyw2T44o9ZORf92OhtSlqLIQE/CbgI\nuL6qLmT5fQzXvXmjqtpZVduqatvSyE1KWpshAT8AHKiq3ZPbt7EceEknuFUDXlXPAE8lOXdy16XA\n3qZdSRrF0KPonwBunhxBfxy4ul1LksYyKOBV9RDgNcGldcYz2aSOGXCpYwZc6pgBlzpmwKWOGXCp\nYwZc6tjQE12m8lbgky0KT/zubQ2LAzd9pG19gJYXhgD4x8b1NzSu/+QcLkrQ+jmc3rD2DwZu5wou\ndcyASx0z4FLHDLjUMQMudcyASx0z4FLHVg14knOTPLTi47kk186jOUmzWfVEl6p6FHg3QJIl4CBw\nR+O+JI1g2l30S4H/qKrvt2hG0rimDfiVwC0tGpE0vsEBnwxcvAL4u//n669f+OCFsbqTNJNpVvDL\ngQeq6rjnua+88MFp4/QmaUbTBPwq3D2X1pWhlw/eBHwAuL1tO5LGNHQu+ovAzzbuRdLIPJNN6pgB\nlzpmwKWOGXCpYwZc6pgBlzpmwKWONZmL/gzwZy0KT1zfeG75sbblAdg/h8doqfXPaFfj+gCbG9d/\nor7XrPa2bVcM2s4VXOqYAZc6ZsCljhlwqWMGXOqYAZc6ZsCljg0d+PD7SR5J8nCSW5Kc0roxSbMb\ncuGDs4BPAtuq6l3AEsvTVSWd4Ibuop8EbExyEnAq8J/tWpI0llUDXlUHgc8BTwJPA/9dVX/fujFJ\nsxuyi/7TwIeBc4CfBzYl+ehxtnt9Lvqr4/cpaQ2G7KK/H/heVR2uqldYnqz6y2/eaOVcdA/NSyeG\nIVl8EnhPklOThOXrk+1r25akMQx5Db4buA14APju5Ht2Nu5L0giGzkX/LPDZxr1IGpkvl6WOGXCp\nYwZc6pgBlzpmwKWOGXCpYwZc6liqavyiyWHg+1N8yxnAD0dvZH7sf/HW+3OYtv+3V9WZq23UJODT\nSrKnqrYtuo+1sv/FW+/PoVX/7qJLHTPgUsdOlICv9zev2P/irffn0KT/E+I1uKQ2TpQVXFIDCw14\nksuSPJrksSTXLbKXtUhydpL7kuydjJW+ZtE9rUWSpSQPJvnaonuZVpLNSW5Lsj/JviTvXXRP02g9\nknxhAU+yBHwJuBw4H7gqyfmL6meNjgKfqqrzgfcAv7cOnwPANazfKT1fBO6uqvOAC1hHz2MeI8kX\nuYJfDDxWVY9X1RHgVpaHO64bVfV0VT0w+fx5lv9znbXYrqaTZAvwQeCGRfcyrSRvAd4H7AKoqiNV\n9exiu5pa05Hkiwz4WcBTK24fYJ2FY6UkW4ELgd2L7WRqXwA+DazHYbjnAIeBmyYvMW5IsmnRTQ01\nj5HkHmQbQZLTgK8A11bVc4vuZ6gkHwIOVdX9i+5ljU4CLgKur6oLgReBdXMsZ+hI8lksMuAHgbNX\n3N4yuW9dSbKB5XDfXFW3L7qfKW0HrkjyBMsvkS5J8uXFtjSVA8CByWBQWB4OetEC+5nWoJHks1hk\nwL8DvDPJOUlOZvngwlcX2M/UJmOkdwH7qurzi+5nWlX1maraUlVbWf75f72qRl1BWqqqZ4Cnkpw7\nuetSYO8CW5pW85Hkg6aqtlBVR5N8HLiH5aOHN1bVI4vqZ422Ax8Dvpvkocl9f1hVdy6wp580nwBu\nniwSjwNXL7ifwapqd5LXRpIfBR5k5DPaPJNN6pgH2aSOGXCpYwZc6pgBlzpmwKWOGXCpYwZc6pgB\nlzr2f6wYEMLwNoS3AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x12adc9390>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for i in range(9):\n",
    "    scores[i]/=counts[i]\n",
    "plt.imshow(scores, cmap='hot', interpolation='nearest')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.970707611738\n",
      "Accuracy on test set:\n",
      "0.915877740783\n",
      "Accuracy on test set:\n",
      "0.908283629449\n"
     ]
    }
   ],
   "source": [
    "mixedpred = ((y_pred)+(y_predxgb))/2\n",
    "\n",
    "for preds in [y_pred, y_predxgb, mixedpred]:\n",
    "    ls = []\n",
    "    for row,values in enumerate(preds):\n",
    "        trueclass = truth.iloc[row]-1\n",
    "        prob = values[trueclass]\n",
    "        ls.append(-np.log(prob+1e-6))\n",
    "\n",
    "    print(\"Accuracy on test set:\")\n",
    "    print(np.mean(ls))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged[\"Text\"]=merged[\"Text\"].str.lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>sentenceWordBag</th>\n",
       "      <th>bigrams</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>wordfeatures</th>\n",
       "      <th>mutType</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>bigramfeatures</th>\n",
       "      <th>all</th>\n",
       "      <th>split</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...</td>\n",
       "      <td>[(Cyclin, M,), (M,, an), (an, orphan), (orphan...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>CBL</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>{'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>CBL</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>[Recent, evidence, has, demonstrated, that, ac...</td>\n",
       "      <td>{'Recent': 1, 'evidence': 1, 'demonstrated': 1...</td>\n",
       "      <td>{'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...</td>\n",
       "      <td>15 Apart from N454D, all residues affected by ...</td>\n",
       "      <td>[15, Apart, from, N454D, ,, all, residues, aff...</td>\n",
       "      <td>{'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...</td>\n",
       "      <td>[(15, Apart), (Apart, from), (from, N454D,), (...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>CBL</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>[Oncogenic, mutations, in, the, monomeric, Cas...</td>\n",
       "      <td>{'Oncogenic': 1, 'mutations': 1, 'monomeric': ...</td>\n",
       "      <td>{'CBL': 1, 'EGFR': 1, 'KIT': 1}</td>\n",
       "      <td>Finally, the third group constituted mutatio...</td>\n",
       "      <td>[Finally, ,, the, third, group, constituted, m...</td>\n",
       "      <td>{'Finally': 1, 'third': 1, 'group': 1, 'consti...</td>\n",
       "      <td>[(Finally,, the), (the, third), (third, group)...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene                                               Text  \\\n",
       "0   0  FAM58A  cyclin-dependent kinases (cdks) regulate a var...   \n",
       "1   1     CBL   abstract background  non-small cell lung canc...   \n",
       "2   2     CBL   abstract background  non-small cell lung canc...   \n",
       "3   3     CBL  recent evidence has demonstrated that acquired...   \n",
       "4   4     CBL  oncogenic mutations in the monomeric casitas b...   \n",
       "\n",
       "              Variation  Class  \\\n",
       "0  Truncating Mutations      1   \n",
       "1                 W802*      2   \n",
       "2                 Q249E      2   \n",
       "3                 N454D      3   \n",
       "4                 L399V      4   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "2  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "3  [Recent, evidence, has, demonstrated, that, ac...   \n",
       "4  [Oncogenic, mutations, in, the, monomeric, Cas...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "2  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "3  {'Recent': 1, 'evidence': 1, 'demonstrated': 1...   \n",
       "4  {'Oncogenic': 1, 'mutations': 1, 'monomeric': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "2  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "3  {'JAK2': 1, 'CBL': 1, 'FLT3': 1, 'MPL': 1, 'NR...   \n",
       "4                    {'CBL': 1, 'EGFR': 1, 'KIT': 1}   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "1   Using select c-CBL somatic mutations such as ...   \n",
       "2   Using select c-CBL somatic mutations such as ...   \n",
       "3  15 Apart from N454D, all residues affected by ...   \n",
       "4    Finally, the third group constituted mutatio...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "2  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "3  [15, Apart, from, N454D, ,, all, residues, aff...   \n",
       "4  [Finally, ,, the, third, group, constituted, m...   \n",
       "\n",
       "                                     sentenceWordBag  \\\n",
       "0  {'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...   \n",
       "1  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "2  {'Using': 1, 'select': 1, 'c-CBL': 1, 'somatic...   \n",
       "3  {'15': 1, 'Apart': 1, 'N454D': 1, 'residues': ...   \n",
       "4  {'Finally': 1, 'third': 1, 'group': 1, 'consti...   \n",
       "\n",
       "                                             bigrams  \\\n",
       "0  [(Cyclin, M,), (M,, an), (an, orphan), (orphan...   \n",
       "1  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "2  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "3  [(15, Apart), (Apart, from), (from, N454D,), (...   \n",
       "4  [(Finally,, the), (the, third), (third, group)...   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        wordfeatures                mutType  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 1, 0, 0]   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 1]   \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "4  [1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 0]   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "2  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "4  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      bigramfeatures  \\\n",
       "0  [1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...   \n",
       "1  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "2  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "3  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "4  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                                 all  split  \n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  False  \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   True  \n",
       "2  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   True  \n",
       "3  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, ...   True  \n",
       "4  [1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, ...   True  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>sentenceWordBag</th>\n",
       "      <th>bigrams</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>wordfeatures</th>\n",
       "      <th>mutType</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>bigramfeatures</th>\n",
       "      <th>all</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...</td>\n",
       "      <td>[(Cyclin, M,), (M,, an), (an, orphan), (orphan...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene                                               Text  \\\n",
       "0   0  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "\n",
       "              Variation  Class  \\\n",
       "0  Truncating Mutations      1   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "\n",
       "                                           genebag  \\\n",
       "0  {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "\n",
       "                                     sentenceWordBag  \\\n",
       "0  {'Cyclin': 1, 'M': 1, 'orphan': 1, 'cyclin': 1...   \n",
       "\n",
       "                                             bigrams  \\\n",
       "0  [(Cyclin, M,), (M,, an), (an, orphan), (orphan...   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        wordfeatures                mutType  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 1, 0, 0]   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      bigramfeatures  \\\n",
       "0  [1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...   \n",
       "\n",
       "                                                 all  \n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def stemList(list):\n",
    "    stemmed = []\n",
    "    for i in list:\n",
    "        stemmed.append(sno.stem(i))\n",
    "    return stemmed\n",
    "\n",
    "def lowerList(list):\n",
    "    lowered = []\n",
    "    for i in list:\n",
    "        lowered.append(i.lower())\n",
    "    return lowered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "sno = nltk.stem.SnowballStemmer('english')\n",
    "merged[\"stemmed\"]=merged[\"tokenized\"].map(stemList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "merged[\"loweredtokens\"]=merged[\"tokenized\"].map(lowerList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>...</th>\n",
       "      <th>bigrams</th>\n",
       "      <th>genesmentioned</th>\n",
       "      <th>geneofinterest</th>\n",
       "      <th>wordfeatures</th>\n",
       "      <th>mutType</th>\n",
       "      <th>sentenceWordFeatures</th>\n",
       "      <th>bigramfeatures</th>\n",
       "      <th>all</th>\n",
       "      <th>stemmed</th>\n",
       "      <th>loweredtokens</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>Cyclin M, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>...</td>\n",
       "      <td>[(Cyclin, M,), (M,, an), (an, orphan), (orphan...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 1, 0, 0]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[cyclin-depend, kinas, (, cdks, ), regul, a, v...</td>\n",
       "      <td>[cyclin-dependent, kinases, (, cdks, ), regula...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>Using select c-CBL somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>...</td>\n",
       "      <td>[(Using, select), (select, c-CBL), (c-CBL, som...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1]</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>[abstract, background, non-smal, cell, lung, c...</td>\n",
       "      <td>[abstract, background, non-small, cell, lung, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene                                               Text  \\\n",
       "0   0  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   1     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "\n",
       "              Variation  Class  \\\n",
       "0  Truncating Mutations      1   \n",
       "1                 W802*      2   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   Cyclin M, an orphan cyclin, is the product of...   \n",
       "1   Using select c-CBL somatic mutations such as ...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "1                        ...                           \n",
       "\n",
       "                                             bigrams  \\\n",
       "0  [(Cyclin, M,), (M,, an), (an, orphan), (orphan...   \n",
       "1  [(Using, select), (select, c-CBL), (c-CBL, som...   \n",
       "\n",
       "                                      genesmentioned  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      geneofinterest  \\\n",
       "0  [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                        wordfeatures                mutType  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 1, 0, 0]   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...  [0, 0, 0, 0, 0, 0, 1]   \n",
       "\n",
       "                                sentenceWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                      bigramfeatures  \\\n",
       "0  [1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, ...   \n",
       "1  [1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                                 all  \\\n",
       "0  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "1  [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "\n",
       "                                             stemmed  \\\n",
       "0  [cyclin-depend, kinas, (, cdks, ), regul, a, v...   \n",
       "1  [abstract, background, non-smal, cell, lung, c...   \n",
       "\n",
       "                                       loweredtokens  \n",
       "0  [cyclin-dependent, kinases, (, cdks, ), regula...  \n",
       "1  [abstract, background, non-small, cell, lung, ...  \n",
       "\n",
       "[2 rows x 21 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "wordcounts = {}\n",
    "for i in merged[\"loweredtokens\"]:\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.9 and v>len(merged)*.1:\n",
    "        abridgedsorted[k]=v\n",
    "merged[\"tkfs\"]=merged[\"loweredtokens\"].map(lambda b: [int(word in b) for word in abridgedsorted.keys()])\n",
    "\n",
    "wordcounts = {}\n",
    "for i in merged[\"stemmed\"]:\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsortedst = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.9 and v>len(merged)*.1:\n",
    "        abridgedsortedst[k]=v\n",
    "merged[\"stfs\"]=merged[\"stemmed\"].map(lambda b: [int(word in b) for word in abridgedsortedst.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.941154965812\n",
      "Accuracy on training set:\n",
      "0.473622472708\n",
      "Accuracy on test set:\n",
      "0.95800805684\n",
      "Accuracy on training set:\n",
      "0.53279318227\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"tkfs\", \"Class\")\n",
    "logistictester(merged, \"stfs\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2623"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(merged[\"stfs\"][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3580"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(merged[\"tkfs\"][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['20', 'like', 'control', 'follow', 'tabl', 'same', 'dna', 'respect', 'line', 'region', 'when', 'acid', 'involv', 'could', 'affect', 'known', 'possibl', 'some', 'figur', 'type', 'thus', 'method', 'perform', '7', 'within', 'tumor', 'presenc', 'case', 'chang', 'repres', 'those', 'normal', 'becaus', 'direct', 'bind', '12', 'target', 'wild-typ', 'b', 'confirm', 'select', '15', 'remain', 'first', 'patient', 'requir', 'analyz', 'role', 'they', 'did', '8', 'amino', 'common', 'generat', 'potenti', 'base', 'whether', 'occur', 'sampl', '9', 'alter', 'obtain', 'predict', 'signal', 'factor', 'accord', 'c', 'wherea', 'independ', 'clinic', 'residu', '50', 'contrast', 'through', 'find', 'respons', 'major', 'total', 'structur', 'group', 'genet', 'experi', 'then', 'there', '14', 'time', 'reveal', 'higher', 'assess', 'molecular', 'under', 'interest', '11', '30', 'therefor', 'multipl', 'model', 'ani', 'locat', '16', 'examin', 'vitro', 'support', '13', 'investig', 'larg', 'caus', 'small', 'lead', 'loss', 'form', 'demonstr', 'would', 'growth', 'mechan', 'interact', 'reduc', 'measur', 'due', 'fig', 'inhibit', 'introduct', '18', 'singl', 'four', 'evid', '24', '19', 'least', 'point', 'encod', 'less', 'character', 'diseas', 'contribut', 'correspond', 'will', 'substitut', 'strong', 'mani', 'here', 'system', '25', 'low', '17', 'if', 'individu', 'set', 'might', 'do', 'analys', 'complex', 'famili', '21', 'complet', 'kinas', 'antibodi', 'initi', 'induc', 'construct', 'lack', 'avail', 'pathway', 'condit', 'carri', 'vector', '100', 'delet', 'regul', 'lower', 'against', 'without', 'absenc', 'anoth', 'et', 'correl', '22', 'combin', 'inhibitor', 'genom', 'evalu', 'togeth', 'phosphoryl', 'abil', 'negat', 'sensit', 'new', 'distinct', 'second', 'note', 'particular', 'transfect', 'promot', 'frequent', 'clear', 'among', 's', '[', ']', 'primari', 'transcript', 'expect', 'decreas', 'final', 'where', 'inform', 'abov', 'overal', 'furthermor', 'exampl', '40', 'tissu', '>', 'out', 'cultur', '26', 'materi', 'treatment', 'while', 'screen', 'deriv', 'product', 'depend', 'phenotyp', 'five', 'conserv', 'pcr', 'constitut', 'consid', 'rang', 'part', 'veri', 'even', 'allow', '<', 'limit', 'given', 'exon', 'over', 'cellular', 'appear', '27', 'clone', '23', 'frequenc', 'd', 'p', 'missens', 'current', 'critic', 'consequ', 'approxim', 'collect', 'variant', 'biolog', '29', '28', 'see', '2a', 'and/or', 'general', '1a', 'valu', 'format', 'differenti', 'downstream', 'comparison', 'establish', 'allel', '35', 'prolifer', 'relev', 'doe', 'should', 'rate', 'mean', 'defin', 'standard', 'process', 'concentr', 'local', '1b', 'except', 'various', 'blot', 'fact', 'oncogen', 'dure', 'per', 'subsequ', 'implic', 'treat', '32', 'code', 'novel', '31', 'next', 'befor', 'disrupt', 'approach', 'vivo', 'detail', '2b', 'i', 'display', '', 'need', 'stabil', 'abl', '3a', 'enhanc', 'taken', 'reaction', 'play', 'transform', 'altern', 'somat', 'conclus', 'incub', 'popul', 'appli', 'receptor', '33', 'western', 'subject', 'extract', 'statist', 'identif', 'confer', 'harbor', 'featur', 'calcul', 'mediat', 'moreov', 'them', 'day', 'mark', 'essenti', 'close', 'exhibit', 'ii', 'member', 'homolog', 'separ', '37', 'six', 'al', '34', 'mutagenesi', 'subset', 'unknown', 'progress', 'kit', 'key', '48', 'still', 'rare', 'prepar', 'design', 'understand', 'plasmid', 'status', 'partial', 'isol', 'refer', 'suffici', 'publish', 'stain', 'cdna', 'pattern', 'mous', 't', 'basi', 'inde', 'g', 'produc', 'fail', 'origin', 'overexpress', 'primer', 'h', 'inactiv', '=', 'experiment', '', 'conform', 'nucleotid', 'panel', '', 'malign', 'yet', 'make', 'event', 'maintain', 'size', 'ad', '3b', 'about', 'surviv', 'sinc', 'probabl', 'propos', '36', 'cycl', 'map', 'amplif', 'seen', 'characterist', 'end', 'resist', 'stabl', 'amount', '60', 'unlik', 'alon', 'wt', 'effici', 'ratio', 'research', 'exist', 'n', 'molecul', 'valid', 'distribut', 'lysat', 'few', 'plate', 'natur', 'reflect', 'gel', 'upon', 'institut', 'despit', \"'s\", 'therapeut', 'secondari', 'databas', 'medium', 'ident', 'review', 'buffer', 'state', 'via', 'chromosom', 'uniqu', 'imag', 'carcinoma', 'account', 'context', 'amplifi', 'tyrosin', 'explain', 'e', 'membran', 'block', 'seri', '4a', 'breast', 'technolog', 'act', 'rather', 'rna', 'supplement', 'pair', 'often', 'manufactur', 'regulatori', 'insert', 'work', 'so', 'green', 'laboratori', 'reduct', 'list', 'revers', 'chain', 'equal', 'link', 'therapi', 'modifi', 'impact', 'mice', 'suppressor', 'near', 'remov', 'suppress', 'red', 'led', '1c', 'cluster', 'none', 'c-termin', 'extens', 'serum', 'properti', 'who', 'integr', 'introduc', '4b', 's1', 'view', 'compon', 'endogen', 'substanti', 'hypothesi', 'focus', 'supplementari', '38', 'motif', 'year', 'seven', 'averag', 'go', 'blue', 'larger', 'fragment', 'repeat', 'codon', 'class', 'aberr', 'full', '39', 'insight', 'purifi', 'achiev', 'address', 'recombin', 'impair', 'prevent', 'made', 'probe', 'greater', 'univers', 'exclud', 'poor', 'defect', 'main', '90', '42', 'eight', '45', 'fraction', 'biochem', 'prior', 'help', 'truncat', '0', '80', 'hour', 'certain', 'below', 'drug', 'summari', 'up', 'right', '41', 'profil', '43', 'access', 'immunoblot', 'protocol', 'proport', 'manner', '2c', 'wide', '0.5', 'estim', 'necessari', 'central', 'substrat', 'earli', 'far', 'stimul', 'version', 'highlight', 'how', 'phase', 'seem', 'lung', '70', 'approv', 'retain', 'entir', 'mm', 'order', 'quantit', 'serv', 'littl', 'much', 'loop', 'mrna', 'strategi', 'notabl', 'degre', 'procedur', 'softwar', 'site-direct', 'splice', 'classifi', 'count', 'left', 'center', 'full-length', '', 'success', 's2', 'percentag', 'http', 'varieti', 'ca', 'exclus', 'section', 'crystal', 'age', 'us', 'core', 'appar', 'variabl', 'toward', 'reason', 'mammalian', 'n-termin', 'wild', '3c', 'term', '0.1', 'marker', 'polymorph', 'abnorm', 'fold', 'open', 'becom', 'gain', 'top', 'diagnosi', 'rapid', 'surfac', 'almost', 'fluoresc', 'grown', 'align', 'share', 'date', 'variat', 'now', 'restrict', 'recogn', 'compris', 'intens', 'influenc', 'nuclear', '44', 'advantag', 'syndrom', 'appropri', 'intern', 'third', 'wash', 'versus', 'side', 'transient', 'min', 'regard', 'reagent', 'across', 'enzym', 'germlin', 'creat', 'vari', 'verifi', 'score', '...', 'fusion', 'invitrogen', 'translat', 'hand', 'bound', 'contact', 'termin', 'divers', 'convers', 'background', 'transfer', '4c', 'affin', 'especi', 'f', 'unclear', 'visual', '5a', 'acquir', 'blood', 'fulli', 'catalyt', 'outcom', 'apoptosi', 'undergo', 'later', 'nine', 'band', 'predomin', 'whose', 'spectrum', 'facilit', 'along', 'recurr', 'half', 'mix', 'discoveri', 'preval', 'random', 'step', 'tumorigenesi', 'program', 'risk', 'inact', 'elev', 'author', '2000', 'lyse', '0.05', 'domin', 'slight', 'thought', 'polymeras', 'download', 'stage', 'whole', 'comprehens', 'patholog', 'improv', 'replac', 'bar', 'instruct', 'copi', 'extent', 'short', 'start', 'extend', 'therebi', 'area', 'accumul', 'volum', 'correct', 'ref', 'ph', 'last', 'agent', 'minim', 'criteria', 'receiv', '200', 'highest', 'advanc', 'stabli', 'unabl', 'consider', 'yield', 'length', 'alreadi', 'immunoprecipit', 'putat', 'inhibitori', 'solut', 'typic', 'ligand', 'relationship', 'r', 'hypothes', 'iii', '5b', 'unit', 'surpris', 'knowledg', 'modul', '46', '+', 'cohort', 'continu', 'harvest', 'transloc', 'better', 'santa', 's3', '47', 'abrog', 'latter', 'recruit', 'expos', 'induct', 'exact', 'primarili', 'cytoplasm', 'trial', 'aris', 'transduct', 'weak', 'dilut', 'quantifi', 'diagnos', 'onto', 'onc', '72', 'broad', 'nm', 'm', 'equival', 'leukemia', '150', 'futur', 'label', 'way', 'disord', 'lysi', 'g', '3d', 'peptid', 'explor', 'curv', 'difficult', 'coloni', 'organ', 'biotechnolog', '1d', 'agreement', 'load', 'embryon', 'promega', 'adjac', 'outsid', 'chemic', 'al.', 'temperatur', 'everi', 'week', 'interpret', 'intermedi', 'overlap', 'discov', 'roch', 'benefit', 'cruz', 'm', 'heterogen', 'document', 'heterozyg', 'parallel', 'histori', 'tool', 'damag', 'minor', 'favor', 'potent', 'question', 'light', 'element', 'effort', 'keyword', '95', 'death', 'suscept', 'public', 'seed', 'interfer', 'lesion', 'empti', 'illustr', 'elucid', 'rais', 'candid', 'fetal', 'qiagen', 'modif', 'presum', 'robust', 'plus', 'capabl', 'duplic', 'particip', '75', 'overnight', 'nor', 'monitor', 'partner', 'triplic', 'extracellular', 'dramat', 'incid', 'summar', 'strike', 'life', 'twice', 'deviat', 'search', 'reach', 'solid', 'schemat', 'compound', 'shift', 'done', 'turn', 'inc.', 'month', 'around', 'respond', 'repair', 'numer', 'conduct', 'basal', 'pathogen', 'comput', 'american', 'ovarian', 'emerg', 'articl', 'briefli', 'histolog', 'rearrang', 'extrem', 'precis', 'sigma', '49', 'phosphatas', 'exposur', 'locus', 'specimen', 'remark', 'explan', 'fibroblast', 'notion', 'fix', '52', 'incorpor', 'call', 'ma', 'pool', 'onlin', 'error', 'speci', 'enrich', 'lost', 'great', 'algorithm', 'long', 'monoclon', 'colon', 'grow', 'until', '1.5', 'definit', 'drive', 'classif', 'phosphat', '53', 'p53', 'upstream', 'doubl', 'densiti', 'again', 'nevertheless', 'de', 'perhap', 'medic', 'neither', 'issu', 'abolish', 's4', 'consent', 'bovin', 'instead', 'must', 'represent', 'flank', 'dose', 'diminish', '0.01', 'match', '51', 'destabil', 'plot', 'transit', 'hydrophob', 'degrad', 'action', '55', 'isoform', '2d', 'intracellular', 'modest', 'portion', 'specul', 'homozyg', 'subclon', 'diagnost', 'proteas', 'distinguish', 'throughout', '-', '2.5', 'slide', 'sds-page', 'digest', 'smaller', 'moder', '2005', 'in-fram', 'physiolog', 'stratagen', 'pathogenesi', 'nonsens', 'metastat', 'mixtur', 'longer', '500', 'assum', 'murin', 'name', 'nacl', 'divid', 'akt', 'consensus', 'v', 'driver', 'melanoma', 'dimer', '65', 'subunit', 'though', 'read', 'paramet', 'infect', '@', 'arrow', 'templat', 'resolv', 'care', 'good', 'scan', 'anim', 'conclud', 'efficaci', 'intron', 'place', 'charg', 'person', 'abund', '58', 'purpos', '64', 'g/ml', 'optim', 'occurr', '2002', 'parent', 'dmem', 'stop', 'immedi', 'neutral', 'network', 'enabl', 'earlier', 'subtyp', 'lane', 'upper', 'power', 'electrophoresi', 'kind', 'genotyp', 'sporad', 'impli', '54', 'analog', 'bioscienc', 'pocket', 'crucial', 'media', 'sourc', '2006', 'util', 'transduc', 'proxim', 'applic', '&', 'box', 'challeng', 'femal', 'cooper', 'bottom', 'weight', 'aim', 'encompass', 'employ', 'median', '4d', 'inherit', 'agaros', 'accompani', 'hospit', 'viabil', '5c', '2001', \"'\", 'depict', 'tag', 'render', 'centrifug', 'coordin', '6a', 'loss-of-funct', 'ng', 'lethal', 'itself', 'chronic', 'toler', 'biosystem', 'interfac', 'deleteri', 'henc', 'prove', 'surround', 'hybrid', 'purchas', 'span', 'give', 'powerpoint', 'best', 'usual', 'exogen', 'synthesi', 'epitheli', 'option', 'recommend', '0.2', 'literatur', 'driven', 'helix', 'assembl', '', 'bp', 'array', '61', 'restor', 'preferenti', 'sd', 'absent', 'possess', 'ring', 'prognosi', 'intact', 'free', 'pbs', 'systemat', 'bear', 'edta', 'cours', 'rt-pcr', '105', 'atp', 'committe', 'effector', 'l', 'colorect', 'instabl', 'egfr', 'adenocarcinoma', '2004', 'migrat', 'elimin', '57', '7.5', 'bead', 'perturb', 'sought', 'replic', 'invas', '56', 'san', 'morpholog', 'forward', 'expand', 'ml', 'minut', '2003', 'arrest', 'polyclon', 'flow', 'recognit', 'dynam', 'oligonucleotid', 'page', 'peripher', 'bond', 'capac', 'terminus', 'true', 'exert', 'descript', 'aggress', 'gain-of-funct', '2008', 'skin', 'instanc', 'offer', 'k', 'simultan', 'male', 'frameshift', 'take', 'nation', 'period', '2010', 'categori', 'liquid', 'acut', 'purif', '2011', 'basic', 'sodium', 'rule', 'ten', 'h.', 'bone', 'chemotherapi', 'yellow', 'what', 'nucleus', 'mass', 'pi3k', '37c', 'unfortun', 'black', '6b', 'luciferas', '', '1999', '#', 'microscop', 'fisher', 'kidney', 'stem', '', 'defici', 'releas', 'depart', 'compos', '0.001', 'attribut', 'packag', 'brain', 's6', 'medicin', 'intrins', 'women', 'room', 'cotransfect', 'white', 'rabbit', 'adult', 'ubiquitin', 'scale', 'intrigu', '300', 'ic50', 'virus', 'problem', 'routin', 'color', 'strand', 'myeloid', 'i.e.', 'classic', 'column', 'middl', 'serin', 'reproduc', 'resembl', 'frame', '2007', 'engin', 'indirect', 'grade', '59', 'supernat', 'mg', 'toxic', 'readili', 'segment', 'assign', 'mutual', 'run', 'global', '1997', 'cancer-associ', '68', 'index', 'promis', 'transactiv', '?', 'unexpect', 'metastasi', 'methyl', 'cdnas', 'understood', 'failur', 'rise', 'content', '1998', '67', 'excess', 'coupl', 'deregul', 'board', '250', 'ng/ml', 'subtl', 'accept', '66', 'fuse', 'hydrogen', 'believ', 'resid', 'fbs', 'germani', 'vast', 'alway', '85', 'lie', 'e.g.', 'sanger', 'concomit', 'rescu', 'obvious', 'compet', '2009', 'compromis', '3', 'mode', 'qualiti', 'adopt', 'text', 's5', 'trigger', 'ras', 'prostat', 'sole', 'synthes', 'yeast', 'aspect', 'denatur', '1994', 'attempt', 'actual', 'whi', 'sds', 'erk', 'braf', 'maximum', 'precipit', 'inject', 'record', 'dual', 'fit', 'otherwis', 'stress', 'gift', 'frozen', 'cover', '96-well', 'beyond', 'usa', '76', 'mapk', 'convent', 'hotspot', 'arginin', 'amersham', 'liver', '4c', '106', 'just', 'filter', 'iv', 'threshold', 'die', 'development', '1.0', 'met', 'ligas', 'switch', 'situ', 'l', 'immunohistochemistri', 'dulbecco', 'clarifi', '71', 'lipofectamin', '*', 'ectop', 'unrel', 'belong', 'reli', 'pten', 'techniqu', 'deplet', 'librari', 'carrier', 'down', 'simpl', 'cascad', 'kras', 'canon', 'likelihood', 'orient', 'benign', 'adjust', 'interv', 'novo', 'prognost', 'manifest', 'eventu', 'unaffect', 'physic', '293t', 'ultim', 'predisposit', 'likewis', 'plasma', 'bodi', 'heterozygos', 'subgroup', 'ensur', 'proper', 'prolong', '2.0', 'pdb', '62', 'x', 'prism', 'tumorigen', 'tandem', 'epiderm', 'avoid', 'resolut', 'gfp', 'scienc', 'helic', 'decis', 'albeit', 'look', 'causal', 'xenograft', 'whom', 'clonal', 'onset', 'implement', 'tp53', '1996', 'mg/ml', 'besid', 'never', 'triton', 'large-scal', 'protect', 'donor', 'lymphocyt', 'polar', '74', 'transmembran', 'regress', 'tris-hcl', 'mechanist', 'segreg', 'enzymat', 'newli', '69', 'penetr', 'underw', '96', 'null', 'dna-bind', 'snp', 'rest', 'tumour', 'tend', 'promin', 'captur', 'complementari', 'begin', 'incomplet', '63', '5', 'spot', 'knockdown', 'gave', '2012', 'biopsi', 'brca1', 'resuspend', 'la', 'viabl', 'idea', 'bank', 'anti-mous', 'upregul', 'coli', 'field', '400', 'complement', 'matrix', 'eagl', 'concord', 'untreat', 'chosen', 'emphas', 'acquisit', 'maxim', 'backbon', 'health', 'downregul', 'hr', 'retrovir', 'discrep', 'sustain', 'largest', 'abstract', 'salt', 'epigenet', 'chromatin', 'reliabl', 'transgen', 'acceler', 'head', 'unpublish', 'bd', 'request', 'cystein', 'pharmacolog', 'predispos', 'regardless', 'millipor', '104', 'mainten', 'oper', 'water', 'diagram', 'hereditari', 'dissect', 'argu', 'strain', 'oncolog', 'dataset', 'break', '2013', 'guidelin', 'bridg', 'quit', 'fall', 'student', 'absolut', 'shorter', 'gray', 'insuffici', 'tail', 'competit', 'exom', 'behavior', 'platform', 'rpmi', 'principl', 'dysregul', 'cytometri', 'nativ', 'signatur', 'uncov', 'squamous', 'anticip', 'synthet', 'inclus', 'co2', 'miss', '0.4', 'soft', 'practic', 'e.', 'connect', 'accur', 'adapt', '120', 'loci', 'overcom', 'alanin', 'manag', 'mo', 'exceed', 'inabl', 'dish', 'lymphoma', 'spread', 'viral', 'fewer', 'neoplasm', 'autophosphoryl', 'tight', 'focal', 'rat', 'delay', 'paraffin-embed', 'epitop', 's.', 'real-tim', 'annot', '1e', 'past', 'chemiluminesc', 'fresh', 'atom', 'his', 'wildtyp', 'simpli', 'minimum', 'leav', 'hallmark', 'lesser', 'kb', 'counterpart', 'categor', 'fashion', 'confid', 'lysin', 'leucin', 'prefer', 'glycin', 'pedigre', 'entri', 'surgeri', 'y', 'ethic', '1995', 'follow-up', 'bacteri', 'repress', 'pancreat', 'administr', 'linker', 'opposit', 'three-dimension', 'prompt', '4e', 'situat', 'fish', 'carlsbad', 'abi', 'attenu', 'gastrointestin', 'project', 'resect', 'x-100', 'exchang', 'unstabl', '78', 'microscopi', 'away', '3e', 'metabol', 'hundr', 'underlin', 'proven', 'undetect', 'quikchang', 'preliminari', 'convert', '0.3', 'coexpress', 'glioblastoma', 'behav', 'citi', 'relaps', 'ongo', 'prospect', 'c.', 'cosmic', 'stronger', 'heterodimer', '2e', 'prematur', 'spontan', '88', 'antigen', 'scientif', 'predictor', 'loui', 'face', 'carcinogenesi', 'late', 'aggreg', 'elut', 'md', 'daili', 'figureopen', 'distanc', 'object', 'tabdownload', 'immunohistochem', '', 'former', 'surgic', 'myc', 'pellet', '10-fold', 'ration', '79', 'vs.', 'preserv', 'mimic', 'serial', 'aa', 'bio-rad', '77', 'peak', 'concurr', 'warrant', 'chose', 'junction', '73', 'consortium', 'trypsin', 'nitrocellulos', 'conjug', '6c', 'dmso', '94', 'prolif', 'kda', 'immunofluoresc', 'nih', 'gap', 'immun', 'colleagu', 'linkag', 'uncertain', 'calf', 'interf', 'histopatholog', 'resourc', 'threonin', 'nuclei', 'denot', 'input', 'mild', 'stromal', 'knockout', 'id', 'van', 'constant', 'pack', 'rational', 'cyclin', 'oncogenesi', 'sort', 'microarray', 'nitrogen', '86', 'concern', 'manual', 'prolin', 'invari', 'matur', '600', 'intervent', 'refin', '0.0001', 'ligat', 'bladder', '5d', 'aspart', '7a', 'outlin', 'worldwid', 'heterodim', 'linear', 'profound', 'heat', 'c', 'g1', 'chromatographi', 'agar', 'opportun', 'radiat', 'too', 'dissoci', 'cytotox', 'easili', 'x-ray', 'orang', 'balanc', '', 'co-transfect', 't-test', 'mgcl2', 'somewhat', 'node', 'assist', 'e-mail', 'divis', '89', 'nsclc', 'evolutionarili', 'foci', 'gastric', 'japan', 'madison', 'postul', 'genbank', 'sigma-aldrich', 'endometri', 'paper', 'come', 'neck', 'mention', 'sign', 'coverag', 'autosom', '7.4', 'neoplast', 'renal', 'preced', 'bioinformat', 'instrument', 'hematopoiet', 'wi', 'sequenti', 'lineag', 'clontech', 'ice', 'quantif', 'hot', 'silico', 'preclin', 'becam', 'million', 'fals', 'anti-rabbit', 'excel', 'check', 'environ', 'double-strand', 'percent', 'thyroid', 'flexibl', 'genome-wid', 'dose-depend', 'conjunct', 'evolut', 'circl', 'underscor', 'histon', 'jolla', 'apart', '0.25', 'discrimin', 'oral', 'diffus', 'bias', 'own', 'recapitul', 'anneal', 'c-terminus', 'tris', '3.5', 'pik3ca', 'phosphatidylinositol', '107', '3f', 'homogen', 'ascertain', 'fundament', 'autom', 'boundari', '', 'embed', 'marrow', 'flag', 'igg', 'persist', 'metastas', 'flag-tag', 'insensit', 'enough', 'written', 'graph', 'unclassifi', 'indistinguish', 'augment', '1993', 'subcellular', 'arm', 'decad', 'guid', 'goal', 'diego', 'manuscript', 'store', '8.0', 'mobil', 'mitogen-activ', 'down-regul', 'potenc', 'j', 'silenc', '3-kinas', 'specifi', 'deposit', 'crystallograph', 'glycerol', 'precursor', 'snps', 'ny', 'baselin', 'communic', 'cross', 'evolutionari', '84', 'cytokin', '81', '87', 'atcc', 'scheme', 'methionin', 'refractori', 'diverg', 'ethanol', 'sirna', 'lipid', 'lee', 'drawn', 'recov', 'up-regul', 'window', 'amplicon', 'virtual', 'actin', 'asterisk', 'hematolog', 'nonsynonym', '92', 'salin', 'reactiv', 'complic', 'enrol', 'legend', '1000', '91', 'mek', 'counsel', 'drosophila', 'nude', 'uk', 'space', 'brca2', 'small-molecul', 'margin', '98', 'kinet', 'cm', 'fourth', 'formalin-fix', 'abbrevi', 'e.g', 'apoptot', 'dominant-neg', 'pa', 'passeng', 'hepe', 'strict', 'atlas', 'attract', '82', 'aliquot', 'angiogenesi', 'durat', '7b', 'silent', 'nras', 'tcga', 'escherichia', 'enter', 'nonetheless', '3.0', 'glutam', 'st.', 'trend', 'preclud', '93', 'solubl', '103', 'valuabl', 'dr.', 'imatinib', 'reminisc', 'pressur', 'invers', 'vehicl', 'nj', 'excis', 'healthi', 'distal', '', 'conceiv', 'weaker', 'horseradish', 'composit', 'g418', 'elsewher', 'agil', 'pronounc', 'mismatch', 'advers', 'high-throughput', 'fax', 'retrovirus', 'file', 'move', 'qualit', '6d', 'adhes', 'suitabl', 'fluorid', 'experienc', 'consecut', 'solvent', 'paradigm', 'foster', 'evidenc', 'aid', '', 'paraffin', 'concept', 'necessarili', '97', 'puromycin', 'e3', 'dead', 'log', 'forc', 'pretreat', 'biomark', 'cleavag', 'kd', 'expans', 'q', 'ortholog', 'machineri', 'school', 'split', 's1a', 'u', 'm.', '24', 'steric', 'arrang', 'infer', '130', 'zhang', 'reconstitut', 'w', 'solv', '1.2', 'phenomenon', 'pcdna3', 'build', 'ethnic', 'immunostain', '4.5', 'ct', 'single-nucleotid', 'thousand', 'mtor', 'progenitor', 'propens', '0.8', 'confluenc', 'european', 'i.e', 'retent', 'brief', 'tract', 'inlineview', 'ba/f3', '140', 'survey', 'subcutan', 'collabor', 'axi', 'd1', 'children', 'germ', 'elicit', 'occupi', 'goat', 'egf', '0.6', 'her', 'p21', 'cell-bas', 'difficulti', 'pathologist', 'ci', 'catalyz', '117', 'ha', 'il-3', 'glioma', 'regimen', 'aml', 's7', 'burden', 'disease-caus', 'layer', 'evolv', 'mutagen', 'nonspecif', 'infiltr', '125', 'permit', 'commerci', 's2a', 'serine/threonin', 'il', 'sens', 'polyacrylamid', 'oxygen', 'accommod', 'england', '83', 'distant', '13', 'infrequ', 'squar', 'causat', 'cocktail', 'submit', 'wors', 'trace', 'unusu', 'escap', 'adaptor', 'mitot', 'hematoxylin', 'glutamin', 'pictur', 'thermo', 'd.', '2-fold', 'contamin', 'co-express', 'popup', 'sarcoma', 'rank', 'ip', 'checkpoint', 'anchorage-independ', 'cut', 'rough', '2.1', 'exploit', '24-well', 'embryo', 'proband', 'slow', 'bcr-abl', 'ligand-independ', 'alpha', 'er', 'off', 'fill', 'hold', '4f', 'hek293t', 'lymph', 'valin', 'noteworthi', 'ns', 's1b', 'grew', 'energi', 'gradient', 'germ-lin', 'compat', 'meet', 'memori', 'anti-flag', 'add', '800', 'childhood', 'raf', 'illumina', 'spatial', 'magnitud', 'conflict', 'starv', 'fugen', 'retrospect', 'mount', 'dot', 'buri', 'engag', 'calcium', 'underli', '2.2', '99', 'phenylalanin', 'proteasom', 'round', 'cosegreg', 'ask', 'mini', 'dephosphoryl', 'equilibrium', 'live', 'violet', 'straightforward', 'mortal', 'loh', 'hela', 'electron', 'owe', 'cancer-rel', 'dye', 'architectur', 'scaffold', 'deep', 'framework', 'decid', 'monom', 'streptomycin', 'etiolog', 'ge', 'atp-bind', 'src', 'transcrib', 'eukaryot', 'high-risk', 'coincid', 'dock', 'proto-oncogen', 'ideal', 'pleas', 'sex', 'harbour', 'chanc', 'compil', 's2b', 'vascular', 'long-term', 'scatter', 'odd', 'cis', 'foundat', '1640', 'young', 'm2', 'depth', 'dri', 's3a', '3.1', 'indel', 'delin', 'erbb2', 'oppos', 'adequ', 'journal', 'dtt', 'accuraci', 'technic', 'rapamycin', 'dnas', 'immort', 'choic', 'erk1/2', 'glu', 'res', 'lobe', 'overview', '1.3', 'glucos', 'clinician', '7c', 'lentivir', 'sulfat', 'gtpase', 'insulin', 'synergist', 'pediatr', 'retriev', '1992', 'stringent', 'phosphotyrosin', 'phone', 'feedback', 'mim', 'compart', 'landscap', 'judg', 'boil', 'rtk', 'sum', 'dark', 'penicillin', 'p-valu', 'magnif', 'endotheli', 'suspens', 'messeng', 'ala', 'keep', 'fast', 'wherebi', 'oncoprotein', 'pierc', 'ion', 'sometim', 'unless', 'ribosom', 'pull-down', 'mammari', '121', 'juxtamembran', 'fcs', 'massiv', 'inspect', 'spectra', 'cytogenet', 'widespread', 'chimer', '1.1', 'extern', 'herein', 'plausibl', 'suppli', '3t3', 'ice-cold', 'cambridg', 'neoplasia', 'lymphoblast', 'elong', 'next-gener', '0.9', 'cleft', 'chen', 'cost', 'suspect', 'web', 'assumpt', 'old', 'li', 'transplant', 'calbiochem', 'n-', 'hormon', 'formal', 'lymphoid', 'irrespect', 'diamet', 'alloster', 'withdraw', 'discret', '4.0', 'healthcar', 'undertaken', 'wang', 'graphpad', 'uniform', 'attent', '-actin', 'pr', 'rotat', 'atyp', 'peroxidas', 'agre', 'bold', 'bsa', 'neighbor', 'unambigu', 'passag', 'greatest', 'del', 'boston', 'contrari', 'cleav', '1991', 'dosag', 'well-character', 'transport', '1.4', '2.3', 'blast', 'raw', '2017', 'well-known', 'spectrometri', 'archiv', 'uncharacter', 'upstat', '3.2', '6-well', 'homologu', 'myeloprolif', '110', 'hyperactiv', 'methodolog', 'log-rank', 'artifici', 'mmol/l', 'motil', 'uncommon', 'ncbi', 'varianc', 'movement', 'intestin', 'world', 'gapdh', 'scenario', 'antisens', 'lowest', '1.8', 'der', 'attach', 'artifact', 'hepatocellular', 'symptom', 'dash', 'controversi', 's3b', '2f', 'catalogu', 'tki', 'immunoglobulin', 'join', 'photograph', '1,000', '2.6', 'histidin', 'bulki', 'interrog', 'anti-ha', 'hit', 'hairpin', 'stratif', 'renilla', 'multiplex', 'mask', 'reson', 'proteolyt', 'cutoff', 'curat', '350', 'tkis', 'shrna', 'synonym', 'superfamili', 'a.', 'cytosol', 'decemb', 'visibl', 'st', 'he', 'groov', 'remodel', 'colleg', 'lake', '-helix', 'n-terminus', 'moieti', 'back', 'california', 'rneasi', 'elig', 'retard', 'biolab', 'juli', 'suffer', 'taq', 'permiss', 'nih3t3', 'dysfunct', 'reprint', 'centr', 'shape', 'confin', 'myelogen', 'prime', 'june', 'cutan', 'princip', 'reliev', '1.6', 'breakpoint', 'vertebr', 'isoleucin', 'unphosphoryl', 'inconsist', 'non-smal', 'eleven', 'doi', '123', 'hope', 'aros', 'know', 'uv', 'nervous', 'master', 'sequence-specif', 'proteolysi', 'aacr', 'pivot', 'ha-tag', 'output', 'lose', '000', 'satur', 'reinforc', 'senesc', 'inflammatori', 'hercul', 'clinicopatholog', 'flt3', 'remiss', 'tyr', 'purpl', 'brct', 'naf', 'quick', '102', 'pancrea', 'govern', 'dissolv', 'sonic', 'entiti', 'chip', 'host', 'intramolecular', 'titrat', 'danver', 'coval', 'pharmaceut', '109', '23', 'polypeptid', 'pymol', 'haploinsuffici', 'caucasian', 'kaplan-mei', 'august', 'protein-protein', 'heterolog', 'thorough', 'phosphopeptid', 'co-occur', 'broader', 'rnas', 'bacteria', 'benefici', 'gefitinib', 'dodecyl', 'pi3k/akt', '|', 'electrostat', 'ripa', 'irradi', 'patch', 'addict', 'discord', 'coeffici', 'turnov', 'platelet-deriv', 'took', 'np-40', '2014', 'antitumor', '0.7', 'administ', 'amino-acid', '293', 'eosin', '160', 'ineffect', 'arrowhead', 'acet', 'tk', 'cold', 'microsatellit', 'electrophoret', 'graphic', 'bidirect', 'ihc', 'arg', 'ribbon', 'occasion', 'ecl', 'manipul', 'ecori', 'cerevisia', '10,000', 'ubiquit', '1/2', 'older', 'confoc', 'tube', 'franc', 'rtks', 'carboxy-termin', 'ile', 'carboxyl', '5.0', '2', 'recur', '2.4', '12.5', 'special', 'tran', 'quantiti', 'phospho-akt', 'val'])"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abridgedsortedst.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['20', 'human', 'mutant', 'same', 'dna', 'number', 'when', 'could', 'table', 'several', 'known', 'proteins', 'some', 'thus', '7', 'associated', 'within', 'presence', 'figure', 'effect', 'those', 'additional', 'because', '12', 'performed', 'wild-type', 'functional', 'function', 'b', 'control', '15', 'likely', 'sequence', 'first', 'important', 'present', 'respectively', 'methods', 'expressed', 'they', 'did', '8', 'amino', 'showed', 'levels', 'whether', 'binding', '9', 'high', 'activation', 'role', 'c', 'whereas', 'result', 'specific', 'increased', '50', 'tumor', 'indicated', 'patients', 'suggest', 'significant', 'through', 'assay', 'analyzed', 'contrast', 'genes', 'sequencing', 'effects', 'acid', 'then', 'there', 'containing', 'region', 'total', 'previous', 'detected', '14', 'determined', 'obtained', 'higher', '11', 'molecular', 'clinical', '30', 'therefore', 'site', 'any', '16', 'multiple', 'vitro', 'show', '13', 'highly', 'type', 'recently', 'small', 'suggesting', 'would', 'resulting', 'lines', 'growth', 'based', 'loss', 'due', 'cases', 'assays', 'independent', 'fig', 'introduction', '18', 'four', 'development', 'single', 'common', 'potential', '24', '19', 'least', 'samples', 'possible', 'significantly', 'tumors', 'genetic', 'less', 'experiments', 'tested', 'mutants', 'provide', 'involved', 'consistent', 'line', 'will', 'here', 'many', '25', 'according', '17', 'if', 'low', 'required', 'indicate', 'might', 'evidence', 'do', 'model', '21', 'under', 'disease', 'large', 'revealed', 'following', 'suggests', 'confirmed', '100', 'level', 'against', 'domains', 'changes', 'kinase', 'lower', 'without', 'affect', 'determine', 'absence', 'available', 'another', 'reduced', 'generated', 'normal', 'target', '22', 'mutated', 'together', 'residues', 'new', 'cancers', 'ability', 'sites', 'et', 'among', 's', '[', ']', 'directly', 'vector', 'group', 'family', 'primary', 'structure', 'second', 'response', 'analyses', 'signaling', 'identify', 'case', 'where', 'examined', 'located', 'point', 'above', 'overall', 'furthermore', 'findings', '40', 'test', 'complex', 'factor', 'expressing', '>', 'out', 'distinct', 'negative', '26', 'relative', 'while', 'predicted', 'corresponding', 'five', 'patient', 'pcr', 'indicating', 'use', 'expected', 'system', 'phosphorylation', 'recent', 'very', '<', 'given', 'over', 'mechanisms', '27', 'included', 'interestingly', 'majority', 'increase', 'even', 'change', '23', 'form', 'd', 'p', 'interaction', 'missense', 'cellular', 'regions', 'differences', 'mechanism', 'critical', 'positive', '28', '29', 'individual', 'transfected', 'treatment', '2a', 'sequences', 'and/or', 'occur', 'suggested', 'part', '1a', 'pathway', 'selected', 'downstream', 'report', 'genomic', 'time', 'see', 'assessed', 'frequency', 'represent', '35', 'residue', 'antibody', 'approximately', 'inhibitor', 'types', 'conserved', 'does', 'should', 'followed', 'affected', 'inhibition', 'measured', 'position', 'acids', 'characterized', 'complete', 'example', 'sample', '1b', 'proliferation', 'various', 'remains', 'conditions', 'active', 'direct', 'contribute', 'set', 'during', 'fact', 'showing', 'per', 'alterations', 'being', 'demonstrated', '32', 'derived', 'novel', 'include', 'significance', 'resulted', '31', 'factors', 'structural', 'next', 'before', 'formation', 'range', 'vivo', 'frequently', '2b', 'i', 'considered', '', 'antibodies', 'comparison', 'materials', 'strong', 'association', '3a', 'difference', 'pathways', 'major', 'taken', 'lead', 'signal', 'support', 'lack', 'somatic', 'cause', 'substitution', 'deletion', 'functions', '33', 'western', 'identification', 'encoding', 'inhibitors', 'shows', 'induced', 'moreover', 'them', 'possibility', 'note', 'observation', 'particular', 'ii', 'provided', '37', 'developed', 'tissue', 'six', '34', 'mutagenesis', 'treated', 'controls', 'unknown', 'al', 'sensitivity', 'key', 'standard', '48', 'still', 'kit', 'information', 'similarly', 'able', 'substitutions', 'status', 'culture', 'strongly', 'transcription', 'variants', 'transfection', 'cdna', 'finally', 'mouse', 'phenotype', 't', 'groups', 'indeed', 'basis', 'g', 'h', 'indicates', 'receptor', '=', '', 'published', 'established', 'assess', '', 'activating', 'clear', 'subset', 'limited', 'targets', 'activated', 'yet', 'sensitive', 'values', 'forms', 'relatively', 'leading', '3b', 'about', 'observations', 'interactions', 'oncogenic', 'since', 'caused', 'models', '36', 'exon', 'decreased', 'seen', 'method', 'stable', 'evaluated', 'specifically', 'related', 'finding', '60', 'constructs', 'alone', 'wt', 'panel', 'defined', 'investigated', 'applied', 'n', 'importance', 'targeted', 'allele', 'few', 'rare', 'demonstrate', 'upon', 'despite', 'altered', \"'s\", 'secondary', 'survival', 'medium', 'research', 'isolated', 'confirm', 'essential', 'via', 'incubated', 'calculated', 'variant', 'sequenced', 'experimental', 'exons', 'nucleotide', 'approach', 'detection', 'rate', 'e', 'studied', 'remaining', 'mean', 'tyrosine', 'regulation', 'series', 'reaction', '4a', 'breast', 'carried', 'relevant', 'sufficient', 'contains', 'rather', 'coding', 'rna', 'times', 'often', 'regulatory', 'amplification', 'so', 'population', 'unique', 'members', 'therapeutic', 'features', 'green', 'phosphorylated', 'mice', 'red', 'size', 'led', '1c', 'none', 'progression', 'general', 'buffer', 'serum', 'c-terminal', 'who', 'enhanced', 'primers', 'selection', 'concentrations', 'reports', 'others', '4b', 's1', 'activate', 'impact', 'induce', 'hypothesis', 'supplementary', '38', 'end', 'manufacturer', 'bind', 'seven', 'reduction', 'added', 'blue', 'endogenous', 'leads', 'larger', 'go', 'mutational', 'value', 'prepared', 'days', '39', 'ratio', 'full', 'products', 'occurs', 'involving', 'correlation', 'combined', 'investigate', 'made', 'suppressor', 'greater', 'purified', 'displayed', 'context', '90', '42', 'lysates', 'currently', 'initial', 'combination', 'eight', '45', 'except', 'comparable', 'cycle', 'failed', 'work', 'prior', 'chain', 'provides', '0', '80', 'responsible', 'pattern', 'database', 'biological', 'view', 'amplified', 'below', 'up', 'potentially', '41', 'activities', '43', 'plasmid', 'average', 'kinases', 'dependent', 'presented', 'subsequent', 'cultured', 'particularly', 'manner', '2c', 'positions', '0.5', 'confer', 'events', 'properties', 'inhibit', 'necessary', 'express', 'concentration', 'targeting', 'therapy', 'alleles', 'alternative', 'hours', 'contained', 'early', 'carcinoma', 'develop', 'far', 'proposed', 'detect', 'generate', 'noted', 'frequent', 'years', 'how', '70', 'current', 'subjected', 'staining', 'play', 'blot', 'understanding', 'representative', 'mm', 'collected', 'constitutive', 'chromosome', 'lung', 'increasing', 'clearly', 'inhibited', 'occurred', 'little', 'summary', 'much', 'mrna', 'plates', 'appears', 'require', 'produced', 'mediated', 'site-directed', 'software', 'characterization', 'deletions', 'gel', 'certain', 'left', 'carrying', 'contain', 'resistance', 'statistical', 'decrease', '', 'full-length', 'state', 's2', 'http', 'university', 'ca', 'screening', 'variety', 'distribution', 'stability', 'conformation', 'us', 'right', 'numbers', 'mammalian', 'overexpression', 'codon', 'biochemical', 'extracted', 'wild', '3c', 'central', 'substrate', 'tissues', 'like', '0.1', 'phase', 'crystal', 'genome', 'marked', 'commonly', 'harboring', 'diagnosis', 'binds', 'almost', 'subsequently', 'top', 'construct', 'n-terminal', 'grown', 'affecting', 'effective', 'now', 'need', 'individuals', 'consequences', 'represents', 'close', 'specificity', 'nuclear', 'exhibited', '44', 'alter', 'versus', 'core', 'product', 'importantly', 'drug', 'min', 'supported', 'implicated', 'identical', 'across', 'correlated', 'partial', 'final', 'germline', 'introduced', '...', 'invitrogen', 'third', 'molecules', 'experiment', 'having', 'conclusion', 'motif', 'bound', 'transcriptional', 'probably', 'date', 'signals', 'amount', 'help', 'version', 'clinically', '4c', 'reveal', 'especially', 'f', 'process', 'predict', 'unclear', 'percentage', 'functionally', 'classified', '5a', 'blood', 'age', 'fully', 'membrane', 'fraction', 'open', 'apoptosis', 'plays', 'examine', 'appropriate', 'testing', 'entire', 'clones', 'reverse', 'nature', 'transformation', 'evaluation', 'nine', 'interact', 'loop', 'completely', 'formed', 'inactivation', 'whose', 'disrupt', 'generation', 'equal', 'structures', 'order', 'supplemented', 'along', 'half', 'promoter', 'spectrum', 'protocol', 'surface', 'evaluate', 'normalized', 'tumorigenesis', 'occurring', 'allow', 'side', 'proportion', 'modified', 'generally', 'inactive', 'encodes', '2000', 'relevance', 'affinity', 'center', 'interest', 'approved', 'includes', '0.05', 'rates', 'location', 'appear', 'catalytic', 'whole', 'class', 'thought', 'risk', 'polymerase', 'later', 'components', 'lysed', 'degree', 'resistant', 'explain', 'characteristics', 'notably', 'thereby', 'discovery', 'primer', 'account', 'alteration', 'technologies', 'assessment', 'promote', 'selective', 'linked', 'fusion', 'ph', 'underlying', 'review', 'elevated', 'criteria', 'detectable', 'act', 'background', 'increases', '200', 'highest', 'exhibit', 'stably', 'short', 'unable', 'extent', 'syndrome', 'supporting', 'roles', 'detailed', 'systems', 'already', 'address', 'hand', 'inhibitory', 'cloned', 'washed', 'institute', 'requires', 'r', 'vectors', 'listed', 'comprehensive', 'iii', '5b', 'gain', 'knowledge', '46', '+', 'aberrant', 'harbor', 'putative', 'responses', 'recombinant', 'maintained', 'better', 'localization', 'needed', 'achieved', 'constitutively', 'santa', 's3', '47', 'instructions', 'malignant', 'screened', 'latter', 'stained', 'volume', 'estimated', 'lacking', 'induction', 'useful', 'designed', 'primarily', 'advantage', 'regulate', 'acquired', 'laboratories', 'ref', 'onto', 'transduction', 'display', 'once', '72', 'diverse', 'nm', 'malignancies', 'm', 'plasmids', 'differentiation', '150', 'molecule', 'future', 'encoded', 'blotting', 'procedures', 'processes', 'make', 'image', 'partially', 'lysis', 'seems', 'approaches', 'quantitative', 'induces', 'g', '3d', 'apparent', 'poor', 'enzyme', 'contact', 'day', 'allows', 'possibly', 'length', 'mainly', 'difficult', 'fold', '1d', 'agreement', 'harvested', 'promega', 'adjacent', 'additionally', 'outside', 'al.', 'fragment', 'complexes', 'largely', 'dominant', 'every', 'strategy', 'cruz', 'relationship', 'roche', 'separate', 'recurrent', 'm', 'equivalent', 'efficiency', 'reference', 'unlikely', 'oncogene', 'phenotypes', 'heterozygous', 'transformed', 'keywords', 'unlike', 'exposed', '95', 'validated', 'variable', 'cohort', 'impaired', 'base', 'inhibits', 'light', 'transferred', 'influence', 'screen', 'clone', 'disruption', 'biotechnology', 'compare', 'collection', 'outcome', 'leukemia', 'empty', 'solution', 'component', 'reporter', 'suppression', 'exact', 'remain', 'allowed', 'agents', 'trials', 'copy', 'figures', 'substantial', 'fetal', 'nearly', 'plus', 'qiagen', 'localized', 'regulated', 'consisting', 'causes', 'block', 'retained', '75', 'near', 'overnight', 'nor', 'reflect', 'cluster', 'temperature', 'separated', 'ligand', 'history', 'implications', 'extracellular', 'statistically', 'last', 'alignment', 'life', 'therapies', 'twice', 'done', 'measure', 'solid', 'differential', 'event', 'tables', 'inc.', 'respect', 'around', 'diagnosed', 'exception', 'conversely', 'bars', 'basal', 'verified', 'interacts', 'follows', 'providing', 'splice', 'ovarian', 'stage', 'incidence', 'independently', 'visualized', 'briefly', 'immunoblotting', 'activates', 'conformational', 'american', 'carcinomas', 'examination', 'repair', 'sigma', 'marker', '49', 'means', 'prevalence', 'locus', 'program', 'splicing', 'transforming', 'understand', 'carry', 'define', 'markedly', '52', 'defects', 'reagent', 'ma', 'parallel', 'intermediate', 'wide', 'notion', 'conclusions', 'regulates', 'variation', 'online', 'defective', 'mixed', 'initially', 'schematic', 'homology', 'seeded', 'reviewed', 'focused', 'developing', 'species', 'remained', 'lost', 'death', 'consequence', 'predominantly', 'long', 'monoclonal', 'frequencies', 'conducted', 'embryonic', 'fixed', 'characteristic', 'families', 'until', 'receptors', '1.5', 'sets', 'investigation', 'rapid', 'points', 'benefit', 'assayed', 'diseases', 'map', 'interesting', 'fluorescence', 'p53', '53', 'upstream', 'documented', 'pathogenic', 'again', 'lesions', 'nevertheless', 'stimulation', 'classification', 'incubation', 'perhaps', 'broad', 'neither', 'minimal', 'de', 's4', 'bovine', 'toward', 'instead', 'must', 'feature', 'step', 'colon', 'reduce', 'appeared', 'contribution', 'insight', 'become', 'prevent', '0.01', '51', 'exposure', 'obtain', 'transient', 'extensive', 'facilitate', 'patterns', 'weeks', 'amounts', 'membranes', 'phosphate', 'representing', 'accumulation', 'slightly', 'download', 'potent', '55', 'months', '2d', 'phosphatase', 'characterize', 'overexpressed', 'material', 'robust', 'peptide', 'medical', 'figs', 'recognized', 'intracellular', 'bands', 'examples', 'immunoprecipitation', 'informed', 'original', 'prediction', 'throughout', '-', '2.5', 'sds-page', 'represented', 'area', 'smaller', 'share', 'hydrophobic', 'consent', 'find', 'received', '2005', 'advanced', 'in-frame', 'homozygous', 'specimens', 'classes', 'initiation', 'abnormalities', 'stratagene', 'pathogenesis', 'involve', 'nonsense', 'metastatic', 'longer', '500', 'triplicate', 'comparing', 'correlate', 'murine', 'nacl', 'candidate', 'transiently', 'representation', 'discovered', 'excluded', 'terms', 'reactions', 'akt', 'article', 'consensus', 'truncated', 'technology', 'focus', 'v', 'section', 'laboratory', 'member', 'promotes', '65', 'counted', 'origin', 'though', 'drugs', 'quantified', 'serve', '@', 'institutional', 'supports', 'causing', 'density', 'explanation', 'good', 'way', 'mediate', 'protease', 'populations', 'intensity', 'weak', 'profile', 'fragments', 'mapped', '58', 'polymorphisms', '64', 'numerous', 'g/ml', 'cytoplasmic', 'validation', 'degradation', 'affects', '2002', 'insights', 'damage', 'dmem', 'involvement', 'earlier', 'elements', 'upper', 'electrophoresis', 'setting', 'accordance', 'double', '54', 'effectively', 'internal', 'media', 'transcript', 'access', '2006', 'allowing', 'inhibiting', '&', 'produce', 'modest', 'score', 'local', 'bottom', 'capable', 'confirming', 'melanoma', 'sporadic', 'median', 'demonstrating', '4d', 'identifying', 'design', 'agarose', 'differ', 'efficacy', 'template', 'pocket', 'plated', '5c', 'turn', 'alternatively', 'removed', 'viability', 'labeled', 'presumably', '2001', \"'\", 'profiles', 'explained', 'exist', 'codons', '6a', 'error', 'loss-of-function', 'ng', 'crucial', 'pair', 'colonies', 'pairs', 'itself', 'stimulated', 'homologous', 'hence', 'main', 'depending', 'blots', 'cloning', 'exclude', 'powerpoint', 'best', 'biosystems', 'poorly', 'biology', 'harbored', 'synthesis', 'polymorphism', 'purchased', 'epithelial', '0.2', 'chronic', 'details', 'driven', 'occurrence', 'helix', '', 'bp', '61', 'sd', 'literature', 'markers', 'hospital', 'absent', 'summarized', 'establish', 'prognosis', 'chromosomal', 'translocation', 'portion', 'pbs', 'intact', 'contributes', 'extraction', 'undergo', 'edta', 'severe', 'usually', 'random', 'colony', 'rapidly', 'repeated', 'repeat', 'substantially', 'rt-pcr', '105', 'closely', 'atp', 'probe', 'abnormal', 'widely', 'free', 'l', 'driver', 'shift', 'colorectal', 'efforts', 'egfr', '2004', 'deleterious', '57', '7.5', 'instability', 'authors', 'involves', 'gels', 'sought', 'regulating', 'normally', '56', 'question', 'integrity', 'san', 'surprisingly', 'parameters', 'evident', 'issue', 'ml', '2003', 'network', 'course', 'rearrangements', 'negatively', 'recruitment', 'forward', 'analyze', 'integrated', 'polyclonal', 'strategies', 'replaced', 'recognition', 'efficiently', 'defect', 'enriched', 'flow', 'mixture', 'interface', 'typically', 'beads', 'peripheral', 'referred', 'familial', 'terminus', 'true', 'fluorescent', 'immunoblot', 'successfully', 'gain-of-function', '2008', 'skin', 'search', 'predictive', 'restriction', 'exclusively', 'k', 'action', 'oncogenes', 'consistently', 'constructed', 'hypothesized', 'consequently', 'immunoprecipitated', 'loading', 'capacity', 'page', 'biosciences', '2010', 'liquid', 'deleted', 'translation', '2011', 'sections', 'sodium', 'describe', 'tests', 'purification', 'truncation', 'h.', 'yellow', 'extracts', 'what', 'nucleus', 'subunit', 'acute', 'susceptibility', 'pi3k', 'demonstrates', '37c', 'reveals', 'unfortunately', 'respond', 'ring', 'ten', '6b', 'luciferase', 'correlates', '', 'genetics', '1999', '#', 'conferred', 'fisher', 'black', 'acts', 'called', 'national', '', 'chemical', 'chemotherapy', 'aggressive', '0.001', 'disrupted', 'determination', 'matched', 'aberrations', 'brain', 'detail', 'arise', 's6', 'motifs', 'blocked', 'women', 'room', 'extended', 'bone', 'monitored', 'committee', 'composed', 'fibroblasts', 'promoting', 'compound', '300', 'seem', 'mapping', 'conservation', 'white', 'basic', 'exogenous', 'treatments', 'array', 'exclusive', 'partner', 'deviation', 'rabbit', 'myeloid', 'i.e.', 'middle', 'transition', 'identity', 'preparation', 'isoforms', '2007', 'minor', 'respective', 'minutes', 'divided', 'pathology', '59', 'production', 'weight', 'blocks', 'mg', 'cycles', 'readily', 'moderate', 'probability', 'mass', 'modeling', 'efficient', 'originally', 'algorithm', 'arrest', 'interacting', 'making', '1997', 'cancer-associated', 'corresponds', 'band', 'makes', 'enzymes', 'created', 'enhance', 'phenotypic', 'stem', 'towards', '68', 'contacts', 'frameshift', 'interfere', 'serine', '?', 'extremely', 'states', 'insertion', 'impair', 'diagnostic', 'metastasis', 'cdnas', 'explore', 'suppress', 'regulator', 'department', 'structurally', 'options', 'female', 'heterogeneity', 'understood', 'correct', 'propose', 'probed', 'kidney', 'encode', 'resolved', 'refs', 'adenocarcinoma', '1998', 'reagents', 'bearing', 'discussed', 'restricted', 'medicine', 'failure', 'inactivating', '67', 'regard', 'considering', '250', 'ng/ml', 'subtle', '66', 'parental', 'heterogeneous', 'hydrogen', 'ic50', 'fbs', 'germany', 'always', '85', 'tool', 'regarding', 'confers', 'e.g.', 'sanger', 'striking', 'selectively', '2009', 'processed', 'accession', '3', 'achieve', 'author', 'text', 'highlighted', 's5', 'ras', 'precise', 'yeast', 'typical', 'improved', 'surrounding', 'genetically', 'quality', '1994', 'panels', 'why', 'dose', 'interpretation', 'sds', 'neutral', 'braf', 'maximum', 'animal', 'maintain', 'modification', 'index', 'consists', 'prostate', 'accompanied', 'dramatic', 'start', 'dual', 'validate', 'otherwise', 'transfections', 'link', 'subcloned', 'frozen', 'transactivation', 'intrinsic', 'clustered', 'erk', 'dimerization', 'beyond', 'global', 'usa', '96-well', '76', 'images', 'growing', 'male', 'mostly', 'amersham', 'condition', '4c', '106', 'vast', 'just', 'frame', 'scale', 'mapk', 'measurements', 'mutually', 'residual', 'physiological', 'liver', 'cultures', '1.0', 'differentially', 'met', 'scored', 'engineered', 'substrates', 'preferentially', 'list', 'hybridization', 'iv', 'situ', 'elucidate', 'l', 'directed', 'fused', 'lacks', 'stabilize', 'probes', 'humans', 'immunohistochemistry', 'curve', 'dulbecco', 'description', 'arginine', '71', 'content', '*', 'unrelated', 'developmental', 'collectively', 'protocols', 'pten', 'lipofectamine', 'exists', 'remarkably', 'remove', 'simple', 'ratios', 'mediates', 'stop', 'flanking', 'malignancy', 'lane', 'ranging', 'clustering', 'accordingly', 'synthesized', 'comparative', 'migration', 'disorders', 'kras', 'locations', 'diminished', 'virus', 'likelihood', 'outcomes', 'adult', 'benign', 'systematic', 'novo', 'speculate', 'ligase', 'down', 'modifications', 'canonical', 'arrows', 'diluted', 'peptides', 'dilution', 'unaffected', '293t', 'compounds', 'employed', 'inducing', 'likewise', 'conclude', 'predisposition', 'plasma', 'heterozygosity', 'transduced', 'concomitant', 'microscope', 'areas', 'immediately', 'purpose', 'cotransfected', 'depicted', 'changed', 'curves', 'prevalent', 'existence', 'lysate', '2.0', 'subject', 'pdb', 'predictions', 'drive', '62', 'create', 'code', 'greatly', 'x', 'prism', 'analogous', 'epidermal', 'gfp', 'observe', 'transcripts', 'albeit', 'give', 'lethal', 'subtypes', 'whom', 'nucleotides', 'correspond', 'varying', 'plate', 'tandem', 'onset', 'tp53', 'regulators', '1996', 'mg/ml', 'repeats', 'depends', 'triton', 'lanes', 'never', 'large-scale', 'inactivated', 'stabilization', '74', 'inherited', 'truncating', 'yielded', 'tris-hcl', 'transmembrane', 'tag', 'rise', 'diversity', 'newly', '69', 'ultimately', 'suppressed', 'insertions', 'underwent', '96', 'null', 'dna-binding', 'infection', 'snp', 'distributed', 'surprising', 'problem', 'scores', 'trial', 'partners', 'stages', 'equally', 'source', 'complementary', 'period', 'conventional', '63', 'terminal', '5', 'reason', 'gave', 'paired', '2012', 'brca1', 'dramatically', 'disorder', 'estimate', 'reasons', 'simultaneously', 'viable', 'la', 'essentially', 'anti-mouse', 'positively', 'clonal', 'resolution', 'care', 'coli', 'variations', 'isoform', 'requirement', 'initiated', '400', 'confirmation', 'matrix', 'abolished', 'cytoplasm', 'gift', 'spanning', 'extension', 'depend', 'controlled', 'highlights', 'retain', 'overlapping', 'chosen', 'coupled', 'resuspended', 'knockdown', 'idea', 'subtype', 'comparisons', 'starting', 'untreated', 'enzymatic', 'profiling', 'doses', 'tolerated', 'prognostic', 'optimal', 'health', 'forming', 'acquisition', 'hr', 'excess', 'blocking', 'eagle', 'depletion', 'largest', 'besides', 'maintaining', 'accessible', 'syndromes', 'strand', 'backbone', 'cohorts', 'determining', 'chromatin', 'contributing', 'effectors', 'centrifugation', 'measurement', 'abrogated', 'retroviral', 'considerable', 'unpublished', 'fail', 'associate', 'encompassing', 'progress', 'serves', 'animals', 'bd', 'plotted', 'regardless', 'chains', 'charged', 'millipore', '104', 'maintenance', 'bank', 'hereditary', 'effort', 'abstract', 'accepted', 'infected', 'similarity', 'rarely', '2013', 'head', 'threshold', 'extensively', 'international', 'ubiquitin', 'procedure', 'addressed', 'epigenetic', 'apparently', 'shorter', 'ectopic', 'transform', 'varied', 'mode', 'digestion', 'tumour', 'examining', 'rpmi', 'improve', 'abrogate', 'cytometry', 'squamous', 'gray', 'salt', 'student', 'rule', 'termed', 'regression', 'native', 'co2', 'board', 'water', '0.4', 'soft', 'explored', 'guidelines', 'e.', 'promising', 'overlap', '120', 'release', 'loci', 'distinguish', 'oncology', 'assigned', 'eventually', 'synthetic', 'mo', 'box', 'homolog', 'fewer', 'assembly', 'prominent', 'inability', 'paraffin-embedded', 'library', 's.', 'run', 'real-time', '1e', 'quite', 'amplifications', 'enrichment', 'slight', 'reach', 'his', 'tools', 'aspects', 'highlight', 'wildtype', 'simply', 'slides', 'proximity', 'minimum', 'stress', 'databases', 'reached', 'activator', 'year', 'removal', 'lesser', 'power', 'illustrated', 'cascade', 'successful', 'kb', 'intron', 'modulate', 'calculate', 'steps', 'fashion', 'insufficient', 'injected', 'agent', 'offer', 'exhibits', 'continued', 'rescue', 'focal', 'y', '1995', 'bonds', 'united', 'viral', 'surgery', 'inclusion', 'past', 'mediating', 'follow-up', 'destabilize', 'three-dimensional', 'solely', '4e', 'predicting', 'alanine', 'leucine', 'combinations', 'carlsbad', 'abi', 'gastrointestinal', 'imaging', 'carriers', 'pancreatic', 'approval', 'hotspot', 'consisted', 'proper', 'x-100', 'fish', 'effector', 'subsets', 'ligands', '78', 'microscopy', 'away', 'instance', '3e', 'grade', 'disrupting', 'rest', 'glycine', 'autophosphorylation', 'morphology', 'fusions', 'replacement', 'abundance', 'category', 'comprised', 'strikingly', 'quikchange', 'preliminary', '0.3', 'opposite', 'bond', 'possess', 'proven', 'unstable', 'epitope', 'c.', 'cosmic', 'stronger', 'appearance', 'platform', '2e', 'ongoing', 'reflects', 'city', 'lysine', 'translational', 'measuring', 'column', 'becomes', 'reflected', '88', 'pathological', 'oligonucleotide', 'enhances', 'suppressors', 'adding', 'administration', 'analyzing', 'louis', 'allelic', 'stimulate', 'orientation', 'severely', 'supernatant', 'carcinogenesis', 'md', 'daily', 'figureopen', 'recommended', 'favor', 'scientific', 'linker', 'personal', 'recognize', 'tabdownload', 'responsive', '', 'invasive', 'preventing', 'myc', 'recombination', '10-fold', 'select', 'indirect', 'cysteine', 'tumorigenic', 'duplication', '79', 'actually', 'vs.', 'rat', 'exome', 'overcome', 'progressive', 'bio-rad', '77', 'isolation', 'chose', 'prevents', 'participate', 'attributed', '73', 'slide', 'emerging', 'intronic', 'comprising', 'nitrocellulose', 'fractions', '6c', 'dmso', '94', 'kda', 'nih', 'availability', 'alignments', 'emerged', 'aa', 'consortium', 'toxicity', 'premature', 'promoted', 'strain', 'lacked', 'uncertain', 'inserted', 'package', 'calf', 'confidence', 'extend', 'body', 'proliferative', 'driving', 'nuclei', 'construction', 'verify', 'van', 'stromal', 'rationale', 'oncogenesis', '86', 'penetrance', 'displaying', 'computational', 'late', 'obvious', 'unexpected', 'favorable', 'conservative', 'great', 'analysed', 'aligned', 'knockout', 'mechanistic', 'variability', 'ends', '600', 'tail', 'donor', 'immunohistochemical', 'stabilized', 'classical', 'threonine', 'reliable', 'processing', '0.0001', 'altering', 'deficient', 'prove', 'substituted', '5d', 'reduces', 'linkage', 'bladder', 'arising', 'lymphoma', 'defining', 'loaded', '7a', 'worldwide', 'categories', 'weakly', 'invasion', 'nitrogen', 'c', 'routine', 'prolonged', 'g1', 'chromatography', 'expanded', 'agar', 'signalling', 'too', 'neoplasms', 'grow', 'easily', 'phosphorylate', 'id', 'counts', 'orange', '', 'assumed', 'informative', 'exchange', 'supplemental', 'differed', 'duplicate', 'mgcl2', 'ranged', 'predominant', 'somewhat', 'element', 'colleagues', 'cyclin', 'inactivate', 'e-mail', 'vary', 'proline', 'color', '89', 'nsclc', 'randomly', 'evolutionarily', 'foci', 'gastric', 'japan', 'ensure', 'translocations', 'radiation', 'diagram', 'madison', 'genbank', 'sigma-aldrich', 'raised', 'shared', 'organization', 'folding', 'transgenic', 'introduce', 'endometrial', 'transfer', 'remarkable', 'namely', 'avoid', 'sustained', 'loops', 'coverage', 'pooled', '7.4', 'conversion', 'dataset', 'neoplastic', 'renal', 'undergoing', 'wi', 'hematopoietic', 'stabilizing', 'physical', 'clontech', 'neck', 'signature', 'moderately', 't-test', 'ice', 'genotype', 'challenge', 'naturally', 'autosomal', 'indicative', 'quantification', 'division', 'silico', 'combining', 'perform', 'became', 'application', 'histological', 'anti-rabbit', 'techniques', 'clinic', 'displays', 'adenocarcinomas', 'suppresses', 'fibroblast', 'incomplete', 'histology', 'abundant', 'powerful', 'susceptible', 'yield', 'percent', 'bar', 'pathogenicity', 'xenograft', 'preclinical', 'genome-wide', 'serial', 'disrupts', 'thyroid', 'recurrence', 'dynamic', 'x-ray', 'jolla', 'apart', 'immunofluorescence', 'generating', 'notable', 'distance', 'million', 'oligonucleotides', 'pool', 'chromosomes', '0.25', 'definition', 'amplify', 'former', 'input', 'hot', 'dose-dependent', 'died', 'own', 'bacterial', 'false', 'c-terminus', 'treating', 'tris', '3.5', 'biologic', 'lies', 'pik3ca', 'phosphatidylinositol', 'centrifuged', 'conjunction', '107', '3f', 'recognizes', 'science', 'objective', 'maximal', '', 'behavior', 'proved', 'ways', 'degrees', 'dependence', 'washing', 'interval', 'copies', 'lesion', 'written', 'enough', 'marrow', 'accounts', 'field', 'placed', 'interfering', '1993', 'flag-tagged', 'evolution', 'subcellular', 'genomes', 'deficiency', 'pedigrees', 'diego', 'settings', 'experience', 'immune', 'igg', 'recorded', '8.0', 'embedded', 'segment', 'precipitated', 'surgical', 'j', 'dynamics', 'antigen', 'fall', 'histone', 'crystallographic', 'flag', 'glycerol', 'sources', 'snps', 'ny', 'indistinguishable', 'mitogen-activated', 'facilitates', 'bases', 'switch', 'baseline', 'potency', 'kind', 'digested', 'manuscript', 'relation', 'evolutionary', 'abolish', 'fresh', '84', 'unit', 'deregulation', 'management', '81', 'limit', '87', 'reads', 'natural', 'overexpressing', 'atcc', 'reading', 'prompted', 'algorithms', 'compromised', 'methionine', 'microarray', 'identifies', 'predictor', 'deregulated', 'bridge', 'statistics', 'spontaneous', 'drawn', 'estimates', 'subunits', 'designated', 'chemiluminescence', 'continuous', 'recruited', 'explaining', 'separately', 'gap', 'severity', 'uses', 'kindly', 'comprise', 'goal', 'refractory', 'intriguingly', 'arrow', 'ethanol', 'place', 'nonsynonymous', '92', 'proximal', 'lee', 'saline', 'stored', 'concordance', 'mild', 'dishes', '1000', '91', 'mek', 'drosophila', 'constitute', 'assessing', 'nude', 'classify', 'believed', 'genotyping', 'conformations', 'experimentally', 'uk', 'receiving', 'downloaded', 'acting', 'brca2', '98', 'cm', 'reducing', 'underlined', 'actin', 'small-molecule', 'decreases', 'formalin-fixed', 'e.g', 'upregulation', 'apoptotic', 'investigations', 'constant', 'profound', '3-kinase', 'contributed', 'fourth', 'glioblastoma', 'minority', 'units', 'modulation', 'biopsy', 'sirna', 'pa', 'insert', 'promoters', 'automated', 'hepes', 'atlas', 'sufficiently', 'alters', 'publication', '82', 'possibilities', 'angiogenesis', 'eluted', 'raises', '7b', 'silent', 'nras', 'tcga', 'escherichia', 'comprises', 'dominant-negative', 'mentioned', 'nonetheless', '3.0', 'differentiated', 'methylation', 'tumours', 'st.', 'utilized', 'clarify', '93', 'ubiquitination', 'dissociation', 'destabilization', '103', 'dr.', 'valuable', 'imatinib', 'practice', 'reminiscent', 'measures', 'modeled', 'limits', 'virtually', 'intriguing', 'indirectly', 'nj', 'charge', 'accounted', 'decisions', 'healthy', '', 'subgroup', 'interference', 'mix', 'analyzer', 'discrepancy', 'weaker', 'horseradish', 'elucidated', 'specimen', 'contexts', 'paper', 'rearrangement', 'project', 'g418', 'subjects', 'incorporation', 'rely', 'helical', 'vehicle', 'terminator', 'concurrent', 'elsewhere', 'agilent', 'pronounced', 'duration', 'cytotoxic', 'xenografts', 'distal', 'entry', 'high-throughput', 'publications', 'assembled', 'spread', 'fax', 'illustrate', 'genomics', 'increasingly', 'associates', 'drivers', '6d', 'standardized', 'treat', 'performing', 'complement', 'fluoride', 'ethics', 'solvent', 'segregation', 'differs', 'routinely', 'subgroups', '', 'dysregulation', 'adhesion', 'paraffin', 'environment', 'hundred', 'aimed', 'necessarily', 'aim', '97', 'e3', 'legend', 'acidic', 'insensitive', 'impairment', 'sizes', 'covering', 'kd', 'exert', 'controlling', 'plot', 'q', 'downregulation', 'spots', 'tend', 'recommendations', 'screens', 'exonic', 's1a', 'u', 'puromycin', 'experienced', 'school', 'biopsies', 'conditional', 'm.', '24', 'peak', 'tightly', 'property', 'accurate', 'partly', 'metastases', 'consider', 'missing', '130', 'scanning', 'zhang', 'w', '1.2', 'purposes', 'pcdna3', 'sequential', '4.5', 'ct', 'single-nucleotide', 'predicts', 'machinery', 'corrected', 'suitable', 'unclassified', 'mtor', 'phenomenon', 'lipid', 'evidenced', 'immunoblotted', 'implemented', 'pressure', 'melanomas', 'resection', 'soluble', 'immunoprecipitates', 'named', 'consecutive', '0.8', 'implying', 'propensity', 'i.e', 'expansion', 'retention', 'issues', 'potently', 'served', 'public', 'fundamental', 'inlineview', 'ba/f3', '140', 'restore', 'axis', 'd1', 'incorporated', 'lineage', 'dimer', 'responsiveness', 'differently', 'germ', 'cleavage', 'brief', 'silencing', 'pedigree', 'children', 'helices', 'egf', '0.6', 'goat', 'p21', 'her', 'cell-based', 'excluding', 'absolute', 'foster', 'european', 'numbering', 'lymphocytes', 'therapeutics', 'ci', 'linking', '117', 'ha', 'undetectable', 'il-3', 'decreasing', 'retains', 'rescued', 's7', 'broadly', 'intervention', 'seemed', 'diffuse', 'exhibiting', 'organs', 'ones', 'existing', 'stabilizes', 'annealing', 'passenger', 'harbors', 'categorized', '125', 'mobility', 's2a', 'serine/threonine', 'considerably', 'exclusion', 'computed', 'il', 'abbreviations', 'disease-causing', 'england', 'split', '83', 'polyacrylamide', '13', 'week', 'boxes', 'worse', 'proliferate', 'directions', 'injection', 'catalytically', 'reflecting', 'mature', 'hematoxylin', 'initiate', 'd.', '2-fold', 'reside', 'acquire', 'histologically', 'thermo', 'popup', 'systemic', 'burden', 'carrier', 'derive', '2.1', 'utility', 'prevented', 'integration', 'anchorage-independent', 'imply', 'glutamine', 'restored', '24-well', 'capture', 'heterodimer', 'relies', 'arrays', 'attractive', 'bcr-abl', 'questions', 'alpha', 'rendered', 'bias', 'limitations', 'hallmark', 'extreme', 'extract', 'maps', 'mimic', 'communication', 'confirms', '4f', 'metabolism', 'hotspots', 'hek293t', 'ligand-independent', 'lymph', 'correctly', 'clusters', 'node', 'ns', 'precursor', 'noteworthy', 'read', 's1b', 'grew', 'germ-line', 'ip', 'histologic', 'adjusted', 'assuming', 'complexity', 'log', 'linear', 'valine', 'count', 'compatible', 'anti-flag', 'nonspecific', '800', 'off', 'oxygen', 'solved', 'childhood', 'raf', 'implies', 'predispose', 'illumina', 'lethality', 'hour', 'mitotic', 'coordinates', 'deposited', 'removing', 'resource', 'individually', 'fugene', 'hypothesize', 'possesses', 'determinant', 'enable', 'magnitude', 'indicator', 'underlie', '2.2', '99', 'sequencer', 'scheme', 'window', 'calcium', 'perturb', 'module', 'situation', 'mini', 'oral', 'counseling', 'equilibrium', 'violet', 'er', 'loh', 'hela', 'distant', 'mortality', 'cancer-related', 'applicable', 'deep', 'deaths', 'break', 'refer', 'principle', 'junction', 'request', 'buried', 'streptomycin', 'ge', 'wells', 'intensities', 'owing', 'initiating', 'breaks', 'atp-binding', 'src', 'consideration', 'adverse', 'normalization', 'checked', 'programs', 'take', 'high-risk', 'phenylalanine', 'columns', 'pharmacological', 'presentation', 'similarities', 'atoms', 'methyl', 's2b', 'scan', 'sex', 'aid', 'apply', 'enrolled', 'long-term', 'biologically', 'please', 'cis', 'mismatch', '1640', 'replication', 'relapse', 'accelerated', 'excellent', 'm2', 's3a', '3.1', 'young', 'face', 'entirely', 'introducing', 'arm', 'graph', 'tract', 'success', 'adopt', 'foundation', 'distinguished', 'erbb2', 'evaluating', 'challenging', 'dtt', 'accuracy', 'dye', 'replicates', 'aml', 'energy', 'rapamycin', 'co-transfected', 'dnas', 'monitoring', 'parts', 'starved', 'erk1/2', 'causative', 'res', 'colored', 'submitted', '1.3', 'emergence', 'diminish', 'glucose', 'phosphorylates', 'postulated', 'supernatants', 'recruit', '7c', 'trend', 'lentiviral', 'sulfate', 'destabilizing', 'limiting', 'insulin', 'taking', 'delay', 'overview', '1992', 'translated', 'glu', 'balance', 'phone', 'feedback', 'mim', 'positioned', 'rtk', 'immunostaining', 'framework', 'responded', 'females', 'penicillin', 'look', 'endothelial', 'gradient', 'ala', 'whereby', 'dilutions', 'vascular', 'references', 'revealing', 'needs', 'phosphotyrosine', 'pierce', 'unless', 'manually', 'dark', 'paradigm', 'mammary', '121', 'reasonable', 'recovered', 'tagged', 'juxtamembrane', 'cooperative', 'fcs', 'triplicates', 'beginning', 'toxic', 'clinicians', 'roughly', 'widespread', 'influenced', '1.1', 'herein', 'resultant', 'landscape', 'eliminate', 'mediator', 'spectra', '3t3', 'ice-cold', 'prospective', 'variables', 'plots', 'cambridge', 'polar', 'leukemias', 'spot', 'next-generation', 'sometimes', '0.9', 'chen', 'detecting', 'cut', 'working', 'lie', 'trigger', 'pull-down', 'old', 'tended', 'uncovered', 'li', 'spliced', 'investigators', 'coexpression', 'calbiochem', 'attempted', 'relapsed', 'n-', 'conferring', 'somatically', 'lymphoid', 'accounting', 'denaturing', 'percentages', 'bioinformatics', '4.0', 'assistance', 'escape', 'stringent', 'progressed', 'healthcare', 'definitive', 'undertaken', 'wang', 'fast', 'graphpad', 'concept', 'option', 'attention', 'correlations', 'coordinated', '-actin', 'perturbations', 'pr', 'cocktail', 'aliquots', 'accumulated', 'accumulate', 'believe', 'magnification', 'bsa', 'presents', 'atypical', 'cooperate', 'technique', 'associations', 'flexible', 'greatest', 'adaptor', 'transcribed', 'irrespective', 'attenuated', 'boston', 'termination', 'proband', 'problems', 'pediatric', 'upregulated', 'contrary', 'peroxidase', 'del', 'recurrently', 'bold', 'orthologs', '1991', 'docking', 'sorting', 'well-characterized', 'phosphatases', 'specified', 'creating', '1.4', '2.3', 'completed', 'interpreted', 'raw', 'immunoblots', 'diagnostics', 'external', '2017', 'accommodate', 'requirements', 'decades', 'well-known', 'morphological', 'relationships', 'heat', 'spectrometry', 'labeling', 'uncharacterized', 'upstate', 'kinetics', 'chance', '3.2', 'packing', 'cosegregation', '6-well', 'myeloproliferative', 'candidates', 'progenitor', '110', 'aspartic', 'log-rank', 'versions', 'mmol/l', 'memorial', 'proportions', 'chimeric', 'slow', 'repression', 'inducible', 'concentrated', 'ncbi', 'closed', 'odds', 'plausible', 'checkpoint', 'immediate', 'ethnic', 'retrovirus', 'world', 'fit', 'leaving', 'considerations', 'amplicons', 'coexpressed', 'counting', 'architecture', 'antisense', 'lowest', '1.8', 'der', 'hepatocellular', 'produces', 'indels', 'surfaces', 'started', 'dosage', 'motility', 'competitive', 'gains', 'unusual', 'centers', 'allosteric', 's3b', 'raising', 'gapdh', '2f', 'choice', 'cleft', 'tki', 'transfectants', 'highlighting', 'decided', 'technical', 'preferential', 'uncommon', 'sense', '1,000', 'instances', 'suppressing', '2.6', 'bulky', 'systematically', 'separation', 'ectopically', 'raise', 'annotated', 'anti-ha', 'hematologic', 'limitation', 'histidine', 'renilla', 'strands', 'proto-oncogene', 'stratification', 'counterpart', '350', 'tkis', 'reproducible', 'shrna', 'backgrounds', 'quantitatively', 'intervals', 'gtpase', 'web', 'sum', 'adopts', 'incubating', 'visualization', 'superfamily', 'enables', 'boundaries', 'dead', 'a.', 'withdrawal', 'december', 'cytokines', 'risks', 'counterparts', 'arises', 'concluded', 'captured', 'demonstration', 'st', 'lymphoblastic', 'he', 'opportunity', 'probable', 'college', 'straightforward', '-helix', 'corresponded', 'n-terminus', 'pathologic', 'california', 'rneasy', 'mark', 'differing', 'constitutes', 'symptoms', 'composition', 'suppressive', 'implementation', 'neighboring', 'taq', 'nih3t3', 'july', 'exceeded', 'applying', 'filter', 'probabilities', 'exerts', 'relatives', 'june', 'steric', '1.6', 'sarcoma', 'expresses', 'folded', 'affinities', 'unphosphorylated', 'down-regulation', 'sensitivities', 'filtered', 'cutaneous', 'outlined', 'lobe', 'isoleucine', 'non-small', 'eleven', 'doi', 'reasoned', '123', 'asterisk', 'guide', 'arose', 'invariant', 'impairs', 'deviations', 'groove', 'atom', 'synonymous', 'pretreatment', 'nervous', 'sequence-specific', 'promise', 'proteolysis', 'aacr', '000', 'lengths', 'assumption', 'inflammatory', 'uv', 'layer', 'variance', 'hercules', 'hairpin', 'master', 'obviously', 'enhancement', 'flt3', 'dashed', 'searched', 'tyr', 'immortalized', 'purple', 'precisely', 'defines', 'brct', 'enabling', 'space', 'naf', 'up-regulation', 'myelogenous', 'adults', 'pancreas', '102', 'facilitated', 'converted', 'modify', 'giving', 'simultaneous', 'replace', 'improvement', 'dimers', 'danvers', '109', '23', 'helpful', 'pymol', 'monitor', 'diameter', 'enhancer', 'males', 'kaplan-meier', 'etiology', 'cytogenetic', 'distributions', 'performance', 'enhancing', 'august', 'strictly', 'ligated', 'protein-protein', 'selectivity', 'proteolytic', 'journal', 'broader', 'visible', 'immunoglobulin', 'clarified', 'compromise', 'distinctive', 'ideal', 'resembling', 'senescence', 'rnas', 'drives', 'bacteria', 'beneficial', 'stain', 'gefitinib', 'locally', 'qualitative', 'dodecyl', 'pi3k/akt', '|', 'intramolecular', 'divergent', 'quantify', 'establishing', 'correction', 'heterologous', 'enabled', 'independence', 'ripa', 'chemicals', 'biolabs', 'turnover', 'platelet-derived', 'took', 'np-40', 'suspected', 'warranted', '2014', 'personalized', '0.7', '293', 'describing', 'eosin', '160', 'gives', 'tk', 'pellets', 'marginally', 'back', 'heterodimerization', 'artificial', 'libraries', 'cold', 'ihc', 'arg', 'double-strand', 'antitumor', 'ecl', 'ecori', 'electrostatic', 'cerevisiae', 'genotypes', '10,000', 'yielding', '1/2', 'month', 'movement', 'older', 'shifted', 'asked', 'damaging', 'confocal', 'covered', 'stimulates', 'france', 'picture', 'rtks', 'dysregulated', 'ile', 'messenger', 'triggered', '5.0', 'parameter', 'participates', '2', 'homologs', 'carboxy-terminal', '2.4', '12.5', 'signs', 'phospho-akt'])"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abridgedsorted.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged[\"tokenizedSentenceslower\"]=merged[\"tokenizedSentences\"].map(lowerList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged[\"stemmedSentences\"]=merged[\"tokenizedSentenceslower\"].map(stemList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "wordcounts = {}\n",
    "for i in merged[\"tokenizedSentenceslower\"]:\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsorted = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.9 and v>len(merged)*.1:\n",
    "        abridgedsorted[k]=v\n",
    "merged[\"tksfs\"]=merged[\"tokenizedSentenceslower\"].map(lambda b: [int(word in b) for word in abridgedsorted.keys()])\n",
    "\n",
    "wordcounts = {}\n",
    "for i in merged[\"stemmedSentences\"]:\n",
    "    words = set(i)\n",
    "    for word in words:\n",
    "        wordcounts[word] = wordcounts.get(word,0)+1\n",
    "sortedwords = sorted(wordcounts.items(), key=lambda c: c[1], reverse=True)\n",
    "abridgedsortedst = {}\n",
    "for (k,v) in sortedwords:\n",
    "    if v< len(merged)*.9 and v>len(merged)*.1:\n",
    "        abridgedsortedst[k]=v\n",
    "merged[\"stsfs\"]=merged[\"stemmedSentences\"].map(lambda b: [int(word in b) for word in abridgedsortedst.keys()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.36261181474\n",
      "Accuracy on training set:\n",
      "1.16974478285\n",
      "Accuracy on test set:\n",
      "1.3672453906\n",
      "Accuracy on training set:\n",
      "1.17405370866\n",
      "Accuracy on test set:\n",
      "1.39396178973\n",
      "Accuracy on training set:\n",
      "1.22210601382\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"tksfs\", \"Class\")\n",
    "logistictester(merged, \"stsfs\", \"Class\")\n",
    "logistictester(merged, \"sentenceWordFeatures\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "594\n",
      "562\n",
      "606\n"
     ]
    }
   ],
   "source": [
    "print(len(merged[\"tksfs\"][0]))\n",
    "print(len(merged[\"stsfs\"][0]))\n",
    "print(len(merged[\"sentenceWordFeatures\"][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.898992950261\n",
      "Accuracy on training set:\n",
      "0.410107793969\n",
      "Accuracy on test set:\n",
      "0.915415076969\n",
      "Accuracy on training set:\n",
      "0.458825115764\n"
     ]
    }
   ],
   "source": [
    "merged[\"tkall\"]= merged[\"tkfs\"]+merged[\"tksfs\"]\n",
    "merged[\"stall\"]= merged[\"stfs\"]+merged[\"stsfs\"]\n",
    "\n",
    "logistictester(merged, \"tkall\", \"Class\")\n",
    "logistictester(merged, \"stall\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'Gene', 'Text', 'Variation', 'Class', 'tokenized', 'wordbag',\n",
       "       'genebag', 'Sentences', 'tokenizedSentences', 'sentenceWordBag',\n",
       "       'bigrams', 'genesmentioned', 'geneofinterest', 'wordfeatures',\n",
       "       'mutType', 'sentenceWordFeatures', 'bigramfeatures', 'all', 'stemmed',\n",
       "       'loweredtokens', 'tkfs', 'stfs', 'split', 'tokenizedSentenceslower',\n",
       "       'stemmedSentences', 'tksfs', 'stsfs', 'tkall', 'stall'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.90323515643\n",
      "Accuracy on training set:\n",
      "0.416149729852\n"
     ]
    }
   ],
   "source": [
    "merged[\"x\"]=merged[\"wordfeatures\"]+merged[\"sentenceWordFeatures\"]\n",
    "logistictester(merged, \"x\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.881107744171\n",
      "Accuracy on training set:\n",
      "0.358872645365\n",
      "Accuracy on test set:\n",
      "0.891478146387\n",
      "Accuracy on training set:\n",
      "0.394260924059\n"
     ]
    }
   ],
   "source": [
    "merged[\"all\"]=merged[\"tkfs\"]+merged[\"genesmentioned\"]+merged[\"geneofinterest\"]+merged[\"mutType\"]+merged[\"tksfs\"]+merged[\"bigramfeatures\"]\n",
    "merged[\"allst\"]=merged[\"stfs\"]+merged[\"genesmentioned\"]+merged[\"geneofinterest\"]+merged[\"mutType\"]+merged[\"stsfs\"]+merged[\"bigramfeatures\"]\n",
    "logistictester(merged, \"all\", \"Class\")\n",
    "logistictester(merged, \"allst\", \"Class\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9291\n",
      "8302\n"
     ]
    }
   ],
   "source": [
    "print(len(merged[\"all\"][0]))\n",
    "print(len(merged[\"allst\"][0]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1128\n",
      "4572\n",
      "Accuracy on test set:\n",
      "1.5570535051\n",
      "Accuracy on training set:\n",
      "1.4535594051\n",
      "Accuracy on test set:\n",
      "1.37573625335\n",
      "Accuracy on training set:\n",
      "1.1235283018\n"
     ]
    }
   ],
   "source": [
    "print(len(merged[\"bigramfeaturesfiltered\"][0]))\n",
    "print(len(merged[\"bigramfeatures\"][0]))\n",
    "\n",
    "logistictester(merged, \"bigramfeaturesfiltered\", \"Class\")\n",
    "logistictester(merged, \"bigramfeatures\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.887567715586\n",
      "Accuracy on training set:\n",
      "0.393044800739\n",
      "Accuracy on test set:\n",
      "0.881107744171\n",
      "Accuracy on training set:\n",
      "0.358872645365\n"
     ]
    }
   ],
   "source": [
    "merged[\"allft\"]=merged[\"tkfs\"]+merged[\"genesmentioned\"]+merged[\"geneofinterest\"]+merged[\"mutType\"]+merged[\"tksfs\"]+merged[\"bigramfeaturesfiltered\"]\n",
    "merged[\"all\"]=merged[\"tkfs\"]+merged[\"genesmentioned\"]+merged[\"geneofinterest\"]+merged[\"mutType\"]+merged[\"tksfs\"]+merged[\"bigramfeatures\"]\n",
    "\n",
    "logistictester(merged, \"allft\", \"Class\")\n",
    "logistictester(merged, \"all\", \"Class\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9291\n",
      "5847\n"
     ]
    }
   ],
   "source": [
    "print(len(merged[\"all\"][0]))\n",
    "print(len(merged[\"allft\"][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "0.92086615935\n",
      "Accuracy on training set:\n",
      "0.199745516484\n",
      "Accuracy on test set:\n",
      "0.887567715586\n",
      "Accuracy on training set:\n",
      "0.393044800739\n",
      "Accuracy on test set:\n",
      "0.925553440886\n",
      "Accuracy on training set:\n",
      "0.532481473998\n",
      "Accuracy on test set:\n",
      "1.27982564631\n",
      "Accuracy on training set:\n",
      "0.083359046324\n",
      "Accuracy on test set:\n",
      "0.881107744171\n",
      "Accuracy on training set:\n",
      "0.358872645365\n",
      "Accuracy on test set:\n",
      "0.911233803094\n",
      "Accuracy on training set:\n",
      "0.484541892687\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"allft\", \"Class\", C=5e-2)\n",
    "logistictester(merged, \"allft\", \"Class\", C=1e-2)\n",
    "logistictester(merged, \"allft\", \"Class\", C=5e-3)\n",
    "logistictester(merged, \"all\", \"Class\", C=5e-1)\n",
    "logistictester(merged, \"all\", \"Class\", C=1e-2)\n",
    "logistictester(merged, \"all\", \"Class\", C=5e-3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy on test set:\n",
      "1.36261181474\n",
      "Accuracy on training set:\n",
      "1.16974478285\n",
      "Accuracy on test set:\n",
      "1.06788325526\n",
      "Accuracy on training set:\n",
      "0.755133174496\n",
      "Accuracy on test set:\n",
      "1.61814665159\n",
      "Accuracy on training set:\n",
      "1.51613091716\n"
     ]
    }
   ],
   "source": [
    "logistictester(merged, \"tksfs\", \"Class\")   #sentences lowered from genes or allele\n",
    "logistictester(merged, \"sentencegWordFeatures\", \"Class\")   # sentencegWordFeatures only from genes\n",
    "logistictester(merged, \"sentencevWordFeatures\", \"Class\")   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "594\n",
      "1696\n",
      "293\n"
     ]
    }
   ],
   "source": [
    "print(len(merged[\"tksfs\"][0]))\n",
    "print(len(merged[\"sentencegWordFeatures\"][0]))\n",
    "print(len(merged[\"sentencevWordFeatures\"][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception ignored in: 'zmq.backend.cython.message.Frame.__dealloc__'\n",
      "Traceback (most recent call last):\n",
      "  File \"zmq/backend/cython/checkrc.pxd\", line 12, in zmq.backend.cython.checkrc._check_rc (zmq/backend/cython/message.c:4294)\n",
      "KeyboardInterrupt\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1989\n",
      "Accuracy on test set:\n",
      "1.03711558563\n",
      "Accuracy on training set:\n",
      "0.698230079295\n"
     ]
    }
   ],
   "source": [
    "merged[\"allsentences\"] = merged[\"sentencegWordFeatures\"] + merged[\"sentencevWordFeatures\"]\n",
    "print(len(merged[\"allsentences\"][0]))\n",
    "logistictester(merged, \"allsentences\", \"Class\")  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Text</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>tokenized</th>\n",
       "      <th>wordbag</th>\n",
       "      <th>genebag</th>\n",
       "      <th>Sentences</th>\n",
       "      <th>tokenizedSentences</th>\n",
       "      <th>...</th>\n",
       "      <th>allft</th>\n",
       "      <th>gSentences</th>\n",
       "      <th>vSentences</th>\n",
       "      <th>tokenizedvSentences</th>\n",
       "      <th>sentencevWordBag</th>\n",
       "      <th>sentencevWordFeatures</th>\n",
       "      <th>tokenizedgSentences</th>\n",
       "      <th>sentencegWordBag</th>\n",
       "      <th>sentencegWordFeatures</th>\n",
       "      <th>allsentences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>[Cyclin-dependent, kinases, (, CDKs, ), regula...</td>\n",
       "      <td>{'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...</td>\n",
       "      <td>{'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}</td>\n",
       "      <td>cyclin m, an orphan cyclin, is the product of...</td>\n",
       "      <td>[Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>cyclin m, an orphan cyclin, is the product of...</td>\n",
       "      <td></td>\n",
       "      <td>[]</td>\n",
       "      <td>{}</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[cyclin, m, ,, an, orphan, cyclin, ,, is, the,...</td>\n",
       "      <td>{'cyclin': 1, 'orphan': 1, 'product': 1, 'fam5...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>[Abstract, Background, Non-small, cell, lung, ...</td>\n",
       "      <td>{'Abstract': 1, 'Background': 1, 'Non-small': ...</td>\n",
       "      <td>{'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...</td>\n",
       "      <td>using select c-cbl somatic mutations such as ...</td>\n",
       "      <td>[Using, select, c-CBL, somatic, mutations, suc...</td>\n",
       "      <td>...</td>\n",
       "      <td>[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...</td>\n",
       "      <td>c-cbl is an e3 ubiquitin ligase and adaptor m...</td>\n",
       "      <td>using select c-cbl somatic mutations such as ...</td>\n",
       "      <td>[using, select, c-cbl, somatic, mutations, suc...</td>\n",
       "      <td>{'using': 1, 'select': 1, 'c-cbl': 1, 'somatic...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, ...</td>\n",
       "      <td>[c-cbl, is, an, e3, ubiquitin, ligase, and, ad...</td>\n",
       "      <td>{'c-cbl': 1, 'e3': 1, 'ubiquitin': 1, 'ligase'...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, ...</td>\n",
       "      <td>[0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows  44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ID    Gene                                               Text  \\\n",
       "0   0  FAM58A  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   1     CBL   Abstract Background  Non-small cell lung canc...   \n",
       "\n",
       "              Variation  Class  \\\n",
       "0  Truncating Mutations      1   \n",
       "1                 W802*      2   \n",
       "\n",
       "                                           tokenized  \\\n",
       "0  [Cyclin-dependent, kinases, (, CDKs, ), regula...   \n",
       "1  [Abstract, Background, Non-small, cell, lung, ...   \n",
       "\n",
       "                                             wordbag  \\\n",
       "0  {'Cyclin-dependent': 1, 'kinases': 1, 'CDKs': ...   \n",
       "1  {'Abstract': 1, 'Background': 1, 'Non-small': ...   \n",
       "\n",
       "                                             genebag  \\\n",
       "0    {'FAM58A': 1, 'CDK8': 1, 'CCND1': 1, 'MYCN': 1}   \n",
       "1  {'EGFR': 1, 'MET': 1, 'KRAS': 1, 'CBL': 1, 'SR...   \n",
       "\n",
       "                                           Sentences  \\\n",
       "0   cyclin m, an orphan cyclin, is the product of...   \n",
       "1   using select c-cbl somatic mutations such as ...   \n",
       "\n",
       "                                  tokenizedSentences  \\\n",
       "0  [Cyclin, M, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [Using, select, c-CBL, somatic, mutations, suc...   \n",
       "\n",
       "                         ...                          \\\n",
       "0                        ...                           \n",
       "1                        ...                           \n",
       "\n",
       "                                               allft  \\\n",
       "0  [1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, ...   \n",
       "1  [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...   \n",
       "\n",
       "                                          gSentences  \\\n",
       "0   cyclin m, an orphan cyclin, is the product of...   \n",
       "1   c-cbl is an e3 ubiquitin ligase and adaptor m...   \n",
       "\n",
       "                                          vSentences  \\\n",
       "0                                                      \n",
       "1   using select c-cbl somatic mutations such as ...   \n",
       "\n",
       "                                 tokenizedvSentences  \\\n",
       "0                                                 []   \n",
       "1  [using, select, c-cbl, somatic, mutations, suc...   \n",
       "\n",
       "                                    sentencevWordBag  \\\n",
       "0                                                 {}   \n",
       "1  {'using': 1, 'select': 1, 'c-cbl': 1, 'somatic...   \n",
       "\n",
       "                               sentencevWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, ...   \n",
       "\n",
       "                                 tokenizedgSentences  \\\n",
       "0  [cyclin, m, ,, an, orphan, cyclin, ,, is, the,...   \n",
       "1  [c-cbl, is, an, e3, ubiquitin, ligase, and, ad...   \n",
       "\n",
       "                                    sentencegWordBag  \\\n",
       "0  {'cyclin': 1, 'orphan': 1, 'product': 1, 'fam5...   \n",
       "1  {'c-cbl': 1, 'e3': 1, 'ubiquitin': 1, 'ligase'...   \n",
       "\n",
       "                               sentencegWordFeatures  \\\n",
       "0  [0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, ...   \n",
       "1  [0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, ...   \n",
       "\n",
       "                                        allsentences  \n",
       "0  [0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, ...  \n",
       "1  [0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, ...  \n",
       "\n",
       "[2 rows x 44 columns]"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "del merged[\"wordbag\"]\n",
    "del merged[\"genebag\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'Gene', 'Text', 'Variation', 'Class', 'tokenized', 'Sentences',\n",
       "       'tokenizedSentences', 'sentenceWordBag', 'bigrams', 'genesmentioned',\n",
       "       'geneofinterest', 'wordfeatures', 'mutType', 'sentenceWordFeatures',\n",
       "       'bigramfeatures', 'all', 'stemmed', 'loweredtokens', 'tkfs', 'stfs',\n",
       "       'split', 'tokenizedSentenceslower', 'stemmedSentences', 'tksfs',\n",
       "       'stsfs', 'tkall', 'stall', 'x', 'allst', 'bigramsfiltered',\n",
       "       'bigramfeaturesfiltered', 'allft', 'gSentences', 'vSentences',\n",
       "       'tokenizedvSentences', 'sentencevWordBag', 'sentencevWordFeatures',\n",
       "       'tokenizedgSentences', 'sentencegWordBag', 'sentencegWordFeatures',\n",
       "       'allsentences'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "del merged[\"genesmentioned\"]\n",
    "del merged[\"geneofinterest\"]\n",
    "del merged[\"x\"]\n",
    "del merged[\"stemmed\"]\n",
    "del merged[\"stfs\"]\n",
    "del merged[\"stemmedSentences\"]\n",
    "del merged[\"stsfs\"]\n",
    "del merged[\"stall\"]\n",
    "del merged[\"allst\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'Gene', 'Text', 'Variation', 'Class', 'tokenized', 'Sentences',\n",
       "       'tokenizedSentences', 'sentenceWordBag', 'bigrams', 'wordfeatures',\n",
       "       'mutType', 'sentenceWordFeatures', 'bigramfeatures', 'all',\n",
       "       'loweredtokens', 'tkfs', 'split', 'tokenizedSentenceslower', 'tksfs',\n",
       "       'tkall', 'bigramsfiltered', 'bigramfeaturesfiltered', 'allft',\n",
       "       'gSentences', 'vSentences', 'tokenizedvSentences', 'sentencevWordBag',\n",
       "       'sentencevWordFeatures', 'tokenizedgSentences', 'sentencegWordBag',\n",
       "       'sentencegWordFeatures', 'allsentences'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df[\"all\"]=df[\"wordfeatures\"]+df[\"genesmentioned\"]+df[\"geneofinterest\"]+df[\"mutType\"]+df[\"sentenceWordFeatures\"]+df[\"bigramfeatures\"]\n",
    "nEmpty"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3426"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nEmpty"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3689"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "39\n",
      "DICER1\n",
      "40\n",
      "DICER1\n",
      "78\n",
      "CCND3\n",
      "95\n",
      "TGFBR1\n",
      "98\n",
      "TGFBR2\n",
      "99\n",
      "TGFBR2\n",
      "284\n",
      "NKX2-1\n",
      "367\n",
      "EPAS1\n",
      "373\n",
      "TP53\n",
      "374\n",
      "TP53\n",
      "375\n",
      "TP53\n",
      "377\n",
      "TP53\n",
      "379\n",
      "TP53\n",
      "383\n",
      "TP53\n",
      "394\n",
      "TP53\n",
      "408\n",
      "TP53\n",
      "411\n",
      "TP53\n",
      "413\n",
      "TP53\n",
      "421\n",
      "TP53\n",
      "424\n",
      "TP53\n",
      "425\n",
      "TP53\n",
      "426\n",
      "TP53\n",
      "429\n",
      "TP53\n",
      "432\n",
      "TP53\n",
      "436\n",
      "TP53\n",
      "442\n",
      "TP53\n",
      "443\n",
      "TP53\n",
      "445\n",
      "TP53\n",
      "451\n",
      "TP53\n",
      "462\n",
      "TP53\n",
      "465\n",
      "TP53\n",
      "485\n",
      "TP53\n",
      "488\n",
      "TP53\n",
      "491\n",
      "TP53\n",
      "505\n",
      "TP53\n",
      "509\n",
      "TP53\n",
      "511\n",
      "TP53\n",
      "512\n",
      "TP53\n",
      "513\n",
      "TP53\n",
      "522\n",
      "TP53\n",
      "526\n",
      "TP53\n",
      "527\n",
      "TP53\n",
      "531\n",
      "TP53\n",
      "617\n",
      "FBXW7\n",
      "626\n",
      "FBXW7\n",
      "627\n",
      "FBXW7\n",
      "647\n",
      "CDKN2A\n",
      "656\n",
      "CDKN2A\n",
      "658\n",
      "CDKN2A\n",
      "672\n",
      "CDKN2A\n",
      "674\n",
      "CDKN2A\n",
      "676\n",
      "CDKN2A\n",
      "681\n",
      "CDKN2A\n",
      "682\n",
      "CDKN2A\n",
      "707\n",
      "ERBB2\n",
      "716\n",
      "ERBB2\n",
      "770\n",
      "ERBB2\n",
      "791\n",
      "ERBB4\n",
      "795\n",
      "ERBB4\n",
      "796\n",
      "ERBB4\n",
      "823\n",
      "BRIP1\n",
      "824\n",
      "BRIP1\n",
      "825\n",
      "BRIP1\n",
      "828\n",
      "BRIP1\n",
      "838\n",
      "ABL1\n",
      "839\n",
      "ABL1\n",
      "840\n",
      "ABL1\n",
      "843\n",
      "ABL1\n",
      "844\n",
      "ABL1\n",
      "845\n",
      "ABL1\n",
      "846\n",
      "ABL1\n",
      "847\n",
      "ABL1\n",
      "848\n",
      "ABL1\n",
      "849\n",
      "ABL1\n",
      "850\n",
      "ABL1\n",
      "851\n",
      "ABL1\n",
      "852\n",
      "ABL1\n",
      "853\n",
      "ABL1\n",
      "855\n",
      "ABL1\n",
      "856\n",
      "ABL1\n",
      "857\n",
      "ABL1\n",
      "859\n",
      "ABL1\n",
      "860\n",
      "ABL1\n",
      "861\n",
      "ABL1\n",
      "862\n",
      "ABL1\n",
      "964\n",
      "ESR1\n",
      "967\n",
      "ESR1\n",
      "971\n",
      "ESR1\n",
      "1154\n",
      "KMT2C\n",
      "1157\n",
      "KMT2C\n",
      "1225\n",
      "PIK3CB\n",
      "1227\n",
      "PIK3CB\n",
      "1245\n",
      "PIK3R1\n",
      "1246\n",
      "PIK3R1\n",
      "1258\n",
      "PIK3R1\n",
      "1259\n",
      "PIK3R1\n",
      "1260\n",
      "PIK3R1\n",
      "1267\n",
      "PIK3R1\n",
      "1268\n",
      "PIK3R1\n",
      "1296\n",
      "HRAS\n",
      "1343\n",
      "AKT1\n",
      "1576\n",
      "RAD54L\n",
      "1964\n",
      "CTNNB1\n",
      "1971\n",
      "CTNNB1\n",
      "1981\n",
      "CTNNB1\n",
      "1996\n",
      "MAP2K1\n",
      "1997\n",
      "MAP2K1\n",
      "2001\n",
      "MAP2K1\n",
      "2005\n",
      "MAP2K1\n",
      "2013\n",
      "MAP2K1\n",
      "2014\n",
      "MAP2K1\n",
      "2017\n",
      "MAP2K1\n",
      "2019\n",
      "MAP2K1\n",
      "2030\n",
      "MAP2K1\n",
      "2032\n",
      "MAP2K2\n",
      "2035\n",
      "MAP2K2\n",
      "2039\n",
      "MAP2K2\n",
      "2040\n",
      "MAP2K2\n",
      "2124\n",
      "CCND1\n",
      "2127\n",
      "CCND1\n",
      "2351\n",
      "AURKA\n",
      "2352\n",
      "AURKA\n",
      "2363\n",
      "STK11\n",
      "2911\n",
      "NFE2L2\n",
      "2914\n",
      "NFE2L2\n",
      "2917\n",
      "NFE2L2\n",
      "2918\n",
      "NFE2L2\n",
      "2929\n",
      "NFE2L2\n",
      "2930\n",
      "NFE2L2\n",
      "2941\n",
      "NFKBIA\n",
      "2954\n",
      "GNAS\n",
      "2955\n",
      "GNAS\n",
      "3108\n",
      "DNMT3B\n",
      "3142\n",
      "KRAS\n",
      "3155\n",
      "KRAS\n",
      "3165\n",
      "RAF1\n",
      "3169\n",
      "RAF1\n",
      "3201\n",
      "RASA1\n",
      "3221\n",
      "NTRK1\n",
      "3231\n",
      "NTRK2\n",
      "3233\n",
      "NTRK2\n",
      "3302\n",
      "RUNX1\n",
      "3304\n",
      "RUNX1\n",
      "3353\n",
      "TP53\n",
      "3389\n",
      "BCL2L11\n",
      "3410\n",
      "ABL1\n",
      "3413\n",
      "MAP2K1\n",
      "3428\n",
      "TP53\n",
      "3510\n",
      "ABL1\n",
      "3589\n",
      "NTRK2\n",
      "3597\n",
      "ABL1\n",
      "3603\n",
      "GLI1\n",
      "3646\n",
      "RUNX1\n",
      "3683\n",
      "MAP2K1\n",
      "3686\n",
      "MAP2K1\n"
     ]
    }
   ],
   "source": [
    "for row,sentence in enumerate(merged[\"gSentences\"]):\n",
    "    if len(sentence)==0:\n",
    "        if len(merged[\"vSentences\"][row])>0:\n",
    "            print(row)\n",
    "            print(merged[\"Gene\"][row])\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for row,sentence in enumerate(merged[\"gSentences\"]):\n",
    "    if len(sentence)==0:\n",
    "        if len(merged[\"vSentences\"][row])>0:\n",
    "            print(row)\n",
    "            print(merged[\"Gene\"][row])\n",
    "            "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
